---
title: "Res_1_screens_070422"
author: "Matt Coelho"
date: "07.04.2022"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
# load packages
library(tidyverse)
library(reshape2)
library(ggrepel)
library(stringr)
library(digest)
library(ggpointdensity)
```

```{r}
# read in gRNA library
guides <- as_tibble(read.csv("Oncogene_BE_ABE_NGN_TWIST_LIB_041121.csv", stringsAsFactors = FALSE, header = TRUE))
guides

# read in off-target NGN analysis from Joel GPU
off_targets <- as_tibble(read.csv("Oncogene_BE_ABE_NGN_TWIST_LIB_092021.offtargets.csv", stringsAsFactors = FALSE, header = TRUE))
off_targets %>% filter(is_dupe == TRUE)

off_targets <- off_targets %>% select(ID, guide, off_target_metric_NG, off_target_summary_NG, polyT, Direction, is_dupe)
  
# join the off targets to the tibble
lib <- left_join(guides, off_targets)

# count guides
guides
lib
lib %>% filter(is_dupe == TRUE)

#  unique guides
lib <- distinct(lib)
lib <- lib %>%
  select(!c(is_dupe, strand, off_target_summary))

# infer strand from gRNA "Direction"
lib <- mutate(lib, strand = ifelse(Direction == "right", "+", ifelse(Direction == "left", "-", "NA")))
lib

# off_target_reports
lib <- separate(lib, off_target_summary_NG, remove = FALSE, into = c("perfect_matches", "one_mismatch", "two_mismatches", "three_mismatches", "four_mismatches", "extra_"), sep = c(","), extra = "merge")
lib <- lib %>% 
  select(!"extra_")

lib <- mutate(lib, perfect_matches = str_remove_all(perfect_matches, "\\{"))
lib <- mutate(lib, four_mismatches = str_remove_all(four_mismatches, "\\}"))

lib <- mutate(lib, perfect_matches = as.numeric(substr(perfect_matches, 4, 50)))
lib <- mutate(lib, one_mismatch = as.numeric(substr(one_mismatch, 4, 50)))
lib <- mutate(lib, two_mismatches = as.numeric(substr(two_mismatches, 4, 50)))
lib <- mutate(lib, three_mismatches = as.numeric(substr(three_mismatches, 4, 50)))
lib <- mutate(lib, four_mismatches = as.numeric(substr(four_mismatches, 4, 50)))

# rename ID to sgRNA_ID
lib <- rename(lib, sgRNA_ID = ID)
lib <- rename(lib, start = start_BEstimate)
lib <- rename(lib, end = end_BEstimate)
lib <- rename(lib, chr = chr_BEstimate)

lib %>% 
  arrange(desc(perfect_matches)) %>% select(Gene, off_target_summary_NG)

lib %>% filter(perfect_matches >1 | is.na(perfect_matches)) 
lib %>% filter(perfect_matches >1) %>% count(Gene)
lib %>% count(Gene)
lib %>% filter(perfect_matches >1) %>% select(Gene, off_target_summary_NG)
lib
```

```{r}
# read in raw gRNA reads counts
raw_reads <- as_tibble(read.csv("raw_read_counts_120422.csv", stringsAsFactors = FALSE, header = TRUE))
raw_reads <- mutate(raw_reads, sgRNA_ID = as.character(sgRNA_ID))

# compute RPM normalised values
# define functions
RPM <- function(x) ((x/sum(x) *1000000) + 1)
average <- function(x,y) ((x + y) /2)
L2FC <- function(x,y) (log2(x/y))
raw_reads <- mutate_at(raw_reads, vars(matches("reads")), list(RPM = RPM))

raw_reads
```

```{r}
# compute average RPM
# HT-29 average
#BE3.9max NGN
raw_reads <- mutate(raw_reads, HT29_CBE_T0_RPM_average = average(HT29_CBE_T0_exp1_reads_RPM, HT29_CBE_T0_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, HT29_CBE_Control_RPM_average = average(HT29_CBE_Control_exp1_reads_RPM, HT29_CBE_Control_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, HT29_CBE_Tram_RPM_average = average(HT29_CBE_Tram_exp1_reads_RPM, HT29_CBE_Tram_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, HT29_CBE_Pict_RPM_average = average(HT29_CBE_Pict_exp1_reads_RPM, HT29_CBE_Pict_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, HT29_CBE_DebCet_RPM_average = average(HT29_CBE_DebCet_exp1_reads_RPM, HT29_CBE_DebCet_exp2_reads_RPM))

#ABE8e NGN
raw_reads <- mutate(raw_reads, HT29_ABE_T0_RPM_average = average(HT29_ABE_T0_exp1_reads_RPM, HT29_ABE_T0_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, HT29_ABE_Control_RPM_average = average(HT29_ABE_Control_exp1_reads_RPM, HT29_ABE_Control_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, HT29_ABE_Tram_RPM_average = average(HT29_ABE_Tram_exp1_reads_RPM, HT29_ABE_Tram_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, HT29_ABE_Pict_RPM_average = average(HT29_ABE_Pict_exp1_reads_RPM, HT29_ABE_Pict_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, HT29_ABE_DebCet_RPM_average = average(HT29_ABE_DebCet_exp1_reads_RPM, HT29_ABE_DebCet_exp2_reads_RPM))

# H23 average
#BE3.9max NGN
raw_reads <- mutate(raw_reads, H23_CBE_T0_RPM_average = average(H23_CBE_T0_exp1_reads_RPM, H23_CBE_T0_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, H23_CBE_Control_RPM_average = average(H23_CBE_Control_exp1_reads_RPM, H23_CBE_Control_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, H23_CBE_Sotor_RPM_average = average(H23_CBE_Sotor_exp1_reads_RPM, H23_CBE_Sotor_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, H23_CBE_Adag_RPM_average = average(H23_CBE_Adag_exp1_reads_RPM, H23_CBE_Adag_exp2_reads_RPM))

#ABE8e NGN
raw_reads <- mutate(raw_reads, H23_ABE_T0_RPM_average = average(H23_ABE_T0_exp1_reads_RPM, H23_ABE_T0_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, H23_ABE_Control_RPM_average = average(H23_ABE_Control_exp1_reads_RPM, H23_ABE_Control_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, H23_ABE_Sotor_RPM_average = average(H23_ABE_Sotor_exp1_reads_RPM, H23_ABE_Sotor_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, H23_ABE_Adag_RPM_average = average(H23_ABE_Adag_exp1_reads_RPM, H23_ABE_Adag_exp2_reads_RPM))


# PC9 average
#BE3.9max NGN
raw_reads <- mutate(raw_reads, PC9_CBE_T0_RPM_average = average(PC9_CBE_T0_exp1_reads_RPM, PC9_CBE_T0_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, PC9_CBE_Control_RPM_average = average(PC9_CBE_Control_exp1_reads_RPM, PC9_CBE_Control_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, PC9_CBE_Gefit_RPM_average = average(PC9_CBE_Gefit_exp1_reads_RPM, PC9_CBE_Gefit_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, PC9_CBE_Osim_RPM_average = average(PC9_CBE_Osim_exp1_reads_RPM, PC9_CBE_Osim_exp2_reads_RPM))

#ABE8e NGN
raw_reads <- mutate(raw_reads, PC9_ABE_T0_RPM_average = average(PC9_ABE_T0_exp1_reads_RPM, PC9_ABE_T0_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, PC9_ABE_Control_RPM_average = average(PC9_ABE_Control_exp1_reads_RPM, PC9_ABE_Control_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, PC9_ABE_Gefit_RPM_average = average(PC9_ABE_Gefit_exp1_reads_RPM, PC9_ABE_Gefit_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, PC9_ABE_Osim_RPM_average = average(PC9_ABE_Osim_exp1_reads_RPM, PC9_ABE_Osim_exp2_reads_RPM))
```

```{r}
# compute L2FC from normalised gRNA counts - single experiments
# HT-29 exp 1
#BE3.9max NGN
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_Control_plasmid_1 = L2FC(HT29_CBE_Control_exp1_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_Control_T0_1 = L2FC(HT29_CBE_Control_exp1_reads_RPM, HT29_CBE_T0_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_Tram_Control_1 = L2FC(HT29_CBE_Tram_exp1_reads_RPM, HT29_CBE_Control_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_Pict_Control_1 = L2FC(HT29_CBE_Pict_exp1_reads_RPM, HT29_CBE_Control_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_DebCet_Control_1 = L2FC(HT29_CBE_DebCet_exp1_reads_RPM, HT29_CBE_Control_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_Tram_plasmid_1 = L2FC(HT29_CBE_Tram_exp1_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_Pict_plasmid_1 = L2FC(HT29_CBE_Pict_exp1_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_DebCet_plasmid_1 = L2FC(HT29_CBE_DebCet_exp1_reads_RPM, plasmid_reads_RPM))

#ABE8e NGN
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_Control_plasmid_1 = L2FC(HT29_ABE_Control_exp1_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_Control_T0_1 = L2FC(HT29_ABE_Control_exp1_reads_RPM, HT29_ABE_T0_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_Tram_Control_1 = L2FC(HT29_ABE_Tram_exp1_reads_RPM, HT29_ABE_Control_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_Pict_Control_1 = L2FC(HT29_ABE_Pict_exp1_reads_RPM, HT29_ABE_Control_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_DebCet_Control_1 = L2FC(HT29_ABE_DebCet_exp1_reads_RPM, HT29_ABE_Control_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_Tram_plasmid_1 = L2FC(HT29_ABE_Tram_exp1_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_Pict_plasmid_1 = L2FC(HT29_ABE_Pict_exp1_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_DebCet_plasmid_1 = L2FC(HT29_ABE_DebCet_exp1_reads_RPM, plasmid_reads_RPM))

# HT-29 exp 2
#BE3.9max NGN
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_Control_plasmid_2 = L2FC(HT29_CBE_Control_exp2_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_Control_T0_2 = L2FC(HT29_CBE_Control_exp2_reads_RPM, HT29_CBE_T0_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_Tram_Control_2 = L2FC(HT29_CBE_Tram_exp2_reads_RPM, HT29_CBE_Control_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_Pict_Control_2 = L2FC(HT29_CBE_Pict_exp2_reads_RPM, HT29_CBE_Control_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_DebCet_Control_2 = L2FC(HT29_CBE_DebCet_exp2_reads_RPM, HT29_CBE_Control_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_Tram_plasmid_2 = L2FC(HT29_CBE_Tram_exp2_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_Pict_plasmid_2 = L2FC(HT29_CBE_Pict_exp2_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_DebCet_plasmid_2 = L2FC(HT29_CBE_DebCet_exp2_reads_RPM, plasmid_reads_RPM))

#ABE8e NGN
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_Control_plasmid_2 = L2FC(HT29_ABE_Control_exp2_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_Control_T0_2 = L2FC(HT29_ABE_Control_exp2_reads_RPM, HT29_ABE_T0_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_Tram_Control_2 = L2FC(HT29_ABE_Tram_exp2_reads_RPM, HT29_ABE_Control_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_Pict_Control_2 = L2FC(HT29_ABE_Pict_exp2_reads_RPM, HT29_ABE_Control_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_DebCet_Control_2 = L2FC(HT29_ABE_DebCet_exp2_reads_RPM, HT29_ABE_Control_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_Tram_plasmid_2 = L2FC(HT29_ABE_Tram_exp2_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_Pict_plasmid_2 = L2FC(HT29_ABE_Pict_exp2_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_DebCet_plasmid_2 = L2FC(HT29_ABE_DebCet_exp2_reads_RPM, plasmid_reads_RPM))


# H23 exp 1
#BE3.9max NGN
raw_reads <- mutate(raw_reads, L2FC_H23_CBE_Control_plasmid_1 = L2FC(H23_CBE_Control_exp1_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_CBE_Control_T0_1 = L2FC(H23_CBE_Control_exp1_reads_RPM, H23_CBE_T0_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_CBE_Sotor_Control_1 = L2FC(H23_CBE_Sotor_exp1_reads_RPM, H23_CBE_Control_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_CBE_Adag_Control_1 = L2FC(H23_CBE_Adag_exp1_reads_RPM, H23_CBE_Control_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_CBE_Sotor_plasmid_1 = L2FC(H23_CBE_Sotor_exp1_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_CBE_Adag_plasmid_1 = L2FC(H23_CBE_Adag_exp1_reads_RPM, plasmid_reads_RPM))

#ABE8e NGN
raw_reads <- mutate(raw_reads, L2FC_H23_ABE_Control_plasmid_1 = L2FC(H23_ABE_Control_exp1_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_ABE_Control_T0_1 = L2FC(H23_ABE_Control_exp1_reads_RPM, H23_ABE_T0_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_ABE_Sotor_Control_1 = L2FC(H23_ABE_Sotor_exp1_reads_RPM, H23_ABE_Control_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_ABE_Adag_Control_1 = L2FC(H23_ABE_Adag_exp1_reads_RPM, H23_ABE_Control_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_ABE_Sotor_plasmid_1 = L2FC(H23_ABE_Sotor_exp1_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_ABE_Adag_plasmid_1 = L2FC(H23_ABE_Adag_exp1_reads_RPM, plasmid_reads_RPM))

# H23 exp 2
#BE3.9max NGN
raw_reads <- mutate(raw_reads, L2FC_H23_CBE_Control_plasmid_2 = L2FC(H23_CBE_Control_exp2_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_CBE_Control_T0_2 = L2FC(H23_CBE_Control_exp2_reads_RPM, H23_CBE_T0_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_CBE_Sotor_Control_2 = L2FC(H23_CBE_Sotor_exp2_reads_RPM, H23_CBE_Control_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_CBE_Adag_Control_2 = L2FC(H23_CBE_Adag_exp2_reads_RPM, H23_CBE_Control_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_CBE_Sotor_plasmid_2 = L2FC(H23_CBE_Sotor_exp2_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_CBE_Adag_plasmid_2 = L2FC(H23_CBE_Adag_exp2_reads_RPM, plasmid_reads_RPM))

#ABE8e NGN
raw_reads <- mutate(raw_reads, L2FC_H23_ABE_Control_plasmid_2 = L2FC(H23_ABE_Control_exp2_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_ABE_Control_T0_2 = L2FC(H23_ABE_Control_exp2_reads_RPM, H23_ABE_T0_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_ABE_Sotor_Control_2 = L2FC(H23_ABE_Sotor_exp2_reads_RPM, H23_ABE_Control_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_ABE_Adag_Control_2 = L2FC(H23_ABE_Adag_exp2_reads_RPM, H23_ABE_Control_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_ABE_Sotor_plasmid_2 = L2FC(H23_ABE_Sotor_exp2_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_ABE_Adag_plasmid_2 = L2FC(H23_ABE_Adag_exp2_reads_RPM, plasmid_reads_RPM))

# PC9 exp 1
#BE3.9max NGN
raw_reads <- mutate(raw_reads, L2FC_PC9_CBE_Control_plasmid_1 = L2FC(PC9_CBE_Control_exp1_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_CBE_Control_T0_1 = L2FC(PC9_CBE_Control_exp1_reads_RPM, PC9_CBE_T0_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_CBE_Gefit_Control_1 = L2FC(PC9_CBE_Gefit_exp1_reads_RPM, PC9_CBE_Control_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_CBE_Osim_Control_1 = L2FC(PC9_CBE_Osim_exp1_reads_RPM, PC9_CBE_Control_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_CBE_Gefit_plasmid_1 = L2FC(PC9_CBE_Gefit_exp1_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_CBE_Osim_plasmid_1 = L2FC(PC9_CBE_Osim_exp1_reads_RPM, plasmid_reads_RPM))

#ABE8e NGN
raw_reads <- mutate(raw_reads, L2FC_PC9_ABE_Control_plasmid_1 = L2FC(PC9_ABE_Control_exp1_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_ABE_Control_T0_1 = L2FC(PC9_ABE_Control_exp1_reads_RPM, PC9_ABE_T0_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_ABE_Gefit_Control_1 = L2FC(PC9_ABE_Gefit_exp1_reads_RPM, PC9_ABE_Control_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_ABE_Osim_Control_1 = L2FC(PC9_ABE_Osim_exp1_reads_RPM, PC9_ABE_Control_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_ABE_Gefit_plasmid_1 = L2FC(PC9_ABE_Gefit_exp1_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_ABE_Osim_plasmid_1 = L2FC(PC9_ABE_Osim_exp1_reads_RPM, plasmid_reads_RPM))

# PC9 exp 2
#BE3.9max NGN
raw_reads <- mutate(raw_reads, L2FC_PC9_CBE_Control_plasmid_2 = L2FC(PC9_CBE_Control_exp2_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_CBE_Control_T0_2 = L2FC(PC9_CBE_Control_exp2_reads_RPM, PC9_CBE_T0_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_CBE_Gefit_Control_2 = L2FC(PC9_CBE_Gefit_exp2_reads_RPM, PC9_CBE_Control_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_CBE_Osim_Control_2 = L2FC(PC9_CBE_Osim_exp2_reads_RPM, PC9_CBE_Control_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_CBE_Gefit_plasmid_2 = L2FC(PC9_CBE_Gefit_exp2_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_CBE_Osim_plasmid_2 = L2FC(PC9_CBE_Osim_exp2_reads_RPM, plasmid_reads_RPM))

#ABE8e NGN
raw_reads <- mutate(raw_reads, L2FC_PC9_ABE_Control_plasmid_2 = L2FC(PC9_ABE_Control_exp2_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_ABE_Control_T0_2 = L2FC(PC9_ABE_Control_exp2_reads_RPM, PC9_ABE_T0_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_ABE_Gefit_Control_2 = L2FC(PC9_ABE_Gefit_exp2_reads_RPM, PC9_ABE_Control_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_ABE_Osim_Control_2 = L2FC(PC9_ABE_Osim_exp2_reads_RPM, PC9_ABE_Control_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_ABE_Gefit_plasmid_2 = L2FC(PC9_ABE_Gefit_exp2_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_ABE_Osim_plasmid_2 = L2FC(PC9_ABE_Osim_exp2_reads_RPM, plasmid_reads_RPM))
```

```{r}
#compute L2FC from average of 2 exps from RPM values
# HT-29 average
#BE3.9max NGN
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_Control_plasmid_average = L2FC(HT29_CBE_Control_RPM_average, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_Control_T0_average = L2FC(HT29_CBE_Control_RPM_average, HT29_CBE_T0_RPM_average))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_Tram_Control_average = L2FC(HT29_CBE_Tram_RPM_average, HT29_CBE_Control_RPM_average))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_Pict_Control_average = L2FC(HT29_CBE_Pict_RPM_average, HT29_CBE_Control_RPM_average))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_DebCet_Control_average = L2FC(HT29_CBE_DebCet_RPM_average, HT29_CBE_Control_RPM_average))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_Tram_plasmid_average = L2FC(HT29_CBE_Tram_RPM_average, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_Pict_plasmid_average = L2FC(HT29_CBE_Pict_RPM_average, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_DebCet_plasmid_average = L2FC(HT29_CBE_DebCet_RPM_average, plasmid_reads_RPM))

#ABE8e NGN
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_Control_plasmid_average = L2FC(HT29_ABE_Control_RPM_average, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_Control_T0_average = L2FC(HT29_ABE_Control_RPM_average, HT29_ABE_T0_RPM_average))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_Tram_Control_average = L2FC(HT29_ABE_Tram_RPM_average, HT29_ABE_Control_RPM_average))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_Pict_Control_average = L2FC(HT29_ABE_Pict_RPM_average, HT29_ABE_Control_RPM_average))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_DebCet_Control_average = L2FC(HT29_ABE_DebCet_RPM_average, HT29_ABE_Control_RPM_average))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_Tram_plasmid_average = L2FC(HT29_ABE_Tram_RPM_average, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_Pict_plasmid_average = L2FC(HT29_ABE_Pict_RPM_average, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_DebCet_plasmid_average = L2FC(HT29_ABE_DebCet_RPM_average, plasmid_reads_RPM))

# H23 average
#BE3.9max NGN
raw_reads <- mutate(raw_reads, L2FC_H23_CBE_Control_plasmid_average = L2FC(H23_CBE_Control_RPM_average, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_CBE_Control_T0_average = L2FC(H23_CBE_Control_RPM_average, H23_CBE_T0_RPM_average))
raw_reads <- mutate(raw_reads, L2FC_H23_CBE_Sotor_Control_average = L2FC(H23_CBE_Sotor_RPM_average, H23_CBE_Control_RPM_average))
raw_reads <- mutate(raw_reads, L2FC_H23_CBE_Adag_Control_average = L2FC(H23_CBE_Adag_RPM_average, H23_CBE_Control_RPM_average))
raw_reads <- mutate(raw_reads, L2FC_H23_CBE_Sotor_plasmid_average = L2FC(H23_CBE_Sotor_RPM_average, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_CBE_Adag_plasmid_average = L2FC(H23_CBE_Adag_RPM_average, plasmid_reads_RPM))

#ABE8e NGN
raw_reads <- mutate(raw_reads, L2FC_H23_ABE_Control_plasmid_average = L2FC(H23_ABE_Control_RPM_average, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_ABE_Control_T0_average = L2FC(H23_ABE_Control_RPM_average, H23_ABE_T0_RPM_average))
raw_reads <- mutate(raw_reads, L2FC_H23_ABE_Sotor_Control_average = L2FC(H23_ABE_Sotor_RPM_average, H23_ABE_Control_RPM_average))
raw_reads <- mutate(raw_reads, L2FC_H23_ABE_Adag_Control_average = L2FC(H23_ABE_Adag_RPM_average, H23_ABE_Control_RPM_average))
raw_reads <- mutate(raw_reads, L2FC_H23_ABE_Sotor_plasmid_average = L2FC(H23_ABE_Sotor_RPM_average, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_ABE_Adag_plasmid_average = L2FC(H23_ABE_Adag_RPM_average, plasmid_reads_RPM))

# PC9 average
#BE3.9max NGN
raw_reads <- mutate(raw_reads, L2FC_PC9_CBE_Control_plasmid_average = L2FC(PC9_CBE_Control_RPM_average, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_CBE_Control_T0_average = L2FC(PC9_CBE_Control_RPM_average, PC9_CBE_T0_RPM_average))
raw_reads <- mutate(raw_reads, L2FC_PC9_CBE_Gefit_Control_average = L2FC(PC9_CBE_Gefit_RPM_average, PC9_CBE_Control_RPM_average))
raw_reads <- mutate(raw_reads, L2FC_PC9_CBE_Osim_Control_average = L2FC(PC9_CBE_Osim_RPM_average, PC9_CBE_Control_RPM_average))
raw_reads <- mutate(raw_reads, L2FC_PC9_CBE_Gefit_plasmid_average = L2FC(PC9_CBE_Gefit_RPM_average, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_CBE_Osim_plasmid_average = L2FC(PC9_CBE_Osim_RPM_average, plasmid_reads_RPM))

#ABE8e NGN
raw_reads <- mutate(raw_reads, L2FC_PC9_ABE_Control_plasmid_average = L2FC(PC9_ABE_Control_RPM_average, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_ABE_Control_T0_average = L2FC(PC9_ABE_Control_RPM_average, PC9_ABE_T0_RPM_average))
raw_reads <- mutate(raw_reads, L2FC_PC9_ABE_Gefit_Control_average = L2FC(PC9_ABE_Gefit_RPM_average, PC9_ABE_Control_RPM_average))
raw_reads <- mutate(raw_reads, L2FC_PC9_ABE_Osim_Control_average = L2FC(PC9_ABE_Osim_RPM_average, PC9_ABE_Control_RPM_average))
raw_reads <- mutate(raw_reads, L2FC_PC9_ABE_Gefit_plasmid_average = L2FC(PC9_ABE_Gefit_RPM_average, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_ABE_Osim_plasmid_average = L2FC(PC9_ABE_Osim_RPM_average, plasmid_reads_RPM))
```

```{r}
# left join gRNA library and read counts

lib <- mutate(lib, Gene = ifelse(sgRNA_type == "intergenic", "N/A", Gene))
lib %>% filter(sgRNA_type =="intergenic") %>% select(Gene, sgRNA_type)
data <- left_join(lib, raw_reads)
data %>% count(sgRNA_type)
```

```{r}
#HT-29
# off-targets pre-filter
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

# CBE control vs T0
plot1 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_HT29_CBE_Control_plasmid_average, y= perfect_matches, colour=Gene)) + geom_point(size=3, position = pos) +
labs(title= "HT-29 CBE: exonic gRNAs", x = "L2FC control vs plasmid", y = "number of off-targets", colour = "gene") +
scale_color_manual(values =c("black", "springgreen4", "orange", "red", "blue", "grey55", "brown", "purple", "pink", "salmon", "royalblue")) +
theme_classic()
plot1
ggsave("FigS1a.eps", width = 6, height = 4, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")


# ABE control vs T0
plot2 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_HT29_ABE_Control_plasmid_average, y= perfect_matches, colour=Gene)) + geom_point(size=3, position = pos) +
labs(title= "HT-29 ABE: exonic gRNAs", x = "L2FC control vs plasmid", y = "number of off-targets", colour = "gene") +
scale_color_manual(values =c("black", "springgreen4", "orange", "red", "blue", "grey55", "brown", "purple", "pink", "salmon", "royalblue")) +
theme_classic()
plot2
ggsave("FigS1b.eps", width = 6, height = 4, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")
```

```{r}
#filter out more that 2 perfect mismatches
data %>% filter(perfect_matches >2 | is.na(perfect_matches))
data %>% filter(perfect_matches >500 | is.na(perfect_matches))
data <- data %>% filter(perfect_matches < 3 | is.na(perfect_matches))
```

```{r}
# quality control plots
# HT-29
plot1 <- ggplot(data, aes(y= plasmid_reads)) + 
geom_histogram() +
labs(title= "plasmid Reads")
plot1

plot2 <- ggplot(data, aes(y= HT29_CBE_T0_exp1_reads)) + 
geom_histogram() +
labs(title= "HT29 CBE T0 reads")
plot2

plot3 <- ggplot(data, aes(y= HT29_ABE_T0_exp1_reads)) + 
geom_histogram() +
labs(title= "HT29 ABE T0 reads")
plot3

plot4 <- ggplot(data, aes(x= L2FC_HT29_CBE_Control_T0_1, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "L2FC_CBE_Control_T0_1") +
theme_classic()
plot4

plot5 <- ggplot(data, aes(x= L2FC_HT29_ABE_Control_T0_1, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "L2FC_ABE_Control_T0_1") +
theme_classic()
plot5

plot6 <- ggplot(data, aes(x= L2FC_HT29_CBE_Control_plasmid_1, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "L2FC_CBE_Control_plasmid_1") +
theme_classic()
plot6

plot7 <- ggplot(data, aes(x= L2FC_HT29_ABE_Control_plasmid_1, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "L2FC_ABE_Control_plasmid_1") +
theme_classic()
plot7

plot8 <- ggplot(data, aes(x= L2FC_HT29_CBE_Control_plasmid_2, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "L2FC_CBE_Control_plasmid_2") +
theme_classic()
plot8

plot9 <- ggplot(data, aes(x= L2FC_HT29_ABE_Control_plasmid_2, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "L2FC_ABE_Control_plasmid_2") +
theme_classic()
plot9
```

```{r}
# quality control plots
# H23
plot1 <- ggplot(data, aes(y= H23_CBE_T0_exp1_reads)) + 
geom_histogram() +
labs(title= "H23 CBE T0 reads")
plot1

plot2 <- ggplot(data, aes(y= H23_ABE_T0_exp1_reads)) + 
geom_histogram() +
labs(title= "H23 ABE T0 reads")
plot2

plot3 <- ggplot(data, aes(x= L2FC_H23_CBE_Control_T0_1, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "L2FC_CBE_Control_T0_1") +
theme_classic()
plot3

plot4 <- ggplot(data, aes(x= L2FC_H23_ABE_Control_T0_1, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "L2FC_ABE_Control_T0_1") +
theme_classic()
plot4

plot5 <- ggplot(data, aes(x= L2FC_H23_CBE_Control_plasmid_1, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "L2FC_CBE_Control_plasmid_1") +
theme_classic()
plot5

plot6 <- ggplot(data, aes(x= L2FC_H23_ABE_Control_plasmid_1, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "L2FC_ABE_Control_plasmid_1") +
theme_classic()
plot6

plot7 <- ggplot(data, aes(x= L2FC_H23_CBE_Control_plasmid_2, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "L2FC_CBE_Control_plasmid_2") +
theme_classic()
plot7

plot8 <- ggplot(data, aes(x= L2FC_H23_ABE_Control_plasmid_2, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "L2FC_ABE_Control_plasmid_2") +
theme_classic()
plot8
```

```{r}
# quality control plots
# PC9
plot1 <- ggplot(data, aes(y= PC9_CBE_T0_exp1_reads)) + 
geom_histogram() +
labs(title= "PC9 CBE T0 reads")
plot1

plot2 <- ggplot(data, aes(y= PC9_ABE_T0_exp1_reads)) + 
geom_histogram() +
labs(title= "PC9 ABE T0 reads")
plot2

plot3 <- ggplot(data, aes(x= L2FC_PC9_CBE_Control_T0_1, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "L2FC_CBE_Control_T0_1") +
theme_classic()
plot3

plot4 <- ggplot(data, aes(x= L2FC_PC9_ABE_Control_T0_1, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "L2FC_ABE_Control_T0_1") +
theme_classic()
plot4

plot5 <- ggplot(data, aes(x= L2FC_PC9_CBE_Control_plasmid_1, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "L2FC_CBE_Control_plasmid_1") +
theme_classic()
plot5

plot6 <- ggplot(data, aes(x= L2FC_PC9_ABE_Control_plasmid_1, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "L2FC_ABE_Control_plasmid_1") +
theme_classic()
plot6

plot8 <- ggplot(data, aes(x= L2FC_PC9_CBE_Control_plasmid_2, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "L2FC_CBE_Control_plasmid_2") +
theme_classic()
plot8

plot9 <- ggplot(data, aes(x= L2FC_PC9_ABE_Control_plasmid_2, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "L2FC_ABE_Control_plasmid_2") +
theme_classic()
plot9
```

```{r}
# check for low read numbers and filter out <20 reads, changed from 100 on CBE only
data %>% filter(plasmid_reads <20)
data %>% filter(HT29_CBE_T0_exp1_reads <20)
data %>% filter(H23_CBE_T0_exp1_reads <20)
data %>% filter(PC9_CBE_T0_exp1_reads <20)

data %>% filter(HT29_ABE_T0_exp1_reads <20)
data %>% filter(H23_ABE_T0_exp1_reads <20)
data %>% filter(PC9_ABE_T0_exp1_reads <20)

data %>% filter(plasmid_reads <20 | HT29_CBE_T0_exp1_reads <20 | H23_CBE_T0_exp1_reads <20 | PC9_CBE_T0_exp1_reads <20 | HT29_ABE_T0_exp1_reads <20 | H23_ABE_T0_exp1_reads <20 | PC9_ABE_T0_exp1_reads <20)
data <- filter(data, plasmid_reads >20 & HT29_CBE_T0_exp1_reads >20 & H23_CBE_T0_exp1_reads >20 & PC9_CBE_T0_exp1_reads >20 & HT29_ABE_T0_exp1_reads >20 & H23_ABE_T0_exp1_reads >20 & PC9_ABE_T0_exp1_reads >20)
```

```{r}
#HT-29
# check L2FC PLASMID vs Gene
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

# CBE control vs plasmid
plot1 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_HT29_CBE_Control_plasmid_1)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "L2FC_CBE_plasmid_T0_1") +
theme_classic()
plot1

# CBE Tram vs plasmid
plot2 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_HT29_CBE_Tram_plasmid_1)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "L2FC_CBE_Tram_plasmid_1") +
theme_classic()
plot2

# CBE Pict vs plasmid
plot3 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_HT29_CBE_Pict_plasmid_1)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "L2FC_CBE_Pict_plasmid_1") +
theme_classic()
plot3

# CBE DebCet vs plasmid
plot4 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_HT29_CBE_DebCet_plasmid_1)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "L2FC_CBE_DebCet_plasmid_1") +
theme_classic()
plot4


# ABE plasmid vs T0
plot5 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_HT29_ABE_Control_plasmid_1)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "L2FC_ABE_plasmid_T0_1") +
theme_classic()
plot5

# ABE Tram vs plasmid
plot6 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_HT29_ABE_Tram_plasmid_1)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "L2FC_ABE_Tram_plasmid_1") +
theme_classic()
plot6

# ABE Pict vs plasmid
plot7 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_HT29_ABE_Pict_plasmid_1)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "L2FC_ABE_Pict_plasmid_1") +
theme_classic()
plot7

# ABE DebCet vs plasmid
plot8 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_HT29_ABE_DebCet_plasmid_1)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "L2FC_ABE_DebCet_plasmid_1") +
theme_classic()
plot8
```

```{r}
# replicate correlation - stats HT-29
r1 <- cor(data$L2FC_HT29_CBE_Control_plasmid_1, data$L2FC_HT29_CBE_Control_plasmid_2, method = "pearson")
r2 <- cor(data$L2FC_HT29_CBE_Tram_plasmid_1, data$L2FC_HT29_CBE_Tram_plasmid_2, method = "pearson")
r3 <- cor(data$L2FC_HT29_CBE_Pict_plasmid_1, data$L2FC_HT29_CBE_Pict_plasmid_2, method = "pearson")
r4 <- cor(data$L2FC_HT29_CBE_DebCet_plasmid_1, data$L2FC_HT29_CBE_DebCet_plasmid_2, method = "pearson")
r5 <- cor(data$L2FC_HT29_ABE_Control_plasmid_1, data$L2FC_HT29_ABE_Control_plasmid_2, method = "pearson")
r6 <- cor(data$L2FC_HT29_ABE_Tram_plasmid_1, data$L2FC_HT29_ABE_Tram_plasmid_2, method = "pearson")
r7 <- cor(data$L2FC_HT29_ABE_Pict_plasmid_1, data$L2FC_HT29_ABE_Pict_plasmid_2, method = "pearson")
r8 <- cor(data$L2FC_HT29_ABE_DebCet_plasmid_1, data$L2FC_HT29_ABE_DebCet_plasmid_2, method = "pearson")
```

```{r}
#HT-29 check correlation between replicates
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

# CBE control vs plasmid
plot1 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_HT29_CBE_Control_plasmid_1, y= L2FC_HT29_CBE_Control_plasmid_2)) + geom_jitter(size=3,  position = pos, colour = "grey35") +
labs(title= "HT-29 - replicate correlation", x = "L2FC CBE control vs plasmid exp.1", y = "L2FC CBE control vs plasmid exp.2") +
geom_smooth(method = lm) +
annotate("text", y = 5, x = 0, label = paste("R = ", round(r1, 2)), parse = FALSE, size = 5) +
theme_bw()
plot1
ggsave("FigS2_1.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

# CBE Tram vs plasmid
plot2 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_HT29_CBE_Tram_plasmid_1, y= L2FC_HT29_CBE_Tram_plasmid_2)) + geom_jitter(size=3,  position = pos, colour = "grey35") +
labs(title= "HT-29 - replicate correlation", x = "L2FC CBE Trametinib vs plasmid exp.1", y = "L2FC CBE Trametinib vs plasmid exp.2") +
geom_smooth(method = lm) +
annotate("text", y = 5, x = 0, label = paste("R = ", round(r2, 2)), parse = FALSE, size = 5) +
theme_bw()
plot2
ggsave("FigS2_2.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

# CBE Pict vs plasmid
plot3 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_HT29_CBE_Pict_plasmid_1, y= L2FC_HT29_CBE_Pict_plasmid_2)) + geom_jitter(size=3,  position = pos, colour = "grey35") +
labs(title= "HT-29 - replicate correlation", x = "L2FC CBE Pictilisib vs plasmid exp.1", y = "L2FC CBE Pictilisib vs plasmid exp.2") +
geom_smooth(method = lm) +
annotate("text", y = 5, x = 0, label = paste("R = ", round(r3, 2)), parse = FALSE, size = 5) +
theme_bw()
plot3
ggsave("FigS2_3.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

# CBE DebCet vs plasmid
plot4 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_HT29_CBE_DebCet_plasmid_1, y= L2FC_HT29_CBE_DebCet_plasmid_2)) + geom_jitter(size=3,  position = pos, colour = "grey35") +
labs(title= "HT-29 - replicate correlation", x = "L2FC CBE Dabrafenib+Cetuximab vs plasmid exp.1", y = "L2FC CBE Dabrafenib+Cetuximab vs plasmid exp.2") +
geom_smooth(method = lm) +
annotate("text", y = 5, x = 0, label = paste("R = ", round(r4, 2)), parse = FALSE, size = 5) +
theme_bw()
plot4
ggsave("FigS2_4.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

# ABE control vs plasmid
plot5 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_HT29_ABE_Control_plasmid_1, y= L2FC_HT29_ABE_Control_plasmid_2)) + geom_jitter(size=3,  position = pos, colour = "grey55") +
labs(title= "HT-29 - replicate correlation", x = "L2FC ABE control vs plasmid exp.1", y = "L2FC ABE control vs plasmid exp.2") +
geom_smooth(method = lm) +
annotate("text", y = 5, x = 0, label = paste("R = ", round(r5, 2)), parse = FALSE, size = 5) +
theme_bw()
plot5
ggsave("FigS2_5.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

# ABE Tram vs plasmid
plot6 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_HT29_ABE_Tram_plasmid_1, y= L2FC_HT29_ABE_Tram_plasmid_2)) + geom_jitter(size=3,  position = pos, colour = "grey55") +
labs(title= "HT-29 - replicate correlation", x = "L2FC ABE Trametinib vs plasmid exp.1", y = "L2FC ABE Trametinib vs plasmid exp.2") +
geom_smooth(method = lm) +
annotate("text", y = 5, x = 0, label = paste("R = ", round(r6, 2)), parse = FALSE, size = 5) +
theme_bw()
plot6
ggsave("FigS2_6.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

# ABE Pict vs plasmid
plot7 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_HT29_ABE_Pict_plasmid_1, y= L2FC_HT29_ABE_Pict_plasmid_2)) + geom_jitter(size=3,  position = pos, colour = "grey55") +
labs(title= "HT-29 - replicate correlation", x = "L2FC ABE Pictilisb vs plasmid exp.1", y = "L2FC ABE Pictilisb vs plasmid exp.2") +
geom_smooth(method = lm) +
annotate("text", y = 5, x = 0, label = paste("R = ", round(r7, 2)), parse = FALSE, size = 5) +
theme_bw()
plot7
ggsave("FigS2_7.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

# ABE DebCet vs plasmid
plot8 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_HT29_ABE_DebCet_plasmid_1, y= L2FC_HT29_ABE_DebCet_plasmid_2)) + geom_jitter(size=3,  position = pos, colour = "grey55") +
labs(title= "HT-29 - replicate correlation", x = "L2FC ABE Dabrafenib+Cetuximab vs plasmid exp.1", y = "L2FC ABE Dabrafenib+Cetuximab vs plasmid exp.2") +
geom_smooth(method = lm) +
annotate("text", y = 5, x = 0, label = paste("R = ", round(r8, 2)), parse = FALSE, size = 5) +
theme_bw()
plot8
ggsave("FigS2_8.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")
```

```{r}
#H23
# check L2FC PLASMID vs Gene
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

# CBE control vs plasmid
plot1 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_H23_CBE_Control_plasmid_1)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "L2FC_CBE_control_plasmid_1") +
theme_classic()
plot1

# CBE Sotor vs plasmid
plot2 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_H23_CBE_Sotor_plasmid_1)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "L2FC_CBE_Sotor_plasmid_1") +
theme_classic()
plot2

# CBE Adag vs plasmid
plot3 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_H23_CBE_Adag_plasmid_1)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "L2FC_CBE_Adag_plasmid_1") +
theme_classic()
plot3


# ABE plasmid vs T0
plot5 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_H23_ABE_Control_plasmid_1)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "L2FC_ABE_plasmid_T0_1") +
theme_classic()
plot5

# ABE Sotor vs plasmid
plot6 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_H23_ABE_Sotor_plasmid_1)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "L2FC_ABE_Sotor_plasmid_1") +
theme_classic()
plot6

# ABE Adag vs plasmid
plot7 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_H23_ABE_Adag_plasmid_1)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "L2FC_ABE_Adag_plasmid_1") +
scale_color_continuous() +
theme_classic()
plot7
```
```{r}
# replicate correlation - stats H23
r9 <- cor(data$L2FC_H23_CBE_Control_plasmid_1, data$L2FC_H23_CBE_Control_plasmid_2, method = "pearson")
r10 <- cor(data$L2FC_H23_CBE_Sotor_plasmid_1, data$L2FC_H23_CBE_Sotor_plasmid_2, method = "pearson")
r11 <- cor(data$L2FC_H23_CBE_Adag_plasmid_1, data$L2FC_H23_CBE_Adag_plasmid_2, method = "pearson")
r12 <- cor(data$L2FC_H23_ABE_Control_plasmid_1, data$L2FC_H23_ABE_Control_plasmid_2, method = "pearson")
r13 <- cor(data$L2FC_H23_ABE_Sotor_plasmid_1, data$L2FC_H23_ABE_Sotor_plasmid_2, method = "pearson")
r14 <- cor(data$L2FC_H23_ABE_Adag_plasmid_1, data$L2FC_H23_ABE_Adag_plasmid_2, method = "pearson")
```

```{r}
#H23 check correlation between replicates
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

# CBE control vs plasmid
plot1 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_H23_CBE_Control_plasmid_1, y= L2FC_H23_CBE_Control_plasmid_2)) + geom_jitter(size=3,  position = pos, colour = "grey35") +
labs(title= "H23 - replicate correlation", x = "L2FC CBE control vs plasmid exp.1", y = "L2FC CBE control vs plasmid exp.2") +
geom_smooth(method = lm) +
annotate("text", y = 5, x = 0, label = paste("R = ", round(r9, 2)), parse = FALSE, size = 5) +
theme_bw()
plot1
ggsave("FigS2_9.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

# CBE Sotor vs plasmid
plot2 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_H23_CBE_Sotor_plasmid_1, y= L2FC_H23_CBE_Sotor_plasmid_2)) + geom_jitter(size=3,  position = pos, colour = "grey35") +
labs(title= "H23 - replicate correlation", x = "L2FC CBE Sotorasib vs plasmid exp.1", y = "L2FC CBE Sotorasib vs plasmid exp.2") +
geom_smooth(method = lm) +
annotate("text", y = 5, x = 0, label = paste("R = ", round(r10, 2)), parse = FALSE, size = 5) +
theme_bw()
plot2
ggsave("FigS2_10.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

# CBE Adag vs plasmid
plot3 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_H23_CBE_Adag_plasmid_1, y= L2FC_H23_CBE_Adag_plasmid_2)) + geom_jitter(size=3,  position = pos, colour = "grey35") +
labs(title= "H23 - replicate correlation", x = "L2FC CBE Adagrasib vs plasmid exp.1", y = "L2FC CBE Adagrasib vs plasmid exp.2") +
geom_smooth(method = lm) +
annotate("text", y = 5, x = 0, label = paste("R = ", round(r11, 2)), parse = FALSE, size = 5) +
theme_bw()
plot3
ggsave("FigS2_11.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")


# ABE control vs plasmid
plot4 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_H23_ABE_Control_plasmid_1, y= L2FC_H23_ABE_Control_plasmid_2)) + geom_jitter(size=3,  position = pos, colour = "grey55") +
labs(title= "H23 - replicate correlation", x = "L2FC ABE control vs plasmid exp.1", y = "L2FC ABE control vs plasmid exp.2") +
geom_smooth(method = lm) +
annotate("text", y = 5, x = 0, label = paste("R = ", round(r12, 2)), parse = FALSE, size = 5) +
theme_bw()
plot4
ggsave("FigS2_12.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

# ABE Sotor vs plasmid
plot5 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_H23_ABE_Sotor_plasmid_1, y= L2FC_H23_ABE_Sotor_plasmid_2)) + geom_jitter(size=3,  position = pos, colour = "grey55") +
labs(title= "H23 - replicate correlation", x = "L2FC ABE Sotorasib vs plasmid exp.1", y = "L2FC ABE Sotorasib vs plasmid exp.2") +
geom_smooth(method = lm) +
annotate("text", y = 5, x = 0, label = paste("R = ", round(r13, 2)), parse = FALSE, size = 5) +
theme_bw()
plot5
ggsave("FigS2_13.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

# ABE Adag vs plasmid
plot6 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_H23_ABE_Adag_plasmid_1, y= L2FC_H23_ABE_Adag_plasmid_2)) + geom_jitter(size=3,  position = pos, colour = "grey55") +
labs(title= "H23 - replicate correlation", x = "L2FC ABE Adagrasib vs plasmid exp.1", y = "L2FC ABE Adagrasib vs plasmid exp.2") +
geom_smooth(method = lm) +
annotate("text", y = 5, x = 0, label = paste("R = ", round(r14, 2)), parse = FALSE, size = 5) +
theme_bw()
plot6
ggsave("FigS2_14.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")
```

```{r}
#PC9
# check L2FC PLASMID vs Gene
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

# CBE control vs plasmid
plot1 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_PC9_CBE_Control_plasmid_1)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "L2FC_CBE_plasmid_T0_1") +
theme_classic()
plot1

# CBE Gefit vs plasmid
plot2 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_PC9_CBE_Gefit_plasmid_1)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "L2FC_CBE_Gefit_plasmid_1") +
theme_classic()
plot2

# CBE Adag vs plasmid
plot3 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_PC9_CBE_Osim_plasmid_1)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "L2FC_CBE_Osim_plasmid_1") +
theme_classic()
plot3


# ABE plasmid vs T0
plot5 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_PC9_ABE_Control_plasmid_1)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "L2FC_ABE_plasmid_T0_1") +
theme_classic()
plot5

# ABE Gefit vs plasmid
plot6 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_PC9_ABE_Gefit_plasmid_1)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "L2FC_ABE_Gefit_plasmid_1") +
theme_classic()
plot6

# ABE Osim vs plasmid
plot7 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_PC9_ABE_Osim_plasmid_1)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "L2FC_ABE_Osim_plasmid_1") +
theme_classic()
plot7
```
```{r}
# replicate correlation - stats PC9
r15 <- cor(data$L2FC_PC9_CBE_Control_plasmid_1, data$L2FC_PC9_CBE_Control_plasmid_2, method = "pearson")
r16 <- cor(data$L2FC_PC9_CBE_Gefit_plasmid_1, data$L2FC_PC9_CBE_Gefit_plasmid_2, method = "pearson")
r17 <- cor(data$L2FC_PC9_CBE_Osim_plasmid_1, data$L2FC_PC9_CBE_Osim_plasmid_2, method = "pearson")
r18 <- cor(data$L2FC_PC9_ABE_Control_plasmid_1, data$L2FC_PC9_ABE_Control_plasmid_2, method = "pearson")
r19 <- cor(data$L2FC_PC9_ABE_Gefit_plasmid_1, data$L2FC_PC9_ABE_Gefit_plasmid_2, method = "pearson")
r20 <- cor(data$L2FC_PC9_ABE_Osim_plasmid_1, data$L2FC_PC9_ABE_Osim_plasmid_2, method = "pearson")
```

```{r}
#PC9 check correlation between replicates
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

# CBE control vs plasmid
plot1 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_PC9_CBE_Control_plasmid_1, y= L2FC_PC9_CBE_Control_plasmid_2)) + geom_jitter(size=3,  position = pos, colour = "grey35") +
labs(title= "PC9 - replicate correlation", x = "L2FC CBE control vs plasmid exp.1", y = "L2FC CBE control vs plasmid exp.2") +
geom_smooth(method = lm) +
annotate("text", y = 5, x = 0, label = paste("R = ", round(r15, 2)), parse = FALSE, size = 5) +
theme_bw()
plot1
ggsave("FigS2_15.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

# CBE Gefit vs plasmid
plot2 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_PC9_CBE_Gefit_plasmid_1, y= L2FC_PC9_CBE_Gefit_plasmid_2)) + geom_jitter(size=3,  position = pos, colour = "grey35") +
labs(title= "PC9 - replicate correlation", x = "L2FC CBE Gefitinib vs plasmid exp.1", y = "L2FC CBE Gefitinib vs plasmid exp.2") +
geom_smooth(method = lm) +
annotate("text", y = 5, x = 0, label = paste("R = ", round(r16, 2)), parse = FALSE, size = 5) +
theme_bw()
plot2
ggsave("FigS2_16.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

# CBE Osim vs plasmid
plot3 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_PC9_CBE_Osim_plasmid_1, y= L2FC_PC9_CBE_Osim_plasmid_2)) + geom_jitter(size=3,  position = pos, colour = "grey35") +
labs(title= "PC9 - replicate correlation", x = "L2FC CBE Osimertinib vs plasmid exp.1", y = "L2FC CBE Osimertinib vs plasmid exp.2") +
geom_smooth(method = lm) +
annotate("text", y = 5, x = 0, label = paste("R = ", round(r17, 2)), parse = FALSE, size = 5) +
theme_bw()
plot3
ggsave("FigS2_17.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

# ABE control vs plasmid
plot4 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_PC9_ABE_Control_plasmid_1, y= L2FC_PC9_ABE_Control_plasmid_2)) + geom_jitter(size=3,  position = pos, colour = "grey55") +
labs(title= "PC9 - replicate correlation", x = "L2FC ABE control vs plasmid exp.1", y = "L2FC ABE control vs plasmid exp.2") +
geom_smooth(method = lm) +
annotate("text", y = 5, x = 0, label = paste("R = ", round(r18, 2)), parse = FALSE, size = 5) +
theme_bw()
plot4
ggsave("FigS2_18.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

# ABE Gefit vs plasmid
plot5 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_PC9_ABE_Gefit_plasmid_1, y= L2FC_PC9_ABE_Gefit_plasmid_2)) + geom_jitter(size=3,  position = pos, colour = "grey55") +
labs(title= "PC9 - replicate correlation", x = "L2FC ABE Gefitinib vs plasmid exp.1", y = "L2FC ABE Gefitinib vs plasmid exp.2") +
geom_smooth(method = lm) +
annotate("text", y = 5, x = 0, label = paste("R = ", round(r19, 2)), parse = FALSE, size = 5) +
theme_bw()
plot5
ggsave("FigS2_19.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

# ABE Osim vs plasmid
plot6 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_PC9_ABE_Osim_plasmid_1, y= L2FC_PC9_ABE_Osim_plasmid_2)) + geom_jitter(size=3,  position = pos, colour = "grey55") +
labs(title= "PC9 - replicate correlation", x = "L2FC ABE Osimertinib vs plasmid exp.1", y = "L2FC ABE Osimertinib vs plasmid exp.2") +
geom_smooth(method = lm) +
annotate("text", y = 5, x = 0, label = paste("R = ", round(r20, 2)), parse = FALSE, size = 5) +
theme_bw()
plot6
ggsave("FigS2_20.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")
```

```{r}
#BEstimate annotations for newest Bestimate tool with 3-9 editing window for CBE and ABE

#read in BEstimate sumamry dfs
AKT1_CBE <- as_tibble(read.csv("BEstimate_protein_050423/CBE_NGN_AKT1_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
MYC_CBE <- as_tibble(read.csv("BEstimate_protein_050423/CBE_NGN_MYC_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
PIK3CA_CBE <- as_tibble(read.csv("BEstimate_protein_050423/CBE_NGN_PIK3CA_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
KRAS_CBE <- as_tibble(read.csv("BEstimate_protein_050423/CBE_NGN_KRAS_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
BCL2_CBE <- as_tibble(read.csv("BEstimate_protein_050423/CBE_NGN_BCL2_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
BRAF_CBE <- as_tibble(read.csv("BEstimate_protein_050423/CBE_NGN_BRAF_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
PARP1_CBE <- as_tibble(read.csv("BEstimate_protein_050423/CBE_NGN_PARP1_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
PARP2_CBE <- as_tibble(read.csv("BEstimate_protein_050423/CBE_NGN_PARP2_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
EGFR_CBE <- as_tibble(read.csv("BEstimate_protein_050423/CBE_NGN_EGFR_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
MAP2K1_CBE <- as_tibble(read.csv("BEstimate_protein_050423/CBE_NGN_MAP2K1_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
MAP2K2_CBE <- as_tibble(read.csv("BEstimate_protein_050423/CBE_NGN_MAP2K2_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))

AKT1_ABE <- as_tibble(read.csv("BEstimate_protein_050423/ABE_NGN_AKT1_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
MYC_ABE <- as_tibble(read.csv("BEstimate_protein_050423/ABE_NGN_MYC_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
PIK3CA_ABE <- as_tibble(read.csv("BEstimate_protein_050423/ABE_NGN_PIK3CA_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
KRAS_ABE <- as_tibble(read.csv("BEstimate_protein_050423/ABE_NGN_KRAS_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
BCL2_ABE <- as_tibble(read.csv("BEstimate_protein_050423/ABE_NGN_BCL2_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
BRAF_ABE <- as_tibble(read.csv("BEstimate_protein_050423/ABE_NGN_BRAF_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
PARP1_ABE <- as_tibble(read.csv("BEstimate_protein_050423/ABE_NGN_PARP1_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
PARP2_ABE <- as_tibble(read.csv("BEstimate_protein_050423/ABE_NGN_PARP2_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
EGFR_ABE <- as_tibble(read.csv("BEstimate_protein_050423/ABE_NGN_EGFR_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
MAP2K1_ABE <- as_tibble(read.csv("BEstimate_protein_050423/ABE_NGN_MAP2K1_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
MAP2K2_ABE <- as_tibble(read.csv("BEstimate_protein_050423/ABE_NGN_MAP2K2_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))

#bind rows of all BEsimate dfs
CBE_BEstimate <- bind_rows(AKT1_CBE, MYC_CBE, BCL2_CBE, BRAF_CBE, MAP2K1_CBE, MAP2K2_CBE, PARP1_CBE, PARP2_CBE, PIK3CA_CBE, EGFR_CBE, KRAS_CBE)
ABE_BEstimate <- bind_rows(AKT1_ABE, MYC_ABE, BCL2_ABE, BRAF_ABE, MAP2K1_ABE, MAP2K2_ABE, PARP1_ABE, PARP2_ABE, PIK3CA_ABE, EGFR_ABE, KRAS_ABE)

# remove data columns that mess up the left join
data <- data %>% select(!c(Transcript_ID, Exon_ID, Direction))

#rename guide and Gene columns
CBE_BEstimate<- rename(CBE_BEstimate, Gene = Hugo_Symbol)
CBE_BEstimate<- rename(CBE_BEstimate, Amino_Acid_Position = Protein_Position)
CBE_BEstimate<- rename(CBE_BEstimate, guide = gRNA_Target_Sequence)

ABE_BEstimate<- rename(ABE_BEstimate, Gene = Hugo_Symbol)
ABE_BEstimate<- rename(ABE_BEstimate, Amino_Acid_Position = Protein_Position)
ABE_BEstimate<- rename(ABE_BEstimate, guide = gRNA_Target_Sequence)

#left join the library and read counts with the BEstimate predictions
data_CBE <- left_join(data, CBE_BEstimate)
data_CBE <- mutate(data_CBE, editor = "CBE")

data_ABE <- left_join(data, ABE_BEstimate)
data_ABE <- mutate(data_ABE, editor = "ABE")

annotated_data <- bind_rows(data_CBE, data_ABE)
annotated_data
```

```{r}
# consolidate gRNA scores for gRNAs based on editor

annotated_data <- mutate(annotated_data, L2FC_HT29_DebCet_plasmid_1 = as.numeric(ifelse(editor == "CBE", L2FC_HT29_CBE_DebCet_plasmid_1, ifelse(editor == "ABE", L2FC_HT29_ABE_DebCet_plasmid_1, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_HT29_Tram_plasmid_1 = as.numeric(ifelse(editor == "CBE", L2FC_HT29_CBE_Tram_plasmid_1, ifelse(editor == "ABE", L2FC_HT29_ABE_Tram_plasmid_1, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_HT29_Pict_plasmid_1 = as.numeric(ifelse(editor == "CBE", L2FC_HT29_CBE_Pict_plasmid_1, ifelse(editor == "ABE", L2FC_HT29_ABE_Pict_plasmid_1, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_HT29_Control_plasmid_1 = as.numeric(ifelse(editor == "CBE", L2FC_HT29_CBE_Control_plasmid_1, ifelse(editor == "ABE", L2FC_HT29_ABE_Control_plasmid_1, "ERROR"))))

annotated_data <- mutate(annotated_data, L2FC_HT29_DebCet_plasmid_2 = as.numeric(ifelse(editor == "CBE", L2FC_HT29_CBE_DebCet_plasmid_2, ifelse(editor == "ABE", L2FC_HT29_ABE_DebCet_plasmid_2, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_HT29_Tram_plasmid_2 = as.numeric(ifelse(editor == "CBE", L2FC_HT29_CBE_Tram_plasmid_2, ifelse(editor == "ABE", L2FC_HT29_ABE_Tram_plasmid_2, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_HT29_Pict_plasmid_2 = as.numeric(ifelse(editor == "CBE", L2FC_HT29_CBE_Pict_plasmid_2, ifelse(editor == "ABE", L2FC_HT29_ABE_Pict_plasmid_2, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_HT29_Control_plasmid_2 = as.numeric(ifelse(editor == "CBE", L2FC_HT29_CBE_Control_plasmid_2, ifelse(editor == "ABE", L2FC_HT29_ABE_Control_plasmid_2, "ERROR"))))

annotated_data <- mutate(annotated_data, L2FC_HT29_DebCet_plasmid_average = as.numeric(ifelse(editor == "CBE", L2FC_HT29_CBE_DebCet_plasmid_average, ifelse(editor == "ABE", L2FC_HT29_ABE_DebCet_plasmid_average, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_HT29_Tram_plasmid_average = as.numeric(ifelse(editor == "CBE", L2FC_HT29_CBE_Tram_plasmid_average, ifelse(editor == "ABE", L2FC_HT29_ABE_Tram_plasmid_average, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_HT29_Pict_plasmid_average = as.numeric(ifelse(editor == "CBE", L2FC_HT29_CBE_Pict_plasmid_average, ifelse(editor == "ABE", L2FC_HT29_ABE_Pict_plasmid_average, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_HT29_Control_plasmid_average = as.numeric(ifelse(editor == "CBE", L2FC_HT29_CBE_Control_plasmid_average, ifelse(editor == "ABE", L2FC_HT29_ABE_Control_plasmid_average, "ERROR"))))

annotated_data <- mutate(annotated_data, L2FC_H23_Adag_plasmid_1 = as.numeric(ifelse(editor == "CBE", L2FC_H23_CBE_Adag_plasmid_1, ifelse(editor == "ABE", L2FC_H23_ABE_Adag_plasmid_1, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_H23_Sotor_plasmid_1 = as.numeric(ifelse(editor == "CBE", L2FC_H23_CBE_Sotor_plasmid_1, ifelse(editor == "ABE", L2FC_H23_ABE_Sotor_plasmid_1, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_H23_Control_plasmid_1 = as.numeric(ifelse(editor == "CBE", L2FC_H23_CBE_Control_plasmid_1, ifelse(editor == "ABE", L2FC_H23_ABE_Control_plasmid_1, "ERROR"))))

annotated_data <- mutate(annotated_data, L2FC_H23_Adag_plasmid_2 = as.numeric(ifelse(editor == "CBE", L2FC_H23_CBE_Adag_plasmid_2, ifelse(editor == "ABE", L2FC_H23_ABE_Adag_plasmid_2, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_H23_Sotor_plasmid_2 = as.numeric(ifelse(editor == "CBE", L2FC_H23_CBE_Sotor_plasmid_2, ifelse(editor == "ABE", L2FC_H23_ABE_Sotor_plasmid_2, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_H23_Control_plasmid_2 = as.numeric(ifelse(editor == "CBE", L2FC_H23_CBE_Control_plasmid_2, ifelse(editor == "ABE", L2FC_H23_ABE_Control_plasmid_2, "ERROR"))))

annotated_data <- mutate(annotated_data, L2FC_H23_Adag_plasmid_average = as.numeric(ifelse(editor == "CBE", L2FC_H23_CBE_Adag_plasmid_average, ifelse(editor == "ABE", L2FC_H23_ABE_Adag_plasmid_average, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_H23_Sotor_plasmid_average = as.numeric(ifelse(editor == "CBE", L2FC_H23_CBE_Sotor_plasmid_average, ifelse(editor == "ABE", L2FC_H23_ABE_Sotor_plasmid_average, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_H23_Control_plasmid_average = as.numeric(ifelse(editor == "CBE", L2FC_H23_CBE_Control_plasmid_average, ifelse(editor == "ABE", L2FC_H23_ABE_Control_plasmid_average, "ERROR"))))

annotated_data <- mutate(annotated_data, L2FC_PC9_Control_plasmid_1 = as.numeric(ifelse(editor == "CBE", L2FC_PC9_CBE_Control_plasmid_1, ifelse(editor == "ABE", L2FC_PC9_ABE_Control_plasmid_1, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_PC9_Osim_plasmid_1 = as.numeric(ifelse(editor == "CBE", L2FC_PC9_CBE_Osim_plasmid_1, ifelse(editor == "ABE", L2FC_PC9_ABE_Osim_plasmid_1, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_PC9_Gefit_plasmid_1 = as.numeric(ifelse(editor == "CBE", L2FC_PC9_CBE_Gefit_plasmid_1, ifelse(editor == "ABE", L2FC_PC9_ABE_Gefit_plasmid_1, "ERROR"))))

annotated_data <- mutate(annotated_data, L2FC_PC9_Control_plasmid_2 = as.numeric(ifelse(editor == "CBE", L2FC_PC9_CBE_Control_plasmid_2, ifelse(editor == "ABE", L2FC_PC9_ABE_Control_plasmid_2, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_PC9_Osim_plasmid_2 = as.numeric(ifelse(editor == "CBE", L2FC_PC9_CBE_Osim_plasmid_2, ifelse(editor == "ABE", L2FC_PC9_ABE_Osim_plasmid_2, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_PC9_Gefit_plasmid_2 = as.numeric(ifelse(editor == "CBE", L2FC_PC9_CBE_Gefit_plasmid_2, ifelse(editor == "ABE", L2FC_PC9_ABE_Gefit_plasmid_2, "ERROR"))))

annotated_data <- mutate(annotated_data, L2FC_PC9_Control_plasmid_average = as.numeric(ifelse(editor == "CBE", L2FC_PC9_CBE_Control_plasmid_average, ifelse(editor == "ABE", L2FC_PC9_ABE_Control_plasmid_average, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_PC9_Osim_plasmid_average = as.numeric(ifelse(editor == "CBE", L2FC_PC9_CBE_Osim_plasmid_average, ifelse(editor == "ABE", L2FC_PC9_ABE_Osim_plasmid_average, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_PC9_Gefit_plasmid_average = as.numeric(ifelse(editor == "CBE", L2FC_PC9_CBE_Gefit_plasmid_average, ifelse(editor == "ABE", L2FC_PC9_ABE_Gefit_plasmid_average, "ERROR"))))

#remove all the old score fields based on CBE or ABE
annotated_data <- annotated_data %>%
  select(!c(L2FC_HT29_CBE_DebCet_plasmid_1, L2FC_HT29_CBE_Tram_plasmid_1, L2FC_HT29_CBE_Control_plasmid_1, L2FC_HT29_CBE_Pict_plasmid_1, L2FC_HT29_ABE_DebCet_plasmid_1, L2FC_HT29_ABE_Tram_plasmid_1, L2FC_HT29_ABE_Control_plasmid_1, L2FC_HT29_ABE_Pict_plasmid_1, L2FC_HT29_CBE_DebCet_plasmid_2, L2FC_HT29_CBE_Tram_plasmid_2, L2FC_HT29_CBE_Control_plasmid_2, L2FC_HT29_CBE_Pict_plasmid_2, L2FC_HT29_ABE_DebCet_plasmid_2, L2FC_HT29_ABE_Tram_plasmid_2, L2FC_HT29_ABE_Control_plasmid_2, L2FC_HT29_ABE_Pict_plasmid_2, L2FC_HT29_CBE_DebCet_plasmid_average, L2FC_HT29_CBE_Tram_plasmid_average, L2FC_HT29_CBE_Control_plasmid_average, L2FC_HT29_CBE_Pict_plasmid_average, L2FC_HT29_ABE_DebCet_plasmid_average, L2FC_HT29_ABE_Tram_plasmid_average, L2FC_HT29_ABE_Control_plasmid_average, L2FC_HT29_ABE_Pict_plasmid_average, L2FC_H23_CBE_Adag_plasmid_1, L2FC_H23_CBE_Sotor_plasmid_1, L2FC_H23_CBE_Control_plasmid_1, L2FC_H23_ABE_Adag_plasmid_1, L2FC_H23_ABE_Sotor_plasmid_1, L2FC_H23_ABE_Control_plasmid_1, L2FC_H23_CBE_Adag_plasmid_2, L2FC_H23_CBE_Sotor_plasmid_2, L2FC_H23_CBE_Control_plasmid_2, L2FC_H23_ABE_Adag_plasmid_2, L2FC_H23_ABE_Sotor_plasmid_2, L2FC_H23_ABE_Control_plasmid_2, L2FC_H23_CBE_Adag_plasmid_average, L2FC_H23_CBE_Sotor_plasmid_average, L2FC_H23_CBE_Control_plasmid_average, L2FC_H23_ABE_Adag_plasmid_average, L2FC_H23_ABE_Sotor_plasmid_average, L2FC_H23_ABE_Control_plasmid_average, L2FC_PC9_CBE_Control_plasmid_1, L2FC_PC9_CBE_Gefit_plasmid_1, L2FC_PC9_CBE_Osim_plasmid_1, L2FC_PC9_ABE_Control_plasmid_1, L2FC_PC9_ABE_Gefit_plasmid_1, L2FC_PC9_ABE_Osim_plasmid_1, L2FC_PC9_CBE_Control_plasmid_2, L2FC_PC9_CBE_Gefit_plasmid_2, L2FC_PC9_CBE_Osim_plasmid_2, L2FC_PC9_ABE_Control_plasmid_2, L2FC_PC9_ABE_Gefit_plasmid_2, L2FC_PC9_ABE_Osim_plasmid_2, L2FC_PC9_CBE_Control_plasmid_average, L2FC_PC9_CBE_Gefit_plasmid_average, L2FC_PC9_CBE_Osim_plasmid_average, L2FC_PC9_ABE_Control_plasmid_average, L2FC_PC9_ABE_Gefit_plasmid_average, L2FC_PC9_ABE_Osim_plasmid_average)) %>%
  distinct()
annotated_data
```

```{r} 
# get annotated_data to classify all non-essential and essential genes as one by making gene_alias
annotated_data %>% count(sgRNA_type)
annotated_data <- mutate(annotated_data, gene_alias = ifelse(sgRNA_type == "non_targeting", "non-targeting", ifelse(sgRNA_type == "splice_essential", "essential", ifelse(sgRNA_type == "splice_nonessential", "non-essential", ifelse(sgRNA_type == "intergenic", "intergenic", Gene)))))
```

```{r}
# Make a simplified amino acid position such that we can plot on a numerical scale
annotated_data <- mutate(annotated_data, Amino_Acid_Position_simple = Amino_Acid_Position)
annotated_data <- mutate(annotated_data, Amino_Acid_Position_simple = as.numeric(str_extract(Amino_Acid_Position_simple, "[[:digit:]]+")))
annotated_data %>% select(Amino_Acid_Position, Amino_Acid_Position_simple)
```

```{r}
# read in z-scores from DrugZ

#HT-29
#CBE
HT29_zscore_CBE_control_plasmid <- as_tibble(read.delim("drugZ_zscores_110423/HT29_REP1_REP2_AVERAGE/Control_CBE_vs_plasmid_foldchange", stringsAsFactors = FALSE, header = TRUE, sep ="\t"))
HT29_zscore_CBE_control_plasmid <- HT29_zscore_CBE_control_plasmid %>% rename(c(sgRNA_ID = X, Gene = GENE, HT29_zscore_control_plasmid = zscore)) %>% select(Gene, sgRNA_ID, HT29_zscore_control_plasmid) %>% mutate(editor = "CBE")

HT29_zscore_CBE_Tram_plasmid <- as_tibble(read.delim("drugZ_zscores_110423/HT29_REP1_REP2_AVERAGE/Tram_CBE_vs_plasmid_foldchange", stringsAsFactors = FALSE, header = TRUE, sep ="\t"))
HT29_zscore_CBE_Tram_plasmid <- HT29_zscore_CBE_Tram_plasmid %>% rename(c(sgRNA_ID = X, Gene = GENE, HT29_zscore_Tram_plasmid = zscore)) %>% select(Gene, sgRNA_ID, HT29_zscore_Tram_plasmid) %>% mutate(editor = "CBE")

HT29_zscore_CBE_Pict_plasmid <- as_tibble(read.delim("drugZ_zscores_110423/HT29_REP1_REP2_AVERAGE/Pict_CBE_vs_plasmid_foldchange", stringsAsFactors = FALSE, header = TRUE, sep ="\t"))
HT29_zscore_CBE_Pict_plasmid <- HT29_zscore_CBE_Pict_plasmid %>% rename(c(sgRNA_ID = X, Gene = GENE, HT29_zscore_Pict_plasmid = zscore)) %>% select(Gene, sgRNA_ID, HT29_zscore_Pict_plasmid) %>% mutate(editor = "CBE")

HT29_zscore_CBE_DebCet_plasmid <- as_tibble(read.delim("drugZ_zscores_110423/HT29_REP1_REP2_AVERAGE/DebCet_CBE_vs_plasmid_foldchange", stringsAsFactors = FALSE, header = TRUE, sep ="\t"))
HT29_zscore_CBE_DebCet_plasmid <- HT29_zscore_CBE_DebCet_plasmid %>% rename(c(sgRNA_ID = X, Gene = GENE, HT29_zscore_DebCet_plasmid = zscore)) %>% select(Gene, sgRNA_ID, HT29_zscore_DebCet_plasmid) %>% mutate(editor = "CBE")

#ABE
HT29_zscore_ABE_control_plasmid <- as_tibble(read.delim("drugZ_zscores_110423/HT29_REP1_REP2_AVERAGE/Control_ABE_vs_plasmid_foldchange", stringsAsFactors = FALSE, header = TRUE, sep ="\t"))
HT29_zscore_ABE_control_plasmid <- HT29_zscore_ABE_control_plasmid %>% rename(c(sgRNA_ID = X, Gene = GENE, HT29_zscore_control_plasmid = zscore)) %>% select(Gene, sgRNA_ID, HT29_zscore_control_plasmid) %>% mutate(editor = "ABE")

HT29_zscore_ABE_Tram_plasmid <- as_tibble(read.delim("drugZ_zscores_110423/HT29_REP1_REP2_AVERAGE/Tram_ABE_vs_plasmid_foldchange", stringsAsFactors = FALSE, header = TRUE, sep ="\t"))
HT29_zscore_ABE_Tram_plasmid <- HT29_zscore_ABE_Tram_plasmid %>% rename(c(sgRNA_ID = X, Gene = GENE, HT29_zscore_Tram_plasmid = zscore)) %>% select(Gene, sgRNA_ID, HT29_zscore_Tram_plasmid) %>% mutate(editor = "ABE")

HT29_zscore_ABE_Pict_plasmid <- as_tibble(read.delim("drugZ_zscores_110423/HT29_REP1_REP2_AVERAGE/Pict_ABE_vs_plasmid_foldchange", stringsAsFactors = FALSE, header = TRUE, sep ="\t"))
HT29_zscore_ABE_Pict_plasmid <- HT29_zscore_ABE_Pict_plasmid %>% rename(c(sgRNA_ID = X, Gene = GENE, HT29_zscore_Pict_plasmid = zscore)) %>% select(Gene, sgRNA_ID, HT29_zscore_Pict_plasmid) %>% mutate(editor = "ABE")

HT29_zscore_ABE_DebCet_plasmid <- as_tibble(read.delim("drugZ_zscores_110423/HT29_REP1_REP2_AVERAGE/DebCet_ABE_vs_plasmid_foldchange", stringsAsFactors = FALSE, header = TRUE, sep ="\t"))
HT29_zscore_ABE_DebCet_plasmid <- HT29_zscore_ABE_DebCet_plasmid %>% rename(c(sgRNA_ID = X, Gene = GENE, HT29_zscore_DebCet_plasmid = zscore)) %>% select(Gene, sgRNA_ID, HT29_zscore_DebCet_plasmid) %>% mutate(editor = "ABE")

#H23
#CBE
H23_zscore_CBE_control_plasmid <- as_tibble(read.delim("drugZ_zscores_110423/H23_REP1_REP2_AVERAGE/Control_CBE_vs_plasmid_foldchange", stringsAsFactors = FALSE, header = TRUE, sep ="\t"))
H23_zscore_CBE_control_plasmid <- H23_zscore_CBE_control_plasmid %>% rename(c(sgRNA_ID = X, Gene = GENE, H23_zscore_control_plasmid = zscore)) %>% select(Gene, sgRNA_ID, H23_zscore_control_plasmid) %>% mutate(editor = "CBE")

H23_zscore_CBE_Adag_plasmid <- as_tibble(read.delim("drugZ_zscores_110423/H23_REP1_REP2_AVERAGE/Adag_CBE_vs_plasmid_foldchange", stringsAsFactors = FALSE, header = TRUE, sep ="\t"))
H23_zscore_CBE_Adag_plasmid <- H23_zscore_CBE_Adag_plasmid %>% rename(c(sgRNA_ID = X, Gene = GENE, H23_zscore_Adag_plasmid = zscore)) %>% select(Gene, sgRNA_ID, H23_zscore_Adag_plasmid) %>% mutate(editor = "CBE")

H23_zscore_CBE_Sotor_plasmid <- as_tibble(read.delim("drugZ_zscores_110423/H23_REP1_REP2_AVERAGE/Sotor_CBE_vs_plasmid_foldchange", stringsAsFactors = FALSE, header = TRUE, sep ="\t"))
H23_zscore_CBE_Sotor_plasmid <- H23_zscore_CBE_Sotor_plasmid %>% rename(c(sgRNA_ID = X, Gene = GENE, H23_zscore_Sotor_plasmid = zscore)) %>% select(Gene, sgRNA_ID, H23_zscore_Sotor_plasmid) %>% mutate(editor = "CBE")

#ABE
H23_zscore_ABE_control_plasmid <- as_tibble(read.delim("drugZ_zscores_110423/H23_REP1_REP2_AVERAGE/Control_ABE_vs_plasmid_foldchange", stringsAsFactors = FALSE, header = TRUE, sep ="\t"))
H23_zscore_ABE_control_plasmid <- H23_zscore_ABE_control_plasmid %>% rename(c(sgRNA_ID = X, Gene = GENE, H23_zscore_control_plasmid = zscore)) %>% select(Gene, sgRNA_ID, H23_zscore_control_plasmid) %>% mutate(editor = "ABE")

H23_zscore_ABE_Adag_plasmid <- as_tibble(read.delim("drugZ_zscores_110423/H23_REP1_REP2_AVERAGE/Adag_ABE_vs_plasmid_foldchange", stringsAsFactors = FALSE, header = TRUE, sep ="\t"))
H23_zscore_ABE_Adag_plasmid <- H23_zscore_ABE_Adag_plasmid %>% rename(c(sgRNA_ID = X, Gene = GENE, H23_zscore_Adag_plasmid = zscore)) %>% select(Gene, sgRNA_ID, H23_zscore_Adag_plasmid) %>% mutate(editor = "ABE")

H23_zscore_ABE_Sotor_plasmid <- as_tibble(read.delim("drugZ_zscores_110423/H23_REP1_REP2_AVERAGE/Sotor_ABE_vs_plasmid_foldchange", stringsAsFactors = FALSE, header = TRUE, sep ="\t"))
H23_zscore_ABE_Sotor_plasmid <- H23_zscore_ABE_Sotor_plasmid %>% rename(c(sgRNA_ID = X, Gene = GENE, H23_zscore_Sotor_plasmid = zscore)) %>% select(Gene, sgRNA_ID, H23_zscore_Sotor_plasmid) %>% mutate(editor = "ABE")

#PC9
#CBE
PC9_zscore_CBE_control_plasmid <- as_tibble(read.delim("drugZ_zscores_110423/PC9_REP1_REP2_AVERAGE/Control_CBE_vs_plasmid_foldchange", stringsAsFactors = FALSE, header = TRUE, sep ="\t"))
PC9_zscore_CBE_control_plasmid <- PC9_zscore_CBE_control_plasmid %>% rename(c(sgRNA_ID = X, Gene = GENE, PC9_zscore_control_plasmid = zscore)) %>% select(Gene, sgRNA_ID, PC9_zscore_control_plasmid) %>% mutate(editor = "CBE")

PC9_zscore_CBE_Gefit_plasmid <- as_tibble(read.delim("drugZ_zscores_110423/PC9_REP1_REP2_AVERAGE/Gefit_CBE_vs_plasmid_foldchange", stringsAsFactors = FALSE, header = TRUE, sep ="\t"))
PC9_zscore_CBE_Gefit_plasmid <- PC9_zscore_CBE_Gefit_plasmid %>% rename(c(sgRNA_ID = X, Gene = GENE, PC9_zscore_Gefit_plasmid = zscore)) %>% select(Gene, sgRNA_ID, PC9_zscore_Gefit_plasmid) %>% mutate(editor = "CBE")

PC9_zscore_CBE_Osim_plasmid <- as_tibble(read.delim("drugZ_zscores_110423/PC9_REP1_REP2_AVERAGE/Osim_CBE_vs_plasmid_foldchange", stringsAsFactors = FALSE, header = TRUE, sep ="\t"))
PC9_zscore_CBE_Osim_plasmid <- PC9_zscore_CBE_Osim_plasmid %>% rename(c(sgRNA_ID = X, Gene = GENE, PC9_zscore_Osim_plasmid = zscore)) %>% select(Gene, sgRNA_ID, PC9_zscore_Osim_plasmid) %>% mutate(editor = "CBE")

#ABE
PC9_zscore_ABE_control_plasmid <- as_tibble(read.delim("drugZ_zscores_110423/PC9_REP1_REP2_AVERAGE/Control_ABE_vs_plasmid_foldchange", stringsAsFactors = FALSE, header = TRUE, sep ="\t"))
PC9_zscore_ABE_control_plasmid <- PC9_zscore_ABE_control_plasmid %>% rename(c(sgRNA_ID = X, Gene = GENE, PC9_zscore_control_plasmid = zscore)) %>% select(Gene, sgRNA_ID, PC9_zscore_control_plasmid) %>% mutate(editor = "ABE")

PC9_zscore_ABE_Gefit_plasmid <- as_tibble(read.delim("drugZ_zscores_110423/PC9_REP1_REP2_AVERAGE/Gefit_ABE_vs_plasmid_foldchange", stringsAsFactors = FALSE, header = TRUE, sep ="\t"))
PC9_zscore_ABE_Gefit_plasmid <- PC9_zscore_ABE_Gefit_plasmid %>% rename(c(sgRNA_ID = X, Gene = GENE, PC9_zscore_Gefit_plasmid = zscore)) %>% select(Gene, sgRNA_ID, PC9_zscore_Gefit_plasmid) %>% mutate(editor = "ABE")

PC9_zscore_ABE_Osim_plasmid <- as_tibble(read.delim("drugZ_zscores_110423/PC9_REP1_REP2_AVERAGE/Osim_ABE_vs_plasmid_foldchange", stringsAsFactors = FALSE, header = TRUE, sep ="\t"))
PC9_zscore_ABE_Osim_plasmid <- PC9_zscore_ABE_Osim_plasmid %>% rename(c(sgRNA_ID = X, Gene = GENE, PC9_zscore_Osim_plasmid = zscore)) %>% select(Gene, sgRNA_ID, PC9_zscore_Osim_plasmid) %>% mutate(editor = "ABE")


CBE_zscores <- left_join(HT29_zscore_CBE_control_plasmid, HT29_zscore_CBE_Tram_plasmid)
CBE_zscores <- left_join(CBE_zscores, HT29_zscore_CBE_Pict_plasmid)
CBE_zscores <- left_join(CBE_zscores, HT29_zscore_CBE_DebCet_plasmid)
CBE_zscores <- left_join(CBE_zscores, H23_zscore_CBE_control_plasmid)
CBE_zscores <- left_join(CBE_zscores, H23_zscore_CBE_Sotor_plasmid)
CBE_zscores <- left_join(CBE_zscores, H23_zscore_CBE_Adag_plasmid)
CBE_zscores <- left_join(CBE_zscores, PC9_zscore_CBE_control_plasmid)
CBE_zscores <- left_join(CBE_zscores, PC9_zscore_CBE_Gefit_plasmid)
CBE_zscores <- left_join(CBE_zscores, PC9_zscore_CBE_Osim_plasmid)

ABE_zscores <- left_join(HT29_zscore_ABE_control_plasmid, HT29_zscore_ABE_Tram_plasmid)
ABE_zscores <- left_join(ABE_zscores, HT29_zscore_ABE_Pict_plasmid)
ABE_zscores <- left_join(ABE_zscores, HT29_zscore_ABE_DebCet_plasmid)
ABE_zscores <- left_join(ABE_zscores, H23_zscore_ABE_control_plasmid)
ABE_zscores <- left_join(ABE_zscores, H23_zscore_ABE_Sotor_plasmid)
ABE_zscores <- left_join(ABE_zscores, H23_zscore_ABE_Adag_plasmid)
ABE_zscores <- left_join(ABE_zscores, PC9_zscore_ABE_control_plasmid)
ABE_zscores <- left_join(ABE_zscores, PC9_zscore_ABE_Gefit_plasmid)
ABE_zscores <- left_join(ABE_zscores, PC9_zscore_ABE_Osim_plasmid)

zscores <- bind_rows(CBE_zscores, ABE_zscores)
zscores
```

```{r}
#left join the dataframe with the drugZ zscores
annotated_data <- left_join(annotated_data, zscores)
```

```{r}
#drugz and L2FC correlation
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

# HT29 vs plasmid
plot1 <- ggplot(annotated_data, aes(x= L2FC_HT29_Control_plasmid_average, y= HT29_zscore_control_plasmid)) + geom_jitter(size=3,  position = pos, colour = "grey35") +
labs(title= "HT29 - zscore L2FC correlation", x = "L2FC_HT29_Control_plasmid_average", y = "HT29_zscore_control_plasmid") +
geom_smooth(method = lm) +
theme_bw()
plot1

plot2 <- ggplot(annotated_data, aes(x= L2FC_HT29_Tram_plasmid_average, y= HT29_zscore_Tram_plasmid)) + geom_jitter(size=3,  position = pos, colour = "grey35") +
labs(title= "HT29 - zscore L2FC correlation", x = "L2FC_HT29_Tram_plasmid_average", y = "HT29_zscore_Tram_plasmid") +
geom_smooth(method = lm) +
theme_bw()
plot2

plot3 <- ggplot(annotated_data, aes(x= L2FC_HT29_Pict_plasmid_average, y= HT29_zscore_Pict_plasmid)) + geom_jitter(size=3,  position = pos, colour = "grey35") +
labs(title= "HT29 - zscore L2FC correlation", x = "L2FC_HT29_Pict_plasmid_average", y = "HT29_zscore_Pict_plasmid") +
geom_smooth(method = lm) +
theme_bw()
plot3

plot4 <- ggplot(annotated_data, aes(x= L2FC_HT29_DebCet_plasmid_average, y= HT29_zscore_DebCet_plasmid)) + geom_jitter(size=3,  position = pos, colour = "grey35") +
labs(title= "HT29 - zscore L2FC correlation", x = "L2FC_HT29_DebCet_plasmid_average", y = "HT29_zscore_DebCet_plasmid") +
geom_smooth(method = lm) +
theme_bw()
plot4

# H23 vs plasmid
plot5 <- ggplot(annotated_data, aes(x= L2FC_H23_Control_plasmid_average, y= H23_zscore_control_plasmid)) + geom_jitter(size=3,  position = pos, colour = "grey35") +
labs(title= "H23 - zscore L2FC correlation", x = "L2FC_H23_Control_plasmid_average", y = "H23_zscore_control_plasmid") +
geom_smooth(method = lm) +
theme_bw()
plot5

plot6 <- ggplot(annotated_data, aes(x= L2FC_H23_Adag_plasmid_average, y= H23_zscore_Adag_plasmid)) + geom_jitter(size=3,  position = pos, colour = "grey35") +
labs(title= "H23 - zscore L2FC correlation", x = "L2FC_H23_Adag_plasmid_average", y = "H23_zscore_Adag_plasmid") +
geom_smooth(method = lm) +
theme_bw()
plot6

plot7 <- ggplot(annotated_data, aes(x= L2FC_H23_Sotor_plasmid_average, y= H23_zscore_Sotor_plasmid)) + geom_jitter(size=3,  position = pos, colour = "grey35") +
labs(title= "H23 - zscore L2FC correlation", x = "L2FC_H23_Sotor_plasmid_average", y = "H23_zscore_Sotor_plasmid") +
geom_smooth(method = lm) +
theme_bw()
plot7

# PC9 vs plasmid
plot8 <- ggplot(annotated_data, aes(x= L2FC_PC9_Control_plasmid_average, y= PC9_zscore_control_plasmid)) + geom_jitter(size=3,  position = pos, colour = "grey35") +
labs(title= "PC9 - zscore L2FC correlation", x = "L2FC_PC9_Control_plasmid_average", y = "PC9_zscore_control_plasmid") +
geom_smooth(method = lm) +
theme_bw()
plot8

plot9 <- ggplot(annotated_data, aes(x= L2FC_PC9_Gefit_plasmid_average, y= PC9_zscore_Gefit_plasmid)) + geom_jitter(size=3,  position = pos, colour = "grey35") +
labs(title= "PC9 - zscore L2FC correlation", x = "L2FC_PC9_Gefit_plasmid_average", y = "PC9_zscore_Gefit_plasmid") +
geom_smooth(method = lm) +
theme_bw()
plot9

plot10 <- ggplot(annotated_data, aes(x= L2FC_PC9_Osim_plasmid_average, y= PC9_zscore_Osim_plasmid, shape = editor, colour = sgRNA_type)) + geom_jitter(size=3,  position = pos) +
labs(title= "PC9 - zscore L2FC correlation", x = "L2FC_PC9_Osim_plasmid_average", y = "PC9_zscore_Osim_plasmid") +
geom_smooth(method = lm) +
theme_bw()
plot10
```

```{r}
# dot plots _ proliferation - HT-29

targets <- c("EGFR", "KRAS", "MYC", "PARP1", "PARP2", "MAP2K1", "MAP2K2", "BCL2", "BRAF", "AKT1", "PIK3CA")

pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

plot1 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=L2FC_HT29_Control_plasmid_average, y= L2FC_HT29_Tram_plasmid_average, colour = Gene, shape = editor)) + 
geom_point(position = pos, alpha = 1, size =3) +
labs(title= "", x="HT-29 control vs plasmid (log2 fold-change)", y = "HT-29 trametinib vs plasmid (log2 fold-change)", colour = "gene") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_vline(xintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_HT29_Tram_plasmid_average > 2.5 & (is.na(Amino_Acid_Position) == FALSE), paste(Amino_Acid_Position, Gene), ifelse( L2FC_HT29_Tram_plasmid_1 > 2.5 & (is.na(Amino_Acid_Position) == TRUE),  paste(most_severe_consequence, Gene), ifelse(L2FC_HT29_Tram_plasmid_average < -1.6 & L2FC_HT29_Control_plasmid_average > - 0.25, paste(most_severe_consequence, Gene), "")))), size = 3, position = pos) +
scale_color_manual( values =c("royalblue", "turquoise3", "orange", "red", "springgreen4", "grey55", "brown", "purple", "pink", "salmon", "black")) +
scale_shape_manual(values =c(19, 1)) +
theme_classic()
plot1
ggsave("Fig4a.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

plot2 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=L2FC_HT29_Control_plasmid_average, y= L2FC_HT29_Pict_plasmid_average, colour = Gene, shape = editor)) + 
geom_point(position = pos, alpha = 1, size =3) +
labs(title= "", x="HT-29 control vs plasmid (log2 fold-change)", y = "HT-29 pictilisib vs plasmid (log2 fold-change)", colour = "gene") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_vline(xintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_HT29_Pict_plasmid_average > 1.2 | L2FC_HT29_Control_plasmid_average > 1.5, paste(Amino_Acid_Position, ",", Gene), "")), size = 3, position = pos) +
scale_color_manual(values =c("royalblue", "turquoise3", "orange", "red", "springgreen4", "grey55", "brown", "purple", "pink", "salmon", "black")) +
scale_shape_manual(values =c(19, 1)) +
theme_classic()
plot2
ggsave("Fig2a.eps", width = 6, height = 4, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

plot3 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=L2FC_HT29_Control_plasmid_average, y= L2FC_HT29_DebCet_plasmid_average, colour = Gene, shape = editor)) + 
geom_point(position = pos, alpha = 1, size =3) +
labs(title= "", x="HT-29 control vs plasmid (log2 fold-change)", y = "HT-29 dabrafenib + cetuximab vs plasmid (log2 fold-change)", colour = "gene") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_vline(xintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_HT29_DebCet_plasmid_average > 2.5 & (is.na(Amino_Acid_Position) == FALSE)| L2FC_HT29_Control_plasmid_average > 1.5 & (is.na(Amino_Acid_Position) == FALSE), paste(Amino_Acid_Position, Gene), ifelse(L2FC_HT29_DebCet_plasmid_average > 2.5 & (is.na(Amino_Acid_Position) == TRUE) | L2FC_HT29_Control_plasmid_average > 1.5 & (is.na(Amino_Acid_Position) == TRUE), paste(most_severe_consequence, Gene), ""))), size = 3, position = pos) +
scale_color_manual(values =c("royalblue", "turquoise3", "orange", "red", "springgreen4", "grey55", "brown", "purple", "pink", "salmon", "black")) +
scale_shape_manual(values =c(19, 1)) +
theme_classic()
plot3
ggsave("Fig4b.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")
```

```{r}
# dot plots _ proliferation - H23

targets <- c("EGFR", "KRAS", "MYC", "PARP1", "PARP2", "MAP2K1", "MAP2K2", "BCL2", "BRAF", "AKT1", "PIK3CA")

pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

plot1 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=L2FC_H23_Control_plasmid_average, y= L2FC_H23_Sotor_plasmid_average, colour = Gene, shape = editor)) + 
geom_point(position = pos, alpha = 0.80, size =3) +
labs(title= "", x="L2FC_H23_Control_plasmid_1", y = "L2FC_H23_Sotor_plasmid_1") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_H23_Sotor_plasmid_average > 2 , paste(Amino_Acid_Position, ",", Gene), "")), size = 3, position = pos) +
#facet_wrap(~Gene, scales ="free") +
scale_color_manual(values =c("black", "springgreen4", "orange", "red", "blue", "grey55", "brown", "purple", "pink", "salmon", "royalblue")) +
scale_shape_manual(values =c(19, 1)) +
theme_classic()
plot1

plot2 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=L2FC_H23_Control_plasmid_average, y= L2FC_H23_Adag_plasmid_average, colour = Gene, shape = editor)) + 
geom_point(position = pos, alpha = 0.80, size =3) +
labs(title= "", x="L2FC_H23_Control_plasmid_1", y = "L2FC_H23_Adag_plasmid_1") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_H23_Adag_plasmid_average > 2.5 , paste(Amino_Acid_Position, ",", Gene), "")), size = 3, position = pos) +
scale_color_manual(values =c("black", "springgreen4", "orange", "red", "blue", "grey55", "brown", "purple", "pink", "salmon", "royalblue")) +
scale_shape_manual(values =c(19, 1)) +
theme_classic()
plot2
```

```{r}
# dot plots _ proliferation - H23 Adag vs Sotor

targets <- c("EGFR", "KRAS", "MYC", "PARP1", "PARP2", "MAP2K1", "MAP2K2", "BCL2", "BRAF", "AKT1", "PIK3CA")

pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

plot1 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=L2FC_H23_Adag_plasmid_1, y= L2FC_H23_Sotor_plasmid_1, colour = Gene, shape = editor)) + 
geom_point(position = pos, alpha = 0.90, size = 3) +
labs(title= "H23 - Sotorasib vs Adagrasib", x="L2FC Adagrasib vs plasmid", y = "L2FC Sotorasib vs plasmid", colour = "gene") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_vline(xintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_H23_Adag_plasmid_1 > 2.2 & (is.na(Amino_Acid_Position) == FALSE)| L2FC_H23_Sotor_plasmid_1 > 2 & (is.na(Amino_Acid_Position) == FALSE), paste(Amino_Acid_Position, Gene), ifelse(L2FC_H23_Adag_plasmid_1 > 2.2 & (is.na(Amino_Acid_Position) == TRUE)| L2FC_H23_Sotor_plasmid_1 > 2 & (is.na(Amino_Acid_Position) == TRUE), paste(most_severe_consequence, Gene), ""))), size = 3, position = pos) +
#facet_wrap(~Gene, scales ="free") +
scale_color_manual(values =c("black", "springgreen4", "orange", "red", "blue", "grey55", "brown", "purple", "pink", "salmon", "royalblue")) +
scale_shape_manual(values =c(19, 1)) +
theme_classic()
plot1

plot2 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=L2FC_H23_Adag_plasmid_2, y= L2FC_H23_Sotor_plasmid_2, colour = Gene, shape = editor)) + 
geom_point(position = pos, alpha = 0.90, size = 3) +
labs(title= "H23 - Sotorasib vs Adagrasib", x="L2FC Adagrasib vs plasmid", y = "L2FC Sotorasib vs plasmid", colour = "gene") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_vline(xintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_H23_Adag_plasmid_2 > 2.2 & (is.na(Amino_Acid_Position) == FALSE)| L2FC_H23_Sotor_plasmid_2 > 2 & (is.na(Amino_Acid_Position) == FALSE), paste(Amino_Acid_Position, Gene), ifelse(L2FC_H23_Adag_plasmid_2 > 2.2 & (is.na(Amino_Acid_Position) == TRUE)| L2FC_H23_Sotor_plasmid_2 > 2 & (is.na(Amino_Acid_Position) == TRUE), paste(most_severe_consequence, Gene), ""))), size = 3, position = pos) +
#facet_wrap(~Gene, scales ="free") +
scale_color_manual(values =c("black", "springgreen4", "orange", "red", "blue", "grey55", "brown", "purple", "pink", "salmon", "royalblue")) +
scale_shape_manual(values =c(19, 1)) +
theme_classic()
plot2

plot3 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=L2FC_H23_Control_plasmid_average, y= L2FC_H23_Sotor_plasmid_average, colour = Gene, shape = editor)) + 
geom_point(position = pos, alpha = 0.90, size = 3) +
labs(title= "H23 - Sotorasib vs plasmid", x="L2FC Control vs plasmid", y = "L2FC Sotorasib vs plasmid", colour = "gene") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_vline(xintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_H23_Sotor_plasmid_average > 2 & (is.na(Amino_Acid_Position) == FALSE)| L2FC_H23_Sotor_plasmid_average > 2 & (is.na(Amino_Acid_Position) == FALSE), paste(Amino_Acid_Position, Gene), ifelse(guide == "TGTCAGCAGGACCACCACAG", paste(Amino_Acid_Position, Gene), ""))), size = 3, position = pos) +
#facet_wrap(~Gene, scales ="free") +
scale_color_manual(values =c("black", "springgreen4", "orange", "red", "blue", "grey55", "brown", "purple", "pink", "salmon", "royalblue")) +
scale_shape_manual(values =c(19, 1)) +
theme_classic()
plot3

plot4 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=L2FC_H23_Control_plasmid_average, y= L2FC_H23_Adag_plasmid_average, colour = Gene, shape = editor)) + 
geom_point(position = pos, alpha = 0.90, size = 3) +
labs(title= "H23 - Adagrasib vs plasmid", x="L2FC Control vs plasmid", y = "L2FC Adagrasib vs plasmid", colour = "gene") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_vline(xintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_H23_Adag_plasmid_average > 2 & (is.na(Amino_Acid_Position) == FALSE)| L2FC_H23_Adag_plasmid_average > 2 & (is.na(Amino_Acid_Position) == FALSE), paste(Amino_Acid_Position, Gene), ifelse(guide == "TGTCAGCAGGACCACCACAG", paste(Amino_Acid_Position, Gene), ""))), size = 3, position = pos) +
#facet_wrap(~Gene, scales ="free") +
scale_color_manual(values =c("black", "springgreen4", "orange", "red", "blue", "grey55", "brown", "purple", "pink", "salmon", "royalblue")) +
scale_shape_manual(values =c(19, 1)) +
theme_classic()
plot4

plot5 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=L2FC_H23_Adag_plasmid_average, y= L2FC_H23_Sotor_plasmid_average, colour = Gene, shape = editor)) + 
geom_point(position = pos, alpha = 1, size = 3) +
labs(title= "", x="H23 adagrasib vs plasmid (log2 fold-change)", y = "H23 sotorasib vs plasmid (log2 fold-change)", colour = "gene") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_vline(xintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_H23_Adag_plasmid_average > 1.8 & (is.na(Amino_Acid_Position) == FALSE)| L2FC_H23_Sotor_plasmid_average > 1.5 & (is.na(Amino_Acid_Position) == FALSE), paste(Amino_Acid_Position, Gene), ifelse(L2FC_H23_Adag_plasmid_average > 1.8 & (is.na(Amino_Acid_Position) == TRUE)| L2FC_H23_Sotor_plasmid_average > 1.5 & (is.na(Amino_Acid_Position) == TRUE), paste(most_severe_consequence, Gene),""))), size = 3, position = pos, max.overlaps = 50) +
scale_color_manual( values =c("royalblue", "turquoise3", "orange", "red", "springgreen4", "grey55", "brown", "purple", "pink", "salmon", "black")) +
scale_shape_manual(values =c(19, 1)) +
theme_classic()
plot5
ggsave("Fig2b.eps", width = 6, height = 4, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")
```

```{r}
# dot plots _ proliferation - PC9 average

targets <- c("EGFR", "KRAS", "MYC", "PARP1", "PARP2", "MAP2K1", "MAP2K2", "BCL2", "BRAF", "AKT1", "PIK3CA")

pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

plot1 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=L2FC_PC9_Control_plasmid_average, y= L2FC_PC9_Gefit_plasmid_average, colour = Gene, shape = editor)) + 
geom_point(position = pos, alpha = 1, size =3) +
labs(title= " ", x="PC9 control vs plasmid (log2 fold-change)", y = "PC9 gefitinib vs plasmid (log2 fold-change)", colour = "gene") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_vline(xintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_PC9_Gefit_plasmid_average > 4 & (is.na(Amino_Acid_Position) == FALSE) | L2FC_PC9_Gefit_plasmid_average < -2.5 & L2FC_PC9_Control_plasmid_average > -0.4 & (is.na(Amino_Acid_Position) == FALSE), paste(Amino_Acid_Position, Gene), ifelse(L2FC_PC9_Gefit_plasmid_average > 4 & (is.na(Amino_Acid_Position) == TRUE) | L2FC_PC9_Gefit_plasmid_average < -2.5 & L2FC_PC9_Control_plasmid_average > -0.4 & (is.na(Amino_Acid_Position) == TRUE), paste(most_severe_consequence, Gene), ifelse(L2FC_PC9_Gefit_plasmid_average >1 & L2FC_PC9_Control_plasmid_average < -1.9, paste(Amino_Acid_Position, Gene), ifelse(L2FC_PC9_Gefit_plasmid_average >1.5 & L2FC_PC9_Control_plasmid_average > 1, paste(Amino_Acid_Position, Gene), ""))))), size = 3, position = pos) +
scale_color_manual( values =c("royalblue", "turquoise3", "orange", "red", "springgreen4", "grey55", "brown", "purple", "pink", "salmon", "black")) +
scale_shape_manual(values =c(19, 1)) +
theme_classic()
plot1
ggsave("Fig2c.eps", width = 6, height = 4, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

plot2 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=L2FC_PC9_Control_plasmid_average, y= L2FC_PC9_Osim_plasmid_average, colour = Gene, shape = editor)) + 
geom_point(position = pos, alpha = 1, size =3) +
labs(title= " ", x="PC9 control vs plasmid (log2 fold-change)", y = "PC9 osimertinib vs plasmid (log2 fold-change)", colour = "gene") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_vline(xintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_PC9_Osim_plasmid_average > 2 & (is.na(Amino_Acid_Position) == FALSE) | L2FC_PC9_Osim_plasmid_average < -2 & L2FC_PC9_Control_plasmid_average > -0.4 & (is.na(Amino_Acid_Position) == FALSE), paste(Amino_Acid_Position, Gene), ifelse(L2FC_PC9_Osim_plasmid_average > 2 & (is.na(Amino_Acid_Position) == TRUE) | L2FC_PC9_Osim_plasmid_average < -2 & L2FC_PC9_Control_plasmid_average > -0.4 & (is.na(Amino_Acid_Position) == TRUE), paste(most_severe_consequence, Gene), ifelse(L2FC_PC9_Osim_plasmid_average >0.5 & L2FC_PC9_Control_plasmid_average < -1.8, paste(Amino_Acid_Position, Gene), ifelse(L2FC_PC9_Osim_plasmid_average >2 & L2FC_PC9_Control_plasmid_average >1, paste(Amino_Acid_Position, Gene), ""))))), size = 3, position = pos) +
scale_color_manual( values =c("royalblue", "turquoise3", "orange", "red", "springgreen4", "grey55", "brown", "purple", "pink", "salmon", "black")) +
scale_shape_manual(values =c(19, 1)) +
theme_classic()
plot2
ggsave("Fig2d.eps", width = 6, height = 4, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")
```

```{r}
# dot plots _ proliferation

targets <- c("EGFR", "KRAS", "MYC", "PARP1", "PARP2", "MAP2K1", "MAP2K2", "BCL2", "BRAF", "AKT1", "PIK3CA")

pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

plot1 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=Amino_Acid_Position_simple, y= L2FC_HT29_Control_plasmid_average, colour = most_severe_consequence)) + 
geom_point(position = pos, shape = 16, alpha = 0.8, size =2) +
labs(title= "", x="amino acid position", y = "L2FC_HT29_Control_plasmid_average") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
facet_wrap(~Gene, scales = "free_x") +
scale_color_manual(values =c("grey65", "black", "black", "orange", "blue", "red", "purple", "grey65", "grey65", "grey65", "grey65")) +
theme_classic()
plot1

plot2 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=Amino_Acid_Position_simple, y= L2FC_H23_Control_plasmid_1, colour = most_severe_consequence)) + 
geom_point(position = pos, shape = 16, alpha = 0.8, size =2) +
labs(title= "", x="amino acid position", y = "L2FC_H23_Control_plasmid_1") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
facet_wrap(~Gene, scales = "free_x") +
scale_color_manual(values =c("grey65", "black", "black", "orange", "blue", "red", "purple", "grey65", "grey65", "grey65", "grey65")) +
theme_classic()
plot2

plot3 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=Amino_Acid_Position_simple, y= L2FC_PC9_Control_plasmid_1, colour = most_severe_consequence)) + 
geom_point(position = pos, shape = 16, alpha = 0.8, size =2) +
labs(title= "", x="amino acid position", y = "L2FC_PC9_Control_plasmid_1") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
facet_wrap(~Gene, scales = "free_x") +
scale_color_manual(values =c("grey65", "black", "black", "orange", "blue", "red", "purple", "grey65", "grey65", "grey65", "grey65")) +
theme_classic()
plot3
```

```{r}
# dot plots _ proliferation - gene essentially

targets <- c("KRAS", "MYC", "BRAF")

pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

plot1 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=L2FC_HT29_Control_plasmid_average, y= L2FC_H23_Control_plasmid_average, colour = Gene)) + 
geom_point(position = pos, shape = 16, alpha = 0.8, size =3) +
labs(title= "", x="L2FC_HT29_Control_plasmid_average", y = "L2FC_H23_Control_plasmid_average") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
scale_color_manual(values =c("red", "orange", "black")) +
theme_bw()
plot1

targets <- c("EGFR", "MYC", "BRAF")

plot2 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=L2FC_HT29_Control_plasmid_average, y= L2FC_PC9_Control_plasmid_average, colour = Gene)) + 
geom_point(position = pos, shape = 16, alpha = 1, size =3) +
labs(title= "", x="HT-29 control vs plasmid (log2 fold-change)", y = "PC9 control vs plasmid (log2 fold-change)", colour = "gene") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_vline(xintercept=c(0,0),linetype="dotted") +
scale_color_manual(values =c("red", "orange", "black")) +
theme_classic()
plot2
ggsave("Fig1c.eps", width = 5, height = 4, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

targets <- c("EGFR", "MYC", "KRAS")

plot3 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=L2FC_PC9_Control_plasmid_average, y= L2FC_H23_Control_plasmid_average, colour = Gene)) + 
geom_point(position = pos, shape = 16, alpha = 0.8, size =3) +
labs(title= "", x="L2FC_PC9_Control_plasmid_average", y = "L2FC_H23_Control_plasmid_average") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
scale_color_manual(values =c("red", "orange", "black")) +
theme_bw()
plot3
```

```{r}
# read in COSMIC counts

COSMIC <- as_tibble(read.csv("COSMIC_11_230822.csv", stringsAsFactors = FALSE, header = TRUE))
COSMIC <- rename(COSMIC, Amino_Acid_Position = Position)
COSMIC <- rename(COSMIC, COSMIC_count = Count)
COSMIC <- rename(COSMIC, Amino_Acid_Change_COSMIC = AA.Mutation)
COSMIC <- rename(COSMIC, CDS_mutation_COSMIC = CDS.Mutation)

COSMIC <- mutate(COSMIC, Amino_Acid_Position = as.character(Amino_Acid_Position))

COSMIC2 <- select(COSMIC, Gene, Amino_Acid_Position, COSMIC_count)
COSMIC2 <- COSMIC2 %>%
  group_by(Gene, Amino_Acid_Position) %>%
  summarise(COSMIC_count = sum(COSMIC_count)) %>% distinct()
COSMIC2
annotated_data_cosmic <- left_join(annotated_data, COSMIC2) %>% distinct()
annotated_data_cosmic %>% filter(Gene == "AKT1" & Amino_Acid_Position =="17")
annotated_data_cosmic %>% filter(COSMIC_count > 100) %>% select(Gene, Amino_Acid_Position, COSMIC_count)
annotated_data_cosmic <- annotated_data_cosmic %>% 
  mutate(COSMIC_count = ifelse(is.na(COSMIC_count) == TRUE, "0", COSMIC_count)) %>%
  mutate(COSMIC_count = as.numeric(COSMIC_count))
```

```{r}
# dot plots _ proliferation - HT-29 COSMIC graph

targets <- c("EGFR", "KRAS", "MYC", "PARP1", "PARP2", "MAP2K1", "MAP2K2", "BCL2", "BRAF", "AKT1", "PIK3CA")

pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

plot1 <- ggplot(subset(annotated_data_cosmic, Gene == "AKT1"), aes(x=Amino_Acid_Position_simple, y= L2FC_HT29_Control_plasmid_average, size = COSMIC_count)) + 
geom_point(position = pos, shape = 1, alpha = 1) +
labs(title = "", x="AKT1 amino acid position", y = "HT-29 control vs plasmid (log2 fold-change)", size = "COSMIC count") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_HT29_Control_plasmid_average > 1 & (is.na(Amino_Acid_Position) == FALSE), paste(Amino_Acid_Position),  "")), position = pos, size = 4) +
scale_size(range = c(3, 10)) +
theme_classic()
plot1
ggsave("Fig1e.eps", width = 6, height = 4.1, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

plot2 <- ggplot(subset(annotated_data_cosmic, Gene == "PIK3CA"), aes(x=Amino_Acid_Position_simple, y= L2FC_HT29_Control_plasmid_average, size = COSMIC_count)) + 
geom_point(position = pos, shape = 1, alpha = 1) +
labs(title = "", x="PIK3CA amino acid position", y = "HT-29 control vs plasmid (log2 fold-change)", size = "COSMIC count") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_HT29_Control_plasmid_average > 1 & (is.na(Amino_Acid_Position) == FALSE), paste(Amino_Acid_Position),  "")), position = pos, size = 4) +
scale_size(range = c(3, 10)) +
theme_classic()
plot2
ggsave("Fig1f.eps", width = 6, height = 4.1, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")
```

```{r}
# mutate BRAF curated_Domain field to simplify for figure
annotated_data %>% filter(Gene == "BRAF") %>%
  select(Gene, curated_Domain) %>%
  count(curated_Domain)

annotated_data <- mutate(annotated_data, curated_Domain = ifelse(curated_Domain == "ATP binding site;Protein kinase" & Gene == "BRAF", "ATP binding site", ifelse(curated_Domain == "Protein kinase;ATP binding site", "ATP binding site", ifelse(curated_Domain == "Breakpoint for translocation to form KIAA1549-BRAF fusion protein;Disordered", "Disordered", ifelse(curated_Domain == "Breakpoint for translocation to form KIAA1549-BRAF fusion protein", "", curated_Domain)))))

annotated_data %>% filter(Gene == "BRAF") %>%
  select(Gene, curated_Domain) %>%
  count(curated_Domain)
```

```{r}
# dot plots _ proliferation
targets <- c("BRAF")
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

plot1 <- ggplot(subset(annotated_data, Gene %in% c(targets) & (is.na(Amino_Acid_Position_simple) ==FALSE)), aes(x=Amino_Acid_Position_simple, y= L2FC_HT29_Control_plasmid_average, shape = most_severe_consequence, colour = curated_Domain)) + 
geom_point(position = pos, alpha = 1, size = 3) +
labs(title= "", x="BRAF amino acid position", y = "HT-29 control vs plasmid (log2 fold-change)", colour = "domain", shape = "consequence") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_HT29_Control_plasmid_average > 0.7 | L2FC_HT29_Control_plasmid_average < -2.5, paste(PTM), "")), size = 3, position = pos) +
scale_color_manual(values =c("grey55", "purple", "black", "royalblue", "orange", "red")) +
scale_shape_manual(values=c(19, 2, 11, 7, 3)) +
theme_classic()
plot1
ggsave("Fig1d.eps", width = 6, height = 4, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")
```

```{r}
# dot plots _ proliferation

targets <- c("EGFR", "KRAS", "MYC", "PARP1", "PARP2", "MAP2K1", "MAP2K2", "BCL2", "BRAF", "AKT1", "PIK3CA")

pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

plot1 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=L2FC_HT29_Control_plasmid_1, y= L2FC_H23_Control_plasmid_1, colour = most_severe_consequence)) + 
geom_point(position = pos, alpha = 0.80, size =3) +
labs(title= "", x="L2FC_HT29_Control_plasmid_1", y = "L2FC_H23_Control_plasmid_1") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
#geom_text_repel(aes(label=ifelse(L2FC_PC9_Gefit_plasmid_1 > 4 | L2FC_PC9_Gefit_plasmid_1 < -2.5 & L2FC_PC9_Control_plasmid_1 > -0.4, paste(Amino_Acid_Position, ",", Gene, ",", Consequence), "")), size = 3, position = pos) +
scale_color_manual(values =c("black", "springgreen4", "orange", "red", "blue", "grey55", "brown", "purple", "pink", "salmon", "royalblue")) +
#scale_shape_manual(values =c(19, 1)) +
facet_wrap(~Gene, scales = "free_x") +
theme_classic()
plot1

plot2 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=L2FC_HT29_Control_plasmid_1, y= L2FC_PC9_Control_plasmid_1, colour = most_severe_consequence)) + 
geom_point(position = pos, alpha = 0.80, size =3) +
labs(title= "", x="L2FC_HT29_Control_plasmid_1", y = "L2FC_PC9_Control_plasmid_1") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
#geom_text_repel(aes(label=ifelse(L2FC_PC9_Gefit_plasmid_1 > 4 | L2FC_PC9_Gefit_plasmid_1 < -2.5 & L2FC_PC9_Control_plasmid_1 > -0.4, paste(Amino_Acid_Position, ",", Gene, ",", Consequence), "")), size = 3, position = pos) +
scale_color_manual(values =c("black", "springgreen4", "orange", "red", "blue", "grey55", "brown", "purple", "pink", "salmon", "royalblue")) +
#scale_shape_manual(values =c(19, 1)) +
facet_wrap(~Gene, scales = "free_x") +
theme_classic()
plot2

plot3 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=L2FC_PC9_Control_plasmid_1, y= L2FC_H23_Control_plasmid_1, colour = most_severe_consequence)) + 
geom_point(position = pos, alpha = 0.80, size =3) +
labs(title= "", x="L2FC_PC9_Control_plasmid_1", y = "L2FC_H23_Control_plasmid_1") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
#geom_text_repel(aes(label=ifelse(L2FC_PC9_Gefit_plasmid_1 > 4 | L2FC_PC9_Gefit_plasmid_1 < -2.5 & L2FC_PC9_Control_plasmid_1 > -0.4, paste(Amino_Acid_Position, ",", Gene, ",", Consequence), "")), size = 3, position = pos) +
scale_color_manual(values =c("black", "springgreen4", "orange", "red", "blue", "grey55", "brown", "purple", "pink", "salmon", "royalblue")) +
#scale_shape_manual(values =c(19, 1)) +
facet_wrap(~Gene, scales = "free_x") +
theme_classic()
plot3
```

```{r}
#stats for violin plots - Student's t-test

#HT29
NT_HT29 <- annotated_data %>%
  filter(gene_alias == "non-targeting") %>%
  select(L2FC_HT29_Control_plasmid_average)

essential_HT29 <- annotated_data %>%
  filter(gene_alias == "essential") %>%
  select(L2FC_HT29_Control_plasmid_average)

t_test_1 <- t.test(NT_HT29, essential_HT29, var.equal = TRUE)
t_test_1

#H23
NT_H23 <- annotated_data %>%
  filter(gene_alias == "non-targeting") %>%
  select(L2FC_H23_Control_plasmid_average)

essential_H23 <- annotated_data %>%
  filter(gene_alias == "essential") %>%
  select(L2FC_H23_Control_plasmid_average)

t_test_2 <- t.test(NT_H23, essential_H23, var.equal = TRUE)
t_test_2

#PC9
NT_PC9 <- annotated_data %>%
  filter(gene_alias == "non-targeting") %>%
  select(L2FC_PC9_Control_plasmid_average)

essential_PC9 <- annotated_data %>%
  filter(gene_alias == "essential") %>%
  select(L2FC_PC9_Control_plasmid_average)

t_test_3 <- t.test(NT_PC9, essential_PC9, var.equal = TRUE)
t_test_3
```

```{r} 
# Violin overview plots Figure 1
# control vs plasmid
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

#HT-29
plot1 <- ggplot(annotated_data, aes(y = forcats::fct_relevel(gene_alias, "PARP1", "PARP2", "PIK3CA", "AKT1", "BCL2", "MAP2K1", "MAP2K2", "BRAF", "KRAS","EGFR","MYC", "essential", "non-essential", "intergenic", "non-targeting") , x = L2FC_HT29_Control_plasmid_average)) +
labs(title= "", y = "", x = "HT-29 control vs plasmid (log2 fold-change)") +
geom_point(size=3, alpha=1, position = pos, shape = 1) +
geom_violin(colour = "salmon", alpha = 1) +
annotate("text", y = 12, x = 1.8, label = paste("p =", formatC(t_test_1$p.value, digits = 3)), parse = FALSE, size = 5) +
theme_classic()
plot1
ggsave("Fig1b.eps", width = 6, height = 5, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")


#H23
plot3 <- ggplot(annotated_data, aes(y = forcats::fct_relevel(gene_alias, "PARP1", "PARP2", "PIK3CA", "AKT1", "BCL2", "MAP2K1", "MAP2K2", "BRAF", "KRAS","EGFR","MYC", "essential", "non-essential", "non-targeting") , x = L2FC_H23_Control_plasmid_average)) +
labs(title= "", y = "", x = "H23 control vs plasmid (log2 fold-change)") +
geom_point(size=3, alpha=1, position = pos, shape = 1) +
geom_violin(colour = "salmon", alpha = 1) +
annotate("text", y = 12, x = 1.8, label = paste("p =", formatC(t_test_2$p.value, digits = 3)), parse = FALSE, size = 5) +
theme_classic()
plot3
ggsave("FigS1b_1.eps", width = 6, height = 5, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

#PC9
plot2 <- ggplot(annotated_data, aes(y = forcats::fct_relevel(gene_alias, "PARP1", "PARP2", "PIK3CA", "AKT1", "BCL2", "MAP2K1", "MAP2K2", "BRAF", "KRAS","EGFR","MYC", "essential", "non-essential", "non-targeting") , x = L2FC_PC9_Control_plasmid_average)) +
labs(title= "", y = "", x = "PC9 control vs plasmid (log2 fold-change)") +
geom_point(size=3, alpha=1, position = pos, shape = 1) +
geom_violin(colour = "salmon", alpha = 1) +
annotate("text", y = 12, x = 1.9, label = paste("p =", formatC(t_test_3$p.value, digits = 3)), parse = FALSE, size = 5) +
theme_classic()
plot2
ggsave("FigS1b_2.eps", width = 6, height = 5, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")
```

```{r}
# primers for Golden Gate arrayed validation studies
# forward_primer
# add a G for the expression from U6 promtoter by replaceing the base at pos. 1 if, it is not a G
annotated_data <- mutate(annotated_data, G_guide = ifelse(substr(guide, 1, 1) == "G", guide, paste0("G", substr(guide, 2, 20))))

annotated_data <- mutate(annotated_data, for_primer = paste0("CACC", G_guide))

# reverse_primer
annotated_data <- mutate(annotated_data, rev_primer = G_guide)
annotated_data <- mutate(annotated_data, rev_primer = str_replace_all(rev_primer, "A", "t"))
annotated_data <- mutate(annotated_data, rev_primer = str_replace_all(rev_primer, "T", "a"))
annotated_data <- mutate(annotated_data, rev_primer = str_replace_all(rev_primer, "C", "g"))
annotated_data <- mutate(annotated_data, rev_primer = str_replace_all(rev_primer, "G", "c"))
annotated_data <- mutate(annotated_data, rev_primer = str_to_upper(rev_primer))

annotated_data <- mutate(annotated_data, rev_primer = paste0(substr(rev_primer, 20, 20), substr(rev_primer, 19, 19),substr(rev_primer, 18, 18),substr(rev_primer, 17, 17),substr(rev_primer, 16, 16),substr(rev_primer, 15, 15), substr(rev_primer, 14, 14), substr(rev_primer, 13, 13), substr(rev_primer, 12, 12), substr(rev_primer, 11, 11), substr(rev_primer, 10, 10), substr(rev_primer, 9, 9), substr(rev_primer, 8, 8), substr(rev_primer, 7, 7), substr(rev_primer, 6, 6), substr(rev_primer, 5, 5), substr(rev_primer, 4, 4), substr(rev_primer, 3, 3), substr(rev_primer, 2, 2), substr(rev_primer, 1, 1)))

annotated_data <- mutate(annotated_data, rev_primer = paste0("AAAC", rev_primer))
```

```{r} 
# investigate BRAF hits
annotated_data %>% 
  filter(Gene == "BRAF") %>%
  filter(L2FC_HT29_Control_plasmid_average > 1.6 | L2FC_PC9_Control_plasmid_average > 1.6 | L2FC_H23_Control_plasmid_average > 1.6) %>%
  select(Gene, guide, most_severe_consequence, Amino_Acid_Position, editor, L2FC_HT29_Control_plasmid_average, L2FC_PC9_Control_plasmid_average, L2FC_H23_Control_plasmid_average, sgRNA_type, for_primer, rev_primer)

annotated_data %>% 
  filter(Gene == "BRAF") %>%
  filter(L2FC_HT29_Control_plasmid_average > 1.6 | L2FC_PC9_Control_plasmid_average > 1.6 | L2FC_H23_Control_plasmid_average > 1.6) %>%
  select(Gene, guide, most_severe_consequence, Amino_Acid_Position, editor, L2FC_HT29_Control_plasmid_average, L2FC_PC9_Control_plasmid_average, L2FC_H23_Control_plasmid_average, sgRNA_type, for_primer, rev_primer) %>% write.csv("BRAF_valid.csv")
```

```{r}
#prime editing validation scope
annotated_data %>% filter(L2FC_PC9_Gefit_plasmid_average > 4 | L2FC_PC9_Osim_plasmid_average > 2) %>% write.csv("Marie_pridict_test_input.csv")
```
