---
title: "Res_1_screens_070422"
author: "Matt Coelho"
date: "07.04.2022"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
# load packages
library(tidyverse)
library(reshape2)
library(ggrepel)
library(stringr)
library(digest)
library(ggpointdensity)

#set max.overlaps of labelling graphs to infinity
options(geom_text_repel.max.overlaps = Inf)
```

```{r}
# read in gRNA library
guides <- as_tibble(read.csv("Oncogene_BE_ABE_NGN_TWIST_LIB_041121.csv", stringsAsFactors = FALSE, header = TRUE))
guides

# read in off-target NGN analysis from Joel GPU
off_targets <- as_tibble(read.csv("Oncogene_BE_ABE_NGN_TWIST_LIB_092021.offtargets.csv", stringsAsFactors = FALSE, header = TRUE))
off_targets %>% filter(is_dupe == TRUE)

off_targets <- off_targets %>% select(ID, guide, off_target_metric_NG, off_target_summary_NG, polyT, Direction, is_dupe)
  
# join the off targets to the tibble
lib <- left_join(guides, off_targets)

# count guides
guides
lib
lib %>% filter(is_dupe == TRUE)

#  unique guides
lib <- distinct(lib)
lib <- lib %>%
  select(!c(is_dupe, strand, off_target_summary))

# infer strand from gRNA "Direction"
lib <- mutate(lib, strand = ifelse(Direction == "right", "+", ifelse(Direction == "left", "-", "NA")))
lib

# off_target_reports
lib <- separate(lib, off_target_summary_NG, remove = FALSE, into = c("perfect_matches", "one_mismatch", "two_mismatches", "three_mismatches", "four_mismatches", "extra_"), sep = c(","), extra = "merge")
lib <- lib %>% 
  select(!"extra_")

lib <- mutate(lib, perfect_matches = str_remove_all(perfect_matches, "\\{"))
lib <- mutate(lib, four_mismatches = str_remove_all(four_mismatches, "\\}"))

lib <- mutate(lib, perfect_matches = as.numeric(substr(perfect_matches, 4, 50)))
lib <- mutate(lib, one_mismatch = as.numeric(substr(one_mismatch, 4, 50)))
lib <- mutate(lib, two_mismatches = as.numeric(substr(two_mismatches, 4, 50)))
lib <- mutate(lib, three_mismatches = as.numeric(substr(three_mismatches, 4, 50)))
lib <- mutate(lib, four_mismatches = as.numeric(substr(four_mismatches, 4, 50)))

# rename ID to sgRNA_ID
lib <- rename(lib, sgRNA_ID = ID)
lib <- rename(lib, start = start_BEstimate)
lib <- rename(lib, end = end_BEstimate)
lib <- rename(lib, chr = chr_BEstimate)

lib %>% 
  arrange(desc(perfect_matches)) %>% select(Gene, off_target_summary_NG)

lib %>% filter(perfect_matches >1 | is.na(perfect_matches)) 
lib %>% filter(perfect_matches >1) %>% count(Gene)
lib %>% count(Gene)
lib %>% filter(perfect_matches >1) %>% select(Gene, off_target_summary_NG)
lib
```

```{r}
# read in raw gRNA reads counts
raw_reads <- as_tibble(read.csv("raw_read_counts_120422.csv", stringsAsFactors = FALSE, header = TRUE))
raw_reads <- mutate(raw_reads, sgRNA_ID = as.character(sgRNA_ID))

# compute RPM normalised values
# define functions
RPM <- function(x) ((x/sum(x) *1000000) + 1)
average <- function(x,y) ((x + y) /2)
L2FC <- function(x,y) (log2(x/y))
```

```{r}
# compute average reads
# HT-29 average
#BE3.9max NGN
raw_reads <- mutate(raw_reads, HT29_CBE_T0_average_reads = average(HT29_CBE_T0_exp1_reads, HT29_CBE_T0_exp2_reads))
raw_reads <- mutate(raw_reads, HT29_CBE_Control_average_reads = average(HT29_CBE_Control_exp1_reads, HT29_CBE_Control_exp2_reads))
raw_reads <- mutate(raw_reads, HT29_CBE_Tram_average_reads = average(HT29_CBE_Tram_exp1_reads, HT29_CBE_Tram_exp2_reads))
raw_reads <- mutate(raw_reads, HT29_CBE_Pict_average_reads = average(HT29_CBE_Pict_exp1_reads, HT29_CBE_Pict_exp2_reads))
raw_reads <- mutate(raw_reads, HT29_CBE_DebCet_average_reads = average(HT29_CBE_DebCet_exp1_reads, HT29_CBE_DebCet_exp2_reads))

#ABE8e NGN
raw_reads <- mutate(raw_reads, HT29_ABE_T0_average_reads = average(HT29_ABE_T0_exp1_reads, HT29_ABE_T0_exp2_reads))
raw_reads <- mutate(raw_reads, HT29_ABE_Control_average_reads = average(HT29_ABE_Control_exp1_reads, HT29_ABE_Control_exp2_reads))
raw_reads <- mutate(raw_reads, HT29_ABE_Tram_average_reads = average(HT29_ABE_Tram_exp1_reads, HT29_ABE_Tram_exp2_reads))
raw_reads <- mutate(raw_reads, HT29_ABE_Pict_average_reads = average(HT29_ABE_Pict_exp1_reads, HT29_ABE_Pict_exp2_reads))
raw_reads <- mutate(raw_reads, HT29_ABE_DebCet_average_reads = average(HT29_ABE_DebCet_exp1_reads, HT29_ABE_DebCet_exp2_reads))

# H23 average
#BE3.9max NGN
raw_reads <- mutate(raw_reads, H23_CBE_T0_average_reads = average(H23_CBE_T0_exp1_reads, H23_CBE_T0_exp2_reads))
raw_reads <- mutate(raw_reads, H23_CBE_Control_average_reads = average(H23_CBE_Control_exp1_reads, H23_CBE_Control_exp2_reads))
raw_reads <- mutate(raw_reads, H23_CBE_Sotor_average_reads = average(H23_CBE_Sotor_exp1_reads, H23_CBE_Sotor_exp2_reads))
raw_reads <- mutate(raw_reads, H23_CBE_Adag_average_reads = average(H23_CBE_Adag_exp1_reads, H23_CBE_Adag_exp2_reads))

#ABE8e NGN
raw_reads <- mutate(raw_reads, H23_ABE_T0_average_reads = average(H23_ABE_T0_exp1_reads, H23_ABE_T0_exp2_reads))
raw_reads <- mutate(raw_reads, H23_ABE_Control_average_reads = average(H23_ABE_Control_exp1_reads, H23_ABE_Control_exp2_reads))
raw_reads <- mutate(raw_reads, H23_ABE_Sotor_average_reads = average(H23_ABE_Sotor_exp1_reads, H23_ABE_Sotor_exp2_reads))
raw_reads <- mutate(raw_reads, H23_ABE_Adag_average_reads = average(H23_ABE_Adag_exp1_reads, H23_ABE_Adag_exp2_reads))


# PC9 average
#BE3.9max NGN
raw_reads <- mutate(raw_reads, PC9_CBE_T0_average_reads = average(PC9_CBE_T0_exp1_reads, PC9_CBE_T0_exp2_reads))
raw_reads <- mutate(raw_reads, PC9_CBE_Control_average_reads = average(PC9_CBE_Control_exp1_reads, PC9_CBE_Control_exp2_reads))
raw_reads <- mutate(raw_reads, PC9_CBE_Gefit_average_reads = average(PC9_CBE_Gefit_exp1_reads, PC9_CBE_Gefit_exp2_reads))
raw_reads <- mutate(raw_reads, PC9_CBE_Osim_average_reads = average(PC9_CBE_Osim_exp1_reads, PC9_CBE_Osim_exp2_reads))

#ABE8e NGN
raw_reads <- mutate(raw_reads, PC9_ABE_T0_average_reads = average(PC9_ABE_T0_exp1_reads, PC9_ABE_T0_exp2_reads))
raw_reads <- mutate(raw_reads, PC9_ABE_Control_average_reads = average(PC9_ABE_Control_exp1_reads, PC9_ABE_Control_exp2_reads))
raw_reads <- mutate(raw_reads, PC9_ABE_Gefit_average_reads = average(PC9_ABE_Gefit_exp1_reads, PC9_ABE_Gefit_exp2_reads))
raw_reads <- mutate(raw_reads, PC9_ABE_Osim_average_reads = average(PC9_ABE_Osim_exp1_reads, PC9_ABE_Osim_exp2_reads))

# MHH-ES-1 average

#ABE8e NGN
raw_reads <- mutate(raw_reads, MHHES1_ABE_T0_average_reads = average(MHHES1_ABE_T0_exp1_reads, MHHES1_ABE_T0_exp2_reads))
raw_reads <- mutate(raw_reads, MHHES1_ABE_Control_average_reads = average(MHHES1_ABE_Control_exp1_reads, MHHES1_ABE_Control_exp2_reads))
raw_reads <- mutate(raw_reads, MHHES1_ABE_Olap_average_reads = average(MHHES1_ABE_Olap_exp1_reads, MHHES1_ABE_Olap_exp2_reads))
raw_reads <- mutate(raw_reads, MHHES1_ABE_Nirap_average_reads = average(MHHES1_ABE_Nirap_exp1_reads, MHHES1_ABE_Nirap_exp2_reads))
```

```{r}
#compute RPM values
raw_reads <- mutate_at(raw_reads, vars(matches("reads")), list(RPM = RPM))
raw_reads
```

```{r}
# compute L2FC from normalised gRNA counts - single experiments
# HT-29 exp 1
#BE3.9max NGN
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_Control_plasmid_1 = L2FC(HT29_CBE_Control_exp1_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_Control_T0_1 = L2FC(HT29_CBE_Control_exp1_reads_RPM, HT29_CBE_T0_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_Tram_Control_1 = L2FC(HT29_CBE_Tram_exp1_reads_RPM, HT29_CBE_Control_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_Pict_Control_1 = L2FC(HT29_CBE_Pict_exp1_reads_RPM, HT29_CBE_Control_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_DebCet_Control_1 = L2FC(HT29_CBE_DebCet_exp1_reads_RPM, HT29_CBE_Control_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_Tram_plasmid_1 = L2FC(HT29_CBE_Tram_exp1_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_Pict_plasmid_1 = L2FC(HT29_CBE_Pict_exp1_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_DebCet_plasmid_1 = L2FC(HT29_CBE_DebCet_exp1_reads_RPM, plasmid_reads_RPM))

#ABE8e NGN
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_Control_plasmid_1 = L2FC(HT29_ABE_Control_exp1_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_Control_T0_1 = L2FC(HT29_ABE_Control_exp1_reads_RPM, HT29_ABE_T0_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_Tram_Control_1 = L2FC(HT29_ABE_Tram_exp1_reads_RPM, HT29_ABE_Control_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_Pict_Control_1 = L2FC(HT29_ABE_Pict_exp1_reads_RPM, HT29_ABE_Control_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_DebCet_Control_1 = L2FC(HT29_ABE_DebCet_exp1_reads_RPM, HT29_ABE_Control_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_Tram_plasmid_1 = L2FC(HT29_ABE_Tram_exp1_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_Pict_plasmid_1 = L2FC(HT29_ABE_Pict_exp1_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_DebCet_plasmid_1 = L2FC(HT29_ABE_DebCet_exp1_reads_RPM, plasmid_reads_RPM))

# HT-29 exp 2
#BE3.9max NGN
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_Control_plasmid_2 = L2FC(HT29_CBE_Control_exp2_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_Control_T0_2 = L2FC(HT29_CBE_Control_exp2_reads_RPM, HT29_CBE_T0_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_Tram_Control_2 = L2FC(HT29_CBE_Tram_exp2_reads_RPM, HT29_CBE_Control_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_Pict_Control_2 = L2FC(HT29_CBE_Pict_exp2_reads_RPM, HT29_CBE_Control_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_DebCet_Control_2 = L2FC(HT29_CBE_DebCet_exp2_reads_RPM, HT29_CBE_Control_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_Tram_plasmid_2 = L2FC(HT29_CBE_Tram_exp2_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_Pict_plasmid_2 = L2FC(HT29_CBE_Pict_exp2_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_DebCet_plasmid_2 = L2FC(HT29_CBE_DebCet_exp2_reads_RPM, plasmid_reads_RPM))

#ABE8e NGN
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_Control_plasmid_2 = L2FC(HT29_ABE_Control_exp2_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_Control_T0_2 = L2FC(HT29_ABE_Control_exp2_reads_RPM, HT29_ABE_T0_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_Tram_Control_2 = L2FC(HT29_ABE_Tram_exp2_reads_RPM, HT29_ABE_Control_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_Pict_Control_2 = L2FC(HT29_ABE_Pict_exp2_reads_RPM, HT29_ABE_Control_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_DebCet_Control_2 = L2FC(HT29_ABE_DebCet_exp2_reads_RPM, HT29_ABE_Control_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_Tram_plasmid_2 = L2FC(HT29_ABE_Tram_exp2_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_Pict_plasmid_2 = L2FC(HT29_ABE_Pict_exp2_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_DebCet_plasmid_2 = L2FC(HT29_ABE_DebCet_exp2_reads_RPM, plasmid_reads_RPM))


# H23 exp 1
#BE3.9max NGN
raw_reads <- mutate(raw_reads, L2FC_H23_CBE_Control_plasmid_1 = L2FC(H23_CBE_Control_exp1_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_CBE_Control_T0_1 = L2FC(H23_CBE_Control_exp1_reads_RPM, H23_CBE_T0_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_CBE_Sotor_Control_1 = L2FC(H23_CBE_Sotor_exp1_reads_RPM, H23_CBE_Control_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_CBE_Adag_Control_1 = L2FC(H23_CBE_Adag_exp1_reads_RPM, H23_CBE_Control_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_CBE_Sotor_plasmid_1 = L2FC(H23_CBE_Sotor_exp1_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_CBE_Adag_plasmid_1 = L2FC(H23_CBE_Adag_exp1_reads_RPM, plasmid_reads_RPM))

#ABE8e NGN
raw_reads <- mutate(raw_reads, L2FC_H23_ABE_Control_plasmid_1 = L2FC(H23_ABE_Control_exp1_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_ABE_Control_T0_1 = L2FC(H23_ABE_Control_exp1_reads_RPM, H23_ABE_T0_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_ABE_Sotor_Control_1 = L2FC(H23_ABE_Sotor_exp1_reads_RPM, H23_ABE_Control_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_ABE_Adag_Control_1 = L2FC(H23_ABE_Adag_exp1_reads_RPM, H23_ABE_Control_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_ABE_Sotor_plasmid_1 = L2FC(H23_ABE_Sotor_exp1_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_ABE_Adag_plasmid_1 = L2FC(H23_ABE_Adag_exp1_reads_RPM, plasmid_reads_RPM))

# H23 exp 2
#BE3.9max NGN
raw_reads <- mutate(raw_reads, L2FC_H23_CBE_Control_plasmid_2 = L2FC(H23_CBE_Control_exp2_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_CBE_Control_T0_2 = L2FC(H23_CBE_Control_exp2_reads_RPM, H23_CBE_T0_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_CBE_Sotor_Control_2 = L2FC(H23_CBE_Sotor_exp2_reads_RPM, H23_CBE_Control_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_CBE_Adag_Control_2 = L2FC(H23_CBE_Adag_exp2_reads_RPM, H23_CBE_Control_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_CBE_Sotor_plasmid_2 = L2FC(H23_CBE_Sotor_exp2_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_CBE_Adag_plasmid_2 = L2FC(H23_CBE_Adag_exp2_reads_RPM, plasmid_reads_RPM))

#ABE8e NGN
raw_reads <- mutate(raw_reads, L2FC_H23_ABE_Control_plasmid_2 = L2FC(H23_ABE_Control_exp2_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_ABE_Control_T0_2 = L2FC(H23_ABE_Control_exp2_reads_RPM, H23_ABE_T0_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_ABE_Sotor_Control_2 = L2FC(H23_ABE_Sotor_exp2_reads_RPM, H23_ABE_Control_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_ABE_Adag_Control_2 = L2FC(H23_ABE_Adag_exp2_reads_RPM, H23_ABE_Control_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_ABE_Sotor_plasmid_2 = L2FC(H23_ABE_Sotor_exp2_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_ABE_Adag_plasmid_2 = L2FC(H23_ABE_Adag_exp2_reads_RPM, plasmid_reads_RPM))

# PC9 exp 1
#BE3.9max NGN
raw_reads <- mutate(raw_reads, L2FC_PC9_CBE_Control_plasmid_1 = L2FC(PC9_CBE_Control_exp1_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_CBE_Control_T0_1 = L2FC(PC9_CBE_Control_exp1_reads_RPM, PC9_CBE_T0_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_CBE_Gefit_Control_1 = L2FC(PC9_CBE_Gefit_exp1_reads_RPM, PC9_CBE_Control_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_CBE_Osim_Control_1 = L2FC(PC9_CBE_Osim_exp1_reads_RPM, PC9_CBE_Control_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_CBE_Gefit_plasmid_1 = L2FC(PC9_CBE_Gefit_exp1_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_CBE_Osim_plasmid_1 = L2FC(PC9_CBE_Osim_exp1_reads_RPM, plasmid_reads_RPM))

#ABE8e NGN
raw_reads <- mutate(raw_reads, L2FC_PC9_ABE_Control_plasmid_1 = L2FC(PC9_ABE_Control_exp1_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_ABE_Control_T0_1 = L2FC(PC9_ABE_Control_exp1_reads_RPM, PC9_ABE_T0_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_ABE_Gefit_Control_1 = L2FC(PC9_ABE_Gefit_exp1_reads_RPM, PC9_ABE_Control_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_ABE_Osim_Control_1 = L2FC(PC9_ABE_Osim_exp1_reads_RPM, PC9_ABE_Control_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_ABE_Gefit_plasmid_1 = L2FC(PC9_ABE_Gefit_exp1_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_ABE_Osim_plasmid_1 = L2FC(PC9_ABE_Osim_exp1_reads_RPM, plasmid_reads_RPM))

# PC9 exp 2
#BE3.9max NGN
raw_reads <- mutate(raw_reads, L2FC_PC9_CBE_Control_plasmid_2 = L2FC(PC9_CBE_Control_exp2_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_CBE_Control_T0_2 = L2FC(PC9_CBE_Control_exp2_reads_RPM, PC9_CBE_T0_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_CBE_Gefit_Control_2 = L2FC(PC9_CBE_Gefit_exp2_reads_RPM, PC9_CBE_Control_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_CBE_Osim_Control_2 = L2FC(PC9_CBE_Osim_exp2_reads_RPM, PC9_CBE_Control_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_CBE_Gefit_plasmid_2 = L2FC(PC9_CBE_Gefit_exp2_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_CBE_Osim_plasmid_2 = L2FC(PC9_CBE_Osim_exp2_reads_RPM, plasmid_reads_RPM))

#ABE8e NGN
raw_reads <- mutate(raw_reads, L2FC_PC9_ABE_Control_plasmid_2 = L2FC(PC9_ABE_Control_exp2_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_ABE_Control_T0_2 = L2FC(PC9_ABE_Control_exp2_reads_RPM, PC9_ABE_T0_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_ABE_Gefit_Control_2 = L2FC(PC9_ABE_Gefit_exp2_reads_RPM, PC9_ABE_Control_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_ABE_Osim_Control_2 = L2FC(PC9_ABE_Osim_exp2_reads_RPM, PC9_ABE_Control_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_ABE_Gefit_plasmid_2 = L2FC(PC9_ABE_Gefit_exp2_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_ABE_Osim_plasmid_2 = L2FC(PC9_ABE_Osim_exp2_reads_RPM, plasmid_reads_RPM))

# PC9 exp 3
#BE3.9max NGN
raw_reads <- mutate(raw_reads, L2FC_PC9_CBE_Gefit_plasmid_3 = L2FC(PC9_CBE_Gefit_exp3_reads_RPM, plasmid_reads_RPM))

# MHH-ES-1 exp 1
#ABE8e NGN
raw_reads <- mutate(raw_reads, L2FC_MHHES1_ABE_Control_plasmid_1 = L2FC(MHHES1_ABE_Control_exp1_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_MHHES1_ABE_Control_T0_1 = L2FC(MHHES1_ABE_Control_exp1_reads_RPM, MHHES1_ABE_T0_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_MHHES1_ABE_Olap_Control_1 = L2FC(MHHES1_ABE_Olap_exp1_reads_RPM, MHHES1_ABE_Control_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_MHHES1_ABE_Nirap_Control_1 = L2FC(MHHES1_ABE_Nirap_exp1_reads_RPM, MHHES1_ABE_Control_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_MHHES1_ABE_Olap_plasmid_1 = L2FC(MHHES1_ABE_Olap_exp1_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_MHHES1_ABE_Nirap_plasmid_1 = L2FC(MHHES1_ABE_Nirap_exp1_reads_RPM, plasmid_reads_RPM))

# MHH-ES-1 exp 2
#ABE8e NGN
raw_reads <- mutate(raw_reads, L2FC_MHHES1_ABE_Control_plasmid_2 = L2FC(MHHES1_ABE_Control_exp2_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_MHHES1_ABE_Control_T0_2 = L2FC(MHHES1_ABE_Control_exp2_reads_RPM, MHHES1_ABE_T0_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_MHHES1_ABE_Olap_Control_2 = L2FC(MHHES1_ABE_Olap_exp2_reads_RPM, MHHES1_ABE_Control_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_MHHES1_ABE_Nirap_Control_2 = L2FC(MHHES1_ABE_Nirap_exp2_reads_RPM, MHHES1_ABE_Control_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_MHHES1_ABE_Olap_plasmid_2 = L2FC(MHHES1_ABE_Olap_exp2_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_MHHES1_ABE_Nirap_plasmid_2 = L2FC(MHHES1_ABE_Nirap_exp2_reads_RPM, plasmid_reads_RPM))

```

```{r}
#compute L2FC from average of 2 exps from RPM values
# HT-29 average
#BE3.9max NGN
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_Control_plasmid_average = L2FC(HT29_CBE_Control_average_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_Control_T0_average = L2FC(HT29_CBE_Control_average_reads_RPM, HT29_CBE_T0_average_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_Tram_Control_average = L2FC(HT29_CBE_Tram_average_reads_RPM, HT29_CBE_Control_average_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_Pict_Control_average = L2FC(HT29_CBE_Pict_average_reads_RPM, HT29_CBE_Control_average_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_DebCet_Control_average = L2FC(HT29_CBE_DebCet_average_reads_RPM, HT29_CBE_Control_average_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_Tram_plasmid_average = L2FC(HT29_CBE_Tram_average_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_Pict_plasmid_average = L2FC(HT29_CBE_Pict_average_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_DebCet_plasmid_average = L2FC(HT29_CBE_DebCet_average_reads_RPM, plasmid_reads_RPM))

#ABE8e NGN
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_Control_plasmid_average = L2FC(HT29_ABE_Control_average_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_Control_T0_average = L2FC(HT29_ABE_Control_average_reads_RPM, HT29_ABE_T0_average_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_Tram_Control_average = L2FC(HT29_ABE_Tram_average_reads_RPM, HT29_ABE_Control_average_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_Pict_Control_average = L2FC(HT29_ABE_Pict_average_reads_RPM, HT29_ABE_Control_average_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_DebCet_Control_average = L2FC(HT29_ABE_DebCet_average_reads_RPM, HT29_ABE_Control_average_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_Tram_plasmid_average = L2FC(HT29_ABE_Tram_average_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_Pict_plasmid_average = L2FC(HT29_ABE_Pict_average_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_DebCet_plasmid_average = L2FC(HT29_ABE_DebCet_average_reads_RPM, plasmid_reads_RPM))

# H23 average
#BE3.9max NGN
raw_reads <- mutate(raw_reads, L2FC_H23_CBE_Control_plasmid_average = L2FC(H23_CBE_Control_average_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_CBE_Control_T0_average = L2FC(H23_CBE_Control_average_reads_RPM, H23_CBE_T0_average_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_CBE_Sotor_Control_average = L2FC(H23_CBE_Sotor_average_reads_RPM, H23_CBE_Control_average_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_CBE_Adag_Control_average = L2FC(H23_CBE_Adag_average_reads_RPM, H23_CBE_Control_average_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_CBE_Sotor_plasmid_average = L2FC(H23_CBE_Sotor_average_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_CBE_Adag_plasmid_average = L2FC(H23_CBE_Adag_average_reads_RPM, plasmid_reads_RPM))

#ABE8e NGN
raw_reads <- mutate(raw_reads, L2FC_H23_ABE_Control_plasmid_average = L2FC(H23_ABE_Control_average_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_ABE_Control_T0_average = L2FC(H23_ABE_Control_average_reads_RPM, H23_ABE_T0_average_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_ABE_Sotor_Control_average = L2FC(H23_ABE_Sotor_average_reads_RPM, H23_ABE_Control_average_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_ABE_Adag_Control_average = L2FC(H23_ABE_Adag_average_reads_RPM, H23_ABE_Control_average_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_ABE_Sotor_plasmid_average = L2FC(H23_ABE_Sotor_average_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_ABE_Adag_plasmid_average = L2FC(H23_ABE_Adag_average_reads_RPM, plasmid_reads_RPM))

# PC9 average
#BE3.9max NGN
raw_reads <- mutate(raw_reads, L2FC_PC9_CBE_Control_plasmid_average = L2FC(PC9_CBE_Control_average_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_CBE_Control_T0_average = L2FC(PC9_CBE_Control_average_reads_RPM, PC9_CBE_T0_average_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_CBE_Gefit_Control_average = L2FC(PC9_CBE_Gefit_average_reads_RPM, PC9_CBE_Control_average_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_CBE_Osim_Control_average = L2FC(PC9_CBE_Osim_average_reads_RPM, PC9_CBE_Control_average_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_CBE_Gefit_plasmid_average = L2FC(PC9_CBE_Gefit_average_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_CBE_Osim_plasmid_average = L2FC(PC9_CBE_Osim_average_reads_RPM, plasmid_reads_RPM))

#ABE8e NGN
raw_reads <- mutate(raw_reads, L2FC_PC9_ABE_Control_plasmid_average = L2FC(PC9_ABE_Control_average_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_ABE_Control_T0_average = L2FC(PC9_ABE_Control_average_reads_RPM, PC9_ABE_T0_average_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_ABE_Gefit_Control_average = L2FC(PC9_ABE_Gefit_average_reads_RPM, PC9_ABE_Control_average_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_ABE_Osim_Control_average = L2FC(PC9_ABE_Osim_average_reads_RPM, PC9_ABE_Control_average_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_ABE_Gefit_plasmid_average = L2FC(PC9_ABE_Gefit_average_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_ABE_Osim_plasmid_average = L2FC(PC9_ABE_Osim_average_reads_RPM, plasmid_reads_RPM))

# MHH-ES-1 average

#ABE8e NGN
raw_reads <- mutate(raw_reads, L2FC_MHHES1_ABE_Control_plasmid_average = L2FC(MHHES1_ABE_Control_average_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_MHHES1_ABE_Control_T0_average = L2FC(MHHES1_ABE_Control_average_reads_RPM, MHHES1_ABE_T0_average_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_MHHES1_ABE_Olap_Control_average = L2FC(MHHES1_ABE_Olap_average_reads_RPM, MHHES1_ABE_Control_average_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_MHHES1_ABE_Nirap_Control_average = L2FC(MHHES1_ABE_Nirap_average_reads_RPM, MHHES1_ABE_Control_average_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_MHHES1_ABE_Olap_plasmid_average = L2FC(MHHES1_ABE_Olap_average_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_MHHES1_ABE_Nirap_plasmid_average = L2FC(MHHES1_ABE_Nirap_average_reads_RPM, plasmid_reads_RPM))
```

```{r}
# from L2FC, define z-scores
# define function
zscore <- function(x) ((x-mean(x))/sd(x))
raw_reads <- mutate_at(raw_reads, vars(matches("L2FC")), list(zscore = zscore))
```

```{r}
# left join gRNA library and read counts
lib <- mutate(lib, Gene = ifelse(sgRNA_type == "intergenic", "N/A", Gene))
lib %>% filter(sgRNA_type =="intergenic") %>% select(Gene, sgRNA_type)
data <- left_join(lib, raw_reads)
data %>% count(sgRNA_type)
```

```{r}
#HT-29
# off-targets pre-filter
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

# CBE control vs T0
plot1 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_HT29_CBE_Control_plasmid_average_zscore, y= perfect_matches, colour=Gene)) + geom_point(size=3, max.overlaps = Inf) +
labs(title= "HT-29 CBE: exonic gRNAs", x = "z-score - control vs plasmid", y = "number of off-targets", colour = "gene") +
scale_color_manual( values =c("royalblue", "turquoise3", "orange", "red", "springgreen4", "grey55", "brown", "purple", "salmon", "pink", "black")) +
theme_classic()
plot1
ggsave("FigS1a.eps", width = 6, height = 4, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")


# ABE control vs T0
plot2 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_HT29_ABE_Control_plasmid_average_zscore, y= perfect_matches, colour=Gene)) + geom_point(size=3,max.overlaps = Inf) +
labs(title= "HT-29 ABE: exonic gRNAs", x = "zscore - control vs plasmid", y = "number of off-targets", colour = "gene") +
scale_color_manual( values =c("royalblue", "turquoise3", "orange", "red", "springgreen4", "grey55", "brown", "purple", "salmon", "pink", "black")) +
theme_classic()
plot2
ggsave("FigS1b.eps", width = 6, height = 4, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")
```

```{r}
#filter out more than 2 perfect mismatches
data %>% filter(perfect_matches >2 | is.na(perfect_matches))
data %>% filter(perfect_matches >500 | is.na(perfect_matches))
data <- data %>% filter(perfect_matches < 3 | is.na(perfect_matches))
```

```{r}
# quality control plots
# HT-29
plot1 <- ggplot(data, aes(y= plasmid_reads)) + 
geom_histogram() +
labs(title= "plasmid Reads")
plot1

plot2 <- ggplot(data, aes(y= HT29_CBE_T0_exp1_reads)) + 
geom_histogram() +
labs(title= "HT29 CBE T0 reads")
plot2

plot3 <- ggplot(data, aes(y= HT29_ABE_T0_exp1_reads)) + 
geom_histogram() +
labs(title= "HT29 ABE T0 reads")
plot3

plot4 <- ggplot(data, aes(x= L2FC_HT29_CBE_Control_T0_1_zscore, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "zscore_CBE_Control_T0_1") +
theme_classic()
plot4

plot5 <- ggplot(data, aes(x= L2FC_HT29_ABE_Control_T0_1_zscore, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "zscore_ABE_Control_T0_1") +
theme_classic()
plot5

plot6 <- ggplot(data, aes(x= L2FC_HT29_CBE_Control_plasmid_1_zscore, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "zscore_CBE_Control_plasmid_1") +
theme_classic()
plot6

plot7 <- ggplot(data, aes(x= L2FC_HT29_ABE_Control_plasmid_1_zscore, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "zscore_ABE_Control_plasmid_1") +
theme_classic()
plot7

plot8 <- ggplot(data, aes(x= L2FC_HT29_CBE_Control_plasmid_2_zscore, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "zscore_CBE_Control_plasmid_2") +
theme_classic()
plot8

plot9 <- ggplot(data, aes(x= L2FC_HT29_ABE_Control_plasmid_2_zscore, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "zscore_ABE_Control_plasmid_2") +
theme_classic()
plot9
```

```{r}
# quality control plots
# H23
plot1 <- ggplot(data, aes(y= H23_CBE_T0_exp1_reads)) + 
geom_histogram() +
labs(title= "H23 CBE T0 reads")
plot1

plot2 <- ggplot(data, aes(y= H23_ABE_T0_exp1_reads)) + 
geom_histogram() +
labs(title= "H23 ABE T0 reads")
plot2

plot3 <- ggplot(data, aes(x= L2FC_H23_CBE_Control_T0_1_zscore, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "zscore_CBE_Control_T0_1") +
theme_classic()
plot3

plot4 <- ggplot(data, aes(x= L2FC_H23_ABE_Control_T0_1_zscore, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "zscore_ABE_Control_T0_1") +
theme_classic()
plot4

plot5 <- ggplot(data, aes(x= L2FC_H23_CBE_Control_plasmid_1_zscore, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "zscore_CBE_Control_plasmid_1") +
theme_classic()
plot5

plot6 <- ggplot(data, aes(x= L2FC_H23_ABE_Control_plasmid_1_zscore, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "zscore_ABE_Control_plasmid_1") +
theme_classic()
plot6

plot7 <- ggplot(data, aes(x= L2FC_H23_CBE_Control_plasmid_2_zscore, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "zscore_CBE_Control_plasmid_2") +
theme_classic()
plot7

plot8 <- ggplot(data, aes(x= L2FC_H23_ABE_Control_plasmid_2_zscore, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "zscore_ABE_Control_plasmid_2") +
theme_classic()
plot8
```

```{r}
# quality control plots
# PC9
plot1 <- ggplot(data, aes(y= PC9_CBE_T0_exp1_reads)) + 
geom_histogram() +
labs(title= "PC9 CBE T0 reads")
plot1

plot2 <- ggplot(data, aes(y= PC9_ABE_T0_exp1_reads)) + 
geom_histogram() +
labs(title= "PC9 ABE T0 reads")
plot2

plot3 <- ggplot(data, aes(x= L2FC_PC9_CBE_Control_T0_1_zscore, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "zscore_CBE_Control_T0_1") +
theme_classic()
plot3

plot4 <- ggplot(data, aes(x= L2FC_PC9_ABE_Control_T0_1_zscore, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "zscore_ABE_Control_T0_1") +
theme_classic()
plot4

plot5 <- ggplot(data, aes(x= L2FC_PC9_CBE_Control_plasmid_1_zscore, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "zscore_CBE_Control_plasmid_1") +
theme_classic()
plot5

plot6 <- ggplot(data, aes(x= L2FC_PC9_ABE_Control_plasmid_1_zscore, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "zscore_ABE_Control_plasmid_1") +
theme_classic()
plot6

plot8 <- ggplot(data, aes(x= L2FC_PC9_CBE_Control_plasmid_2_zscore, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "zscore_CBE_Control_plasmid_2") +
theme_classic()
plot8

plot9 <- ggplot(data, aes(x= L2FC_PC9_ABE_Control_plasmid_2_zscore, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "zscore_ABE_Control_plasmid_2") +
theme_classic()
plot9
```

```{r}
# quality control plots
# MHH-ES-1
plot1 <- ggplot(data, aes(y= MHHES1_ABE_T0_exp1_reads)) + 
geom_histogram() +
labs(title= "MHHES1 ABE T0 reads")
plot1

plot2 <- ggplot(data, aes(x= L2FC_MHHES1_ABE_Control_T0_1_zscore, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "zscore_ABE_Control_T0_1") +
theme_classic()
plot2

plot3 <- ggplot(data, aes(x= L2FC_MHHES1_ABE_Control_plasmid_1_zscore, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "zscore_ABE_Control_plasmid_1") +
theme_classic()
plot3

plot4 <- ggplot(data, aes(x= L2FC_MHHES1_ABE_Control_plasmid_2_zscore, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "zscore_ABE_Control_plasmid_2") +
theme_classic()
plot4
```

```{r}
# check for low read numbers and filter out <100 reads
data %>% filter(plasmid_reads <100)
data %>% filter(HT29_CBE_T0_exp1_reads <100)
data %>% filter(H23_CBE_T0_exp1_reads <100)
data %>% filter(PC9_CBE_T0_exp1_reads <100)

data %>% filter(HT29_ABE_T0_exp1_reads <100)
data %>% filter(H23_ABE_T0_exp1_reads <100)
data %>% filter(PC9_ABE_T0_exp1_reads <100)
data %>% filter(MHHES1_ABE_T0_exp1_reads <100)

data %>% filter(plasmid_reads <100 | HT29_CBE_T0_exp1_reads <100 | H23_CBE_T0_exp1_reads <100 | PC9_CBE_T0_exp1_reads <100 | HT29_ABE_T0_exp1_reads <100 | H23_ABE_T0_exp1_reads <100 | PC9_ABE_T0_exp1_reads <100 |MHHES1_ABE_T0_exp1_reads <100)

data <- filter(data, plasmid_reads >100 & HT29_CBE_T0_exp1_reads >100 & H23_CBE_T0_exp1_reads >100 & PC9_CBE_T0_exp1_reads >100 & HT29_ABE_T0_exp1_reads >100 & H23_ABE_T0_exp1_reads >100 & PC9_ABE_T0_exp1_reads >100 & MHHES1_ABE_T0_exp1_reads >100)
```

```{r}
#HT-29
# check L2FC PLASMID vs Gene
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

# CBE control vs plasmid
plot1 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_HT29_CBE_Control_plasmid_1_zscore)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "L2FC_CBE_plasmid_T0_1_zscore") +
theme_classic()
plot1

# CBE Tram vs plasmid
plot2 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_HT29_CBE_Tram_plasmid_1_zscore)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "L2FC_CBE_Tram_plasmid_1_zscore") +
theme_classic()
plot2

# CBE Pict vs plasmid
plot3 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_HT29_CBE_Pict_plasmid_1_zscore)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "L2FC_CBE_Pict_plasmid_1_zscore") +
theme_classic()
plot3

# CBE DebCet vs plasmid
plot4 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_HT29_CBE_DebCet_plasmid_1_zscore)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "L2FC_CBE_DebCet_plasmid_1_zscore") +
theme_classic()
plot4


# ABE plasmid vs T0
plot5 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_HT29_ABE_Control_plasmid_1_zscore)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "L2FC_ABE_plasmid_T0_1_zscore") +
theme_classic()
plot5

# ABE Tram vs plasmid
plot6 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_HT29_ABE_Tram_plasmid_1_zscore)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "L2FC_ABE_Tram_plasmid_1_zscore") +
theme_classic()
plot6

# ABE Pict vs plasmid
plot7 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_HT29_ABE_Pict_plasmid_1_zscore)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "L2FC_ABE_Pict_plasmid_1_zscore") +
theme_classic()
plot7

# ABE DebCet vs plasmid
plot8 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_HT29_ABE_DebCet_plasmid_1_zscore)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "L2FC_ABE_DebCet_plasmid_1_zscore") +
theme_classic()
plot8
```

```{r}
# replicate correlation - stats HT-29
r1 <- cor(data$L2FC_HT29_CBE_Control_plasmid_1_zscore, data$L2FC_HT29_CBE_Control_plasmid_2, method = "pearson")
r2 <- cor(data$L2FC_HT29_CBE_Tram_plasmid_1_zscore, data$L2FC_HT29_CBE_Tram_plasmid_2_zscore, method = "pearson")
r3 <- cor(data$L2FC_HT29_CBE_Pict_plasmid_1_zscore, data$L2FC_HT29_CBE_Pict_plasmid_2_zscore, method = "pearson")
r4 <- cor(data$L2FC_HT29_CBE_DebCet_plasmid_1_zscore, data$L2FC_HT29_CBE_DebCet_plasmid_2_zscore, method = "pearson")
r5 <- cor(data$L2FC_HT29_ABE_Control_plasmid_1_zscore, data$L2FC_HT29_ABE_Control_plasmid_2_zscore, method = "pearson")
r6 <- cor(data$L2FC_HT29_ABE_Tram_plasmid_1_zscore, data$L2FC_HT29_ABE_Tram_plasmid_2_zscore, method = "pearson")
r7 <- cor(data$L2FC_HT29_ABE_Pict_plasmid_1_zscore, data$L2FC_HT29_ABE_Pict_plasmid_2_zscore, method = "pearson")
r8 <- cor(data$L2FC_HT29_ABE_DebCet_plasmid_1_zscore, data$L2FC_HT29_ABE_DebCet_plasmid_2_zscore, method = "pearson")
```

```{r}
#HT-29 check correlation between replicates
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

# CBE control vs plasmid
plot1 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_HT29_CBE_Control_plasmid_1_zscore, y= L2FC_HT29_CBE_Control_plasmid_2_zscore)) + 
geom_pointdensity(size=3,  position = pos) +
labs(title= "HT-29 - replicate correlation", x = "zscore CBE control vs plasmid exp.1", y = "zscore CBE control vs plasmid exp.2", colour = "density") +
geom_smooth(method = lm) +
annotate("text", y = 5, x = 0, label = paste("r = ", round(r1, 2)), parse = FALSE, size = 7) +
theme_bw(base_size = 15)
plot1
ggsave("FigS2_1.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

# CBE Tram vs plasmid
plot2 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_HT29_CBE_Tram_plasmid_1_zscore, y= L2FC_HT29_CBE_Tram_plasmid_2_zscore)) + geom_pointdensity(size=3,  position = pos) +
labs(title= "HT-29 - replicate correlation", x = "zscore CBE Trametinib vs plasmid exp.1", y = "zscore CBE Trametinib vs plasmid exp.2", colour = "density") +
geom_smooth(method = lm) +
annotate("text", y = 5, x = 0, label = paste("r = ", round(r2, 2)), parse = FALSE, size = 7) +
theme_bw(base_size = 15)
plot2
ggsave("FigS2_2.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

# CBE Pict vs plasmid
plot3 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_HT29_CBE_Pict_plasmid_1_zscore, y= L2FC_HT29_CBE_Pict_plasmid_2_zscore)) + geom_pointdensity(size=3,  position = pos) +
labs(title= "HT-29 - replicate correlation", x = "zscore CBE Pictilisib vs plasmid exp.1", y = "zscore CBE Pictilisib vs plasmid exp.2", colour = "density") +
geom_smooth(method = lm) +
annotate("text", y = 5, x = 0, label = paste("r = ", round(r3, 2)), parse = FALSE, size = 7) +
theme_bw(base_size = 15)
plot3
ggsave("FigS2_3.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

# CBE DebCet vs plasmid
plot4 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_HT29_CBE_DebCet_plasmid_1_zscore, y= L2FC_HT29_CBE_DebCet_plasmid_2_zscore)) + 
geom_pointdensity(size=3,  position = pos) +
labs(title= "HT-29 - replicate correlation", x = "zscore CBE Dabrafenib+Cetuximab vs plasmid exp.1", y = "zscore CBE Dabrafenib+Cetuximab vs plasmid exp.2", colour = "density") +
geom_smooth(method = lm) +
annotate("text", y = 5, x = 0, label = paste("r = ", round(r4, 2)), parse = FALSE, size = 7) +
theme_bw(base_size = 15)
plot4
ggsave("FigS2_4.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

# ABE control vs plasmid
plot5 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_HT29_ABE_Control_plasmid_1_zscore, y= L2FC_HT29_ABE_Control_plasmid_2_zscore)) +
geom_pointdensity(size=3,  position = pos) +
scale_color_viridis_c() +
labs(title= "HT-29 - replicate correlation", x = "zscore ABE control vs plasmid exp.1", y = "zscore ABE control vs plasmid exp.2", colour = "density") +
geom_smooth(method = lm) +
annotate("text", y = 5, x = 0, label = paste("r = ", round(r5, 2)), parse = FALSE, size = 7) +
theme_bw(base_size = 15)
plot5
ggsave("FigS2_5.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

# ABE Tram vs plasmid
plot6 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_HT29_ABE_Tram_plasmid_1_zscore, y= L2FC_HT29_ABE_Tram_plasmid_2_zscore)) + geom_pointdensity(size=3,  position = pos) +
scale_color_viridis_c() +
labs(title= "HT-29 - replicate correlation", x = "zscore ABE Trametinib vs plasmid exp.1", y = "zscore ABE Trametinib vs plasmid exp.2", colour = "density") +
geom_smooth(method = lm) +
annotate("text", y = 5, x = 0, label = paste("r = ", round(r6, 2)), parse = FALSE, size = 7) +
theme_bw(base_size = 15)
plot6
ggsave("FigS2_6.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

# ABE Pict vs plasmid
plot7 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_HT29_ABE_Pict_plasmid_1_zscore, y= L2FC_HT29_ABE_Pict_plasmid_2_zscore)) + geom_pointdensity(size=3,  position = pos) +
scale_color_viridis_c() +
labs(title= "HT-29 - replicate correlation", x = "zscore ABE Pictilisb vs plasmid exp.1", y = "zscore ABE Pictilisb vs plasmid exp.2", colour = "density") +
geom_smooth(method = lm) +
annotate("text", y = 5, x = 0, label = paste("r = ", round(r7, 2)), parse = FALSE, size = 7) +
theme_bw(base_size = 15)
plot7
ggsave("FigS2_7.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

# ABE DebCet vs plasmid
plot8 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_HT29_ABE_DebCet_plasmid_1_zscore, y= L2FC_HT29_ABE_DebCet_plasmid_2_zscore)) + 
geom_pointdensity(size=3,  position = pos) +
scale_color_viridis_c() +
labs(title= "HT-29 - replicate correlation", x = "zscore ABE Dabrafenib+Cetuximab vs plasmid exp.1", y = "zscore ABE Dabrafenib+Cetuximab vs plasmid exp.2", colour = "density") +
geom_smooth(method = lm) +
annotate("text", y = 7, x = 0, label = paste("r = ", round(r8, 2)), parse = FALSE, size = 7) +
theme_bw(base_size = 15)
plot8
ggsave("FigS2_8.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")
```

```{r}
#H23
# check L2FC PLASMID vs Gene
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

# CBE control vs plasmid
plot1 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_H23_CBE_Control_plasmid_1_zscore)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "zscore_CBE_control_plasmid_1") +
theme_classic()
plot1

# CBE Sotor vs plasmid
plot2 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_H23_CBE_Sotor_plasmid_1_zscore)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "zscore_CBE_Sotor_plasmid_1") +
theme_classic()
plot2

# CBE Adag vs plasmid
plot3 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_H23_CBE_Adag_plasmid_1_zscore)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "zscore_CBE_Adag_plasmid_1") +
theme_classic()
plot3


# ABE plasmid vs T0
plot5 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_H23_ABE_Control_plasmid_1_zscore)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "zscore_ABE_plasmid_T0_1") +
theme_classic()
plot5

# ABE Sotor vs plasmid
plot6 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_H23_ABE_Sotor_plasmid_1_zscore)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "zscore_ABE_Sotor_plasmid_1") +
theme_classic()
plot6

# ABE Adag vs plasmid
plot7 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_H23_ABE_Adag_plasmid_1_zscore)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "zscore_ABE_Adag_plasmid_1") +
scale_color_continuous() +
theme_classic()
plot7
```

```{r}
# replicate correlation - stats H23
r9 <- cor(data$L2FC_H23_CBE_Control_plasmid_1_zscore, data$L2FC_H23_CBE_Control_plasmid_2_zscore, method = "pearson")
r10 <- cor(data$L2FC_H23_CBE_Sotor_plasmid_1_zscore, data$L2FC_H23_CBE_Sotor_plasmid_2_zscore, method = "pearson")
r11 <- cor(data$L2FC_H23_CBE_Adag_plasmid_1_zscore, data$L2FC_H23_CBE_Adag_plasmid_2_zscore, method = "pearson")
r12 <- cor(data$L2FC_H23_ABE_Control_plasmid_1_zscore, data$L2FC_H23_ABE_Control_plasmid_2_zscore, method = "pearson")
r13 <- cor(data$L2FC_H23_ABE_Sotor_plasmid_1_zscore, data$L2FC_H23_ABE_Sotor_plasmid_2_zscore, method = "pearson")
r14 <- cor(data$L2FC_H23_ABE_Adag_plasmid_1_zscore, data$L2FC_H23_ABE_Adag_plasmid_2_zscore, method = "pearson")
```

```{r}
#H23 check correlation between replicates
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

# CBE control vs plasmid
plot1 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_H23_CBE_Control_plasmid_1_zscore, y= L2FC_H23_CBE_Control_plasmid_2_zscore)) +
geom_pointdensity(size=3,  position = pos) +
labs(title= "H23 - replicate correlation", x = "zscore CBE control vs plasmid exp.1", y = "zscore CBE control vs plasmid exp.2", colour = "density") +
geom_smooth(method = lm) +
annotate("text", y = 5, x = 0, label = paste("r = ", round(r9, 2)), parse = FALSE, size = 7) +
theme_bw(base_size = 15)
plot1
ggsave("FigS2_9.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

# CBE Sotor vs plasmid
plot2 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_H23_CBE_Sotor_plasmid_1_zscore, y= L2FC_H23_CBE_Sotor_plasmid_2_zscore)) + geom_pointdensity(size=3,  position = pos) +
labs(title= "H23 - replicate correlation", x = "zscore CBE Sotorasib vs plasmid exp.1", y = "zscore CBE Sotorasib vs plasmid exp.2", colour = "density") +
geom_smooth(method = lm) +
annotate("text", y = 5, x = 0, label = paste("r = ", round(r10, 2)), parse = FALSE, size = 7) +
theme_bw(base_size = 15)
plot2
ggsave("FigS2_10.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

# CBE Adag vs plasmid
plot3 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_H23_CBE_Adag_plasmid_1_zscore, y= L2FC_H23_CBE_Adag_plasmid_2_zscore)) + geom_pointdensity(size=3,  position = pos) +
labs(title= "H23 - replicate correlation", x = "zscore CBE Adagrasib vs plasmid exp.1", y = "zscore CBE Adagrasib vs plasmid exp.2", colour = "density") +
geom_smooth(method = lm) +
annotate("text", y = 5, x = 0, label = paste("r = ", round(r11, 2)), parse = FALSE, size = 7) +
theme_bw(base_size = 15)
plot3
ggsave("FigS2_11.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")


# ABE control vs plasmid
plot4 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_H23_ABE_Control_plasmid_1_zscore, y= L2FC_H23_ABE_Control_plasmid_2_zscore)) +
geom_pointdensity(size=3,  position = pos) +
scale_color_viridis_c() +
labs(title= "H23 - replicate correlation", x = "zscore ABE control vs plasmid exp.1", y = "zscore ABE control vs plasmid exp.2", colour = "density") +
geom_smooth(method = lm) +
annotate("text", y = 7, x = 0, label = paste("r = ", round(r12, 2)), parse = FALSE, size = 7) +
theme_bw(base_size = 15)
plot4
ggsave("FigS2_12.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

# ABE Sotor vs plasmid
plot5 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_H23_ABE_Sotor_plasmid_1_zscore, y= L2FC_H23_ABE_Sotor_plasmid_2_zscore)) + geom_pointdensity(size=3,  position = pos) +
scale_color_viridis_c() +
labs(title= "H23 - replicate correlation", x = "zscore ABE Sotorasib vs plasmid exp.1", y = "zscore ABE Sotorasib vs plasmid exp.2", colour = "density") +
geom_smooth(method = lm) +
annotate("text", y = 7, x = 0, label = paste("r = ", round(r13, 2)), parse = FALSE, size = 7) +
theme_bw(base_size = 15)
plot5
ggsave("FigS2_13.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

# ABE Adag vs plasmid
plot6 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_H23_ABE_Adag_plasmid_1_zscore, y= L2FC_H23_ABE_Adag_plasmid_2_zscore)) + geom_pointdensity(size=3,  position = pos) +
scale_color_viridis_c() +
labs(title= "H23 - replicate correlation", x = "zscore ABE Adagrasib vs plasmid exp.1", y = "zscore ABE Adagrasib vs plasmid exp.2", colour = "density") +
geom_smooth(method = lm) +
annotate("text", y = 7, x = 0, label = paste("r = ", round(r14, 2)), parse = FALSE, size = 7) +
theme_bw(base_size = 15)
plot6
ggsave("FigS2_14.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")
```

```{r}
#PC9
# check L2FC PLASMID vs Gene
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

# CBE control vs plasmid
plot1 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_PC9_CBE_Control_plasmid_1_zscore)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "zscore_CBE_plasmid_T0_1") +
theme_classic()
plot1

# CBE Gefit vs plasmid
plot2 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_PC9_CBE_Gefit_plasmid_1_zscore)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "zscore_CBE_Gefit_plasmid_1") +
theme_classic()
plot2

# CBE Adag vs plasmid
plot3 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_PC9_CBE_Osim_plasmid_1_zscore)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "zscore_CBE_Osim_plasmid_1") +
theme_classic()
plot3


# ABE plasmid vs T0
plot5 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_PC9_ABE_Control_plasmid_1_zscore)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "zscore_ABE_plasmid_T0_1") +
theme_classic()
plot5

# ABE Gefit vs plasmid
plot6 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_PC9_ABE_Gefit_plasmid_1_zscore)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "zscore_ABE_Gefit_plasmid_1") +
theme_classic()
plot6

# ABE Osim vs plasmid
plot7 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_PC9_ABE_Osim_plasmid_1_zscore)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "zscore_ABE_Osim_plasmid_1") +
theme_classic()
plot7
```

```{r}
# replicate correlation - stats PC9
r15 <- cor(data$L2FC_PC9_CBE_Control_plasmid_1_zscore, data$L2FC_PC9_CBE_Control_plasmid_2_zscore, method = "pearson")
r16 <- cor(data$L2FC_PC9_CBE_Gefit_plasmid_1_zscore, data$L2FC_PC9_CBE_Gefit_plasmid_2_zscore, method = "pearson")
r17 <- cor(data$L2FC_PC9_CBE_Gefit_plasmid_1_zscore, data$L2FC_PC9_CBE_Gefit_plasmid_3_zscore, method = "pearson")
r18 <- cor(data$L2FC_PC9_CBE_Gefit_plasmid_2_zscore, data$L2FC_PC9_CBE_Gefit_plasmid_3_zscore, method = "pearson")
r19 <- cor(data$L2FC_PC9_CBE_Osim_plasmid_1_zscore, data$L2FC_PC9_CBE_Osim_plasmid_2_zscore, method = "pearson")
r20 <- cor(data$L2FC_PC9_ABE_Control_plasmid_1_zscore, data$L2FC_PC9_ABE_Control_plasmid_2_zscore, method = "pearson")
r21 <- cor(data$L2FC_PC9_ABE_Gefit_plasmid_1_zscore, data$L2FC_PC9_ABE_Gefit_plasmid_2_zscore, method = "pearson")
r22 <- cor(data$L2FC_PC9_ABE_Osim_plasmid_1_zscore, data$L2FC_PC9_ABE_Osim_plasmid_2_zscore, method = "pearson")
```

```{r}
#PC9 check correlation between replicates
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

# CBE control vs plasmid
plot1 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_PC9_CBE_Control_plasmid_1_zscore, y= L2FC_PC9_CBE_Control_plasmid_2_zscore)) + 
geom_pointdensity(size=3,  position = pos) +
labs(title= "PC9 - replicate correlation", x = "zscore CBE control vs plasmid exp.1", y = "zscore CBE control vs plasmid exp.2", colour = "density") +
geom_smooth(method = lm) +
annotate("text", y = 5, x = 0, label = paste("r = ", round(r15, 2)), parse = FALSE, size = 7) +
theme_bw(base_size = 15)
plot1
ggsave("FigS2_15.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

# CBE exp 1/2 Gefit vs plasmid
plot2 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_PC9_CBE_Gefit_plasmid_1_zscore, y= L2FC_PC9_CBE_Gefit_plasmid_2_zscore)) + geom_pointdensity(size=3,  position = pos) +
labs(title= "PC9 - replicate correlation", x = "zscore CBE Gefitinib vs plasmid exp.1", y = "zscore CBE Gefitinib vs plasmid exp.2", colour = "density") +
geom_smooth(method = lm) +
annotate("text", y = 13, x = 0, label = paste("r = ", round(r16, 2)), parse = FALSE, size = 7) +
theme_bw(base_size = 15)
plot2
ggsave("FigS2_16.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

# CBE exp 1/3 Gefit vs plasmid
plot3 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_PC9_CBE_Gefit_plasmid_1_zscore, y= L2FC_PC9_CBE_Gefit_plasmid_3_zscore)) + geom_pointdensity(size=3,  position = pos) +
labs(title= "PC9 - replicate correlation", x = "zscore CBE Gefitinib vs plasmid exp.1", y = "zscore CBE Gefitinib vs plasmid exp.3", colour = "density") +
geom_smooth(method = lm) +
annotate("text", y = 12, x = 0, label = paste("r = ", round(r17, 2)), parse = FALSE, size = 7) +
theme_bw(base_size = 15)
plot3
ggsave("FigS2_XXX.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

# CBE exp 2/3 Gefit vs plasmid
plot4 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_PC9_CBE_Gefit_plasmid_2_zscore, y= L2FC_PC9_CBE_Gefit_plasmid_3_zscore)) + geom_pointdensity(size=3,  position = pos) +
labs(title= "PC9 - replicate correlation", x = "zscore CBE Gefitinib vs plasmid exp.2", y = "zscore CBE Gefitinib vs plasmid exp.3", colour = "density") +
geom_smooth(method = lm) +
annotate("text", y = 12, x = 0, label = paste("r = ", round(r18, 2)), parse = FALSE, size = 7) +
theme_bw(base_size = 15)
plot4

# CBE Osim vs plasmid
plot5 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_PC9_CBE_Osim_plasmid_1_zscore, y= L2FC_PC9_CBE_Osim_plasmid_2_zscore)) + geom_pointdensity(size=3,  position = pos) +
labs(title= "PC9 - replicate correlation", x = "zscore CBE Osimertinib vs plasmid exp.1", y = "zscore CBE Osimertinib vs plasmid exp.2", colour = "density") +
geom_smooth(method = lm) +
annotate("text", y = 5, x = 0, label = paste("r = ", round(r19, 2)), parse = FALSE, size = 7) +
theme_bw(base_size = 15)
plot5
ggsave("FigS2_17.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

# ABE control vs plasmid
plot6 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_PC9_ABE_Control_plasmid_1_zscore, y= L2FC_PC9_ABE_Control_plasmid_2_zscore)) +
geom_pointdensity(size=3,  position = pos) +
scale_color_viridis_c() +
labs(title= "PC9 - replicate correlation", x = "zscore ABE control vs plasmid exp.1", y = "zscore ABE control vs plasmid exp.2", colour = "density") +
geom_smooth(method = lm) +
annotate("text", y = 5, x = 0, label = paste("r = ", round(r20, 2)), parse = FALSE, size = 7) +
theme_bw(base_size = 15)
plot6
ggsave("FigS2_18.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

# ABE Gefit vs plasmid
plot7 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_PC9_ABE_Gefit_plasmid_1_zscore, y= L2FC_PC9_ABE_Gefit_plasmid_2_zscore)) +
geom_pointdensity(size=3,  position = pos) +
scale_color_viridis_c() +
labs(title= "PC9 - replicate correlation", x = "zscore ABE Gefitinib vs plasmid exp.1", y = "zscore ABE Gefitinib vs plasmid exp.2", colour = "density") +
geom_smooth(method = lm) +
annotate("text", y = 5, x = 0, label = paste("r = ", round(r21, 2)), parse = FALSE, size = 7) +
theme_bw(base_size = 15)
plot7
ggsave("FigS2_19.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

# ABE Osim vs plasmid
plot8 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_PC9_ABE_Osim_plasmid_1_zscore, y= L2FC_PC9_ABE_Osim_plasmid_2_zscore)) +
geom_pointdensity(size=3,  position = pos) +
scale_color_viridis_c() +
labs(title= "PC9 - replicate correlation", x = "zscore ABE Osimertinib vs plasmid exp.1", y = "zscore ABE Osimertinib vs plasmid exp.2", colour = "density") +
geom_smooth(method = lm) +
annotate("text", y = 5, x = 0, label = paste("r = ", round(r22, 2)), parse = FALSE, size = 7) +
theme_bw(base_size = 15)
plot8
ggsave("FigS2_20.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")
```

```{r}
# MHHES1
# check L2FC PLASMID vs Gene
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

# ABE plasmid vs T0
plot1 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_MHHES1_ABE_Control_plasmid_1_zscore)) + geom_jitter(size=3, alpha=1, position = pos, colour = "black") +
labs(title= "all gRNAs", x = "gene", y = "control vs plasmid (z-score)") +
theme_classic()
plot1

# ABE Olap vs plasmid
plot2 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_MHHES1_ABE_Olap_plasmid_1_zscore)) + geom_jitter(size=3, alpha=1, position = pos, colour = "blue") +
labs(title= "all gRNAs", x = "gene", y = "olaparib vs plasmid (z-score)") +
theme_classic()
plot2

# ABE Nirap vs plasmid
plot3 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_MHHES1_ABE_Nirap_plasmid_1_zscore)) + geom_jitter(size=3, alpha=1, position = pos, colour = "red") +
labs(title= "all gRNAs", x = "gene", y = "niraparib vs plasmid (z-score)") +
theme_classic()
plot3
```

```{r}
# replicate correlation - stats MHH-ES-1
r23 <- cor(data$L2FC_MHHES1_ABE_Control_plasmid_1_zscore, data$L2FC_MHHES1_ABE_Control_plasmid_2_zscore, method = "pearson")
r24 <- cor(data$L2FC_MHHES1_ABE_Olap_plasmid_1_zscore, data$L2FC_MHHES1_ABE_Olap_plasmid_2_zscore, method = "pearson")
r25 <- cor(data$L2FC_MHHES1_ABE_Nirap_plasmid_1_zscore, data$L2FC_MHHES1_ABE_Nirap_plasmid_2_zscore, method = "pearson")
```

```{r}
# MHH-ES-1 check correlation between replicates
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

# ABE control vs plasmid
plot1 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_MHHES1_ABE_Control_plasmid_1_zscore, y= L2FC_MHHES1_ABE_Control_plasmid_2_zscore)) +
geom_pointdensity(size=3,  position = pos) +
scale_color_viridis_c() +
labs(title= "MHH-ES-1 - replicate correlation", x = "zscore ABE control vs plasmid exp.1", y = "zscore ABE control vs plasmid exp.2", colour = "density") +
geom_smooth(method = lm) +
annotate("text", y = 5, x = 0, label = paste("r = ", round(r23, 2)), parse = FALSE, size = 7) +
theme_bw(base_size = 15)
plot1
ggsave("FigS2_21.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

# ABE Olap vs plasmid
plot6 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_MHHES1_ABE_Olap_plasmid_1_zscore, y= L2FC_MHHES1_ABE_Olap_plasmid_2_zscore)) + 
geom_pointdensity(size=3,  position = pos) +
scale_color_viridis_c() +
labs(title= "MHH-ES-1 - replicate correlation", x = "zscore ABE Olaparib vs plasmid exp.1", y = "zscore ABE Olaparib vs plasmid exp.2", colour = "density") +
geom_smooth(method = lm) +
annotate("text", y = 5, x = 0, label = paste("r = ", round(r24, 2)), parse = FALSE, size = 7) +
theme_bw(base_size = 15)
plot6
ggsave("FigS2_22.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

# ABE Nirap vs plasmid
plot7 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_MHHES1_ABE_Nirap_plasmid_1_zscore, y= L2FC_MHHES1_ABE_Nirap_plasmid_2_zscore)) + geom_pointdensity(size=3,  position = pos) +
scale_color_viridis_c() +
labs(title= "MHH-ES-1 - replicate correlation", x = "zscore ABE Niraparib vs plasmid exp.1", y = "zscore ABE Niraparib vs plasmid exp.2", colour = "density") +
geom_smooth(method = lm) +
annotate("text", y = 5, x = 0, label = paste("r = ", round(r25, 2)), parse = FALSE, size = 7) +
theme_bw(base_size = 15)
plot7
ggsave("FigS2_23.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")
```

```{r}
#BEstimate annotations for newest Bestimate tool with 3-9 editing window for CBE and ABE

#read in BEstimate sumamry dfs
AKT1_CBE <- as_tibble(read.csv("BEstimate_protein_050423/CBE_NGN_AKT1_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
MYC_CBE <- as_tibble(read.csv("BEstimate_protein_050423/CBE_NGN_MYC_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
PIK3CA_CBE <- as_tibble(read.csv("BEstimate_protein_050423/CBE_NGN_PIK3CA_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
KRAS_CBE <- as_tibble(read.csv("BEstimate_protein_050423/CBE_NGN_KRAS_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
BCL2_CBE <- as_tibble(read.csv("BEstimate_protein_050423/CBE_NGN_BCL2_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
BRAF_CBE <- as_tibble(read.csv("BEstimate_protein_050423/CBE_NGN_BRAF_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
PARP1_CBE <- as_tibble(read.csv("BEstimate_protein_050423/CBE_NGN_PARP1_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
PARP2_CBE <- as_tibble(read.csv("BEstimate_protein_050423/CBE_NGN_PARP2_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
EGFR_CBE <- as_tibble(read.csv("BEstimate_protein_050423/CBE_NGN_EGFR_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
MAP2K1_CBE <- as_tibble(read.csv("BEstimate_protein_050423/CBE_NGN_MAP2K1_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
MAP2K2_CBE <- as_tibble(read.csv("BEstimate_protein_050423/CBE_NGN_MAP2K2_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))

AKT1_ABE <- as_tibble(read.csv("BEstimate_protein_050423/ABE_NGN_AKT1_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
MYC_ABE <- as_tibble(read.csv("BEstimate_protein_050423/ABE_NGN_MYC_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
PIK3CA_ABE <- as_tibble(read.csv("BEstimate_protein_050423/ABE_NGN_PIK3CA_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
KRAS_ABE <- as_tibble(read.csv("BEstimate_protein_050423/ABE_NGN_KRAS_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
BCL2_ABE <- as_tibble(read.csv("BEstimate_protein_050423/ABE_NGN_BCL2_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
BRAF_ABE <- as_tibble(read.csv("BEstimate_protein_050423/ABE_NGN_BRAF_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
PARP1_ABE <- as_tibble(read.csv("BEstimate_protein_050423/ABE_NGN_PARP1_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
PARP2_ABE <- as_tibble(read.csv("BEstimate_protein_050423/ABE_NGN_PARP2_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
EGFR_ABE <- as_tibble(read.csv("BEstimate_protein_050423/ABE_NGN_EGFR_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
MAP2K1_ABE <- as_tibble(read.csv("BEstimate_protein_050423/ABE_NGN_MAP2K1_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
MAP2K2_ABE <- as_tibble(read.csv("BEstimate_protein_050423/ABE_NGN_MAP2K2_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))

#bind rows of all BEsimate dfs
CBE_BEstimate <- bind_rows(AKT1_CBE, MYC_CBE, BCL2_CBE, BRAF_CBE, MAP2K1_CBE, MAP2K2_CBE, PARP1_CBE, PARP2_CBE, PIK3CA_CBE, EGFR_CBE, KRAS_CBE)
ABE_BEstimate <- bind_rows(AKT1_ABE, MYC_ABE, BCL2_ABE, BRAF_ABE, MAP2K1_ABE, MAP2K2_ABE, PARP1_ABE, PARP2_ABE, PIK3CA_ABE, EGFR_ABE, KRAS_ABE)

# remove data columns that mess up the left join
data <- data %>% select(!c(Transcript_ID, Exon_ID, Direction))

#rename guide and Gene columns
CBE_BEstimate<- rename(CBE_BEstimate, Gene = Hugo_Symbol)
CBE_BEstimate<- rename(CBE_BEstimate, Amino_Acid_Position = Protein_Position)
CBE_BEstimate<- rename(CBE_BEstimate, guide = gRNA_Target_Sequence)

ABE_BEstimate<- rename(ABE_BEstimate, Gene = Hugo_Symbol)
ABE_BEstimate<- rename(ABE_BEstimate, Amino_Acid_Position = Protein_Position)
ABE_BEstimate<- rename(ABE_BEstimate, guide = gRNA_Target_Sequence)

#left join the library and read counts with the BEstimate predictions
data_CBE <- left_join(data, CBE_BEstimate)
data_CBE <- mutate(data_CBE, editor = "CBE")

data_ABE <- left_join(data, ABE_BEstimate)
data_ABE <- mutate(data_ABE, editor = "ABE")

annotated_data <- bind_rows(data_CBE, data_ABE)
annotated_data
```

```{r}
# consolidate gRNA scores for gRNAs based on editor

annotated_data <- mutate(annotated_data, L2FC_HT29_DebCet_plasmid_1_zscore = as.numeric(ifelse(editor == "CBE", L2FC_HT29_CBE_DebCet_plasmid_1_zscore, ifelse(editor == "ABE", L2FC_HT29_ABE_DebCet_plasmid_1_zscore, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_HT29_Tram_plasmid_1_zscore = as.numeric(ifelse(editor == "CBE", L2FC_HT29_CBE_Tram_plasmid_1_zscore, ifelse(editor == "ABE", L2FC_HT29_ABE_Tram_plasmid_1_zscore, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_HT29_Pict_plasmid_1_zscore = as.numeric(ifelse(editor == "CBE", L2FC_HT29_CBE_Pict_plasmid_1_zscore, ifelse(editor == "ABE", L2FC_HT29_ABE_Pict_plasmid_1_zscore, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_HT29_Control_plasmid_1_zscore = as.numeric(ifelse(editor == "CBE", L2FC_HT29_CBE_Control_plasmid_1_zscore, ifelse(editor == "ABE", L2FC_HT29_ABE_Control_plasmid_1_zscore, "ERROR"))))

annotated_data <- mutate(annotated_data, L2FC_HT29_DebCet_plasmid_2_zscore = as.numeric(ifelse(editor == "CBE", L2FC_HT29_CBE_DebCet_plasmid_2_zscore, ifelse(editor == "ABE", L2FC_HT29_ABE_DebCet_plasmid_2_zscore, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_HT29_Tram_plasmid_2_zscore = as.numeric(ifelse(editor == "CBE", L2FC_HT29_CBE_Tram_plasmid_2_zscore, ifelse(editor == "ABE", L2FC_HT29_ABE_Tram_plasmid_2_zscore, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_HT29_Pict_plasmid_2_zscore = as.numeric(ifelse(editor == "CBE", L2FC_HT29_CBE_Pict_plasmid_2_zscore, ifelse(editor == "ABE", L2FC_HT29_ABE_Pict_plasmid_2_zscore, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_HT29_Control_plasmid_2_zscore = as.numeric(ifelse(editor == "CBE", L2FC_HT29_CBE_Control_plasmid_2_zscore, ifelse(editor == "ABE", L2FC_HT29_ABE_Control_plasmid_2_zscore, "ERROR"))))

annotated_data <- mutate(annotated_data, L2FC_HT29_DebCet_plasmid_average_zscore = as.numeric(ifelse(editor == "CBE", L2FC_HT29_CBE_DebCet_plasmid_average_zscore, ifelse(editor == "ABE", L2FC_HT29_ABE_DebCet_plasmid_average_zscore, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_HT29_Tram_plasmid_average_zscore = as.numeric(ifelse(editor == "CBE", L2FC_HT29_CBE_Tram_plasmid_average_zscore, ifelse(editor == "ABE", L2FC_HT29_ABE_Tram_plasmid_average_zscore, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_HT29_Pict_plasmid_average_zscore = as.numeric(ifelse(editor == "CBE", L2FC_HT29_CBE_Pict_plasmid_average_zscore, ifelse(editor == "ABE", L2FC_HT29_ABE_Pict_plasmid_average_zscore, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_HT29_Control_plasmid_average_zscore = as.numeric(ifelse(editor == "CBE", L2FC_HT29_CBE_Control_plasmid_average_zscore, ifelse(editor == "ABE", L2FC_HT29_ABE_Control_plasmid_average_zscore, "ERROR"))))

annotated_data <- mutate(annotated_data, L2FC_H23_Adag_plasmid_1_zscore = as.numeric(ifelse(editor == "CBE", L2FC_H23_CBE_Adag_plasmid_1_zscore, ifelse(editor == "ABE", L2FC_H23_ABE_Adag_plasmid_1_zscore, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_H23_Sotor_plasmid_1_zscore = as.numeric(ifelse(editor == "CBE", L2FC_H23_CBE_Sotor_plasmid_1_zscore, ifelse(editor == "ABE", L2FC_H23_ABE_Sotor_plasmid_1_zscore, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_H23_Control_plasmid_1_zscore = as.numeric(ifelse(editor == "CBE", L2FC_H23_CBE_Control_plasmid_1_zscore, ifelse(editor == "ABE", L2FC_H23_ABE_Control_plasmid_1_zscore, "ERROR"))))

annotated_data <- mutate(annotated_data, L2FC_H23_Adag_plasmid_2_zscore = as.numeric(ifelse(editor == "CBE", L2FC_H23_CBE_Adag_plasmid_2_zscore, ifelse(editor == "ABE", L2FC_H23_ABE_Adag_plasmid_2_zscore, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_H23_Sotor_plasmid_2_zscore = as.numeric(ifelse(editor == "CBE", L2FC_H23_CBE_Sotor_plasmid_2_zscore, ifelse(editor == "ABE", L2FC_H23_ABE_Sotor_plasmid_2_zscore, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_H23_Control_plasmid_2_zscore = as.numeric(ifelse(editor == "CBE", L2FC_H23_CBE_Control_plasmid_2_zscore, ifelse(editor == "ABE", L2FC_H23_ABE_Control_plasmid_2_zscore, "ERROR"))))

annotated_data <- mutate(annotated_data, L2FC_H23_Adag_plasmid_average_zscore = as.numeric(ifelse(editor == "CBE", L2FC_H23_CBE_Adag_plasmid_average_zscore, ifelse(editor == "ABE", L2FC_H23_ABE_Adag_plasmid_average_zscore, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_H23_Sotor_plasmid_average_zscore = as.numeric(ifelse(editor == "CBE", L2FC_H23_CBE_Sotor_plasmid_average_zscore, ifelse(editor == "ABE", L2FC_H23_ABE_Sotor_plasmid_average_zscore, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_H23_Control_plasmid_average_zscore = as.numeric(ifelse(editor == "CBE", L2FC_H23_CBE_Control_plasmid_average_zscore, ifelse(editor == "ABE", L2FC_H23_ABE_Control_plasmid_average_zscore, "ERROR"))))

annotated_data <- mutate(annotated_data, L2FC_PC9_Control_plasmid_1_zscore = as.numeric(ifelse(editor == "CBE", L2FC_PC9_CBE_Control_plasmid_1_zscore, ifelse(editor == "ABE", L2FC_PC9_ABE_Control_plasmid_1_zscore, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_PC9_Osim_plasmid_1_zscore = as.numeric(ifelse(editor == "CBE", L2FC_PC9_CBE_Osim_plasmid_1_zscore, ifelse(editor == "ABE", L2FC_PC9_ABE_Osim_plasmid_1_zscore, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_PC9_Gefit_plasmid_1_zscore = as.numeric(ifelse(editor == "CBE", L2FC_PC9_CBE_Gefit_plasmid_1_zscore, ifelse(editor == "ABE", L2FC_PC9_ABE_Gefit_plasmid_1_zscore, "ERROR"))))

annotated_data <- mutate(annotated_data, L2FC_PC9_Control_plasmid_2_zscore = as.numeric(ifelse(editor == "CBE", L2FC_PC9_CBE_Control_plasmid_2_zscore, ifelse(editor == "ABE", L2FC_PC9_ABE_Control_plasmid_2_zscore, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_PC9_Osim_plasmid_2_zscore = as.numeric(ifelse(editor == "CBE", L2FC_PC9_CBE_Osim_plasmid_2_zscore, ifelse(editor == "ABE", L2FC_PC9_ABE_Osim_plasmid_2_zscore, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_PC9_Gefit_plasmid_2_zscore = as.numeric(ifelse(editor == "CBE", L2FC_PC9_CBE_Gefit_plasmid_2_zscore, ifelse(editor == "ABE", L2FC_PC9_ABE_Gefit_plasmid_2_zscore, "ERROR"))))

annotated_data <- mutate(annotated_data, L2FC_PC9_Control_plasmid_average_zscore = as.numeric(ifelse(editor == "CBE", L2FC_PC9_CBE_Control_plasmid_average_zscore, ifelse(editor == "ABE", L2FC_PC9_ABE_Control_plasmid_average_zscore, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_PC9_Osim_plasmid_average_zscore = as.numeric(ifelse(editor == "CBE", L2FC_PC9_CBE_Osim_plasmid_average_zscore, ifelse(editor == "ABE", L2FC_PC9_ABE_Osim_plasmid_average_zscore, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_PC9_Gefit_plasmid_average_zscore = as.numeric(ifelse(editor == "CBE", L2FC_PC9_CBE_Gefit_plasmid_average_zscore, ifelse(editor == "ABE", L2FC_PC9_ABE_Gefit_plasmid_average_zscore, "ERROR"))))

annotated_data <- mutate(annotated_data, L2FC_MHHES1_Olap_plasmid_1_zscore = as.numeric(ifelse(editor == "ABE", L2FC_MHHES1_ABE_Olap_plasmid_1_zscore, "ERROR")))
annotated_data <- mutate(annotated_data, L2FC_MHHES1_Nirap_plasmid_1_zscore = as.numeric(ifelse(editor == "ABE", L2FC_MHHES1_ABE_Nirap_plasmid_1_zscore, "ERROR")))
annotated_data <- mutate(annotated_data, L2FC_MHHES1_Control_plasmid_1_zscore = as.numeric(ifelse(editor == "ABE", L2FC_MHHES1_ABE_Control_plasmid_1_zscore, "ERROR")))

annotated_data <- mutate(annotated_data, L2FC_MHHES1_Olap_plasmid_2_zscore = as.numeric(ifelse(editor == "ABE", L2FC_MHHES1_ABE_Olap_plasmid_2_zscore, "ERROR")))
annotated_data <- mutate(annotated_data, L2FC_MHHES1_Nirap_plasmid_2_zscore = as.numeric(ifelse(editor == "ABE", L2FC_MHHES1_ABE_Nirap_plasmid_2_zscore, "ERROR")))
annotated_data <- mutate(annotated_data, L2FC_MHHES1_Control_plasmid_2_zscore = as.numeric(ifelse(editor == "ABE", L2FC_MHHES1_ABE_Control_plasmid_2_zscore, "ERROR")))

annotated_data <- mutate(annotated_data, L2FC_MHHES1_Olap_plasmid_average_zscore = as.numeric(ifelse(editor == "ABE", L2FC_MHHES1_ABE_Olap_plasmid_average_zscore, "ERROR")))
annotated_data <- mutate(annotated_data, L2FC_MHHES1_Nirap_plasmid_average_zscore = as.numeric(ifelse(editor == "ABE", L2FC_MHHES1_ABE_Nirap_plasmid_average_zscore, "ERROR")))
annotated_data <- mutate(annotated_data, L2FC_MHHES1_Control_plasmid_average_zscore = as.numeric(ifelse(editor == "ABE", L2FC_MHHES1_ABE_Control_plasmid_average_zscore, "ERROR")))

```

```{r} 
# get annotated_data to classify all non-essential and essential genes as one by making gene_alias
annotated_data %>% count(sgRNA_type)
annotated_data <- mutate(annotated_data, gene_alias = ifelse(sgRNA_type == "non_targeting", "non-targeting", ifelse(sgRNA_type == "splice_essential", "essential", ifelse(sgRNA_type == "splice_nonessential", "non-essential", ifelse(sgRNA_type == "intergenic", "intergenic", Gene)))))
annotated_data %>% count(gene_alias)
```

```{r}
# Make a simplified amino acid position such that we can plot on a numerical scale
annotated_data <- mutate(annotated_data, Amino_Acid_Position_simple = Amino_Acid_Position)
annotated_data <- mutate(annotated_data, Amino_Acid_Position_simple = as.numeric(str_extract(Amino_Acid_Position_simple, "[[:digit:]]+")))
annotated_data %>% select(Amino_Acid_Position, Amino_Acid_Position_simple)
```
```{r}
#read in MAGeCK datasets from Shriram
#HT29
MAGeCK_control_CBE_HT29 <- as_tibble(read.table("MAGeCK/results_HT29/plasmid_vs_Control_CBE_HT29_exp_combined.sgrna_summary.txt", sep="\t", stringsAsFactors = FALSE, header = TRUE))
MAGeCK_control_ABE_HT29 <- as_tibble(read.table("MAGeCK/results_HT29/plasmid_vs_Control_ABE_HT29_exp_combined.sgrna_summary.txt", sep="\t", stringsAsFactors = FALSE, header = TRUE))
MAGeCK_DebCet_CBE_HT29 <- as_tibble(read.table("MAGeCK/results_HT29/plasmid_vs_DebCet_CBE_HT29_exp_combined.sgrna_summary.txt", sep="\t", stringsAsFactors = FALSE, header = TRUE))
MAGeCK_DebCet_ABE_HT29 <- as_tibble(read.table("MAGeCK/results_HT29/plasmid_vs_DebCet_ABE_HT29_exp_combined.sgrna_summary.txt", sep="\t", stringsAsFactors = FALSE, header = TRUE))
MAGeCK_Pict_CBE_HT29 <- as_tibble(read.table("MAGeCK/results_HT29/plasmid_vs_Pict_CBE_HT29_exp_combined.sgrna_summary.txt", sep="\t", stringsAsFactors = FALSE, header = TRUE))
MAGeCK_Pict_ABE_HT29 <- as_tibble(read.table("MAGeCK/results_HT29/plasmid_vs_Pict_ABE_HT29_exp_combined.sgrna_summary.txt", sep="\t", stringsAsFactors = FALSE, header = TRUE))
MAGeCK_Tram_CBE_HT29 <- as_tibble(read.table("MAGeCK/results_HT29/plasmid_vs_Tram_CBE_HT29_exp_combined.sgrna_summary.txt", sep="\t", stringsAsFactors = FALSE, header = TRUE))
MAGeCK_Tram_ABE_HT29 <- as_tibble(read.table("MAGeCK/results_HT29/plasmid_vs_Tram_ABE_HT29_exp_combined.sgrna_summary.txt", sep="\t", stringsAsFactors = FALSE, header = TRUE))

MAGeCK_control_CBE_HT29 <- MAGeCK_control_CBE_HT29 %>% mutate(editor = "CBE")
MAGeCK_control_ABE_HT29 <- MAGeCK_control_ABE_HT29 %>% mutate(editor = "ABE")
MAGeCK_DebCet_CBE_HT29 <- MAGeCK_DebCet_CBE_HT29 %>% mutate(editor = "CBE")
MAGeCK_DebCet_ABE_HT29 <- MAGeCK_DebCet_ABE_HT29 %>% mutate(editor = "ABE")
MAGeCK_Pict_CBE_HT29 <- MAGeCK_Pict_CBE_HT29 %>% mutate(editor = "CBE")
MAGeCK_Pict_ABE_HT29 <- MAGeCK_Pict_ABE_HT29 %>% mutate(editor = "ABE")
MAGeCK_Tram_CBE_HT29 <- MAGeCK_Tram_CBE_HT29 %>% mutate(editor = "CBE")
MAGeCK_Tram_ABE_HT29 <- MAGeCK_Tram_ABE_HT29 %>% mutate(editor = "ABE")

MAGeCK_control_CBE_HT29 <- MAGeCK_control_CBE_HT29 %>% 
  rename_with(~ if_else(. %in% c("sgrna", "editor"), ., paste0(., "_MAGeCK_HT29_control")), everything()) %>%
  rename(sgRNA_ID = sgrna)
MAGeCK_control_ABE_HT29 <- MAGeCK_control_ABE_HT29 %>% 
  rename_with(~ if_else(. %in% c("sgrna", "editor"), ., paste0(., "_MAGeCK_HT29_control")), everything()) %>%
  rename(sgRNA_ID = sgrna)
MAGeCK_DebCet_CBE_HT29 <- MAGeCK_DebCet_CBE_HT29 %>% 
  rename_with(~ if_else(. %in% c("sgrna", "editor"), ., paste0(., "_MAGeCK_HT29_DebCet")), everything()) %>%
  rename(sgRNA_ID = sgrna)
MAGeCK_DebCet_ABE_HT29 <- MAGeCK_DebCet_ABE_HT29 %>% 
  rename_with(~ if_else(. %in% c("sgrna", "editor"), ., paste0(., "_MAGeCK_HT29_DebCet")), everything()) %>%
  rename(sgRNA_ID = sgrna)
MAGeCK_Pict_CBE_HT29 <- MAGeCK_Pict_CBE_HT29 %>% 
  rename_with(~ if_else(. %in% c("sgrna", "editor"), ., paste0(., "_MAGeCK_HT29_Pict")), everything()) %>%
  rename(sgRNA_ID = sgrna)
MAGeCK_Pict_ABE_HT29 <- MAGeCK_Pict_ABE_HT29 %>% 
  rename_with(~ if_else(. %in% c("sgrna", "editor"), ., paste0(., "_MAGeCK_HT29_Pict")), everything()) %>%
  rename(sgRNA_ID = sgrna)
MAGeCK_Tram_CBE_HT29 <- MAGeCK_Tram_CBE_HT29 %>% 
  rename_with(~ if_else(. %in% c("sgrna", "editor"), ., paste0(., "_MAGeCK_HT29_Tram")), everything()) %>%
  rename(sgRNA_ID = sgrna)
MAGeCK_Tram_ABE_HT29 <- MAGeCK_Tram_ABE_HT29 %>% 
  rename_with(~ if_else(. %in% c("sgrna", "editor"), ., paste0(., "_MAGeCK_HT29_Tram")), everything()) %>%
  rename(sgRNA_ID = sgrna)

#bind rows
MAGeCK_control_HT29 <- bind_rows(MAGeCK_control_CBE_HT29, MAGeCK_control_ABE_HT29)
MAGeCK_Tram_HT29 <- bind_rows(MAGeCK_Tram_CBE_HT29, MAGeCK_Tram_ABE_HT29)
MAGeCK_Pict_HT29 <- bind_rows(MAGeCK_Pict_CBE_HT29, MAGeCK_Pict_ABE_HT29)
MAGeCK_DebCet_HT29 <- bind_rows(MAGeCK_DebCet_CBE_HT29, MAGeCK_DebCet_ABE_HT29)

#join to data
MAGeCK_HT29 <- left_join(MAGeCK_control_HT29, MAGeCK_Tram_HT29)
MAGeCK_HT29 <- left_join(MAGeCK_HT29, MAGeCK_Pict_HT29)
MAGeCK_HT29 <- left_join(MAGeCK_HT29, MAGeCK_DebCet_HT29)

annotated_data <- left_join(annotated_data, MAGeCK_HT29)

#mutate pvalue to give one value
annotated_data <- annotated_data %>%
  mutate(p_value_MAGeCK_HT29_control = ifelse(high_in_treatment_MAGeCK_HT29_control == "True", p.high_MAGeCK_HT29_control, p.low_MAGeCK_HT29_control))
annotated_data <- annotated_data %>%
  mutate(p_value_MAGeCK_HT29_Tram = ifelse(high_in_treatment_MAGeCK_HT29_Tram == "True", p.high_MAGeCK_HT29_Tram, p.low_MAGeCK_HT29_Tram))
annotated_data <- annotated_data %>%
  mutate(p_value_MAGeCK_HT29_Pict = ifelse(high_in_treatment_MAGeCK_HT29_Pict == "True", p.high_MAGeCK_HT29_Pict, p.low_MAGeCK_HT29_Pict))
annotated_data <- annotated_data %>%
  mutate(p_value_MAGeCK_HT29_DebCet = ifelse(high_in_treatment_MAGeCK_HT29_DebCet == "True", p.high_MAGeCK_HT29_DebCet, p.low_MAGeCK_HT29_DebCet))
```

```{r}
#read in MAGeCK datasets from Shriram
#H23
MAGeCK_control_CBE_H23 <- as_tibble(read.table("MAGeCK/results_H23/plasmid_vs_H23_Control_CBE_exp_combined.sgrna_summary.txt", sep="\t", stringsAsFactors = FALSE, header = TRUE))
MAGeCK_control_ABE_H23 <- as_tibble(read.table("MAGeCK/results_H23/plasmid_vs_H23_Control_ABE_exp_combined.sgrna_summary.txt", sep="\t", stringsAsFactors = FALSE, header = TRUE))
MAGeCK_Adag_CBE_H23 <- as_tibble(read.table("MAGeCK/results_H23/plasmid_vs_H23_Adag_CBE_exp_combined.sgrna_summary.txt", sep="\t", stringsAsFactors = FALSE, header = TRUE))
MAGeCK_Adag_ABE_H23 <- as_tibble(read.table("MAGeCK/results_H23/plasmid_vs_H23_Adag_ABE_exp_combined.sgrna_summary.txt", sep="\t", stringsAsFactors = FALSE, header = TRUE))
MAGeCK_Sotor_CBE_H23 <- as_tibble(read.table("MAGeCK/results_H23/plasmid_vs_H23_Sotor_CBE_exp_combined.sgrna_summary.txt", sep="\t", stringsAsFactors = FALSE, header = TRUE))
MAGeCK_Sotor_ABE_H23 <- as_tibble(read.table("MAGeCK/results_H23/plasmid_vs_H23_Sotor_ABE_exp_combined.sgrna_summary.txt", sep="\t", stringsAsFactors = FALSE, header = TRUE))

MAGeCK_control_CBE_H23 <- MAGeCK_control_CBE_H23 %>% mutate(editor = "CBE")
MAGeCK_control_ABE_H23 <- MAGeCK_control_ABE_H23 %>% mutate(editor = "ABE")
MAGeCK_Adag_CBE_H23 <- MAGeCK_Adag_CBE_H23 %>% mutate(editor = "CBE")
MAGeCK_Adag_ABE_H23 <- MAGeCK_Adag_ABE_H23 %>% mutate(editor = "ABE")
MAGeCK_Sotor_CBE_H23 <- MAGeCK_Sotor_CBE_H23 %>% mutate(editor = "CBE")
MAGeCK_Sotor_ABE_H23 <- MAGeCK_Sotor_ABE_H23 %>% mutate(editor = "ABE")

MAGeCK_control_CBE_H23 <- MAGeCK_control_CBE_H23 %>% 
  rename_with(~ if_else(. %in% c("sgrna", "editor"), ., paste0(., "_MAGeCK_H23_control")), everything()) %>%
  rename(sgRNA_ID = sgrna)
MAGeCK_control_ABE_H23 <- MAGeCK_control_ABE_H23 %>% 
  rename_with(~ if_else(. %in% c("sgrna", "editor"), ., paste0(., "_MAGeCK_H23_control")), everything()) %>%
  rename(sgRNA_ID = sgrna)
MAGeCK_Adag_CBE_H23 <- MAGeCK_Adag_CBE_H23 %>% 
  rename_with(~ if_else(. %in% c("sgrna", "editor"), ., paste0(., "_MAGeCK_H23_Adag")), everything()) %>%
  rename(sgRNA_ID = sgrna)
MAGeCK_Adag_ABE_H23 <- MAGeCK_Adag_ABE_H23 %>% 
  rename_with(~ if_else(. %in% c("sgrna", "editor"), ., paste0(., "_MAGeCK_H23_Adag")), everything()) %>%
  rename(sgRNA_ID = sgrna)
MAGeCK_Sotor_CBE_H23 <- MAGeCK_Sotor_CBE_H23 %>% 
  rename_with(~ if_else(. %in% c("sgrna", "editor"), ., paste0(., "_MAGeCK_H23_Sotor")), everything()) %>%
  rename(sgRNA_ID = sgrna)
MAGeCK_Sotor_ABE_H23 <- MAGeCK_Sotor_ABE_H23 %>% 
  rename_with(~ if_else(. %in% c("sgrna", "editor"), ., paste0(., "_MAGeCK_H23_Sotor")), everything()) %>%
  rename(sgRNA_ID = sgrna)

#bind rows
MAGeCK_control_H23 <- bind_rows(MAGeCK_control_CBE_H23, MAGeCK_control_ABE_H23)
MAGeCK_Sotor_H23 <- bind_rows(MAGeCK_Sotor_CBE_H23, MAGeCK_Sotor_ABE_H23)
MAGeCK_Adag_H23 <- bind_rows(MAGeCK_Adag_CBE_H23, MAGeCK_Adag_ABE_H23)

#join to data
MAGeCK_H23 <- left_join(MAGeCK_control_H23, MAGeCK_Sotor_H23)
MAGeCK_H23 <- left_join(MAGeCK_H23, MAGeCK_Adag_H23)

annotated_data <- left_join(annotated_data, MAGeCK_H23)

#mutate pvalue to give one value
annotated_data <- annotated_data %>%
  mutate(p_value_MAGeCK_H23_control = ifelse(high_in_treatment_MAGeCK_H23_control == "True", p.high_MAGeCK_H23_control, p.low_MAGeCK_H23_control))
annotated_data <- annotated_data %>%
  mutate(p_value_MAGeCK_H23_Sotor = ifelse(high_in_treatment_MAGeCK_H23_Sotor == "True", p.high_MAGeCK_H23_Sotor, p.low_MAGeCK_H23_Sotor))
annotated_data <- annotated_data %>%
  mutate(p_value_MAGeCK_H23_Adag = ifelse(high_in_treatment_MAGeCK_H23_Adag == "True", p.high_MAGeCK_H23_Adag, p.low_MAGeCK_H23_Adag))
```

```{r}
#read in MAGeCK datasets from Shriram
#PC9
MAGeCK_control_CBE_PC9 <- as_tibble(read.table("MAGeCK/results_PC9/plasmid_vs_Control_CBE_PC9_exp_combined.sgrna_summary.txt", sep="\t", stringsAsFactors = FALSE, header = TRUE))
MAGeCK_control_ABE_PC9 <- as_tibble(read.table("MAGeCK/results_PC9/plasmid_vs_Control_ABE_PC9_exp_combined.sgrna_summary.txt", sep="\t", stringsAsFactors = FALSE, header = TRUE))
MAGeCK_Osim_CBE_PC9 <- as_tibble(read.table("MAGeCK/results_PC9/plasmid_vs_Osim_CBE_PC9_exp_combined.sgrna_summary.txt", sep="\t", stringsAsFactors = FALSE, header = TRUE))
MAGeCK_Osim_ABE_PC9 <- as_tibble(read.table("MAGeCK/results_PC9/plasmid_vs_Osim_ABE_PC9_exp_combined.sgrna_summary.txt", sep="\t", stringsAsFactors = FALSE, header = TRUE))
MAGeCK_Gefit_CBE_PC9 <- as_tibble(read.table("MAGeCK/results_PC9/plasmid_vs_Gefit_CBE_PC9_exp_combined.sgrna_summary.txt", sep="\t", stringsAsFactors = FALSE, header = TRUE))
MAGeCK_Gefit_ABE_PC9 <- as_tibble(read.table("MAGeCK/results_PC9/plasmid_vs_Gefit_ABE_PC9_exp_combined.sgrna_summary.txt", sep="\t", stringsAsFactors = FALSE, header = TRUE))

MAGeCK_control_CBE_PC9 <- MAGeCK_control_CBE_PC9 %>% mutate(editor = "CBE")
MAGeCK_control_ABE_PC9 <- MAGeCK_control_ABE_PC9 %>% mutate(editor = "ABE")
MAGeCK_Osim_CBE_PC9 <- MAGeCK_Osim_CBE_PC9 %>% mutate(editor = "CBE")
MAGeCK_Osim_ABE_PC9 <- MAGeCK_Osim_ABE_PC9 %>% mutate(editor = "ABE")
MAGeCK_Gefit_CBE_PC9 <- MAGeCK_Gefit_CBE_PC9 %>% mutate(editor = "CBE")
MAGeCK_Gefit_ABE_PC9 <- MAGeCK_Gefit_ABE_PC9 %>% mutate(editor = "ABE")

MAGeCK_control_CBE_PC9 <- MAGeCK_control_CBE_PC9 %>% 
  rename_with(~ if_else(. %in% c("sgrna", "editor"), ., paste0(., "_MAGeCK_PC9_control")), everything()) %>%
  rename(sgRNA_ID = sgrna)
MAGeCK_control_ABE_PC9 <- MAGeCK_control_ABE_PC9 %>% 
  rename_with(~ if_else(. %in% c("sgrna", "editor"), ., paste0(., "_MAGeCK_PC9_control")), everything()) %>%
  rename(sgRNA_ID = sgrna)
MAGeCK_Osim_CBE_PC9 <- MAGeCK_Osim_CBE_PC9 %>% 
  rename_with(~ if_else(. %in% c("sgrna", "editor"), ., paste0(., "_MAGeCK_PC9_Osim")), everything()) %>%
  rename(sgRNA_ID = sgrna)
MAGeCK_Osim_ABE_PC9 <- MAGeCK_Osim_ABE_PC9 %>% 
  rename_with(~ if_else(. %in% c("sgrna", "editor"), ., paste0(., "_MAGeCK_PC9_Osim")), everything()) %>%
  rename(sgRNA_ID = sgrna)
MAGeCK_Gefit_CBE_PC9 <- MAGeCK_Gefit_CBE_PC9 %>% 
  rename_with(~ if_else(. %in% c("sgrna", "editor"), ., paste0(., "_MAGeCK_PC9_Gefit")), everything()) %>%
  rename(sgRNA_ID = sgrna)
MAGeCK_Gefit_ABE_PC9 <- MAGeCK_Gefit_ABE_PC9 %>% 
  rename_with(~ if_else(. %in% c("sgrna", "editor"), ., paste0(., "_MAGeCK_PC9_Gefit")), everything()) %>%
  rename(sgRNA_ID = sgrna)

#bind rows
MAGeCK_control_PC9 <- bind_rows(MAGeCK_control_CBE_PC9, MAGeCK_control_ABE_PC9)
MAGeCK_Gefit_PC9 <- bind_rows(MAGeCK_Gefit_CBE_PC9, MAGeCK_Gefit_ABE_PC9)
MAGeCK_Osim_PC9 <- bind_rows(MAGeCK_Osim_CBE_PC9, MAGeCK_Osim_ABE_PC9)

#join to data
MAGeCK_PC9 <- left_join(MAGeCK_control_PC9, MAGeCK_Gefit_PC9)
MAGeCK_PC9 <- left_join(MAGeCK_PC9, MAGeCK_Osim_PC9)

annotated_data <- left_join(annotated_data, MAGeCK_PC9)

#mutate pvalue to give one value
annotated_data <- annotated_data %>%
  mutate(p_value_MAGeCK_PC9_control = ifelse(high_in_treatment_MAGeCK_PC9_control == "True", p.high_MAGeCK_PC9_control, p.low_MAGeCK_PC9_control))
annotated_data <- annotated_data %>%
  mutate(p_value_MAGeCK_PC9_Gefit = ifelse(high_in_treatment_MAGeCK_PC9_Gefit == "True", p.high_MAGeCK_PC9_Gefit, p.low_MAGeCK_PC9_Gefit))
annotated_data <- annotated_data %>%
  mutate(p_value_MAGeCK_PC9_Osim = ifelse(high_in_treatment_MAGeCK_PC9_Osim == "True", p.high_MAGeCK_PC9_Osim, p.low_MAGeCK_PC9_Osim))
```

```{r}
#read in MAGeCK datasets from Shriram
#MHHES1
MAGeCK_control_ABE_MHHES1 <- as_tibble(read.table("MAGeCK/results_MHHES1/plasmid_vs_MHHES1_ABE_cntrl_exp_combined.sgrna_summary.txt", sep="\t", stringsAsFactors = FALSE, header = TRUE))
MAGeCK_Nirap_ABE_MHHES1 <- as_tibble(read.table("MAGeCK/results_MHHES1/plasmid_vs_MHHES1_ABE_nirap_exp_combined.sgrna_summary.txt", sep="\t", stringsAsFactors = FALSE, header = TRUE))
MAGeCK_Olap_ABE_MHHES1 <- as_tibble(read.table("MAGeCK/results_MHHES1/plasmid_vs_MHHES1_ABE_olap_exp_combined.sgrna_summary.txt", sep="\t", stringsAsFactors = FALSE, header = TRUE))

MAGeCK_control_ABE_MHHES1 <- MAGeCK_control_ABE_MHHES1 %>% mutate(editor = "ABE")
MAGeCK_Nirap_ABE_MHHES1 <- MAGeCK_Nirap_ABE_MHHES1 %>% mutate(editor = "ABE")
MAGeCK_Olap_ABE_MHHES1 <- MAGeCK_Olap_ABE_MHHES1 %>% mutate(editor = "ABE")

MAGeCK_control_ABE_MHHES1 <- MAGeCK_control_ABE_MHHES1 %>% 
  rename_with(~ if_else(. %in% c("sgrna", "editor"), ., paste0(., "_MAGeCK_MHHES1_control")), everything()) %>%
  rename(sgRNA_ID = sgrna)
MAGeCK_Nirap_ABE_MHHES1 <- MAGeCK_Nirap_ABE_MHHES1 %>% 
  rename_with(~ if_else(. %in% c("sgrna", "editor"), ., paste0(., "_MAGeCK_MHHES1_Nirap")), everything()) %>%
  rename(sgRNA_ID = sgrna)
MAGeCK_Olap_ABE_MHHES1 <- MAGeCK_Olap_ABE_MHHES1 %>% 
  rename_with(~ if_else(. %in% c("sgrna", "editor"), ., paste0(., "_MAGeCK_MHHES1_Olap")), everything()) %>%
  rename(sgRNA_ID = sgrna)

#bind rows
MAGeCK_control_MHHES1 <- MAGeCK_control_ABE_MHHES1
MAGeCK_Olap_MHHES1 <- MAGeCK_Olap_ABE_MHHES1
MAGeCK_Nirap_MHHES1 <- MAGeCK_Nirap_ABE_MHHES1

#join to data
MAGeCK_MHHES1 <- left_join(MAGeCK_control_MHHES1, MAGeCK_Olap_MHHES1)
MAGeCK_MHHES1 <- left_join(MAGeCK_MHHES1, MAGeCK_Nirap_MHHES1)

annotated_data <- left_join(annotated_data, MAGeCK_MHHES1)

#mutate pvalue to give one value
annotated_data <- annotated_data %>%
  mutate(p_value_MAGeCK_MHHES1_control = ifelse(high_in_treatment_MAGeCK_MHHES1_control == "True", p.high_MAGeCK_MHHES1_control, p.low_MAGeCK_MHHES1_control))
annotated_data <- annotated_data %>%
  mutate(p_value_MAGeCK_MHHES1_Olap = ifelse(high_in_treatment_MAGeCK_MHHES1_Olap == "True", p.high_MAGeCK_MHHES1_Olap, p.low_MAGeCK_MHHES1_Olap))
annotated_data <- annotated_data %>%
  mutate(p_value_MAGeCK_MHHES1_Nirap = ifelse(high_in_treatment_MAGeCK_MHHES1_Nirap == "True", p.high_MAGeCK_MHHES1_Nirap, p.low_MAGeCK_MHHES1_Nirap))
```


```{r}
# dot plots _ proliferation - HT-29
# using gene_alias and FDR
targets <- c("EGFR", "KRAS", "MYC", "PARP1", "PARP2", "MAP2K1", "MAP2K2", "BCL2", "BRAF", "AKT1", "PIK3CA")

annotated_data$gene_alias <- factor(annotated_data$gene_alias, levels = c("AKT1", "BCL2", "BRAF", "EGFR", "KRAS", "MAP2K1", "MAP2K2", "MYC", "PARP1", "PARP2", "PIK3CA", "non-essential", "intergenic",  "non-targeting", "essential"))

pos <- position_jitter(width = 0.2, height = 0.2)

plot1 <- ggplot(subset(annotated_data, gene_alias %in% c(targets)), aes(x=L2FC_HT29_Control_plasmid_average_zscore, y= L2FC_HT29_Tram_plasmid_average_zscore, colour = gene_alias, shape = editor)) +
geom_point(position = pos, alpha = 1, size = 3) +
labs(title= "", x="HT-29 control vs plasmid (z-score)", y = "HT-29 trametinib vs plasmid (z-score)", colour = "gene", shape = "editor") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_vline(xintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_HT29_Tram_plasmid_average_zscore > 9 & (is.na(Amino_Acid_Position) == FALSE) | L2FC_HT29_Control_plasmid_average_zscore > 3 & (is.na(Amino_Acid_Position) == FALSE), paste(Gene,Amino_Acid_Position),
                      ifelse(L2FC_HT29_Tram_plasmid_average_zscore > 4 & (is.na(Amino_Acid_Position) == FALSE) | L2FC_HT29_Control_plasmid_average_zscore > 3 & (is.na(Amino_Acid_Position) == FALSE), paste(Amino_Acid_Position), 
                             ifelse( L2FC_HT29_Tram_plasmid_average_zscore > 4 & (is.na(Amino_Acid_Position) == TRUE) & (is.na(most_severe_consequence) == FALSE), paste(Gene, most_severe_consequence), ifelse(L2FC_HT29_Tram_plasmid_average_zscore < -3.5 & L2FC_HT29_Control_plasmid_average_zscore > - 0.25, paste(Gene, most_severe_consequence),                                                                                                                                                                  ifelse(L2FC_HT29_Tram_plasmid_average_zscore > 3 & L2FC_HT29_Control_plasmid_average_zscore < -3 & (is.na(Amino_Acid_Position) == FALSE), paste(Amino_Acid_Position), "")))))), size = 3, position = pos, max.overlaps = Inf, show.legend = FALSE) +
scale_color_manual(values =c("royalblue", "turquoise3", "orange", "red", "springgreen4", "grey55","brown", "purple", "salmon","pink", "black", "grey75", "red3")) +
scale_shape_manual(values =c(19, 1)) +
theme_classic()
plot1 <- plot1 +
  geom_rug(data = subset(annotated_data, gene_alias %in% c("non-targeting", "essential")), aes(x = L2FC_HT29_Control_plasmid_average_zscore, y = L2FC_HT29_Tram_plasmid_average_zscore, colour = gene_alias), show.legend = FALSE)
plot1

plot2 <- ggplot(subset(annotated_data, gene_alias %in% c(targets)), aes(x=L2FC_HT29_Control_plasmid_average_zscore, y= L2FC_HT29_Pict_plasmid_average_zscore, colour = gene_alias, shape = editor)) + 
geom_point(position = pos, alpha = 1, size =3) +
labs(title= "", x="HT-29 control vs plasmid (z-score)", y = "HT-29 pictilisib vs plasmid (z-score)", colour = "gene", shape = "editor") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_vline(xintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_HT29_Pict_plasmid_average_zscore > 4 & (is.na(Amino_Acid_Position) == FALSE) | L2FC_HT29_Control_plasmid_average_zscore > 4 & (is.na(Amino_Acid_Position) == FALSE), paste(Gene, Amino_Acid_Position), "")), size = 3, position = pos, max.overlaps = Inf, show.legend = FALSE) +
scale_color_manual(values =c("royalblue", "turquoise3", "orange", "red", "springgreen4", "grey55","brown", "purple", "salmon","pink", "black", "grey75", "red3")) +
scale_shape_manual(values =c(19, 1)) +
theme_classic()
plot2 <- plot2 +
  geom_rug(data = subset(annotated_data, gene_alias %in% c("non-targeting", "essential")), aes(x = L2FC_HT29_Control_plasmid_average_zscore, y = L2FC_HT29_Pict_plasmid_average_zscore, colour = gene_alias), show.legend = FALSE)
plot2

pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

plot3 <- ggplot(subset(annotated_data, gene_alias %in% c(targets)), aes(x=L2FC_HT29_Control_plasmid_average_zscore, y= L2FC_HT29_DebCet_plasmid_average_zscore, colour = gene_alias, shape = editor)) + 
geom_point(position = pos, alpha = 1, size = 3) +
labs(title= "", x="HT-29 control vs plasmid (z-score)", y = "HT-29 dabrafenib + cetuximab vs plasmid (z-score)", colour = "gene", shape = "editor") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_vline(xintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label= ifelse(L2FC_HT29_DebCet_plasmid_average_zscore > 10 & (is.na(Amino_Acid_Position) == FALSE) | L2FC_HT29_Control_plasmid_average_zscore > 3 & (is.na(Amino_Acid_Position) == FALSE), paste(Gene, Amino_Acid_Position),
                      ifelse(L2FC_HT29_DebCet_plasmid_average_zscore > 5 & (is.na(Amino_Acid_Position) == FALSE) | L2FC_HT29_Control_plasmid_average_zscore > 3 & (is.na(Amino_Acid_Position) == FALSE), paste(Amino_Acid_Position), 
                                 ifelse(L2FC_HT29_DebCet_plasmid_average_zscore > 3 & (is.na(Amino_Acid_Position) == TRUE) & (is.na(most_severe_consequence) == FALSE) | L2FC_HT29_Control_plasmid_average_zscore > 3 & (is.na(Amino_Acid_Position) == TRUE) & (is.na(most_severe_consequence) == FALSE), paste(most_severe_consequence), "")))), size = 3, position = pos, max.overlaps = Inf, show.legend = FALSE) +
scale_color_manual(values =c("royalblue", "turquoise3", "orange", "red", "springgreen4", "grey55","brown", "purple", "salmon","pink", "black", "grey75", "red3")) +
scale_shape_manual(values =c(19, 1)) +
theme_classic()
plot3 <- plot3 +
  geom_rug(data = subset(annotated_data, gene_alias %in% c("non-targeting", "essential")), aes(x = L2FC_HT29_Control_plasmid_average_zscore, y = L2FC_HT29_DebCet_plasmid_average_zscore, colour = gene_alias), show.legend = FALSE)
plot3
```

```{r}
#for conference

plot1 <- ggplot(subset(annotated_data, gene_alias %in% c(targets)), aes(x=L2FC_HT29_Control_plasmid_average_zscore, y= L2FC_HT29_Tram_plasmid_average_zscore, colour = gene_alias, shape = editor)) +
geom_point(position = pos, alpha = 1, size = 3) +
labs(title= "", x="HT-29 control vs plasmid (z-score)", y = "HT-29 trametinib vs plasmid (z-score)", colour = "gene", shape = "editor") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_vline(xintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_HT29_Tram_plasmid_average_zscore > 4 & (is.na(Amino_Acid_Position) == FALSE) | L2FC_HT29_Control_plasmid_average_zscore > 3 & (is.na(Amino_Acid_Position) == FALSE), paste(Gene), ifelse( L2FC_HT29_Tram_plasmid_average_zscore > 4 & (is.na(Amino_Acid_Position) == TRUE) & (is.na(most_severe_consequence) == FALSE), paste(Gene, most_severe_consequence), ifelse(L2FC_HT29_Tram_plasmid_average_zscore < -3.5 & L2FC_HT29_Control_plasmid_average_zscore > - 0.25, paste(Gene, most_severe_consequence), ifelse(L2FC_HT29_Tram_plasmid_average_zscore > 3 & L2FC_HT29_Control_plasmid_average_zscore < -3 & (is.na(Gene) == FALSE), paste(Gene), ""))))), size = 3, position = pos, max.overlaps = Inf, show.legend = FALSE) +
scale_color_manual(values =c("royalblue", "turquoise3", "orange", "red", "springgreen4", "grey55","brown", "purple", "salmon","pink", "black", "grey75", "red3")) +
scale_shape_manual(values =c(19, 1)) +
theme_classic()
plot1 <- plot1 +
  geom_rug(data = subset(annotated_data, gene_alias %in% c("non-targeting", "essential")), aes(x = L2FC_HT29_Control_plasmid_average_zscore, y = L2FC_HT29_Tram_plasmid_average_zscore, colour = gene_alias), show.legend = FALSE)
plot1
```

```{r}
#for NG editors

plot1 <- ggplot(subset(annotated_data, gene_alias %in% c(targets)), aes(x=L2FC_HT29_Control_plasmid_average_zscore, y= L2FC_HT29_Tram_plasmid_average_zscore, colour = gene_alias, shape = editor)) +
geom_point(position = pos, alpha = 1, size = 1) +
labs(title= "", x="HT-29 control vs plasmid (z-score)", y = "HT-29 trametinib vs plasmid (z-score)", colour = "gene", shape = "editor") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_vline(xintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_HT29_Tram_plasmid_average_zscore > 4 & (is.na(Amino_Acid_Position) == FALSE) | L2FC_HT29_Control_plasmid_average_zscore > 3 & (is.na(Amino_Acid_Position) == FALSE), paste(Gene), ifelse( L2FC_HT29_Tram_plasmid_average_zscore > 4 & (is.na(Amino_Acid_Position) == TRUE) & (is.na(most_severe_consequence) == FALSE), paste(Gene, most_severe_consequence), ifelse(L2FC_HT29_Tram_plasmid_average_zscore < -3.5 & L2FC_HT29_Control_plasmid_average_zscore > - 0.25, paste(Gene, most_severe_consequence), ifelse(L2FC_HT29_Tram_plasmid_average_zscore > 3 & L2FC_HT29_Control_plasmid_average_zscore < -3 & (is.na(Gene) == FALSE), paste(Gene), ""))))), size = 3, position = pos, max.overlaps = Inf, show.legend = FALSE) +
scale_color_manual(values =c("royalblue", "turquoise3", "orange", "red", "springgreen4", "grey55","brown", "purple", "salmon","pink", "black", "grey75", "red3")) +
scale_shape_manual(values =c(2, 1)) +
theme_classic()
plot1 <- plot1 +
  geom_rug(data = subset(annotated_data, gene_alias %in% c("non-targeting", "essential")), aes(x = L2FC_HT29_Control_plasmid_average_zscore, y = L2FC_HT29_Tram_plasmid_average_zscore, colour = gene_alias), show.legend = FALSE)
plot1
```

```{r}
#check correaltion of hits in categories of sgRNA_type
#HT29 Tram

plot1 <- ggplot(annotated_data, aes(x=FDR_MAGeCK_HT29_DebCet, y= L2FC_HT29_DebCet_plasmid_average_zscore, colour = p_value_MAGeCK_HT29_DebCet)) +
geom_point(alpha = 1) +
labs(title= "", x="FDR_MAGeCK_HT29_DebCet", y = "L2FC_HT29_DebCet_plasmid_average_zscore", size = "FDR (dabrafenib + cetuximab)", colour = "p_value_MAGeCK_HT29_DebCet") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_vline(xintercept=c(0.1),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_HT29_DebCet_plasmid_average_zscore > 5 & (is.na(Amino_Acid_Position) == FALSE) | L2FC_HT29_Control_plasmid_average_zscore > 3 & (is.na(Amino_Acid_Position) == FALSE), paste(Amino_Acid_Position), ifelse(L2FC_HT29_DebCet_plasmid_average_zscore > 3 & (is.na(Amino_Acid_Position) == TRUE) & (is.na(most_severe_consequence) == FALSE) | L2FC_HT29_Control_plasmid_average_zscore > 3 & (is.na(Amino_Acid_Position) == TRUE) & (is.na(most_severe_consequence) == FALSE), paste(most_severe_consequence), ""))), size = 3, position = pos, max.overlaps = Inf) +
scale_shape_manual(values =c(19, 1)) +
scale_size_continuous(trans = "reverse", range = c(0.01, 3)) +
theme_classic()
plot1

plot2 <- ggplot(annotated_data, aes(x=L2FC_HT29_DebCet_plasmid_average_zscore, y= LFC_MAGeCK_HT29_DebCet, colour = p_value_MAGeCK_HT29_DebCet, size = FDR_MAGeCK_HT29_DebCet, shape = editor)) +
geom_point(position = pos, alpha = 1) +
labs(title= "", x="FDR_MAGeCK_HT29_DebCet", y = "L2FC_HT29_DebCet_plasmid_average_zscore", size = "FDR (dabrafenib + cetuximab)", colour = "p_value_MAGeCK_HT29_DebCet", size = "FDR_MAGeCK_HT29_DebCet", shape = "editor") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_vline(xintercept=c(0.1),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_HT29_DebCet_plasmid_average_zscore > 5 & (is.na(Amino_Acid_Position) == FALSE) | L2FC_HT29_Control_plasmid_average_zscore > 3 & (is.na(Amino_Acid_Position) == FALSE), paste(Amino_Acid_Position), ifelse(L2FC_HT29_DebCet_plasmid_average_zscore > 3 & (is.na(Amino_Acid_Position) == TRUE) & (is.na(most_severe_consequence) == FALSE) | L2FC_HT29_Control_plasmid_average_zscore > 3 & (is.na(Amino_Acid_Position) == TRUE) & (is.na(most_severe_consequence) == FALSE), paste(most_severe_consequence), ""))), size = 3, position = pos, max.overlaps = Inf) +
scale_shape_manual(values =c(19, 1)) +
scale_size_continuous(trans = "reverse", range = c(0.01, 3)) +
theme_classic()
plot2

plot3 <- ggplot(annotated_data, aes(x= L2FC_HT29_DebCet_plasmid_1_zscore, y= L2FC_HT29_DebCet_plasmid_2_zscore, colour = FDR_MAGeCK_HT29_DebCet, size = p_value_MAGeCK_HT29_DebCet)) +
geom_point(alpha = 1) +
labs(title= "", x="L2FC_HT29_DebCet_plasmid_1_zscore", y = "L2FC_HT29_DebCet_plasmid_2_zscore", colour = "FDR_MAGeCK_HT29_DebCet", size = "p_value_MAGeCK_HT29_DebCet") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_vline(xintercept=c(0.1),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_HT29_DebCet_plasmid_average_zscore > 5 & (is.na(Amino_Acid_Position) == FALSE) | L2FC_HT29_Control_plasmid_average_zscore > 3 & (is.na(Amino_Acid_Position) == FALSE), paste(Amino_Acid_Position), ifelse(L2FC_HT29_DebCet_plasmid_average_zscore > 3 & (is.na(Amino_Acid_Position) == TRUE) & (is.na(most_severe_consequence) == FALSE) | L2FC_HT29_Control_plasmid_average_zscore > 3 & (is.na(Amino_Acid_Position) == TRUE) & (is.na(most_severe_consequence) == FALSE), paste(most_severe_consequence), ""))), size = 3, position = pos, max.overlaps = Inf) +
scale_shape_manual(values =c(19, 1)) +
scale_size_continuous(trans = "reverse", range = c(0.01, 3)) +
theme_classic()
plot3

plot4 <- ggplot(annotated_data, aes(x= L2FC_HT29_Control_plasmid_1_zscore, y= L2FC_HT29_Control_plasmid_2_zscore, colour = FDR_MAGeCK_HT29_control, size = p_value_MAGeCK_HT29_control)) +
geom_point(alpha = 1) +
labs(title= "", x="L2FC_HT29_Control_plasmid_1_zscore", y = "L2FC_HT29_Control_plasmid_2_zscore", colour = "FDR_MAGeCK_HT29_DebCet", size = "p_value_MAGeCK_HT29_control") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_vline(xintercept=c(0.1),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_HT29_DebCet_plasmid_average_zscore > 5 & (is.na(Amino_Acid_Position) == FALSE) | L2FC_HT29_Control_plasmid_average_zscore > 3 & (is.na(Amino_Acid_Position) == FALSE), paste(Amino_Acid_Position), ifelse(L2FC_HT29_DebCet_plasmid_average_zscore > 3 & (is.na(Amino_Acid_Position) == TRUE) & (is.na(most_severe_consequence) == FALSE) | L2FC_HT29_Control_plasmid_average_zscore > 3 & (is.na(Amino_Acid_Position) == TRUE) & (is.na(most_severe_consequence) == FALSE), paste(most_severe_consequence), ""))), size = 3, position = pos, max.overlaps = Inf) +
scale_shape_manual(values =c(19, 1)) +
scale_size_continuous(trans = "reverse", range = c(0.01, 3)) +
theme_classic()
plot4
```

```{r}
# H23 with rug plot 

plot1 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=L2FC_H23_Adag_plasmid_average_zscore, y= L2FC_H23_Sotor_plasmid_average_zscore, colour = Gene, shape = editor)) + 
geom_point(position = pos, alpha = 1, size = 3) +
labs(title= "", x="H23 adagrasib vs plasmid (z-score)", y = "H23 sotorasib vs plasmid (z-score)", colour = "gene") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_vline(xintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_H23_Adag_plasmid_average_zscore > 4.2 & (is.na(Amino_Acid_Position) == FALSE) & Gene %in% c("KRAS", "MAP2K1", "MAP2K2") | L2FC_H23_Sotor_plasmid_average_zscore > 4.2 & (is.na(Amino_Acid_Position) == FALSE) & Gene %in% c("KRAS", "MAP2K1", "MAP2K2"), paste(Amino_Acid_Position), "")), size = 3, position = pos, max.overlaps = Inf, show.legend = FALSE) +
scale_color_manual(values =c( "grey75", "red3", "royalblue", "turquoise3", "orange", "red", "springgreen4", "grey55","brown", "purple", "salmon","pink", "black")) +
scale_shape_manual(values =c(19, 1)) +
theme_classic() +
geom_rug(data = subset(annotated_data, gene_alias %in% c("non-targeting", "essential")), aes(x = L2FC_H23_Adag_plasmid_average_zscore, y = L2FC_H23_Sotor_plasmid_average_zscore, colour = gene_alias), show.legend = FALSE)
plot1
```


```{r}
# dot plots _ proliferation - PC9 average
# with rug plots

targets <- c("EGFR", "KRAS", "MYC", "PARP1", "PARP2", "MAP2K1", "MAP2K2", "BCL2", "BRAF", "AKT1", "PIK3CA")

pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

plot1 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=L2FC_PC9_Control_plasmid_average_zscore, y= L2FC_PC9_Gefit_plasmid_average_zscore, colour = Gene, shape = editor)) + 
geom_point(position = pos, alpha = 1, size =3) +
labs(title= " ", x="PC9 control vs plasmid (z-score)", y = "PC9 gefitinib vs plasmid (z-score)", colour = "gene") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_vline(xintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_PC9_Gefit_plasmid_average_zscore > 12 & (is.na(Amino_Acid_Position) == FALSE), paste(Gene, Amino_Acid_Position), ifelse(L2FC_PC9_Gefit_plasmid_average_zscore < -3 & L2FC_PC9_Control_plasmid_average_zscore > -2 & (is.na(Amino_Acid_Position) == FALSE), paste(Gene, Amino_Acid_Position, most_severe_consequence), ifelse(L2FC_PC9_Gefit_plasmid_average_zscore > 12 & (is.na(Amino_Acid_Position) == TRUE) | L2FC_PC9_Gefit_plasmid_average_zscore < -3 & L2FC_PC9_Control_plasmid_average_zscore > -2 & (is.na(Amino_Acid_Position) == TRUE), paste(Gene, most_severe_consequence), "")))), size = 3, position = pos, max.overlaps = Inf, show.legend = FALSE) +
scale_color_manual(values =c("grey75", "red3", "royalblue", "turquoise3", "orange", "red", "springgreen4", "grey55","brown", "purple", "salmon","pink", "black")) +
scale_shape_manual(values =c(19, 1)) +
theme_classic() +
  geom_rug(data = subset(annotated_data, gene_alias %in% c("non-targeting", "essential")), aes(x = L2FC_PC9_Control_plasmid_average_zscore, y = L2FC_PC9_Gefit_plasmid_average_zscore, colour = gene_alias), show.legend = FALSE)
plot1

plot2 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=L2FC_PC9_Control_plasmid_average_zscore, y= L2FC_PC9_Osim_plasmid_average_zscore, colour = Gene, shape = editor)) + 
geom_point(position = pos, alpha = 1, size =3) +
labs(title= " ", x="PC9 control vs plasmid (z-score)", y = "PC9 osimertinib vs plasmid (z-score)", colour = "gene") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_vline(xintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_PC9_Osim_plasmid_average_zscore > 5.5 & (is.na(Amino_Acid_Position) == FALSE), paste(Amino_Acid_Position), ifelse(L2FC_PC9_Osim_plasmid_average_zscore < -3 & L2FC_PC9_Control_plasmid_average_zscore > -2 & (is.na(Amino_Acid_Position) == FALSE), paste(Gene, Amino_Acid_Position, most_severe_consequence), ifelse(L2FC_PC9_Osim_plasmid_average_zscore > 5.5 & (is.na(Amino_Acid_Position) == TRUE) | L2FC_PC9_Osim_plasmid_average_zscore < -3 & L2FC_PC9_Control_plasmid_average_zscore > -2 & (is.na(Amino_Acid_Position) == TRUE), paste(most_severe_consequence), "")))), size = 3, position = pos, max.overlaps = Inf, show.legend = FALSE) +
scale_color_manual(values =c("grey75", "red3", "royalblue", "turquoise3", "orange", "red", "springgreen4", "grey55","brown", "purple", "salmon","pink", "black" )) +
scale_shape_manual(values =c(19, 1)) +
theme_classic() +
  geom_rug(data = subset(annotated_data, gene_alias %in% c("non-targeting", "essential")), aes(x = L2FC_PC9_Control_plasmid_average_zscore, y = L2FC_PC9_Osim_plasmid_average_zscore, colour = gene_alias), show.legend = FALSE)
plot2
```

```{r}
# dot plots _ proliferation - PC9 average - for presentations
# with rug plots

targets <- c("EGFR", "KRAS", "MYC", "PARP1", "PARP2", "MAP2K1", "MAP2K2", "BCL2", "BRAF", "AKT1", "PIK3CA")

pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

plot1 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=L2FC_PC9_Control_plasmid_average_zscore, y= L2FC_PC9_Gefit_plasmid_average_zscore, colour = Gene, shape = editor)) + 
geom_point(position = pos, alpha = 1, size =3) +
labs(title= " ", x="PC9 control vs plasmid (z-score)", y = "PC9 gefitinib vs plasmid (z-score)", colour = "gene") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_vline(xintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_PC9_Gefit_plasmid_average_zscore > 12 & (is.na(Amino_Acid_Position) == FALSE), paste(Gene, Amino_Acid_Position), ifelse(L2FC_PC9_Gefit_plasmid_average_zscore < -3.5 & L2FC_PC9_Control_plasmid_average_zscore > -0.5 & (is.na(Amino_Acid_Position) == FALSE), paste(Gene, most_severe_consequence), ifelse(L2FC_PC9_Gefit_plasmid_average_zscore > 12 & (is.na(Amino_Acid_Position) == TRUE) | L2FC_PC9_Gefit_plasmid_average_zscore < -3.5 & L2FC_PC9_Control_plasmid_average_zscore > -0.5 & (is.na(Amino_Acid_Position) == TRUE), paste(Gene, most_severe_consequence), "")))), size = 3, position = pos, max.overlaps = Inf, show.legend = FALSE) +
scale_color_manual(values =c("grey75", "red3", "royalblue", "turquoise3", "orange", "red", "springgreen4", "grey55","brown", "purple", "salmon","pink", "black")) +
scale_shape_manual(values =c(19, 1)) +
theme_classic() +
  geom_rug(data = subset(annotated_data, gene_alias %in% c("non-targeting", "essential")), aes(x = L2FC_PC9_Control_plasmid_average_zscore, y = L2FC_PC9_Gefit_plasmid_average_zscore, colour = gene_alias), show.legend = FALSE)
plot1

plot2 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=L2FC_PC9_Control_plasmid_average_zscore, y= L2FC_PC9_Osim_plasmid_average_zscore, colour = Gene, shape = editor)) + 
geom_point(position = pos, alpha = 1, size =3) +
labs(title= " ", x="PC9 control vs plasmid (z-score)", y = "PC9 osimertinib vs plasmid (z-score)", colour = "gene") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_vline(xintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_PC9_Osim_plasmid_average_zscore > 5.5 & (is.na(Amino_Acid_Position) == FALSE), paste(Gene), ifelse(L2FC_PC9_Osim_plasmid_average_zscore < -5 & L2FC_PC9_Control_plasmid_average_zscore > -1 & (is.na(Amino_Acid_Position) == FALSE), paste(Gene, most_severe_consequence), ifelse(L2FC_PC9_Osim_plasmid_average_zscore > 5.5 & (is.na(Amino_Acid_Position) == TRUE) | L2FC_PC9_Osim_plasmid_average_zscore < -5 & L2FC_PC9_Control_plasmid_average_zscore > -1 & (is.na(Amino_Acid_Position) == TRUE), paste(most_severe_consequence), "")))), size = 3, position = pos, max.overlaps = Inf, show.legend = FALSE) +
scale_color_manual(values =c("grey75", "red3", "royalblue", "turquoise3", "orange", "red", "springgreen4", "grey55","brown", "purple", "salmon","pink", "black" )) +
scale_shape_manual(values =c(19, 1)) +
theme_classic() +
  geom_rug(data = subset(annotated_data, gene_alias %in% c("non-targeting", "essential")), aes(x = L2FC_PC9_Control_plasmid_average_zscore, y = L2FC_PC9_Osim_plasmid_average_zscore, colour = gene_alias), show.legend = FALSE)
plot2
```
```{r}
# dot plots _ proliferation - PC9 average - for presentations
# with rug plots

targets <- c("EGFR", "KRAS", "MYC", "PARP1", "PARP2", "MAP2K1", "MAP2K2", "BCL2", "BRAF", "AKT1", "PIK3CA")

plot1 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=L2FC_PC9_Control_plasmid_average_zscore, y= L2FC_PC9_Gefit_plasmid_average_zscore, colour = Gene, shape = editor)) + 
geom_point(alpha = 1, size =3) +
labs(title= " ", x="PC9 control vs plasmid (z-score)", y = "PC9 gefitinib vs plasmid (z-score)", colour = "gene") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_vline(xintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_PC9_Gefit_plasmid_average_zscore > 12 & (is.na(Amino_Acid_Position) == FALSE), paste(Gene, Amino_Acid_Position), ifelse(L2FC_PC9_Gefit_plasmid_average_zscore < -3.5 & L2FC_PC9_Control_plasmid_average_zscore > -0.5 & (is.na(Amino_Acid_Position) == FALSE), paste(Gene, most_severe_consequence), ifelse(L2FC_PC9_Gefit_plasmid_average_zscore > 12 & (is.na(Amino_Acid_Position) == TRUE) | L2FC_PC9_Gefit_plasmid_average_zscore < -3.5 & L2FC_PC9_Control_plasmid_average_zscore > -0.5 & (is.na(Amino_Acid_Position) == TRUE), paste(Gene, most_severe_consequence), "")))), size = 3, position = pos, max.overlaps = Inf, show.legend = FALSE) +
scale_color_manual(values =c("grey75", "red3", "royalblue", "turquoise3", "orange", "red", "springgreen4", "grey55","brown", "purple", "salmon","pink", "black")) +
scale_shape_manual(values =c(19, 1)) +
theme_classic() +
  geom_rug(data = subset(annotated_data, gene_alias %in% c("non-targeting", "essential")), aes(x = L2FC_PC9_Control_plasmid_average_zscore, y = L2FC_PC9_Gefit_plasmid_average_zscore, colour = gene_alias), show.legend = FALSE)
plot1

plot2 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=L2FC_PC9_Control_plasmid_average_zscore, y= L2FC_PC9_Osim_plasmid_average_zscore, colour = Gene, shape = editor)) + 
geom_point(alpha = 1, size =3) +
labs(title= " ", x="PC9 control vs plasmid (z-score)", y = "PC9 osimertinib vs plasmid (z-score)", colour = "gene") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_vline(xintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_PC9_Osim_plasmid_average_zscore > 5.5 & (is.na(Amino_Acid_Position) == FALSE), paste(Gene), ifelse(L2FC_PC9_Osim_plasmid_average_zscore < -5 & L2FC_PC9_Control_plasmid_average_zscore > -1 & (is.na(Amino_Acid_Position) == FALSE), paste(Gene, most_severe_consequence), ifelse(L2FC_PC9_Osim_plasmid_average_zscore > 5.5 & (is.na(Amino_Acid_Position) == TRUE) | L2FC_PC9_Osim_plasmid_average_zscore < -5 & L2FC_PC9_Control_plasmid_average_zscore > -1 & (is.na(Amino_Acid_Position) == TRUE), paste(most_severe_consequence), "")))), size = 3, position = pos, max.overlaps = Inf, show.legend = FALSE) +
scale_color_manual(values =c("grey75", "red3", "royalblue", "turquoise3", "orange", "red", "springgreen4", "grey55","brown", "purple", "salmon","pink", "black" )) +
scale_shape_manual(values =c(19, 1)) +
theme_classic() +
  geom_rug(data = subset(annotated_data, gene_alias %in% c("non-targeting", "essential")), aes(x = L2FC_PC9_Control_plasmid_average_zscore, y = L2FC_PC9_Osim_plasmid_average_zscore, colour = gene_alias), show.legend = FALSE)
plot2
```

```{r}
# dot plots _ proliferation - HT-29 - ***FOR NG EDITORS***
# using gene_alias and FDR
targets <- c("EGFR", "KRAS", "MYC", "PARP1", "PARP2", "MAP2K1", "MAP2K2", "BCL2", "BRAF", "AKT1", "PIK3CA")

annotated_data$gene_alias <- factor(annotated_data$gene_alias, levels = c("AKT1", "BCL2", "BRAF", "EGFR", "KRAS", "MAP2K1", "MAP2K2", "MYC", "PARP1", "PARP2", "PIK3CA", "non-essential", "intergenic",  "non-targeting", "essential"))

pos <- position_jitter(width = 0.2, height = 0.2)

plot1 <- ggplot(subset(annotated_data, gene_alias %in% c(targets)), aes(x=L2FC_HT29_Control_plasmid_average_zscore, y= L2FC_HT29_Tram_plasmid_average_zscore, colour = gene_alias, shape = editor)) +
geom_point(position = pos, alpha = 1, size = 1) +
labs(title= "", x="HT-29 control vs plasmid (z-score)", y = "HT-29 trametinib vs plasmid (z-score)", colour = "gene", shape = "editor") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_vline(xintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_HT29_Tram_plasmid_average_zscore > 9 & (is.na(Amino_Acid_Position) == FALSE) | L2FC_HT29_Control_plasmid_average_zscore > 3 & (is.na(Amino_Acid_Position) == FALSE), paste(Gene,Amino_Acid_Position),
                      ifelse(L2FC_HT29_Tram_plasmid_average_zscore > 4 & (is.na(Amino_Acid_Position) == FALSE) | L2FC_HT29_Control_plasmid_average_zscore > 3 & (is.na(Amino_Acid_Position) == FALSE), paste(Amino_Acid_Position), 
                             ifelse( L2FC_HT29_Tram_plasmid_average_zscore > 4 & (is.na(Amino_Acid_Position) == TRUE) & (is.na(most_severe_consequence) == FALSE), paste(Gene, most_severe_consequence), ifelse(L2FC_HT29_Tram_plasmid_average_zscore < -3.5 & L2FC_HT29_Control_plasmid_average_zscore > - 0.25, paste(Gene, most_severe_consequence),                                                                                                                                                                  ifelse(L2FC_HT29_Tram_plasmid_average_zscore > 3 & L2FC_HT29_Control_plasmid_average_zscore < -3 & (is.na(Amino_Acid_Position) == FALSE), paste(Amino_Acid_Position), "")))))), size = 3, position = pos, max.overlaps = Inf, show.legend = FALSE) +
scale_color_manual(values =c("royalblue", "turquoise3", "orange", "red", "springgreen4", "grey55","brown", "purple", "salmon","pink", "black", "grey75", "red3")) +
scale_shape_manual(values =c(2, 1)) +
theme_classic()
plot1 <- plot1 +
  geom_rug(data = subset(annotated_data, gene_alias %in% c("non-targeting", "essential")), aes(x = L2FC_HT29_Control_plasmid_average_zscore, y = L2FC_HT29_Tram_plasmid_average_zscore, colour = gene_alias), show.legend = FALSE)
plot1
ggsave("Fig4a.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

plot2 <- ggplot(subset(annotated_data, gene_alias %in% c(targets)), aes(x=L2FC_HT29_Control_plasmid_average_zscore, y= L2FC_HT29_Pict_plasmid_average_zscore, colour = gene_alias, shape = editor)) + 
geom_point(position = pos, alpha = 1, size =1) +
labs(title= "", x="HT-29 control vs plasmid (z-score)", y = "HT-29 pictilisib vs plasmid (z-score)", colour = "gene", shape = "editor") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_vline(xintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_HT29_Pict_plasmid_average_zscore > 4 & (is.na(Amino_Acid_Position) == FALSE) | L2FC_HT29_Control_plasmid_average_zscore > 4 & (is.na(Amino_Acid_Position) == FALSE), paste(Gene, Amino_Acid_Position), "")), size = 3, position = pos, max.overlaps = Inf, show.legend = FALSE) +
scale_color_manual(values =c("royalblue", "turquoise3", "orange", "red", "springgreen4", "grey55","brown", "purple", "salmon","pink", "black", "grey75", "red3")) +
scale_shape_manual(values =c(2, 1)) +
theme_classic()
plot2 <- plot2 +
  geom_rug(data = subset(annotated_data, gene_alias %in% c("non-targeting", "essential")), aes(x = L2FC_HT29_Control_plasmid_average_zscore, y = L2FC_HT29_Pict_plasmid_average_zscore, colour = gene_alias), show.legend = FALSE)
plot2
ggsave("Fig2a.eps", width = 6, height = 4, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

plot3 <- ggplot(subset(annotated_data, gene_alias %in% c(targets)), aes(x=L2FC_HT29_Control_plasmid_average_zscore, y= L2FC_HT29_DebCet_plasmid_average_zscore, colour = gene_alias, shape = editor)) + 
geom_point(position = pos, alpha = 1, size = 1) +
labs(title= "", x="HT-29 control vs plasmid (z-score)", y = "HT-29 dabrafenib + cetuximab vs plasmid (z-score)", colour = "gene", shape = "editor") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_vline(xintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label= ifelse(L2FC_HT29_DebCet_plasmid_average_zscore > 10 & (is.na(Amino_Acid_Position) == FALSE) | L2FC_HT29_Control_plasmid_average_zscore > 3 & (is.na(Amino_Acid_Position) == FALSE), paste(Gene, Amino_Acid_Position),
                      ifelse(L2FC_HT29_DebCet_plasmid_average_zscore > 5 & (is.na(Amino_Acid_Position) == FALSE) | L2FC_HT29_Control_plasmid_average_zscore > 3 & (is.na(Amino_Acid_Position) == FALSE), paste(Amino_Acid_Position), 
                                 ifelse(L2FC_HT29_DebCet_plasmid_average_zscore > 3 & (is.na(Amino_Acid_Position) == TRUE) & (is.na(most_severe_consequence) == FALSE) | L2FC_HT29_Control_plasmid_average_zscore > 3 & (is.na(Amino_Acid_Position) == TRUE) & (is.na(most_severe_consequence) == FALSE), paste(most_severe_consequence), "")))), size = 3, position = pos, max.overlaps = Inf, show.legend = FALSE) +
scale_color_manual(values =c("royalblue", "turquoise3", "orange", "red", "springgreen4", "grey55","brown", "purple", "salmon","pink", "black", "grey75", "red3")) +
scale_shape_manual(values =c(2, 1)) +
theme_classic()
plot3 <- plot3 +
  geom_rug(data = subset(annotated_data, gene_alias %in% c("non-targeting", "essential")), aes(x = L2FC_HT29_Control_plasmid_average_zscore, y = L2FC_HT29_DebCet_plasmid_average_zscore, colour = gene_alias), show.legend = FALSE)
plot3
ggsave("Fig4b.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

```

```{r}
# dot plots _ proliferation - PC9 average *** FOR NG EDITORS***
# with rug plots

targets <- c("EGFR", "KRAS", "MYC", "PARP1", "PARP2", "MAP2K1", "MAP2K2", "BCL2", "BRAF", "AKT1", "PIK3CA")

pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

plot1 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=L2FC_PC9_Control_plasmid_average_zscore, y= L2FC_PC9_Gefit_plasmid_average_zscore, colour = Gene, shape = editor)) + 
geom_point(position = pos, alpha = 1, size =1) +
labs(title= " ", x="PC9 control vs plasmid (z-score)", y = "PC9 gefitinib vs plasmid (z-score)", colour = "gene") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_vline(xintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_PC9_Gefit_plasmid_average_zscore > 12 & (is.na(Amino_Acid_Position) == FALSE), paste(Gene, Amino_Acid_Position), ifelse(L2FC_PC9_Gefit_plasmid_average_zscore < -3.5 & L2FC_PC9_Control_plasmid_average_zscore > -0.5 & (is.na(Amino_Acid_Position) == FALSE), paste(Gene, Amino_Acid_Position, most_severe_consequence), ifelse(L2FC_PC9_Gefit_plasmid_average_zscore > 12 & (is.na(Amino_Acid_Position) == TRUE) | L2FC_PC9_Gefit_plasmid_average_zscore < -3.5 & L2FC_PC9_Control_plasmid_average_zscore > -0.5 & (is.na(Amino_Acid_Position) == TRUE), paste(Gene, most_severe_consequence), "")))), size = 3, position = pos, max.overlaps = Inf, show.legend = FALSE) +
scale_color_manual(values =c("grey75", "red3", "royalblue", "turquoise3", "orange", "red", "springgreen4", "grey55","brown", "purple", "salmon","pink", "black")) +
scale_shape_manual(values =c(2, 1)) +
theme_classic() +
  geom_rug(data = subset(annotated_data, gene_alias %in% c("non-targeting", "essential")), aes(x = L2FC_PC9_Control_plasmid_average_zscore, y = L2FC_PC9_Gefit_plasmid_average_zscore, colour = gene_alias), show.legend = FALSE)
plot1
ggsave("Fig2c.eps", width = 6, height = 4, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

plot2 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=L2FC_PC9_Control_plasmid_average_zscore, y= L2FC_PC9_Osim_plasmid_average_zscore, colour = Gene, shape = editor)) + 
geom_point(position = pos, alpha = 1, size =1) +
labs(title= " ", x="PC9 control vs plasmid (z-score)", y = "PC9 osimertinib vs plasmid (z-score)", colour = "gene") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_vline(xintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_PC9_Osim_plasmid_average_zscore > 5.5 & (is.na(Amino_Acid_Position) == FALSE), paste(Amino_Acid_Position), ifelse(L2FC_PC9_Osim_plasmid_average_zscore < -5 & L2FC_PC9_Control_plasmid_average_zscore > -1 & (is.na(Amino_Acid_Position) == FALSE), paste(Gene, Amino_Acid_Position, most_severe_consequence), ifelse(L2FC_PC9_Osim_plasmid_average_zscore > 5.5 & (is.na(Amino_Acid_Position) == TRUE) | L2FC_PC9_Osim_plasmid_average_zscore < -5 & L2FC_PC9_Control_plasmid_average_zscore > -1 & (is.na(Amino_Acid_Position) == TRUE), paste(most_severe_consequence), "")))), size = 3, position = pos, max.overlaps = Inf, show.legend = FALSE) +
scale_color_manual(values =c("grey75", "red3", "royalblue", "turquoise3", "orange", "red", "springgreen4", "grey55","brown", "purple", "salmon","pink", "black" )) +
scale_shape_manual(values =c(2, 1)) +
theme_classic() +
  geom_rug(data = subset(annotated_data, gene_alias %in% c("non-targeting", "essential")), aes(x = L2FC_PC9_Control_plasmid_average_zscore, y = L2FC_PC9_Osim_plasmid_average_zscore, colour = gene_alias), show.legend = FALSE)
plot2
ggsave("Fig2d.eps", width = 6, height = 4, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

```

```{r}
# H23 with rug plot ***FOR NG EDITORS***

plot1 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=L2FC_H23_Adag_plasmid_average_zscore, y= L2FC_H23_Sotor_plasmid_average_zscore, colour = Gene, shape = editor)) + 
geom_point(position = pos, alpha = 1, size = 1) +
labs(title= "", x="H23 adagrasib vs plasmid (z-score)", y = "H23 sotorasib vs plasmid (z-score)", colour = "gene") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_vline(xintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_H23_Adag_plasmid_average_zscore > 4.2 & (is.na(Amino_Acid_Position) == FALSE) & Gene %in% c("KRAS", "MAP2K1", "MAP2K2") | L2FC_H23_Sotor_plasmid_average_zscore > 4.2 & (is.na(Amino_Acid_Position) == FALSE) & Gene %in% c("KRAS", "MAP2K1", "MAP2K2"), paste(Amino_Acid_Position), "")), size = 3, position = pos, max.overlaps = Inf, show.legend = FALSE) +
scale_color_manual(values =c( "grey75", "red3", "royalblue", "turquoise3", "orange", "red", "springgreen4", "grey55","brown", "purple", "salmon","pink", "black")) +
scale_shape_manual(values =c(2, 1)) +
theme_classic() +
geom_rug(data = subset(annotated_data, gene_alias %in% c("non-targeting", "essential")), aes(x = L2FC_H23_Adag_plasmid_average_zscore, y = L2FC_H23_Sotor_plasmid_average_zscore, colour = gene_alias), show.legend = FALSE)
plot1
ggsave("Fig2b.eps", width = 6, height = 4, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

```

```{r}
# investigate BCL2 driver mutation outlier - does not correlate between replicates
# impose a 10 fold change difference between reads like in Cancer Cell paper - like a 50 fold difference

annotated_data %>% 
  filter(L2FC_MHHES1_Control_plasmid_average_zscore >3) %>%
  select(Gene, guide, MHHES1_ABE_Control_exp1_reads, MHHES1_ABE_Control_exp2_reads)

annotated_data %>% 
  filter(MHHES1_ABE_Control_exp1_reads > 10*MHHES1_ABE_Control_exp2_reads & MHHES1_ABE_Control_exp2_reads != 0 | MHHES1_ABE_Control_exp2_reads > 10*MHHES1_ABE_Control_exp1_reads & MHHES1_ABE_Control_exp1_reads != 0) %>%
  select(Gene, guide, MHHES1_ABE_Control_exp1_reads, MHHES1_ABE_Control_exp2_reads, MHHES1_ABE_T0_exp1_reads, MHHES1_ABE_T0_exp2_reads) 

outliers <- annotated_data %>% 
  filter(MHHES1_ABE_Control_exp1_reads > 10*MHHES1_ABE_Control_exp2_reads & MHHES1_ABE_Control_exp2_reads != 0 | MHHES1_ABE_Control_exp2_reads > 10*MHHES1_ABE_Control_exp1_reads & MHHES1_ABE_Control_exp1_reads != 0) %>%
  select(sgRNA_ID, Gene, guide, MHHES1_ABE_Control_exp1_reads, MHHES1_ABE_Control_exp2_reads, MHHES1_ABE_T0_exp1_reads, MHHES1_ABE_T0_exp2_reads) %>%
  select(sgRNA_ID)

test_clean_MHHES1_data <- annotated_data %>%
  filter(sgRNA_ID %in% c(outliers$sgRNA_ID) == FALSE)
  
annotated_data %>% 
  filter(MHHES1_ABE_Olap_exp1_reads > 10*MHHES1_ABE_Olap_exp2_reads & MHHES1_ABE_Olap_exp2_reads != 0 | MHHES1_ABE_Control_exp2_reads > 10*MHHES1_ABE_Control_exp1_reads & MHHES1_ABE_Control_exp1_reads != 0) %>%
  select(Gene, guide, MHHES1_ABE_Olap_exp1_reads, MHHES1_ABE_Olap_exp2_reads, MHHES1_ABE_T0_exp1_reads, MHHES1_ABE_T0_exp2_reads)

test_clean_MHHES1_data %>% 
  filter(MHHES1_ABE_Olap_exp1_reads > 10*MHHES1_ABE_Olap_exp2_reads & MHHES1_ABE_Olap_exp2_reads != 0 | MHHES1_ABE_Control_exp2_reads > 10*MHHES1_ABE_Control_exp1_reads & MHHES1_ABE_Control_exp1_reads != 0) %>%
  select(Gene, guide, MHHES1_ABE_Olap_exp1_reads, MHHES1_ABE_Olap_exp2_reads, MHHES1_ABE_T0_exp1_reads, MHHES1_ABE_T0_exp2_reads)

annotated_data %>% 
  filter(MHHES1_ABE_Nirap_exp1_reads > 10*MHHES1_ABE_Nirap_exp2_reads & MHHES1_ABE_Nirap_exp2_reads != 0 | MHHES1_ABE_Nirap_exp2_reads > 10*MHHES1_ABE_Control_exp1_reads & MHHES1_ABE_Control_exp1_reads != 0) %>%
  select(Gene, guide, MHHES1_ABE_Nirap_exp1_reads, MHHES1_ABE_Nirap_exp2_reads, MHHES1_ABE_T0_exp1_reads, MHHES1_ABE_T0_exp2_reads)

test_clean_MHHES1_data %>% 
  filter(MHHES1_ABE_Nirap_exp1_reads > 10*MHHES1_ABE_Nirap_exp2_reads & MHHES1_ABE_Nirap_exp2_reads != 0 | MHHES1_ABE_Nirap_exp2_reads > 10*MHHES1_ABE_Control_exp1_reads & MHHES1_ABE_Control_exp1_reads != 0) %>%
  select(Gene, guide, MHHES1_ABE_Nirap_exp1_reads, MHHES1_ABE_Nirap_exp2_reads, MHHES1_ABE_T0_exp1_reads, MHHES1_ABE_T0_exp2_reads)

#check hit read counts
test_clean_MHHES1_data %>% 
  filter(L2FC_MHHES1_Control_plasmid_average_zscore >3 | L2FC_MHHES1_Olap_plasmid_average_zscore > 3 | L2FC_MHHES1_Nirap_plasmid_average_zscore > 3) %>%
  select(sgRNA_ID, Gene, guide, MHHES1_ABE_Control_exp1_reads, MHHES1_ABE_Control_exp2_reads, MHHES1_ABE_Olap_exp1_reads, MHHES1_ABE_Olap_exp2_reads, MHHES1_ABE_Nirap_exp1_reads, MHHES1_ABE_Nirap_exp2_reads)
```

```{r}
# investigate PARP1/2 hits

test_clean_MHHES1_data %>% filter(L2FC_MHHES1_Control_plasmid_average_zscore >3) %>%
  select(Gene, G_guide, most_severe_consequence, Amino_Acid_Position, editor, Edit_Location)

test_clean_MHHES1_data %>% filter(L2FC_MHHES1_Olap_plasmid_average_zscore >3) %>%
  select(Gene, G_guide, most_severe_consequence, Amino_Acid_Position, editor, Protein_Change)

test_clean_MHHES1_data %>% filter(L2FC_MHHES1_Nirap_plasmid_average_zscore >3) %>%
  select(Gene, G_guide, most_severe_consequence, Amino_Acid_Position, editor, Protein_Change)

test_clean_MHHES1_data %>% filter(L2FC_MHHES1_Olap_plasmid_average_zscore < -3 & L2FC_MHHES1_Control_plasmid_average_zscore > -1) %>%
  select(Gene, G_guide, most_severe_consequence, Amino_Acid_Position, editor, Protein_Change)

test_clean_MHHES1_data %>% filter(L2FC_MHHES1_Nirap_plasmid_average_zscore < -3 & L2FC_MHHES1_Control_plasmid_average_zscore > -1) %>%
  select(Gene, G_guide, most_severe_consequence, Amino_Acid_Position, editor, Protein_Change)

test_clean_MHHES1_data %>% filter(L2FC_MHHES1_Olap_plasmid_average_zscore < -3 & L2FC_MHHES1_Nirap_plasmid_average_zscore < -3 & L2FC_MHHES1_Control_plasmid_average_zscore > -0.8) %>%
  select(Gene, Amino_Acid_Position, editor, sgRNA_ID, L2FC_MHHES1_Olap_plasmid_average_zscore, L2FC_MHHES1_Nirap_plasmid_average_zscore, L2FC_MHHES1_Control_plasmid_average_zscore)

PARP_sensitors <- test_clean_MHHES1_data %>% filter(L2FC_MHHES1_Olap_plasmid_average_zscore < -3 & L2FC_MHHES1_Nirap_plasmid_average_zscore < -3 & L2FC_MHHES1_Control_plasmid_average_zscore > -1) %>% 
  select(sgRNA_ID)

PARP_resistors <- test_clean_MHHES1_data %>% filter(L2FC_MHHES1_Olap_plasmid_average_zscore > 3 & L2FC_MHHES1_Olap_plasmid_1_zscore >2 & L2FC_MHHES1_Olap_plasmid_2_zscore > 2 | L2FC_MHHES1_Nirap_plasmid_average_zscore > 3 & L2FC_MHHES1_Nirap_plasmid_1_zscore >2 & L2FC_MHHES1_Nirap_plasmid_2_zscore >2) %>%
  select(sgRNA_ID)

test_clean_MHHES1_data %>% filter(L2FC_MHHES1_Olap_plasmid_average_zscore > 3 | L2FC_MHHES1_Nirap_plasmid_average_zscore > 3) %>%
  select(Gene, most_severe_consequence, Amino_Acid_Position, Protein_Change)

```

```{r}
# dot plots _ proliferation - MHH-ES-1

targets <- c("EGFR", "KRAS", "MYC", "PARP1", "PARP2", "MAP2K1", "MAP2K2", "BCL2", "BRAF", "AKT1", "PIK3CA")

pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

plot1 <- ggplot(subset(test_clean_MHHES1_data, Gene %in% c(targets)), aes(x=L2FC_MHHES1_Control_plasmid_average_zscore, y= L2FC_MHHES1_Olap_plasmid_average_zscore, colour = Gene, shape = editor)) + 
geom_point(position = pos, alpha = 1, size =3) +
labs(title= "", x="MHH-ES-1 control vs plasmid (z-score)", y = "MHH-ES-1 olaparib vs plasmid (z-score)", colour = "gene") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_vline(xintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_MHHES1_Olap_plasmid_average_zscore > 3.1 & (is.na(Amino_Acid_Position) == FALSE), paste(Amino_Acid_Position), ifelse(L2FC_MHHES1_Olap_plasmid_1_zscore > 3.1 & (is.na(Amino_Acid_Position) == TRUE),  paste(most_severe_consequence, Gene), ifelse(L2FC_MHHES1_Control_plasmid_average_zscore > 3.5, paste(most_severe_consequence, Gene), ifelse(sgRNA_ID %in% PARP_sensitors$sgRNA_ID, paste(Gene, Amino_Acid_Position), ""))))), size = 3, position = pos, max.overlaps = Inf) +
scale_color_manual( values =c("royalblue", "turquoise3", "orange", "red", "springgreen4", "grey55", "brown", "purple", "salmon", "pink", "black")) +
scale_shape_manual(values =c(19, 1)) +
theme_classic() 
plot1
ggsave("Fig2c1.eps", width = 6, height = 4, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

plot2 <- ggplot(subset(test_clean_MHHES1_data, Gene %in% c(targets)), aes(x=L2FC_MHHES1_Control_plasmid_average_zscore, y= L2FC_MHHES1_Nirap_plasmid_average_zscore, colour = Gene, shape = editor)) + 
geom_point(position = pos, alpha = 1, size =3) +
labs(title= "", x="MHH-ES-1 control vs plasmid (z-score)", y = "MHH-ES-1 niraparib vs plasmid (z-score)", colour = "gene") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_vline(xintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_MHHES1_Nirap_plasmid_average_zscore > 4 & (is.na(Amino_Acid_Position) == FALSE), paste(Amino_Acid_Position), ifelse( L2FC_MHHES1_Nirap_plasmid_1_zscore > 4 & (is.na(Amino_Acid_Position) == TRUE),  paste(most_severe_consequence, Gene), ifelse(L2FC_MHHES1_Control_plasmid_average_zscore > 4, paste(most_severe_consequence, Gene), ifelse(sgRNA_ID %in% PARP_sensitors$sgRNA_ID, paste(Gene, Amino_Acid_Position), ""))))), size = 3, position = pos, max.overlaps = Inf) +
scale_color_manual( values =c("royalblue", "turquoise3", "orange", "red", "springgreen4", "grey55", "brown", "purple", "salmon", "pink", "black")) +
scale_shape_manual(values =c(19, 1)) +
theme_classic()
plot2
ggsave("Fig2c2.eps", width = 6, height = 4, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")
```

```{r}
# dot plots _ proliferation

targets <- c("EGFR", "KRAS", "MYC", "PARP1", "PARP2", "MAP2K1", "MAP2K2", "BCL2", "BRAF", "AKT1", "PIK3CA")

pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

plot1 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=Amino_Acid_Position_simple, y= L2FC_HT29_Control_plasmid_average_zscore, colour = most_severe_consequence)) + 
geom_point(position = pos, shape = 16, alpha = 0.8, size =2) +
labs(title= "", x="amino acid position", y = "L2FC_HT29_Control_plasmid_average") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
facet_wrap(~Gene, scales = "free_x") +
scale_color_manual(values =c("grey65", "black", "black", "orange", "blue", "red", "purple", "grey65", "grey65", "grey65", "grey65")) +
theme_classic()
plot1

plot2 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=Amino_Acid_Position_simple, y= L2FC_H23_Control_plasmid_1_zscore, colour = most_severe_consequence)) + 
geom_point(position = pos, shape = 16, alpha = 0.8, size =2) +
labs(title= "", x="amino acid position", y = "L2FC_H23_Control_plasmid_1") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
facet_wrap(~Gene, scales = "free_x") +
scale_color_manual(values =c("grey65", "black", "black", "orange", "blue", "red", "purple", "grey65", "grey65", "grey65", "grey65")) +
theme_classic()
plot2

plot3 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=Amino_Acid_Position_simple, y= L2FC_PC9_Control_plasmid_1_zscore, colour = most_severe_consequence)) + 
geom_point(position = pos, shape = 16, alpha = 0.8, size =2) +
labs(title= "", x="amino acid position", y = "L2FC_PC9_Control_plasmid_1") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
facet_wrap(~Gene, scales = "free_x") +
scale_color_manual(values =c("grey65", "black", "black", "orange", "blue", "red", "purple", "grey65", "grey65", "grey65", "grey65")) +
theme_classic()
plot3

plot4 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=Amino_Acid_Position_simple, y= L2FC_MHHES1_Control_plasmid_1_zscore, colour = most_severe_consequence)) + 
geom_point(position = pos, shape = 16, alpha = 0.8, size =2) +
labs(title= "", x="amino acid position", y = "L2FC_MHHES1_Control_plasmid_1") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
facet_wrap(~Gene, scales = "free_x") +
scale_color_manual(values =c("grey65", "black", "black", "orange", "blue", "red", "purple", "grey65", "grey65", "grey65", "grey65")) +
theme_classic()
plot4
```

```{r}
# dot plots _ proliferation - gene essentially

targets <- c("KRAS", "MYC", "BRAF")

pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

plot1 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=L2FC_HT29_Control_plasmid_average_zscore, y= L2FC_H23_Control_plasmid_average_zscore, colour = Gene)) + 
geom_point(position = pos, shape = 16, alpha = 0.8, size =3) +
labs(title= "", x="L2FC_HT29_Control_plasmid_average", y = "L2FC_H23_Control_plasmid_average") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
scale_color_manual(values =c("red", "orange", "black")) +
theme_bw()
plot1

targets <- c("EGFR", "MYC", "BRAF")

plot2 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=L2FC_HT29_Control_plasmid_average_zscore, y= L2FC_PC9_Control_plasmid_average_zscore, colour = Gene)) + 
geom_point(position = pos, shape = 16, alpha = 1, size =3) +
labs(title= "", x="HT-29 control vs plasmid (z-score)", y = "PC9 control vs plasmid (z-score)", colour = "gene") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_vline(xintercept=c(0,0),linetype="dotted") +
scale_color_manual(values =c("red", "orange", "black")) +
theme_classic()
plot2
ggsave("Fig1c.eps", width = 5, height = 4, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

targets <- c("EGFR", "MYC", "KRAS")

plot3 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=L2FC_PC9_Control_plasmid_average_zscore, y= L2FC_H23_Control_plasmid_average_zscore, colour = Gene)) + 
geom_point(position = pos, shape = 16, alpha = 0.8, size =3) +
labs(title= "", x="L2FC_PC9_Control_plasmid_average", y = "L2FC_H23_Control_plasmid_average") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
scale_color_manual(values =c("red", "orange", "black")) +
theme_bw()
plot3

targets <- c("BRAF", "MYC", "PARP1")

plot4 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=L2FC_HT29_Control_plasmid_average_zscore, y= L2FC_MHHES1_Control_plasmid_average_zscore, colour = Gene)) + 
geom_point(position = pos, shape = 16, alpha = 0.8, size =3) +
labs(title= "", x="L2FC_HT29_Control_plasmid_average", y = "L2FC_MHHES1_Control_plasmid_average") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
scale_color_manual(values =c("red", "orange", "black")) +
theme_bw()
plot4
```


```{r}
# mutate BRAF curated_Domain field to simplify for figure
annotated_data %>% filter(Gene == "BRAF") %>%
  select(Gene, curated_Domain) %>%
  count(curated_Domain)

annotated_data <- mutate(annotated_data, curated_Domain = ifelse(curated_Domain == "ATP binding site;Protein kinase" & Gene == "BRAF", "ATP binding site", ifelse(curated_Domain == "Protein kinase;ATP binding site", "ATP binding site", ifelse(curated_Domain == "Breakpoint for translocation to form KIAA1549-BRAF fusion protein;Disordered", "Disordered", ifelse(curated_Domain == "Breakpoint for translocation to form KIAA1549-BRAF fusion protein", "", curated_Domain)))))

annotated_data %>% filter(Gene == "BRAF") %>%
  select(Gene, curated_Domain) %>%
  count(curated_Domain)
```

```{r}
# dot plots _ proliferation - BRAF
targets <- c("BRAF")
pos <- position_jitter(seed = 3, width = 0.2, height = 0.2)

plot1 <- ggplot(subset(annotated_data, Gene %in% c(targets) & (is.na(Amino_Acid_Position_simple) ==FALSE)), aes(x=Amino_Acid_Position_simple, y= L2FC_HT29_Control_plasmid_average_zscore, shape = most_severe_consequence, colour = curated_Domain)) + 
geom_point(position = pos, alpha = 1, size = 3) +
labs(title= "", x="BRAF amino acid position", y = "HT-29 control vs plasmid (z-score)", colour = "domain", shape = "consequence") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_HT29_Control_plasmid_average_zscore < -3 & most_severe_consequence == "missense", paste(PTM), ifelse(L2FC_HT29_Control_plasmid_average_zscore > 2, paste(Amino_Acid_Position), ""))), size = 4, position = pos, max.overlaps = Inf) +
scale_color_manual(values =c("grey55", "purple", "black", "royalblue", "orange", "red")) +
scale_shape_manual(values=c(19, 2, 11, 7, 3)) +
theme_classic()
plot1
ggsave("Fig1d.eps", width = 6, height = 4, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")
```

```{r}
# BRAF mutation investigate
annotated_data %>% filter(L2FC_HT29_Control_plasmid_average_zscore > 2 & Gene =="BRAF") %>% select(Amino_Acid_Position, Protein_Change, most_severe_consequence, L2FC_HT29_Control_plasmid_average_zscore, Edit_Location)
```

```{r}
# figures for domains in BRAF with LoF 

annotated_data %>%
  filter(Gene == "BRAF") %>%
  filter(most_severe_consequence =="missense") %>%
  filter(L2FC_HT29_Control_plasmid_average_zscore < -2) %>%
  select(Gene, most_severe_consequence, L2FC_HT29_Control_plasmid_average_zscore, curated_Domain) %>%
  distinct()

total<-annotated_data %>%
  filter(Gene == "BRAF") %>%
  filter(most_severe_consequence =="missense") %>%
  filter(L2FC_HT29_Control_plasmid_average_zscore < -2) %>%
  select(Gene, most_severe_consequence, L2FC_HT29_Control_plasmid_average_zscore, curated_Domain) %>%
  distinct() %>%
  count()

RBD<-annotated_data %>%
  filter(Gene == "BRAF") %>%
  filter(most_severe_consequence =="missense") %>%
  filter(L2FC_HT29_Control_plasmid_average_zscore < -2) %>%
  select(Gene, most_severe_consequence, L2FC_HT29_Control_plasmid_average_zscore, curated_Domain) %>%
  filter(curated_Domain =="RBD") %>%
  distinct() %>% 
  count()

kinase<-annotated_data %>%
  filter(Gene == "BRAF") %>%
  filter(most_severe_consequence =="missense") %>%
  filter(L2FC_HT29_Control_plasmid_average_zscore < -2) %>%
  select(Gene, most_severe_consequence, L2FC_HT29_Control_plasmid_average_zscore, curated_Domain) %>%
  filter(curated_Domain =="Protein kinase") %>%
  distinct() %>% 
  count()

#RBD
RBD/total*100
#kinase
kinase/total*100
```

```{r}
# dot plots _ proliferation - MYC
targets <- c("MYC")
pos <- position_jitter(seed = 3, width = 0.2, height = 0.2)

plot1 <- ggplot(subset(annotated_data, Gene %in% c(targets) & (is.na(Amino_Acid_Position_simple) ==FALSE)), aes(x=Amino_Acid_Position_simple, y= L2FC_HT29_Control_plasmid_average_zscore, shape = most_severe_consequence, colour = curated_Domain)) + 
geom_point(position = pos, alpha = 1, size = 3) +
labs(title= "", x="MYC amino acid position", y = "HT-29 control vs plasmid (z-score)", colour = "domain", shape = "consequence") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_HT29_Control_plasmid_average_zscore < -5 & most_severe_consequence == "missense", paste(Amino_Acid_Position), ifelse(L2FC_HT29_Control_plasmid_average_zscore > -1, paste(PTM), ""))), size = 4, position = pos, max.overlaps = Inf) +
scale_color_manual(values =c("black", "purple", "grey45", "royalblue", "orange", "red")) +
scale_shape_manual(values=c(1, 19, 2, 11, 7, 3)) +
theme_classic()
plot1
```

```{r}
# mutate PARP1/2 curated_Domain field to simplify for figure
test_clean_MHHES1_data %>% filter(Gene %in% c("PARP1", "PARP2")) %>%
  select(Gene, curated_Domain) %>%
  count(curated_Domain)

test_clean_MHHES1_data <- mutate(test_clean_MHHES1_data, curated_Domain = ifelse(curated_Domain == "Automodification domain;Disordered" & Gene %in% c("PARP1", "PARP2"), "Automodification domain", ifelse(curated_Domain %in% c("Cleavage; by caspase-3 and caspase", "Cleavage; by caspase", "Cleavage; by caspase-8", "Disordered;Cleavage; by caspase-3 and caspase", "Cleavage; by caspase-3 and caspase-7") & Gene %in% c("PARP1", "PARP2"), "Caspase cleavage", ifelse(curated_Domain %in% c("Disordered;Nuclear localization signal", "Nuclear localization signal;Disordered	", "Nuclear localization signal;Disordered") & Gene %in% c("PARP1", "PARP2"), "Nuclear localization signal", ifelse(curated_Domain %in% c("Disordered;PARP-type", "Disordered;PARP-type 2") & Gene %in% c("PARP1", "PARP2"), "Disordered",ifelse(curated_Domain %in% c("Zinc ribbon;PADR1 zinc-binding", "PADR1 zinc-binding;Zinc ribbon") & Gene %in% c("PARP1", "PARP2"), "Zinc ribbon", ifelse(curated_Domain %in% c("Disordered;PARP-type", "Disordered;PARP-type 2") & Gene %in% c("PARP1", "PARP2"), "Disordered",ifelse(curated_Domain %in% c("PARP catalytic;NAD(+) binding site", "PNAD(+) binding site;PARP catalytic", "NAD(+) binding site;PARP catalytic") & Gene %in% c("PARP1", "PARP2"), "PARP catalytic", curated_Domain))))))))

test_clean_MHHES1_data %>% filter(Gene == "PARP2") %>%
  select(Gene, curated_Domain) %>%
  count(curated_Domain)
```

```{r}
# dot plots _ proliferation - PARP1/2

targets <- c("PARP1", "PARP2")
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

plot1 <- ggplot(subset(test_clean_MHHES1_data, Gene %in% c(targets) & (is.na(Amino_Acid_Position_simple) ==FALSE)), aes(x=Amino_Acid_Position_simple, y= L2FC_MHHES1_Control_plasmid_average_zscore, shape = most_severe_consequence, colour = curated_Domain)) + 
geom_point(position = pos, alpha = 1, size = 3) +
labs(title= "", x="PARP1/2 amino acid position", y = "MHH-ES-1 control vs plasmid (z-score)", colour = "domain", shape = "consequence") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
#geom_text_repel(aes(label=ifelse(L2FC_MHHES1_Control_plasmid_average_zscore > -4, paste(PTM), "")), size = 3, position = pos, max.overlaps = Inf) +
scale_color_manual(values =c("grey55", "purple", "black", "royalblue", "orange", "red", "salmon", "red3", "blue3", "pink3", "yellow4", "grey35", "grey75", "orange3", "green3")) +
scale_shape_manual(values=c(1, 19, 2, 11, 7, 3)) +
theme_classic() +
facet_wrap(~Gene, scales = "free_x", nrow = 2)
plot1
ggsave("FigS6a.eps", width = 6, height = 4, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

plot2 <- ggplot(subset(test_clean_MHHES1_data, Gene %in% c(targets) & (is.na(Amino_Acid_Position_simple) ==FALSE)), aes(x=Amino_Acid_Position_simple, y= L2FC_MHHES1_Olap_plasmid_average_zscore, shape = most_severe_consequence, colour = curated_Domain)) + 
geom_point(position = pos, alpha = 1, size = 3) +
labs(title= "", x="PARP1/2 amino acid position", y = "MHH-ES-1 olaparib vs plasmid (z-score)", colour = "domain", shape = "consequence") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_MHHES1_Olap_plasmid_average_zscore > 3, paste(Amino_Acid_Position), ifelse(sgRNA_ID %in% PARP_sensitors$sgRNA_ID, paste(Amino_Acid_Position), ""))), size = 3, position = pos, max.overlaps = Inf) +
scale_color_manual(values =c("grey55", "purple", "black", "royalblue", "orange", "red", "salmon", "red3", "blue3", "pink3", "yellow4", "grey35", "grey75", "orange3", "green3")) +
scale_shape_manual(values=c(1, 19, 2, 11, 7, 3)) +
theme_classic() +
facet_wrap(~Gene, scales = "free_x", nrow = 2)
plot2
ggsave("FigS6b.eps", width = 6, height = 4, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

plot3 <- ggplot(subset(test_clean_MHHES1_data, Gene %in% c(targets) & (is.na(Amino_Acid_Position_simple) ==FALSE)), aes(x=Amino_Acid_Position_simple, y= L2FC_MHHES1_Nirap_plasmid_average_zscore, shape = most_severe_consequence, colour = curated_Domain)) + 
geom_point(position = pos, alpha = 1, size = 3) +
labs(title= "", x="PARP1/2 amino acid position", y = "MHH-ES-1 niraparib vs plasmid (z-score)", colour = "domain", shape = "consequence") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_MHHES1_Nirap_plasmid_average_zscore > 3, paste(Amino_Acid_Position), ifelse(sgRNA_ID %in% PARP_sensitors$sgRNA_ID, paste(Amino_Acid_Position), ""))), size = 3, position = pos, max.overlaps = Inf) +
scale_color_manual(values =c("grey55", "purple", "black", "royalblue", "orange", "red", "salmon", "red3", "blue3", "pink3", "yellow4", "grey35", "grey75", "orange3", "green3")) +
scale_shape_manual(values=c(1, 19, 2, 11, 7, 3)) +
theme_classic() +
facet_wrap(~Gene, scales = "free_x", nrow = 2)
plot3
ggsave("FigS6c.eps", width = 6, height = 4, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")
```

```{r}
# dot plots _ proliferation - PARP1/2
targets <- c("PARP1", "PARP2")
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

plot1 <- ggplot(subset(test_clean_MHHES1_data, Gene %in% c(targets) & (is.na(Amino_Acid_Position_simple) ==FALSE)), aes(x=Amino_Acid_Position_simple, y= L2FC_MHHES1_Control_plasmid_average_zscore, colour = Gene)) + 
geom_point(position = pos, alpha = 1, size = 3) +
labs(title= "", x="PARP1/2 amino acid position", y = "MHH-ES-1 control vs plasmid (z-score)", colour = "gene") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
scale_color_manual(values =c("salmon", "pink2")) +
scale_shape_manual(values=c(19, 2, 11, 7, 3)) +
theme_classic() +
facet_wrap(~Gene, scales = "free_x", ncol = 2) +
scale_y_continuous(limits = c(-6, 5))
plot1
ggsave("Fig3d_1.eps", width = 11, height = 4, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

plot2 <- ggplot(subset(test_clean_MHHES1_data, Gene %in% c(targets) & (is.na(Amino_Acid_Position_simple) ==FALSE)), aes(x=Amino_Acid_Position_simple, y= L2FC_MHHES1_Olap_plasmid_average_zscore, colour = Gene)) + 
geom_point(position = pos, alpha = 1, size = 3) +
labs(title= "", x="PARP1/2 amino acid position", y = "MHH-ES-1 olaparib vs plasmid (z-score)", colour = "gene") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(sgRNA_ID %in% PARP_sensitors$sgRNA_ID, paste(Amino_Acid_Position), "")), size = 4.5, position = pos, max.overlaps = Inf) +
scale_color_manual(values =c("salmon", "pink2")) +
scale_shape_manual(values=c(19, 2, 11, 7, 3)) +
theme_classic() +
facet_wrap(~Gene, scales = "free_x", ncol = 2)+
scale_y_continuous(limits = c(-6, 5))
plot2
ggsave("Fig3d_2.eps", width = 11, height = 4, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

plot3 <- ggplot(subset(test_clean_MHHES1_data, Gene %in% c(targets) & (is.na(Amino_Acid_Position_simple) ==FALSE)), aes(x=Amino_Acid_Position_simple, y= L2FC_MHHES1_Nirap_plasmid_average_zscore, colour = Gene)) + 
geom_point(position = pos, alpha = 1, size =3) +
labs(title= "", x="PARP1/2 amino acid position", y = "MHH-ES-1 niraparib vs plasmid (z-score)", colour = "gene") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(sgRNA_ID %in% PARP_sensitors$sgRNA_ID, paste(Amino_Acid_Position), ifelse(sgRNA_ID %in% PARP_resistors$sgRNA_ID & L2FC_MHHES1_Nirap_plasmid_average_zscore >4 , paste(Amino_Acid_Position), ifelse(sgRNA_ID %in% PARP_resistors$sgRNA_ID & Amino_Acid_Position == "1" , paste(Amino_Acid_Position, most_severe_consequence), "")))), size = 4.5, position = pos, max.overlaps = Inf) +
scale_color_manual(values =c("salmon", "pink2")) +
scale_shape_manual(values=c(19, 2, 11, 7, 3)) +
theme_classic() +
facet_wrap(~Gene, scales = "free_x", ncol = 2) +
scale_y_continuous(limits = c(-6, 5))
plot3
ggsave("Fig3d_3.eps", width = 11, height = 4, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")
```
```{r}
# dot plots _ proliferation - PARP1/2 - for presentations 
targets <- c("PARP1")
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

plot1 <- ggplot(subset(test_clean_MHHES1_data, Gene %in% c(targets) & (is.na(Amino_Acid_Position_simple) ==FALSE)), aes(x=Amino_Acid_Position_simple, y= L2FC_MHHES1_Control_plasmid_average_zscore, colour = Gene)) + 
geom_point(position = pos, alpha = 1, size = 3) +
labs(title= "", x="PARP1/2 amino acid position", y = "MHH-ES-1 control vs plasmid (z-score)", colour = "gene") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
scale_color_manual(values =c("salmon")) +
scale_shape_manual(values=c(19, 2, 11, 7, 3)) +
theme_classic() +
scale_y_continuous(limits = c(-6, 5))
plot1

plot2 <- ggplot(subset(test_clean_MHHES1_data, Gene %in% c(targets) & (is.na(Amino_Acid_Position_simple) ==FALSE)), aes(x=Amino_Acid_Position_simple, y= L2FC_MHHES1_Olap_plasmid_average_zscore, colour = Gene)) + 
geom_point(position = pos, alpha = 1, size = 3) +
labs(title= "", x="PARP1/2 amino acid position", y = "MHH-ES-1 olaparib vs plasmid (z-score)", colour = "gene") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
scale_color_manual(values =c("salmon")) +
scale_shape_manual(values=c(19, 2, 11, 7, 3)) +
theme_classic() +
scale_y_continuous(limits = c(-6, 5))
plot2

plot3 <- ggplot(subset(test_clean_MHHES1_data, Gene %in% c(targets) & (is.na(Amino_Acid_Position_simple) ==FALSE)), aes(x=Amino_Acid_Position_simple, y= L2FC_MHHES1_Nirap_plasmid_average_zscore, colour = Gene)) + 
geom_point(position = pos, alpha = 1, size =3) +
labs(title= "", x="PARP1/2 amino acid position", y = "MHH-ES-1 niraparib vs plasmid (z-score)", colour = "gene") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
scale_color_manual(values =c("salmon")) +
scale_shape_manual(values=c(19, 2, 11, 7, 3)) +
theme_classic() +
scale_y_continuous(limits = c(-6, 5))
plot3
```

``` {r}
#for Josep AZ olaparib
test_clean_MHHES1_data %>% 
  filter(L2FC_MHHES1_Olap_plasmid_average_zscore > 2) %>% 
  filter(Gene %in% c("PARP1", "PARP2")) %>%
  select(Gene, most_severe_consequence, Amino_Acid_Position, Protein_Change) %>%
  write.csv("olaparib_resistors_AZ_091023.csv")
  

test_clean_MHHES1_data %>% 
  filter(L2FC_MHHES1_Olap_plasmid_average_zscore < -3 & L2FC_MHHES1_Control_plasmid_average_zscore > -1) %>% 
  filter(Gene %in% c("PARP1", "PARP2")) %>%
  select(Gene, most_severe_consequence, Amino_Acid_Position, Protein_Change) %>%
    write.csv("olaparib_sensitisors_AZ_091023.csv")
```

```{r}
# primers for Golden Gate arrayed validation studies - PARP validation
# forward_primer
# add a G for the expression from U6 promoter by replacing the base at pos. 1 if, it is not a G
test_clean_MHHES1_data <- mutate(test_clean_MHHES1_data, G_guide = ifelse(substr(guide, 1, 1) == "G", guide, paste0("G", substr(guide, 2, 20))))

test_clean_MHHES1_data <- mutate(test_clean_MHHES1_data, for_primer = paste0("CACC", G_guide))

# reverse_primer
test_clean_MHHES1_data <- mutate(test_clean_MHHES1_data, rev_primer = G_guide)
test_clean_MHHES1_data <- mutate(test_clean_MHHES1_data, rev_primer = str_replace_all(rev_primer, "A", "t"))
test_clean_MHHES1_data <- mutate(test_clean_MHHES1_data, rev_primer = str_replace_all(rev_primer, "T", "a"))
test_clean_MHHES1_data <- mutate(test_clean_MHHES1_data, rev_primer = str_replace_all(rev_primer, "C", "g"))
test_clean_MHHES1_data <- mutate(test_clean_MHHES1_data, rev_primer = str_replace_all(rev_primer, "G", "c"))
test_clean_MHHES1_data <- mutate(test_clean_MHHES1_data, rev_primer = str_to_upper(rev_primer))

test_clean_MHHES1_data <- mutate(test_clean_MHHES1_data, rev_primer = paste0(substr(rev_primer, 20, 20), substr(rev_primer, 19, 19),substr(rev_primer, 18, 18),substr(rev_primer, 17, 17),substr(rev_primer, 16, 16),substr(rev_primer, 15, 15), substr(rev_primer, 14, 14), substr(rev_primer, 13, 13), substr(rev_primer, 12, 12), substr(rev_primer, 11, 11), substr(rev_primer, 10, 10), substr(rev_primer, 9, 9), substr(rev_primer, 8, 8), substr(rev_primer, 7, 7), substr(rev_primer, 6, 6), substr(rev_primer, 5, 5), substr(rev_primer, 4, 4), substr(rev_primer, 3, 3), substr(rev_primer, 2, 2), substr(rev_primer, 1, 1)))

test_clean_MHHES1_data <- mutate(test_clean_MHHES1_data, rev_primer = paste0("AAAC", rev_primer))

test_clean_MHHES1_data %>% filter(L2FC_MHHES1_Olap_plasmid_average_zscore > 3 & L2FC_MHHES1_Olap_plasmid_1_zscore >2 & L2FC_MHHES1_Olap_plasmid_2_zscore > 2 | L2FC_MHHES1_Nirap_plasmid_average_zscore > 3 & L2FC_MHHES1_Nirap_plasmid_1_zscore >2 & L2FC_MHHES1_Nirap_plasmid_2_zscore >2) %>%
  select(sgRNA_ID, Gene, G_guide, most_severe_consequence, Amino_Acid_Position, Protein_Change, L2FC_MHHES1_Control_plasmid_average_zscore, L2FC_MHHES1_Control_plasmid_1_zscore, L2FC_MHHES1_Control_plasmid_2_zscore, L2FC_MHHES1_Olap_plasmid_average_zscore, L2FC_MHHES1_Olap_plasmid_1_zscore, L2FC_MHHES1_Olap_plasmid_2_zscore, L2FC_MHHES1_Nirap_plasmid_average_zscore, L2FC_MHHES1_Nirap_plasmid_1_zscore, L2FC_MHHES1_Nirap_plasmid_2_zscore, curated_Domain, MHHES1_ABE_Control_exp1_reads, MHHES1_ABE_Control_exp2_reads, MHHES1_ABE_Olap_exp1_reads, MHHES1_ABE_Olap_exp2_reads, MHHES1_ABE_Nirap_exp1_reads, MHHES1_ABE_Nirap_exp2_reads, for_primer, rev_primer) %>% 
  write.csv("PARP_resistance.csv")

test_clean_MHHES1_data %>% filter(L2FC_MHHES1_Olap_plasmid_average_zscore < -3 & L2FC_MHHES1_Nirap_plasmid_average_zscore < -3 & L2FC_MHHES1_Control_plasmid_average_zscore > -1) %>%
  select(sgRNA_ID, Gene, G_guide, most_severe_consequence, Amino_Acid_Position, Protein_Change, L2FC_MHHES1_Control_plasmid_average_zscore, L2FC_MHHES1_Control_plasmid_1_zscore, L2FC_MHHES1_Control_plasmid_2_zscore, L2FC_MHHES1_Olap_plasmid_average_zscore, L2FC_MHHES1_Olap_plasmid_1_zscore, L2FC_MHHES1_Olap_plasmid_2_zscore, L2FC_MHHES1_Nirap_plasmid_average_zscore, L2FC_MHHES1_Nirap_plasmid_1_zscore, L2FC_MHHES1_Nirap_plasmid_2_zscore, curated_Domain, MHHES1_ABE_Control_exp1_reads, MHHES1_ABE_Control_exp2_reads, MHHES1_ABE_Olap_exp1_reads, MHHES1_ABE_Olap_exp2_reads, MHHES1_ABE_Nirap_exp1_reads, MHHES1_ABE_Nirap_exp2_reads, for_primer, rev_primer) %>% 
  write.csv("PARP_sensitisors.csv")
```

```{r}
# dot plots _ proliferation

targets <- c("EGFR", "KRAS", "MYC", "PARP1", "PARP2", "MAP2K1", "MAP2K2", "BCL2", "BRAF", "AKT1", "PIK3CA")

pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

plot1 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=L2FC_HT29_Control_plasmid_1_zscore, y= L2FC_H23_Control_plasmid_1_zscore, colour = most_severe_consequence)) + 
geom_point(position = pos, alpha = 0.80, size =3) +
labs(title= "", x="L2FC_HT29_Control_plasmid_1", y = "L2FC_H23_Control_plasmid_1") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
#geom_text_repel(aes(label=ifelse(L2FC_PC9_Gefit_plasmid_1 > 4 | L2FC_PC9_Gefit_plasmid_1 < -2.5 & L2FC_PC9_Control_plasmid_1 > -0.4, paste(Amino_Acid_Position, ",", Gene, ",", Consequence), "")), size = 3, position = pos) +
scale_color_manual(values =c("black", "springgreen4", "orange", "red", "blue", "grey55", "brown", "purple", "pink", "salmon", "royalblue")) +
#scale_shape_manual(values =c(19, 1)) +
facet_wrap(~Gene, scales = "free_x") +
theme_classic()
plot1

plot2 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=L2FC_HT29_Control_plasmid_1_zscore, y= L2FC_PC9_Control_plasmid_1_zscore, colour = most_severe_consequence)) + 
geom_point(position = pos, alpha = 0.80, size =3) +
labs(title= "", x="L2FC_HT29_Control_plasmid_1", y = "L2FC_PC9_Control_plasmid_1") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
#geom_text_repel(aes(label=ifelse(L2FC_PC9_Gefit_plasmid_1 > 4 | L2FC_PC9_Gefit_plasmid_1 < -2.5 & L2FC_PC9_Control_plasmid_1 > -0.4, paste(Amino_Acid_Position, ",", Gene, ",", Consequence), "")), size = 3, position = pos) +
scale_color_manual(values =c("black", "springgreen4", "orange", "red", "blue", "grey55", "brown", "purple", "pink", "salmon", "royalblue")) +
#scale_shape_manual(values =c(19, 1)) +
facet_wrap(~Gene, scales = "free_x") +
theme_classic()
plot2

plot3 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=L2FC_PC9_Control_plasmid_1_zscore, y= L2FC_H23_Control_plasmid_1_zscore, colour = most_severe_consequence)) + 
geom_point(position = pos, alpha = 0.80, size =3) +
labs(title= "", x="L2FC_PC9_Control_plasmid_1", y = "L2FC_H23_Control_plasmid_1") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
#geom_text_repel(aes(label=ifelse(L2FC_PC9_Gefit_plasmid_1 > 4 | L2FC_PC9_Gefit_plasmid_1 < -2.5 & L2FC_PC9_Control_plasmid_1 > -0.4, paste(Amino_Acid_Position, ",", Gene, ",", Consequence), "")), size = 3, position = pos) +
scale_color_manual(values =c("black", "springgreen4", "orange", "red", "blue", "grey55", "brown", "purple", "pink", "salmon", "royalblue")) +
#scale_shape_manual(values =c(19, 1)) +
facet_wrap(~Gene, scales = "free_x") +
theme_classic()
plot3
```

```{r}
#stats for violin plots - Student's t-test

#HT29
NT_HT29 <- annotated_data %>%
  filter(gene_alias == "non-targeting") %>%
  select(L2FC_HT29_Control_plasmid_average_zscore)

essential_HT29 <- annotated_data %>%
  filter(gene_alias == "essential") %>%
  select(L2FC_HT29_Control_plasmid_average_zscore)

t_test_1 <- t.test(NT_HT29, essential_HT29, var.equal = TRUE)
t_test_1

#H23
NT_H23 <- annotated_data %>%
  filter(gene_alias == "non-targeting") %>%
  select(L2FC_H23_Control_plasmid_average_zscore)

essential_H23 <- annotated_data %>%
  filter(gene_alias == "essential") %>%
  select(L2FC_H23_Control_plasmid_average_zscore)

t_test_2 <- t.test(NT_H23, essential_H23, var.equal = TRUE)
t_test_2

#PC9
NT_PC9 <- annotated_data %>%
  filter(gene_alias == "non-targeting") %>%
  select(L2FC_PC9_Control_plasmid_average_zscore)

essential_PC9 <- annotated_data %>%
  filter(gene_alias == "essential") %>%
  select(L2FC_PC9_Control_plasmid_average_zscore)

t_test_3 <- t.test(NT_PC9, essential_PC9, var.equal = TRUE)
t_test_3

# MHH-ES-1
NT_MHHES1 <- test_clean_MHHES1_data %>%
  filter(gene_alias == "non-targeting") %>%
  select(L2FC_MHHES1_Control_plasmid_average_zscore)

essential_MHHES1 <- test_clean_MHHES1_data %>%
  filter(gene_alias == "essential") %>%
  select(L2FC_MHHES1_Control_plasmid_average_zscore)

t_test_4 <- t.test(NT_MHHES1, essential_MHHES1, var.equal = TRUE)
t_test_4
```

```{r} 
# Violin overview plots Figure 1
# control vs plasmid
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

#HT-29
plot1 <- ggplot(annotated_data, aes(y = forcats::fct_relevel(gene_alias, "PARP1", "PARP2", "PIK3CA", "AKT1", "BCL2", "MAP2K1", "MAP2K2", "BRAF", "KRAS","EGFR","MYC", "essential", "non-essential", "intergenic", "non-targeting") , x = L2FC_HT29_Control_plasmid_average_zscore)) +
labs(title= "", y = "", x = "HT-29 control vs plasmid (z-score)") +
geom_point(size=3, alpha=1, position = pos, shape = 1) +
geom_violin(colour = "salmon", alpha = 1) +
annotate("text", y = 12.2, x = 3.2, label = paste0("p=", formatC(t_test_1$p.value, digits = 2)), parse = FALSE, size = 4) +
theme_classic()
plot1

#H23
plot2 <- ggplot(annotated_data, aes(y = forcats::fct_relevel(gene_alias, "PARP1", "PARP2", "PIK3CA", "AKT1", "BCL2", "MAP2K1", "MAP2K2", "BRAF", "KRAS","EGFR","MYC", "essential", "non-essential", "non-targeting") , x = L2FC_H23_Control_plasmid_average_zscore)) +
labs(title= "", y = "", x = "H23 control vs plasmid (z-score)") +
geom_point(size=3, alpha=1, position = pos, shape = 1) +
geom_violin(colour = "salmon", alpha = 1) +
annotate("text", y = 12.2, x = 2.9, label = paste0("p=", formatC(t_test_2$p.value, digits = 2)), parse = FALSE, size = 4) +
theme_classic()
plot2

#PC9
plot3 <- ggplot(annotated_data, aes(y = forcats::fct_relevel(gene_alias, "PARP1", "PARP2", "PIK3CA", "AKT1", "BCL2", "MAP2K1", "MAP2K2", "BRAF", "KRAS","EGFR","MYC", "essential", "non-essential", "non-targeting") , x = L2FC_PC9_Control_plasmid_average_zscore)) +
labs(title= "", y = "", x = "PC9 control vs plasmid (z-score)") +
geom_point(size=3, alpha=1, position = pos, shape = 1) +
geom_violin(colour = "salmon", alpha = 1) +
annotate("text", y = 12.2, x = 2.45, label = paste("p =", formatC(t_test_3$p.value, digits = 2)), parse = FALSE, size = 4) +
theme_classic()
plot3

# MHH-ES-1
plot4 <- ggplot(test_clean_MHHES1_data, aes(y = forcats::fct_relevel(gene_alias, "PARP1", "PARP2", "PIK3CA", "AKT1", "BCL2", "MAP2K1", "MAP2K2", "BRAF", "KRAS","EGFR","MYC", "essential", "non-essential", "non-targeting") , x = L2FC_MHHES1_Control_plasmid_average_zscore)) +
labs(title= "", y = "", x = "MHH-ES-1 control vs plasmid (z-score)") +
geom_point(size=3, alpha=1, position = pos, shape = 1) +
geom_violin(colour = "salmon", alpha = 1) +
annotate("text", y = 12.2, x = 2.4, label = paste0("p=", formatC(t_test_4$p.value, digits = 2)), parse = FALSE, size = 4) +
theme_classic()
plot4
```

```{r} 
# boxplot overview plots Figure 1
# control vs plasmid
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

#HT-29
plot1 <- ggplot(annotated_data, aes(y = forcats::fct_relevel(gene_alias, "PARP1", "PARP2", "PIK3CA", "AKT1", "BCL2", "MAP2K1", "MAP2K2", "BRAF", "KRAS","EGFR","MYC", "essential", "non-essential", "intergenic", "non-targeting") , x = L2FC_HT29_Control_plasmid_average_zscore)) +
labs(title= "", y = "", x = "HT-29 control vs plasmid (z-score)") +
geom_point(size=3, alpha=1, position = pos, shape = 1) +
geom_boxplot(colour = "salmon", alpha = 0) +
annotate("text", y = 12.1, x = 3.35, label = paste0("p = ", formatC(t_test_1$p.value, digits = 2)), parse = FALSE, size = 4) +
theme_classic()
plot1
ggsave("Fig1b.eps", width = 6.2, height = 5, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

#H23
plot2 <- ggplot(annotated_data, aes(y = forcats::fct_relevel(gene_alias, "PARP1", "PARP2", "PIK3CA", "AKT1", "BCL2", "MAP2K1", "MAP2K2", "BRAF", "KRAS","EGFR","MYC", "essential", "non-essential", "non-targeting") , x = L2FC_H23_Control_plasmid_average_zscore)) +
labs(title= "", y = "", x = "H23 control vs plasmid (z-score)") +
geom_point(size=3, alpha=1, position = pos, shape = 1) +
geom_boxplot(colour = "salmon", alpha = 0) +
annotate("text", y = 12.1, x = 2.9, label = paste0("p=", formatC(t_test_2$p.value, digits = 2)), parse = FALSE, size = 3) +
theme_classic()
plot2
ggsave("FigS1b_1.eps", width = 6.2, height = 5, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

#PC9
plot3 <- ggplot(annotated_data, aes(y = forcats::fct_relevel(gene_alias, "PARP1", "PARP2", "PIK3CA", "AKT1", "BCL2", "MAP2K1", "MAP2K2", "BRAF", "KRAS","EGFR","MYC", "essential", "non-essential", "non-targeting") , x = L2FC_PC9_Control_plasmid_average_zscore)) +
labs(title= "", y = "", x = "PC9 control vs plasmid (z-score)") +
geom_point(size=3, alpha=1, position = pos, shape = 1) +
geom_boxplot(colour = "salmon", alpha = 0) +
annotate("text", y = 12.1, x = 2.4, label = paste("p=", formatC(t_test_3$p.value, digits = 2)), parse = FALSE, size = 3) +
theme_classic()
plot3
ggsave("FigS1b_2.eps", width = 6.2, height = 5, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

# MHH-ES-1
plot4 <- ggplot(test_clean_MHHES1_data, aes(y = forcats::fct_relevel(gene_alias, "PARP1", "PARP2", "PIK3CA", "AKT1", "BCL2", "MAP2K1", "MAP2K2", "BRAF", "KRAS","EGFR","MYC", "essential", "non-essential", "non-targeting") , x = L2FC_MHHES1_Control_plasmid_average_zscore)) +
labs(title= "", y = "", x = "MHH-ES-1 control vs plasmid (z-score)") +
geom_point(size=3, alpha=1, position = pos, shape = 1) +
geom_boxplot(colour = "salmon", alpha = 0) +
annotate("text", y = 12.1, x = 2.4, label = paste0("p=", formatC(t_test_4$p.value, digits = 2)), parse = FALSE, size = 3) +
theme_classic()
plot4
ggsave("FigS1b_3.eps", width = 6.2, height = 5, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")
```
```{r}

NT_HT29
essential_HT29

NT_H23
essential_H23

NT_PC9
essential_PC9

NT_MHHES1 %>% 
  filter(is.na(L2FC_MHHES1_Control_plasmid_average_zscore) == FALSE)

essential_MHHES1 %>% 
  filter(is.na(L2FC_MHHES1_Control_plasmid_average_zscore) == FALSE)
```

```{r}
# primers for Golden Gate arrayed validation studies
# forward_primer
# add a G for the expression from U6 promoter by replacing the base at pos. 1 if, it is not a G
annotated_data <- mutate(annotated_data, G_guide = ifelse(substr(guide, 1, 1) == "G", guide, paste0("G", substr(guide, 2, 20))))

annotated_data <- mutate(annotated_data, for_primer = paste0("CACC", G_guide))

# reverse_primer
annotated_data <- mutate(annotated_data, rev_primer = G_guide)
annotated_data <- mutate(annotated_data, rev_primer = str_replace_all(rev_primer, "A", "t"))
annotated_data <- mutate(annotated_data, rev_primer = str_replace_all(rev_primer, "T", "a"))
annotated_data <- mutate(annotated_data, rev_primer = str_replace_all(rev_primer, "C", "g"))
annotated_data <- mutate(annotated_data, rev_primer = str_replace_all(rev_primer, "G", "c"))
annotated_data <- mutate(annotated_data, rev_primer = str_to_upper(rev_primer))

annotated_data <- mutate(annotated_data, rev_primer = paste0(substr(rev_primer, 20, 20), substr(rev_primer, 19, 19),substr(rev_primer, 18, 18),substr(rev_primer, 17, 17),substr(rev_primer, 16, 16),substr(rev_primer, 15, 15), substr(rev_primer, 14, 14), substr(rev_primer, 13, 13), substr(rev_primer, 12, 12), substr(rev_primer, 11, 11), substr(rev_primer, 10, 10), substr(rev_primer, 9, 9), substr(rev_primer, 8, 8), substr(rev_primer, 7, 7), substr(rev_primer, 6, 6), substr(rev_primer, 5, 5), substr(rev_primer, 4, 4), substr(rev_primer, 3, 3), substr(rev_primer, 2, 2), substr(rev_primer, 1, 1)))

annotated_data <- mutate(annotated_data, rev_primer = paste0("AAAC", rev_primer))
```

```{r}
# read in raw gRNA reads counts from sc validation library
raw_reads_scvalid <- as_tibble(read.csv("scvRes1_readcounts_270423.csv", stringsAsFactors = FALSE, header = TRUE))
raw_reads_scvalid <- mutate(raw_reads_scvalid, sgRNA_ID = as.character(sgRNA_ID))

# compute RPM normalised values
raw_reads_scvalid <- mutate_at(raw_reads_scvalid, vars(matches("reads")), list(RPM = RPM))
raw_reads_scvalid
```

```{r}
# compute L2FC from normalised gRNA counts - single experiments. -scResvalid library small validation screen

# PC9
#BE3.9max NGN
raw_reads_scvalid <- mutate(raw_reads_scvalid, L2FC_PC9_CBE_Control_plasmid_scres1valid = L2FC(PC9_CBE_cntrl_scres1valid_reads_RPM, sc_valid_plasmid_reads_RPM))
raw_reads_scvalid <- mutate(raw_reads_scvalid, L2FC_PC9_CBE_Control_T0_scres1valid = L2FC(PC9_CBE_cntrl_scres1valid_reads_RPM, PC9_CBE_T0_scres1valid_reads))
raw_reads_scvalid <- mutate(raw_reads_scvalid, L2FC_PC9_CBE_Gefit_Control_scres1valid = L2FC(PC9_CBE_gefit_scres1valid_reads_RPM, PC9_CBE_cntrl_scres1valid_reads_RPM))
raw_reads_scvalid <- mutate(raw_reads_scvalid, L2FC_PC9_CBE_Osim_Control_scres1valid = L2FC(PC9_CBE_osim_scres1valid_reads_RPM, PC9_CBE_cntrl_scres1valid_reads_RPM))
raw_reads_scvalid <- mutate(raw_reads_scvalid, L2FC_PC9_CBE_Gefit_plasmid_scres1valid = L2FC(PC9_CBE_gefit_scres1valid_reads_RPM, sc_valid_plasmid_reads_RPM))
raw_reads_scvalid <- mutate(raw_reads_scvalid, L2FC_PC9_CBE_Osim_plasmid_scres1valid = L2FC(PC9_CBE_osim_scres1valid_reads_RPM, sc_valid_plasmid_reads_RPM))

#ABE8e NGN
raw_reads_scvalid <- mutate(raw_reads_scvalid, L2FC_PC9_ABE_Control_plasmid_scres1valid = L2FC(PC9_ABE_cntrl_scres1valid_reads_RPM, sc_valid_plasmid_reads_RPM))
raw_reads_scvalid <- mutate(raw_reads_scvalid, L2FC_PC9_ABE_Control_T0_scres1valid = L2FC(PC9_ABE_cntrl_scres1valid_reads_RPM, PC9_ABE_T0_scres1valid_reads))
raw_reads_scvalid <- mutate(raw_reads_scvalid, L2FC_PC9_ABE_Gefit_Control_scres1valid = L2FC(PC9_ABE_gefit_scres1valid_reads_RPM, PC9_ABE_cntrl_scres1valid_reads_RPM))
raw_reads_scvalid <- mutate(raw_reads_scvalid, L2FC_PC9_ABE_Osim_Control_scres1valid = L2FC(PC9_ABE_osim_scres1valid_reads_RPM, PC9_ABE_cntrl_scres1valid_reads_RPM))
raw_reads_scvalid <- mutate(raw_reads_scvalid, L2FC_PC9_ABE_Gefit_plasmid_scres1valid = L2FC(PC9_ABE_gefit_scres1valid_reads_RPM, sc_valid_plasmid_reads_RPM))
raw_reads_scvalid <- mutate(raw_reads_scvalid, L2FC_PC9_ABE_Osim_plasmid_scres1valid = L2FC(PC9_ABE_osim_scres1valid_reads_RPM, sc_valid_plasmid_reads_RPM))

raw_reads_scvalid
```

```{r}
# generate z scores for sc validation library
raw_reads_scvalid <- mutate_at(raw_reads_scvalid, vars(matches("L2FC")), list(zscore = zscore))
raw_reads_scvalid
```

```{r}
#check zscore correlation with L2FC

pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

# CBE gefit vs plasmid
plot1 <- ggplot(raw_reads_scvalid, aes(x= L2FC_PC9_CBE_Gefit_plasmid_scres1valid, y= L2FC_PC9_CBE_Gefit_plasmid_scres1valid_zscore)) + geom_jitter(size=3,  position = pos, colour = "grey35") +
labs(title= "PC9 - zscore L2FC correlation", x = "L2FC_PC9_CBE_Gefit_plasmid_scres1valid", y = "L2FC_PC9_CBE_Gefit_plasmid_scres1valid_zscore") +
geom_smooth(method = lm) +
theme_bw()
plot1

# ABE osim vs plasmid
plot2 <- ggplot(raw_reads_scvalid, aes(x= L2FC_PC9_ABE_Osim_plasmid_scres1valid, y= L2FC_PC9_ABE_Osim_plasmid_scres1valid_zscore)) + geom_jitter(size=3,  position = pos, colour = "grey35") +
labs(title= "PC9 - zscore L2FC correlation", x = "L2FC_PC9_ABE_Osim_plasmid_scres1valid", y = "L2FC_PC9_ABE_Osim_plasmid_scres1valid_zscore") +
geom_smooth(method = lm) +
theme_bw()
plot2
```

```{r}
#left_join the sc validation library results
annotated_data <- left_join(annotated_data, raw_reads_scvalid)
```

```{r}
#PC9 check correlation between large screens and validation library
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

# CBE gefit vs plasmid
plot1 <- ggplot(annotated_data, aes(x= L2FC_PC9_Gefit_plasmid_1_zscore, y= L2FC_PC9_CBE_Gefit_plasmid_scres1valid_zscore)) + geom_jitter(size=3,  position = pos, colour = "grey35") +
labs(title= "PC9 - replicate correlation", x = "L2FC_PC9_Gefit_plasmid_1", y = "L2FC_PC9_CBE_Gefit_plasmid_scres1valid") +
geom_smooth(method = lm) +
theme_bw()
plot1

plot2 <- ggplot(annotated_data, aes(x= L2FC_PC9_Gefit_plasmid_2_zscore, y= L2FC_PC9_CBE_Gefit_plasmid_scres1valid_zscore)) + geom_jitter(size=3,  position = pos, colour = "grey35") +
labs(title= "PC9 - replicate correlation", x = "L2FC_PC9_Gefit_plasmid_2", y = "L2FC_PC9_CBE_Gefit_plasmid_scres1valid") +
geom_smooth(method = lm) +
theme_bw()
plot2

plot3 <- ggplot(annotated_data, aes(x= L2FC_PC9_Gefit_plasmid_average_zscore, y= L2FC_PC9_CBE_Gefit_plasmid_scres1valid_zscore)) + geom_jitter(size=3,  position = pos, colour = "grey35") +
labs(title= "PC9 - replicate correlation", x = "L2FC_PC9_Gefit_plasmid_average", y = "L2FC_PC9_CBE_Gefit_plasmid_scres1valid") +
geom_smooth(method = lm) +
theme_bw()
plot3

# CBE osim vs plasmid
plot4 <- ggplot(annotated_data, aes(x= L2FC_PC9_Osim_plasmid_average_zscore, y= L2FC_PC9_CBE_Osim_plasmid_scres1valid_zscore)) + geom_jitter(size=3,  position = pos, colour = "grey35") +
labs(title= "PC9 - replicate correlation", x = "L2FC_PC9_Osim_plasmid_average", y = "L2FC_PC9_CBE_Osim_plasmid_scres1valid") +
geom_smooth(method = lm) +
theme_bw()
plot4

# ABE gefit vs plasmid
plot5 <- ggplot(annotated_data, aes(x= L2FC_PC9_Gefit_plasmid_1_zscore, y= L2FC_PC9_ABE_Gefit_plasmid_scres1valid_zscore)) + geom_jitter(size=3,  position = pos, colour = "grey35") +
labs(title= "PC9 - replicate correlation", x = "L2FC_PC9_Gefit_plasmid_1", y = "L2FC_PC9_ABE_Gefit_plasmid_scres1valid") +
geom_smooth(method = lm) +
theme_bw()
plot5

plot6 <- ggplot(annotated_data, aes(x= L2FC_PC9_Gefit_plasmid_2_zscore, y= L2FC_PC9_ABE_Gefit_plasmid_scres1valid_zscore)) + geom_jitter(size=3,  position = pos, colour = "grey35") +
labs(title= "PC9 - replicate correlation", x = "L2FC_PC9_Gefit_plasmid_2", y = "L2FC_PC9_ABE_Gefit_plasmid_scres1valid") +
geom_smooth(method = lm) +
theme_bw()
plot6

plot7 <- ggplot(annotated_data, aes(x= L2FC_PC9_Gefit_plasmid_average_zscore, y= L2FC_PC9_ABE_Gefit_plasmid_scres1valid_zscore)) + geom_jitter(size=3,  position = pos, colour = "grey35") +
labs(title= "PC9 - replicate correlation", x = "L2FC_PC9_Gefit_plasmid_average", y = "L2FC_PC9_ABE_Gefit_plasmid_scres1valid") +
geom_smooth(method = lm) +
theme_bw()
plot7

# ABE osim vs plasmid
plot8 <- ggplot(annotated_data, aes(x= L2FC_PC9_Osim_plasmid_average_zscore, y= L2FC_PC9_ABE_Osim_plasmid_scres1valid_zscore)) + geom_jitter(size=3,  position = pos, colour = "grey35") +
labs(title= "PC9 - replicate correlation", x = "L2FC_PC9_Osim_plasmid_average", y = "L2FC_PC9_ABE_Osim_plasmid_scres1valid") +
geom_smooth(method = lm) +
theme_bw()
plot8
```

```{r}
# consolidate gRNA scores for gRNAs based on editor from sc valid library

annotated_data <- mutate(annotated_data, L2FC_PC9_Control_plasmid_scres1valid_zscore = as.numeric(ifelse(editor == "CBE", L2FC_PC9_CBE_Control_plasmid_scres1valid_zscore, ifelse(editor == "ABE", L2FC_PC9_ABE_Control_plasmid_scres1valid_zscore, "ERROR"))))

annotated_data <- mutate(annotated_data, L2FC_PC9_Gefit_plasmid_scres1valid_zscore = as.numeric(ifelse(editor == "CBE", L2FC_PC9_CBE_Gefit_plasmid_scres1valid_zscore, ifelse(editor == "ABE", L2FC_PC9_ABE_Gefit_plasmid_scres1valid_zscore, "ERROR"))))

annotated_data <- mutate(annotated_data, L2FC_PC9_Osim_plasmid_scres1valid_zscore = as.numeric(ifelse(editor == "CBE", L2FC_PC9_CBE_Osim_plasmid_scres1valid_zscore, ifelse(editor == "ABE", L2FC_PC9_ABE_Osim_plasmid_scres1valid_zscore, "ERROR"))))


#remove all the old score fields based on CBE or ABE
annotated_data <- annotated_data %>%
  select(!c(L2FC_PC9_CBE_Gefit_plasmid_scres1valid_zscore, L2FC_PC9_ABE_Gefit_plasmid_scres1valid_zscore, L2FC_PC9_CBE_Osim_plasmid_scres1valid_zscore, L2FC_PC9_ABE_Osim_plasmid_scres1valid_zscore)) %>%
  distinct()
annotated_data
```

```{r}
# replicate correlation - stats PC9 big screen vs validation screen
overlapping_data <- annotated_data %>% filter(is.na(L2FC_PC9_Gefit_plasmid_scres1valid_zscore) == FALSE & is.na(L2FC_PC9_Osim_plasmid_scres1valid_zscore) == FALSE)
overlapping_data
r1 <- cor(overlapping_data$L2FC_PC9_Gefit_plasmid_average_zscore, overlapping_data$L2FC_PC9_Gefit_plasmid_scres1valid_zscore, method = "pearson")
r2 <- cor(overlapping_data$L2FC_PC9_Osim_plasmid_average_zscore, overlapping_data$L2FC_PC9_Osim_plasmid_scres1valid_zscore, method = "pearson")
```

```{r}
#hits overlapping with validation screen
#gefit
gefit_hits <- annotated_data %>% 
  filter(L2FC_PC9_Gefit_plasmid_average_zscore >3 & L2FC_PC9_Gefit_plasmid_scres1valid_zscore >3 | L2FC_PC9_Gefit_plasmid_average_zscore < -2.5 & L2FC_PC9_Gefit_plasmid_scres1valid_zscore < -2.5 & sgRNA_type == "exonic") %>%
  select(guide, Gene, most_severe_consequence, Amino_Acid_Position)

#osim
osim_hits <- annotated_data %>% 
  filter(L2FC_PC9_Osim_plasmid_average_zscore >3 & L2FC_PC9_Osim_plasmid_scres1valid_zscore >3 | L2FC_PC9_Osim_plasmid_average_zscore < -3 & L2FC_PC9_Osim_plasmid_scres1valid_zscore < -3 & sgRNA_type == "exonic") %>%
  select(guide, Gene, most_severe_consequence, Amino_Acid_Position)
```

```{r}
#PC9 check correlation between large screens and validation library
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

# gefit vs plasmid
plot1 <- ggplot(annotated_data, aes(x= L2FC_PC9_Gefit_plasmid_average_zscore, y= L2FC_PC9_Gefit_plasmid_scres1valid_zscore)) + 
geom_pointdensity(size=3,  position = pos) +
scale_colour_viridis_c() +
labs(title= "PC9 - validation screens", x = "original screens: \nPC9 gefitinib vs plasmid (z-score)", y = "validation screens: \nPC9 gefitinib vs plasmid (z-score)", colour = "density") +
geom_smooth(method = lm) +
annotate("text", y = 7, x = 0, label = paste("r = ", round(r1, 2)), parse = FALSE, size = 7) +
geom_text_repel(aes(label=ifelse(L2FC_PC9_Gefit_plasmid_average_zscore > 10 & L2FC_PC9_Gefit_plasmid_scres1valid_zscore >3 | L2FC_PC9_Gefit_plasmid_average_zscore < -3 & L2FC_PC9_Gefit_plasmid_scres1valid_zscore < -2 & Gene == "EGFR", paste0(Gene, ",", Amino_Acid_Position, ",", most_severe_consequence), "")), size = 3, position = pos, max.overlaps = Inf) +
theme_bw(base_size = 15)
plot1
ggsave("FigS5a_1.eps", width = 6, height = 5, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

# osim vs plasmid
plot2 <- ggplot(annotated_data, aes(x= L2FC_PC9_Osim_plasmid_average_zscore, y= L2FC_PC9_Osim_plasmid_scres1valid_zscore)) + 
geom_pointdensity(size=3,  position = pos) +
scale_colour_viridis_c() +
labs(title= "PC9 - validation screens",  x = "original screens: \nPC9 osimertinib vs plasmid (z-score)", y = "validation screens: \nPC9 osimertinib vs plasmid (z-score)", colour = "density") +
geom_smooth(method = lm) +
annotate("text", y = 5, x = -9, label = paste("r = ", round(r2, 2)), parse = FALSE, size = 7) +
geom_text_repel(aes(label=ifelse(L2FC_PC9_Osim_plasmid_average_zscore > 3 & L2FC_PC9_Osim_plasmid_scres1valid_zscore >3 | L2FC_PC9_Osim_plasmid_average_zscore < -4 & L2FC_PC9_Osim_plasmid_scres1valid_zscore < -4 & Gene =="EGFR", paste0(Gene, ",", Amino_Acid_Position, ",", most_severe_consequence), "")), size = 3, position = pos, max.overlaps = Inf) +
theme_bw(base_size = 15)
plot2
ggsave("FigS5a_2.eps", width = 6, height = 5, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")
```

```{r}
#prime editing validation scope
annotated_data %>% 
filter(L2FC_PC9_Gefit_plasmid_average_zscore > 2 & L2FC_PC9_Gefit_plasmid_scres1valid_zscore > 2| L2FC_PC9_Osim_plasmid_average_zscore > 2 & L2FC_PC9_Osim_plasmid_scres1valid_zscore > 2) %>%
  select(Gene, most_severe_consequence, Amino_Acid_Position, guide, L2FC_PC9_Gefit_plasmid_average_zscore, L2FC_PC9_Gefit_plasmid_scres1valid_zscore, L2FC_PC9_Osim_plasmid_average_zscore, L2FC_PC9_Osim_plasmid_scres1valid_zscore, off_target_summary_NG, editor) %>%
  write.csv("Marie_pridict_test_input.csv")
```

```{r}
# Ryan AZ list 08.05.23
annotated_data %>% 
  mutate(gefitinib_resistance = ifelse(L2FC_PC9_Gefit_plasmid_average_zscore > 1.5 & L2FC_PC9_Gefit_plasmid_scres1valid_zscore > 1.5, TRUE, FALSE)) %>%
  mutate(osimertinib_resistance = ifelse(L2FC_PC9_Osim_plasmid_average_zscore > 1.5 & L2FC_PC9_Osim_plasmid_scres1valid_zscore > 1.5, TRUE, FALSE)) %>%
filter(L2FC_PC9_Gefit_plasmid_average_zscore > 1.5 & L2FC_PC9_Gefit_plasmid_scres1valid_zscore > 1 | L2FC_PC9_Osim_plasmid_average_zscore > 1 & L2FC_PC9_Osim_plasmid_scres1valid_zscore > 1.5) %>%
  select(Gene, most_severe_consequence, Amino_Acid_Position, Protein_Change, Transcript_ID, guide, L2FC_PC9_Gefit_plasmid_average_zscore, L2FC_PC9_Gefit_plasmid_scres1valid_zscore, L2FC_PC9_Osim_plasmid_average_zscore, L2FC_PC9_Osim_plasmid_scres1valid_zscore, off_target_summary_NG, editor, gefitinib_resistance, osimertinib_resistance) %>%
  write.csv("AZ_res_mutations.csv")
```

```{r}
#clean up annotated data dataframe
#remove rows with ABE or CBE
filtered_annotated_data <- annotated_data %>% select(-(contains("ABE") | contains("CBE")))
filtered_test_clean_MHHES1_data <- test_clean_MHHES1_data %>% select(-(contains("ABE") | contains("CBE")))

# add variant classes and count each class of variant

#1,2 Dabcet
filtered_annotated_data <- filtered_annotated_data %>% 
  mutate(variant_class_Debcet = ifelse(L2FC_HT29_DebCet_plasmid_1_zscore > 1 & L2FC_HT29_DebCet_plasmid_2_zscore > 1  & L2FC_HT29_DebCet_plasmid_average_zscore > 2 & L2FC_HT29_Control_plasmid_1_zscore < -1 & L2FC_HT29_Control_plasmid_2_zscore < -1 & L2FC_HT29_Control_plasmid_average_zscore < -2, "drug addiction", ifelse(L2FC_HT29_DebCet_plasmid_1_zscore > 1 & L2FC_HT29_DebCet_plasmid_2_zscore > 1 & L2FC_HT29_DebCet_plasmid_average_zscore > 2 & L2FC_HT29_Control_plasmid_1_zscore > -1 & L2FC_HT29_Control_plasmid_2_zscore > -1 & L2FC_HT29_Control_plasmid_average_zscore > -2 & L2FC_HT29_Control_plasmid_1_zscore < 1 & L2FC_HT29_Control_plasmid_2_zscore < 1 & L2FC_HT29_Control_plasmid_average_zscore < 2, "canonical drug resistance", ifelse(L2FC_HT29_Control_plasmid_1_zscore > 1 & L2FC_HT29_Control_plasmid_2_zscore > 1 & L2FC_HT29_Control_plasmid_average_zscore > 2, "driver", ifelse(L2FC_HT29_DebCet_plasmid_1_zscore < -1 & L2FC_HT29_DebCet_plasmid_2_zscore < -1 & L2FC_HT29_DebCet_plasmid_average_zscore < -2 & L2FC_HT29_Control_plasmid_1_zscore > -1 & L2FC_HT29_Control_plasmid_2_zscore > -1 & L2FC_HT29_Control_plasmid_average_zscore > -2, "drug-sensitising", "control") ))))

filtered_annotated_data %>%
  filter(variant_class_Debcet == "drug-sensitising")

#3 Tram
filtered_annotated_data <- filtered_annotated_data %>% 
  mutate(variant_class_Tram= ifelse(L2FC_HT29_Tram_plasmid_1_zscore > 1 & L2FC_HT29_Tram_plasmid_2_zscore > 1  & L2FC_HT29_Tram_plasmid_average_zscore > 2 & L2FC_HT29_Control_plasmid_1_zscore < -1 & L2FC_HT29_Control_plasmid_2_zscore < -1 & L2FC_HT29_Control_plasmid_average_zscore < -2, "drug addiction", ifelse(L2FC_HT29_Tram_plasmid_1_zscore > 1 & L2FC_HT29_Tram_plasmid_2_zscore > 1 & L2FC_HT29_Tram_plasmid_average_zscore > 2 & L2FC_HT29_Control_plasmid_1_zscore > -1 & L2FC_HT29_Control_plasmid_2_zscore > -1 & L2FC_HT29_Control_plasmid_average_zscore > -2 & L2FC_HT29_Control_plasmid_1_zscore < 1 & L2FC_HT29_Control_plasmid_2_zscore < 1 & L2FC_HT29_Control_plasmid_average_zscore < 2, "canonical drug resistance", ifelse(L2FC_HT29_Control_plasmid_1_zscore > 1 & L2FC_HT29_Control_plasmid_2_zscore > 1 & L2FC_HT29_Control_plasmid_average_zscore > 2, "driver", ifelse(L2FC_HT29_Tram_plasmid_1_zscore < -1 & L2FC_HT29_Tram_plasmid_2_zscore < -1 & L2FC_HT29_Tram_plasmid_average_zscore < -2 & L2FC_HT29_Control_plasmid_1_zscore > -1 & L2FC_HT29_Control_plasmid_2_zscore > -1 & L2FC_HT29_Control_plasmid_average_zscore > -2, "drug-sensitising","control") ))))

filtered_annotated_data %>%
  filter(variant_class_Tram == "drug-sensitising")

#4 Pict
filtered_annotated_data <- filtered_annotated_data %>% 
  mutate(variant_class_Pict= ifelse(L2FC_HT29_Pict_plasmid_1_zscore > 1 & L2FC_HT29_Pict_plasmid_2_zscore > 1  & L2FC_HT29_Pict_plasmid_average_zscore > 2 & L2FC_HT29_Control_plasmid_1_zscore < -1 & L2FC_HT29_Control_plasmid_2_zscore < -1 & L2FC_HT29_Control_plasmid_average_zscore < -2, "drug addiction", ifelse(L2FC_HT29_Pict_plasmid_1_zscore > 1 & L2FC_HT29_Pict_plasmid_2_zscore > 1 & L2FC_HT29_Pict_plasmid_average_zscore > 2 & L2FC_HT29_Control_plasmid_1_zscore > -1 & L2FC_HT29_Control_plasmid_2_zscore > -1 & L2FC_HT29_Control_plasmid_average_zscore > -2 & L2FC_HT29_Control_plasmid_1_zscore < 1 & L2FC_HT29_Control_plasmid_2_zscore < 1 & L2FC_HT29_Control_plasmid_average_zscore < 2, "canonical drug resistance", ifelse(L2FC_HT29_Control_plasmid_1_zscore > 1 & L2FC_HT29_Control_plasmid_2_zscore > 1 & L2FC_HT29_Control_plasmid_average_zscore > 2, "driver", ifelse(L2FC_HT29_Pict_plasmid_1_zscore < -1 & L2FC_HT29_Pict_plasmid_2_zscore < -1 & L2FC_HT29_Pict_plasmid_average_zscore < -2 & L2FC_HT29_Control_plasmid_1_zscore > -1 & L2FC_HT29_Control_plasmid_2_zscore > -1 & L2FC_HT29_Control_plasmid_average_zscore > -2, "drug-sensitising", "control") ))))

filtered_annotated_data %>%
  filter(variant_class_Pict == "drug-sensitising")

#5 Sotor
filtered_annotated_data <- filtered_annotated_data %>% 
  mutate(variant_class_Sotor= ifelse(L2FC_H23_Sotor_plasmid_1_zscore > 1 & L2FC_H23_Sotor_plasmid_2_zscore > 1  & L2FC_H23_Sotor_plasmid_average_zscore > 2 & L2FC_H23_Control_plasmid_1_zscore < -1 & L2FC_H23_Control_plasmid_2_zscore < -1 & L2FC_H23_Control_plasmid_average_zscore < -2, "drug addiction", ifelse(L2FC_H23_Sotor_plasmid_1_zscore > 1 & L2FC_H23_Sotor_plasmid_2_zscore > 1 & L2FC_H23_Sotor_plasmid_average_zscore > 2 & L2FC_H23_Control_plasmid_1_zscore > -1 & L2FC_H23_Control_plasmid_2_zscore > -1 & L2FC_H23_Control_plasmid_average_zscore > -2 & L2FC_H23_Control_plasmid_1_zscore < 1 & L2FC_H23_Control_plasmid_2_zscore < 1 & L2FC_H23_Control_plasmid_average_zscore < 2, "canonical drug resistance", ifelse(L2FC_H23_Control_plasmid_1_zscore > 1 & L2FC_H23_Control_plasmid_2_zscore > 1 & L2FC_H23_Control_plasmid_average_zscore > 2, "driver", ifelse(L2FC_H23_Sotor_plasmid_1_zscore < -1 & L2FC_H23_Sotor_plasmid_2_zscore < -1 & L2FC_H23_Sotor_plasmid_average_zscore < -2 & L2FC_H23_Control_plasmid_1_zscore > -1 & L2FC_H23_Control_plasmid_2_zscore > -1 & L2FC_H23_Control_plasmid_average_zscore > -2, "drug-sensitising", "control") ))))

filtered_annotated_data %>%
  filter(variant_class_Sotor == "drug-sensitising")

#6 Adag
filtered_annotated_data <- filtered_annotated_data %>% 
  mutate(variant_class_Adag= ifelse(L2FC_H23_Adag_plasmid_1_zscore > 1 & L2FC_H23_Adag_plasmid_2_zscore > 1  & L2FC_H23_Adag_plasmid_average_zscore > 2 & L2FC_H23_Control_plasmid_1_zscore < -1 & L2FC_H23_Control_plasmid_2_zscore < -1 & L2FC_H23_Control_plasmid_average_zscore < -2, "drug addiction", ifelse(L2FC_H23_Adag_plasmid_1_zscore > 1 & L2FC_H23_Adag_plasmid_2_zscore > 1 & L2FC_H23_Adag_plasmid_average_zscore > 2 & L2FC_H23_Control_plasmid_1_zscore > -1 & L2FC_H23_Control_plasmid_2_zscore > -1 & L2FC_H23_Control_plasmid_average_zscore > -2 & L2FC_H23_Control_plasmid_1_zscore < 1 & L2FC_H23_Control_plasmid_2_zscore < 1 & L2FC_H23_Control_plasmid_average_zscore < 2, "canonical drug resistance", ifelse(L2FC_H23_Control_plasmid_1_zscore > 1 & L2FC_H23_Control_plasmid_2_zscore > 1 & L2FC_H23_Control_plasmid_average_zscore > 2, "driver", ifelse(L2FC_H23_Adag_plasmid_1_zscore < -1 & L2FC_H23_Adag_plasmid_2_zscore < -1 & L2FC_H23_Adag_plasmid_average_zscore < -2 & L2FC_H23_Control_plasmid_1_zscore > -1 & L2FC_H23_Control_plasmid_2_zscore > -1 & L2FC_H23_Control_plasmid_average_zscore > -2, "drug-sensitising", "control") ))))

filtered_annotated_data %>%
  filter(variant_class_Adag == "drug-sensitising")

#7 Gefit
filtered_annotated_data <- filtered_annotated_data %>% 
  mutate(variant_class_Gefit= ifelse(L2FC_PC9_Gefit_plasmid_1_zscore > 1 & L2FC_PC9_Gefit_plasmid_2_zscore > 1  & L2FC_PC9_Gefit_plasmid_average_zscore > 2 & L2FC_PC9_Control_plasmid_1_zscore < -1 & L2FC_PC9_Control_plasmid_2_zscore < -1 & L2FC_PC9_Control_plasmid_average_zscore < -2, "drug addiction", ifelse(L2FC_PC9_Gefit_plasmid_1_zscore > 1 & L2FC_PC9_Gefit_plasmid_2_zscore > 1 & L2FC_PC9_Gefit_plasmid_average_zscore > 2 & L2FC_PC9_Control_plasmid_1_zscore > -1 & L2FC_PC9_Control_plasmid_2_zscore > -1 & L2FC_PC9_Control_plasmid_average_zscore > -2 & L2FC_PC9_Control_plasmid_1_zscore < 1 & L2FC_PC9_Control_plasmid_2_zscore < 1 & L2FC_PC9_Control_plasmid_average_zscore < 2, "canonical drug resistance", ifelse(L2FC_PC9_Control_plasmid_1_zscore > 1 & L2FC_PC9_Control_plasmid_2_zscore > 1 & L2FC_PC9_Control_plasmid_average_zscore > 2, "driver", ifelse(L2FC_PC9_Gefit_plasmid_1_zscore < -1 & L2FC_PC9_Gefit_plasmid_2_zscore < -1 & L2FC_PC9_Gefit_plasmid_average_zscore < -2 & L2FC_PC9_Control_plasmid_1_zscore > -1 & L2FC_PC9_Control_plasmid_2_zscore > -1 & L2FC_PC9_Control_plasmid_average_zscore > -2, "drug-sensitising", "control") ))))

filtered_annotated_data %>%
  filter(variant_class_Gefit == "drug-sensitising")

#8 Osim
filtered_annotated_data <- filtered_annotated_data %>% 
  mutate(variant_class_Osim= ifelse(L2FC_PC9_Osim_plasmid_1_zscore > 1 & L2FC_PC9_Osim_plasmid_2_zscore > 1  & L2FC_PC9_Osim_plasmid_average_zscore > 2 & L2FC_PC9_Control_plasmid_1_zscore < -1 & L2FC_PC9_Control_plasmid_2_zscore < -1 & L2FC_PC9_Control_plasmid_average_zscore < -2, "drug addiction", ifelse(L2FC_PC9_Osim_plasmid_1_zscore > 1 & L2FC_PC9_Osim_plasmid_2_zscore > 1 & L2FC_PC9_Osim_plasmid_average_zscore > 2 & L2FC_PC9_Control_plasmid_1_zscore > -1 & L2FC_PC9_Control_plasmid_2_zscore > -1 & L2FC_PC9_Control_plasmid_average_zscore > -2 & L2FC_PC9_Control_plasmid_1_zscore < 1 & L2FC_PC9_Control_plasmid_2_zscore < 1 & L2FC_PC9_Control_plasmid_average_zscore < 2, "canonical drug resistance", ifelse(L2FC_PC9_Control_plasmid_1_zscore > 1 & L2FC_PC9_Control_plasmid_2_zscore > 1 & L2FC_PC9_Control_plasmid_average_zscore > 2, "driver", ifelse(L2FC_PC9_Osim_plasmid_1_zscore < -1 & L2FC_PC9_Osim_plasmid_2_zscore < -1 & L2FC_PC9_Osim_plasmid_average_zscore < -2 & L2FC_PC9_Control_plasmid_1_zscore > -1 & L2FC_PC9_Control_plasmid_2_zscore > -1 & L2FC_PC9_Control_plasmid_average_zscore > -2, "drug-sensitising", "control") ))))

filtered_annotated_data %>%
  filter(variant_class_Osim == "drug-sensitising")

#9 Olap
filtered_test_clean_MHHES1_data <- filtered_test_clean_MHHES1_data %>% 
  mutate(variant_class_Olap= ifelse(L2FC_MHHES1_Olap_plasmid_1_zscore > 1 & L2FC_MHHES1_Olap_plasmid_2_zscore > 1  & L2FC_MHHES1_Olap_plasmid_average_zscore > 2 & L2FC_MHHES1_Control_plasmid_1_zscore < -1 & L2FC_MHHES1_Control_plasmid_2_zscore < -1 & L2FC_MHHES1_Control_plasmid_average_zscore < -2, "drug addiction", ifelse(L2FC_MHHES1_Olap_plasmid_1_zscore > 1 & L2FC_MHHES1_Olap_plasmid_2_zscore > 1 & L2FC_MHHES1_Olap_plasmid_average_zscore > 2 & L2FC_MHHES1_Control_plasmid_1_zscore > -1 & L2FC_MHHES1_Control_plasmid_2_zscore > -1 & L2FC_MHHES1_Control_plasmid_average_zscore > -2 & L2FC_MHHES1_Control_plasmid_1_zscore < 1 & L2FC_MHHES1_Control_plasmid_2_zscore < 1 & L2FC_MHHES1_Control_plasmid_average_zscore < 2, "canonical drug resistance", ifelse(L2FC_MHHES1_Control_plasmid_1_zscore > 1 & L2FC_MHHES1_Control_plasmid_2_zscore > 1 & L2FC_MHHES1_Control_plasmid_average_zscore > 2, "driver", ifelse(L2FC_MHHES1_Olap_plasmid_1_zscore < -1 & L2FC_MHHES1_Olap_plasmid_2_zscore < -1 & L2FC_MHHES1_Olap_plasmid_average_zscore < -2 & L2FC_MHHES1_Control_plasmid_1_zscore > -1 & L2FC_MHHES1_Control_plasmid_2_zscore > -1 & L2FC_MHHES1_Control_plasmid_average_zscore > -2, "drug-sensitising", "control") ))))

filtered_test_clean_MHHES1_data %>%
  filter(variant_class_Olap == "drug-sensitising")

#10 Nirap
filtered_test_clean_MHHES1_data <- filtered_test_clean_MHHES1_data %>% 
  mutate(variant_class_Nirap= ifelse(L2FC_MHHES1_Nirap_plasmid_1_zscore > 1 & L2FC_MHHES1_Nirap_plasmid_2_zscore > 1  & L2FC_MHHES1_Nirap_plasmid_average_zscore > 2 & L2FC_MHHES1_Control_plasmid_1_zscore < -1 & L2FC_MHHES1_Control_plasmid_2_zscore < -1 & L2FC_MHHES1_Control_plasmid_average_zscore < -2, "drug addiction", ifelse(L2FC_MHHES1_Nirap_plasmid_1_zscore > 1 & L2FC_MHHES1_Nirap_plasmid_2_zscore > 1 & L2FC_MHHES1_Nirap_plasmid_average_zscore > 2 & L2FC_MHHES1_Control_plasmid_1_zscore > -1 & L2FC_MHHES1_Control_plasmid_2_zscore > -1 & L2FC_MHHES1_Control_plasmid_average_zscore > -2 & L2FC_MHHES1_Control_plasmid_1_zscore < 1 & L2FC_MHHES1_Control_plasmid_2_zscore < 1 & L2FC_MHHES1_Control_plasmid_average_zscore < 2, "canonical drug resistance", ifelse(L2FC_MHHES1_Control_plasmid_1_zscore > 1 & L2FC_MHHES1_Control_plasmid_2_zscore > 1 & L2FC_MHHES1_Control_plasmid_average_zscore > 2, "driver", ifelse(L2FC_MHHES1_Nirap_plasmid_1_zscore < -1 & L2FC_MHHES1_Nirap_plasmid_2_zscore < -1 & L2FC_MHHES1_Nirap_plasmid_average_zscore < -2 & L2FC_MHHES1_Control_plasmid_1_zscore > -1 & L2FC_MHHES1_Control_plasmid_2_zscore > -1 & L2FC_MHHES1_Control_plasmid_average_zscore > -2, "drug-sensitising", "control") ))))

filtered_test_clean_MHHES1_data %>%
  filter(variant_class_Nirap == "drug-sensitising")
```

```{r}
#number of each class
filtered_annotated_data %>% count(variant_class_Debcet, Gene, gene_alias)
filtered_annotated_data %>% count(variant_class_Tram, Gene, gene_alias)
filtered_annotated_data %>% count(variant_class_Pict, Gene, gene_alias)
filtered_annotated_data %>% count(variant_class_Sotor, Gene, gene_alias)
filtered_annotated_data %>% count(variant_class_Adag, Gene, gene_alias)
filtered_annotated_data %>% count(variant_class_Gefit, Gene, gene_alias)
filtered_annotated_data %>% count(variant_class_Osim, Gene, gene_alias)

filtered_test_clean_MHHES1_data %>% count(variant_class_Olap, Gene, gene_alias)
filtered_test_clean_MHHES1_data %>% count(variant_class_Nirap, Gene, gene_alias)
```

```{r}
#number of lof mutations
#HT-29
filtered_annotated_data %>% 
  filter(Gene %in% c(targets)) %>% 
  filter(most_severe_consequence == "missense") %>%
  filter(L2FC_HT29_Control_plasmid_1_zscore < -1 & L2FC_HT29_Control_plasmid_2_zscore < -1 & L2FC_HT29_Control_plasmid_average_zscore < -2)

#H23
filtered_annotated_data %>% 
  filter(Gene %in% c(targets)) %>% 
  filter(most_severe_consequence == "missense") %>%
  filter(L2FC_H23_Control_plasmid_1_zscore < -1 & L2FC_H23_Control_plasmid_2_zscore < -1 & L2FC_H23_Control_plasmid_average_zscore < -2)

#PC9
filtered_annotated_data %>% 
  filter(Gene %in% c(targets)) %>% 
  filter(most_severe_consequence == "missense") %>%
  filter(L2FC_PC9_Control_plasmid_1_zscore < -1 & L2FC_PC9_Control_plasmid_2_zscore < -1 & L2FC_PC9_Control_plasmid_average_zscore < -2)

#MHHES1
filtered_test_clean_MHHES1_data %>% 
  filter(Gene %in% c(targets)) %>% 
  filter(most_severe_consequence == "missense") %>%
  filter(L2FC_MHHES1_Control_plasmid_1_zscore < -1 & L2FC_MHHES1_Control_plasmid_2_zscore < -1 & L2FC_MHHES1_Control_plasmid_average_zscore < -2)
``` 

```{r}
#number of gof mutations
#HT-29
filtered_annotated_data %>% 
  filter(Gene %in% c(targets)) %>% 
  filter(most_severe_consequence == "missense") %>%
  filter(L2FC_HT29_Control_plasmid_1_zscore > 1 & L2FC_HT29_Control_plasmid_2_zscore > 1 & L2FC_HT29_Control_plasmid_average_zscore > 2)

#H23
filtered_annotated_data %>% 
  filter(Gene %in% c(targets)) %>% 
  filter(most_severe_consequence == "missense") %>%
  filter(L2FC_H23_Control_plasmid_1_zscore > 1 & L2FC_H23_Control_plasmid_2_zscore > 1 & L2FC_H23_Control_plasmid_average_zscore > 2)

#PC9
filtered_annotated_data %>% 
  filter(Gene %in% c(targets)) %>% 
  filter(most_severe_consequence == "missense") %>%
  filter(L2FC_PC9_Control_plasmid_1_zscore > 1 & L2FC_PC9_Control_plasmid_2_zscore > 1 & L2FC_PC9_Control_plasmid_average_zscore > 2)

#MHHES1
filtered_test_clean_MHHES1_data %>% 
  filter(Gene %in% c(targets)) %>% 
  filter(most_severe_consequence == "missense") %>%
  filter(L2FC_MHHES1_Control_plasmid_1_zscore > 1 & L2FC_MHHES1_Control_plasmid_2_zscore > 1 & L2FC_MHHES1_Control_plasmid_average_zscore > 2)
```

```{r}
# in text write for HT-29 how many lof, gof mutations there are and how many trametinib variant classes there were

#LOF
HT_29_case_study <- filtered_annotated_data %>% 
  mutate(LOF_GOF = ifelse(Gene %in% c(targets) & most_severe_consequence == "missense" & L2FC_HT29_Control_plasmid_1_zscore < -1 & L2FC_HT29_Control_plasmid_2_zscore < -1 & L2FC_HT29_Control_plasmid_average_zscore < -2, "LOF", "" ))

#GOF
HT_29_case_study <- HT_29_case_study %>% 
  mutate(LOF_GOF = ifelse(Gene %in% c(targets) & most_severe_consequence == "missense" & L2FC_HT29_Control_plasmid_1_zscore > 1 & L2FC_HT29_Control_plasmid_2_zscore > 1 & L2FC_HT29_Control_plasmid_average_zscore > 2, "GOF", LOF_GOF ))

#write csv
write.csv(HT_29_case_study, "HT_29_case_study.csv")

HT_29_case_study %>% filter(LOF_GOF =="LOF")
HT_29_case_study %>% filter(LOF_GOF =="GOF")
HT_29_case_study %>% count(variant_class_Tram, Gene)
```

```{r}
#all hits for Figure 6

validation_set_HT29 <- filtered_annotated_data %>% 
  filter(L2FC_HT29_Control_plasmid_average_zscore >1.5 | L2FC_HT29_Tram_plasmid_average_zscore >1.5 | L2FC_HT29_Pict_plasmid_average_zscore >1.5 | L2FC_HT29_DebCet_plasmid_average_zscore >1.5) %>%
  filter(sgRNA_type == "exonic")

validation_set_H23 <- filtered_annotated_data %>% 
  filter(L2FC_H23_Control_plasmid_average_zscore >1.5 | L2FC_H23_Adag_plasmid_average_zscore >1.5 | L2FC_H23_Sotor_plasmid_average_zscore >1.5) %>%
  filter(sgRNA_type == "exonic")

validation_set_PC9 <- filtered_annotated_data %>% 
  filter(L2FC_PC9_Control_plasmid_average_zscore >1.5 | L2FC_PC9_Gefit_plasmid_average_zscore >1.5 & L2FC_PC9_Gefit_plasmid_scres1valid_zscore > 1.5 | L2FC_PC9_Osim_plasmid_average_zscore >1.5 & L2FC_PC9_Osim_plasmid_scres1valid_zscore > 1.5 | sgRNA_ID %in% c("156dcfe646185f7d34b04987763174c762f065c9", "53b63e6975f3b2790703e4fc6191d23902a5b9e8", "6034219e040dd7b9c2e6224f5d36039f1bdd9e07", "34131bd6ecb323bf952225a891399241b58b3131")) %>%
  filter(sgRNA_type == "exonic")

resistance_hits <- bind_rows(validation_set_HT29, validation_set_H23, validation_set_PC9) %>%
  filter(is.na(Amino_Acid_Position) == FALSE) %>%
  distinct()

resistance_hits <- select(resistance_hits, c(Gene, guide, editor, Amino_Acid_Position, Protein_Change, sgRNA_type, most_severe_consequence, off_target_summary_NG, sgRNA_ID, L2FC_HT29_Control_plasmid_1_zscore, L2FC_HT29_Control_plasmid_2_zscore, L2FC_HT29_Control_plasmid_average_zscore, L2FC_HT29_Tram_plasmid_1_zscore, L2FC_HT29_Tram_plasmid_2_zscore, L2FC_HT29_Tram_plasmid_average_zscore, L2FC_HT29_Pict_plasmid_1_zscore, L2FC_HT29_Pict_plasmid_2_zscore, L2FC_HT29_Pict_plasmid_average_zscore, L2FC_HT29_DebCet_plasmid_1_zscore, L2FC_HT29_DebCet_plasmid_2_zscore, L2FC_HT29_DebCet_plasmid_average_zscore, L2FC_H23_Control_plasmid_1_zscore, L2FC_H23_Control_plasmid_2_zscore, L2FC_H23_Control_plasmid_average_zscore, L2FC_H23_Adag_plasmid_1_zscore, L2FC_H23_Adag_plasmid_2_zscore, L2FC_H23_Adag_plasmid_average_zscore, L2FC_H23_Sotor_plasmid_1_zscore, L2FC_H23_Sotor_plasmid_2_zscore, L2FC_H23_Sotor_plasmid_average_zscore, L2FC_PC9_Control_plasmid_1_zscore, L2FC_PC9_Control_plasmid_2_zscore, L2FC_PC9_Control_plasmid_average_zscore, L2FC_PC9_Gefit_plasmid_1_zscore, L2FC_PC9_Gefit_plasmid_2_zscore, L2FC_PC9_Gefit_plasmid_average_zscore, L2FC_PC9_Osim_plasmid_1_zscore, L2FC_PC9_Osim_plasmid_2_zscore, L2FC_PC9_Osim_plasmid_average_zscore, L2FC_PC9_Gefit_plasmid_scres1valid_zscore, L2FC_PC9_Osim_plasmid_scres1valid_zscore, variant_class_Tram, variant_class_Debcet, variant_class_Pict, variant_class_Gefit, variant_class_Osim, variant_class_Adag, variant_class_Sotor))
```

```{r}
# previous annotations for Magda analysis - change to use zscore and check consistency between replicates

sc_HT29_debcet <- filtered_annotated_data %>% filter(is.na(L2FC_PC9_Gefit_plasmid_scres1valid_zscore) == FALSE & is.na(L2FC_PC9_Osim_plasmid_scres1valid_zscore) == FALSE) %>% select(Gene, sgRNA_ID, sgRNA_type, chr, start, end, guide, off_target_summary_NG, Amino_Acid_Position, most_severe_consequence, Amino_Acid_Position, Amino_Acid_Position_simple, editor, G_guide, L2FC_HT29_DebCet_plasmid_1_zscore, L2FC_HT29_DebCet_plasmid_2_zscore, L2FC_HT29_DebCet_plasmid_average_zscore, L2FC_HT29_Control_plasmid_1_zscore, L2FC_HT29_Control_plasmid_2_zscore, L2FC_HT29_Control_plasmid_average_zscore)

#rename Gene N/A to non-targeting
sc_HT29_debcet <- mutate(sc_HT29_debcet, Gene = ifelse(Gene =="N/A", "non_targeting", Gene))

#add binary resistant classifier
sc_HT29_debcet <- mutate(sc_HT29_debcet, resistance_mutation = ifelse(L2FC_HT29_DebCet_plasmid_1_zscore >1 & L2FC_HT29_DebCet_plasmid_2_zscore >1 & L2FC_HT29_DebCet_plasmid_average_zscore > 2, TRUE, FALSE))

#add a shorter sgRNA_ID with gene name
sc_HT29_debcet <- mutate(sc_HT29_debcet, ID = paste0(Gene, "_", substr(sgRNA_ID, 0,6)))

#add variant class
sc_HT29_debcet <- mutate(sc_HT29_debcet, variant_class = ifelse(L2FC_HT29_DebCet_plasmid_1_zscore > 1 & L2FC_HT29_DebCet_plasmid_2_zscore > 1  & L2FC_HT29_DebCet_plasmid_average_zscore > 2 & L2FC_HT29_Control_plasmid_1_zscore < -1 & L2FC_HT29_Control_plasmid_2_zscore < -1 & L2FC_HT29_Control_plasmid_average_zscore < -2, "drug addiction", ifelse(L2FC_HT29_DebCet_plasmid_1_zscore > 1 & L2FC_HT29_DebCet_plasmid_2_zscore > 1 & L2FC_HT29_DebCet_plasmid_average_zscore > 2 & L2FC_HT29_Control_plasmid_1_zscore > -1 & L2FC_HT29_Control_plasmid_2_zscore > -1 & L2FC_HT29_Control_plasmid_average_zscore > -2 & L2FC_HT29_Control_plasmid_1_zscore < 1 & L2FC_HT29_Control_plasmid_2_zscore < 1 & L2FC_HT29_Control_plasmid_average_zscore < 2, "canonical drug resistance", ifelse(L2FC_HT29_Control_plasmid_1_zscore > 1 & L2FC_HT29_Control_plasmid_2_zscore > 1 & L2FC_HT29_Control_plasmid_average_zscore > 2, "driver", "control") )))

write.csv(sc_HT29_debcet, "sc_HT29_debcet_annotated_270623.csv")
```

```{r}
# annotate resistance hits

resistance_hits <- mutate(resistance_hits, gefitinib_resistance = ifelse(L2FC_PC9_Gefit_plasmid_1_zscore > 1 & L2FC_PC9_Gefit_plasmid_2_zscore > 1 & L2FC_PC9_Gefit_plasmid_average_zscore > 2 & L2FC_PC9_Gefit_plasmid_scres1valid_zscore > 2, TRUE, FALSE))
resistance_hits <- mutate(resistance_hits, osimertinib_resistance = ifelse(L2FC_PC9_Osim_plasmid_1_zscore > 1 & L2FC_PC9_Osim_plasmid_2_zscore > 1 &  L2FC_PC9_Osim_plasmid_average_zscore > 2 & L2FC_PC9_Osim_plasmid_scres1valid_zscore > 2, TRUE, FALSE))
resistance_hits <- mutate(resistance_hits, trametinib_resistance = ifelse(L2FC_HT29_Tram_plasmid_1_zscore > 1 & L2FC_HT29_Tram_plasmid_2_zscore > 1 & L2FC_HT29_Tram_plasmid_average_zscore > 2, TRUE, FALSE))
resistance_hits <- mutate(resistance_hits, pictilisib_resistance = ifelse(L2FC_HT29_Pict_plasmid_1_zscore > 1 & L2FC_HT29_Pict_plasmid_2_zscore > 1 & L2FC_HT29_Pict_plasmid_average_zscore > 2, TRUE, FALSE))
resistance_hits <- mutate(resistance_hits, debcet_resistance = ifelse(L2FC_HT29_DebCet_plasmid_1_zscore > 1 & L2FC_HT29_DebCet_plasmid_2_zscore > 1 & L2FC_HT29_DebCet_plasmid_average_zscore > 2, TRUE, FALSE))
resistance_hits <- mutate(resistance_hits, adagrasib_resistance = ifelse(L2FC_H23_Adag_plasmid_1_zscore > 1 & L2FC_H23_Adag_plasmid_2_zscore > 1 & L2FC_H23_Adag_plasmid_average_zscore > 2, TRUE, FALSE))
resistance_hits <- mutate(resistance_hits, sotorasib_resistance = ifelse(L2FC_H23_Sotor_plasmid_1_zscore > 1 & L2FC_H23_Sotor_plasmid_2_zscore > 1 & L2FC_H23_Sotor_plasmid_average_zscore > 2, TRUE, FALSE))

resistance_hits
resistance_hits %>% write.csv("resistance_hits.csv")
```

```{r}
# read in resistance regions for validation arrayed amplicons seq
amplicons <- as_tibble(read.csv("Res1_amplicon_seq/res1_ampliconseq_hits.csv", stringsAsFactors = FALSE, header = TRUE))

amplicon_valid <- left_join(amplicons, filtered_annotated_data)

amplicon_valid <- select(amplicon_valid, c(Gene, G_guide, guide, sgRNA_ID, for_primer, rev_primer, chr, start, end, Location, editor, strand, Amino_Acid_Position, Protein_Change))
amplicon_valid %>% write.csv("Res1_amplicon_seq/ramplicon_valid.csv")
```

```{r}
# write a table of resistance hits that i have amplicon sequencing validation for

primer_metadata <- as_tibble(read.csv("amp_primer_metadata.csv", stringsAsFactors = FALSE, header = TRUE))
amplicon_to_gRNA <- as_tibble(read.csv("010823_amplicon_to_gRNA.csv", stringsAsFactors = FALSE, header = TRUE))
energy_distances <- as_tibble(read.csv("energy_distances_080524.csv", stringsAsFactors = FALSE, header = TRUE))

amplicon_valid <- amplicon_valid %>% filter(sgRNA_ID != "82dc01c1370b9615c1c25c4711aef34db87a3e7a")

Supplementary_Table_2 <- left_join(amplicon_valid, resistance_hits)
Supplementary_Table_2 <- left_join(Supplementary_Table_2, amplicon_to_gRNA)
Supplementary_Table_2 <- left_join(Supplementary_Table_2, primer_metadata)

Supplementary_Table_2 <- mutate(Supplementary_Table_2, ID = paste0(Gene, "_", substr(sgRNA_ID, 1,6)))
Supplementary_Table_2 <- left_join(Supplementary_Table_2, energy_distances)

write.csv(Supplementary_Table_2, "Supplementary_Table_2.csv")
```

```{r}
# analyse the amplicon sequencing data
# filter on >1000 mut reads, depth>1000 reads and, >0.1 vaf
#exclude edit == MAP2K2 p.F213C p. V64V, as pre-existing in HT-29

amp_seq_reference <- as_tibble(read.csv("amp_seq_referencing_220823.csv", stringsAsFactors = FALSE, header = TRUE))
amplicon_seq_data <- as_tibble(read.csv("3189_amplicon_var_report_22_08_2023.csv", stringsAsFactors = FALSE, header = TRUE))

amplicon_seq_data <- amplicon_seq_data %>% filter(depth>1000) %>% distinct()

amp_seq_results <- left_join(amplicon_seq_data, amp_seq_reference)
amp_seq_results <- amp_seq_results %>% mutate(edit_position = ifelse(strand == "-", end-pos_amp, pos_amp-start))
amp_seq_results <- amp_seq_results %>% mutate(edit = paste(Gene, protein))
amp_seq_results <- amp_seq_results %>% mutate(ref_alt = paste0(ref, ">", alt))

#exclude edit == MAP2K2 p.F213C, as pre-existing in HT-29
amp_seq_results <- amp_seq_results %>% filter(protein %in% c("p.F213C") == FALSE)
amp_seq_results <- amp_seq_results %>% filter(protein %in% c("p.V64V") == FALSE )
amp_seq_results <- amp_seq_results %>% filter(protein %in% c("p.Q787Q") == FALSE)
amp_seq_results
```

```{r}
#count gRNAs in amp seq
amp_seq_results <- amp_seq_results %>% mutate(guide = substr(G_guide, 2,20))
amp_seq_results %>% select(guide) %>% distinct() %>% count()
amp_seq_results %>% select(guide) %>% distinct()
# 46 gRNAs but actually 44 gRNAs with an edit with more than 10 % VAF

# filter vaf by 0.1
amp_seq_results <- amp_seq_results %>% filter(vaf>0.1) %>% distinct()
amp_seq_results
amp_seq_results %>% select(guide) %>% distinct()

# ampseq analysis
#number of variants
amp_seq_results_CBE <- filter(amp_seq_results, editor == "CBE")
amp_seq_results_ABE <- filter(amp_seq_results, editor == "ABE")
```

```{r}
# calculate editing efficiency in the window and out the window
amp_seq_results %>% summarise(mean = mean(vaf), sd = sd(vaf))
amp_seq_results %>% filter(edit_position >=4 & edit_position <=9) %>% summarise(median = median(vaf), sd = sd(vaf))
amp_seq_results %>% filter(edit_position <4 | edit_position >9) %>% summarise(median = median(vaf), sd = sd(vaf))

# calculate number of unexpectged base edits
amp_seq_results %>% select(dna, ref_alt, editor) %>% distinct() %>% count()
amp_seq_results %>% select(dna, ref_alt, editor) %>% filter(editor == "CBE") %>% filter(ref_alt %in% c("C>T", "G>A")) %>% distinct() %>% count()
amp_seq_results %>% select(dna, ref_alt, editor) %>% filter(editor == "ABE") %>% filter(ref_alt %in% c("A>G", "T>C")) %>% distinct() %>% count()
137-39-93
```

```{r}
#write supp table3
write.csv(amp_seq_results, "Supplementary_Table_3.csv")
amp_seq_results %>% filter(effect %in% c("silent") == FALSE) %>% write.csv("non_syn_Supplementary_Table_3.csv")
```

```{r}
# make graphs of variants from amplicon seq of VAF colour against the edit position
plot1 <- ggplot(subset(amp_seq_results, is.na(editor)==FALSE), aes(x= edit_position, y= edit, fill = vaf)) +
  geom_tile() +
theme_classic()+
  labs(y="", x ="edit position", fill = "VAF") +
  geom_vline(xintercept = 4, linetype="dotted", colour = "grey65") +
  geom_vline(xintercept = 9, linetype="dotted", colour = "grey65") +
    scale_fill_gradient(low = "grey80", high = "red") +
  facet_wrap(~editor)
plot1

plot2 <- ggplot(subset(amp_seq_results, is.na(editor)==FALSE), aes(x= edit_position, y= edit, colour = vaf, shape = ref_alt)) +
  geom_point() +
theme_classic()+
  theme(legend.position = "none") +
  labs(y="", x ="edit position", colour = "VAF", shape = "") +
  geom_vline(xintercept = 4, linetype="dotted", colour = "grey65") +
  geom_vline(xintercept = 9, linetype="dotted", colour = "grey65") +
    scale_color_continuous(low = "grey80", high = "red") +
    scale_shape_manual(values =c(19, 15, 15, 8, 19)) +
  facet_wrap(~editor)
plot2
ggsave("FigS7a.eps", width = 4.5, height = 16, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

plot3 <- ggplot(subset(amp_seq_results, is.na(editor)==FALSE & effect %in% c("missense", "ess_splice")), aes(x= edit_position, y= edit, fill = vaf)) +
  geom_tile() +
theme_classic()+
  labs(y="", x ="edit position", fill = "VAF") +
  geom_vline(xintercept = 4, linetype="dotted", colour = "grey65") +
  geom_vline(xintercept = 9, linetype="dotted", colour = "grey65") +
    scale_fill_gradient(low = "grey80", high = "red") +
  facet_wrap(~editor)
plot3

plot4 <- ggplot(subset(amp_seq_results_ABE, is.na(editor)==FALSE), aes(x= edit_position, y= guide, fill = ref_alt, alpha = vaf)) +
  geom_tile() +
theme_classic()+
  labs(y="", x ="edit position", fill = "", alpha = "VAF") +
  geom_vline(xintercept = 4, linetype="dotted", colour = "grey65") +
  geom_vline(xintercept = 9, linetype="dotted", colour = "grey65") +
  facet_wrap(~Gene)
plot4

plot5 <- ggplot(subset(amp_seq_results_CBE, is.na(editor)==FALSE), aes(x= edit_position, y= guide, fill = ref_alt, alpha = vaf)) +
  geom_tile() +
theme_classic()+
  labs(y="", x ="edit position", fill = "", alpha = "VAF") +
  geom_vline(xintercept = 4, linetype="dotted", colour = "grey65") +
  geom_vline(xintercept = 9, linetype="dotted", colour = "grey65") +
  facet_wrap(~Gene)
plot5
```

```{r}
# join amp_seq_results and Supplementary_Table_2

amp_seq_results %>% head
Supplementary_Table_2_join <-  Supplementary_Table_2 %>% select(!c(Amino_Acid_Position, chr, start, end, strand, guide, amplicon))
Discussion_table <- left_join(amp_seq_results, Supplementary_Table_2_join)
Discussion_table

#write Discussion_table
write.csv(Discussion_table, "Discussion_table.csv")
```

```{r}
# Discussion Table graphs

pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

#control
plot1 <- ggplot(Discussion_table, aes(x= ed, y= L2FC_HT29_Control_plasmid_average_zscore, colour = vaf)) + 
geom_point(size=3,  position = pos) +
labs(title= "HT29", x = "perturb-seq ed", y = "vL2FC_HT29_Control_plasmid_average_zscore", colour = "VAF") +
#geom_text_repel(aes(label=edit), size = 3, position = pos, max.overlaps = Inf) +
theme_bw(base_size = 15)
plot1

plot2 <- ggplot(Discussion_table, aes(x= diffusion_score, y= L2FC_HT29_Control_plasmid_average_zscore, colour = vaf)) + 
geom_point(size=3,  position = pos) +
labs(title= "HT29", x = "perturb-seq diffusion_score", y = "vL2FC_HT29_Control_plasmid_average_zscore", colour = "VAF") +
theme_bw(base_size = 15)
plot2

plot3 <- ggplot(Discussion_table, aes(x= PFS_outcome_score, y= L2FC_HT29_Control_plasmid_average_zscore, colour = vaf)) + 
geom_point(size=3,  position = pos) +
labs(title= "HT29", x = "perturb-seq PFS_outcome_score", y = "vL2FC_HT29_Control_plasmid_average_zscore", colour = "VAF") +
theme_bw(base_size = 15)
plot3

#drug
plot4 <- ggplot(Discussion_table, aes(x= ed, y= L2FC_HT29_DebCet_plasmid_average_zscore, colour = vaf)) + 
geom_point(size=3,  position = pos) +
labs(title= "HT29", x = "perturb-seq ed", y = "vL2FC_HT29_DebCet_plasmid_average_zscore", colour = "VAF") +
theme_bw(base_size = 15)
plot4

plot5 <- ggplot(Discussion_table, aes(x= ed, y= L2FC_HT29_DebCet_plasmid_average_zscore, colour = variant_class_Debcet)) + 
geom_point(size=3,  position = pos) +
labs(title= "HT29", x = "perturb-seq ed", y = "vL2FC_HT29_DebCet_plasmid_average_zscore", colour = "VAF") +
theme_bw(base_size = 15)
plot5

plot6 <- ggplot(Discussion_table, aes(x= ed, y= L2FC_HT29_Control_plasmid_average_zscore, colour = variant_class_Debcet)) + 
geom_point(size=3,  position = pos) +
labs(title= "HT29", x = "perturb-seq ed", y = "vL2FC_HT29_Control_plasmid_average_zscore", colour = "VAF") +
theme_bw(base_size = 15)
plot6

plot7 <- ggplot(Discussion_table, aes(x= diffusion_score, y= L2FC_HT29_DebCet_plasmid_average_zscore, colour = vaf)) + 
geom_point(size=3,  position = pos) +
labs(title= "HT29", x = "perturb-seq diffusion_score", y = "vL2FC_HT29_DebCet_plasmid_average_zscore", colour = "VAF") +
theme_bw(base_size = 15)
plot7

plot8 <- ggplot(Discussion_table, aes(x= PFS_outcome_score, y= L2FC_HT29_DebCet_plasmid_average_zscore, colour = vaf)) + 
geom_point(size=3,  position = pos) +
labs(title= "HT29", x = "perturb-seq PFS_outcome_score", y = "vL2FC_HT29_DebCet_plasmid_average_zscore", colour = "VAF") +
theme_bw(base_size = 15)
plot8
```

```{r}
# make hit list for NHS George Burghel

NHS_cases <- as_tibble(read.csv("NHS_in_house_sanger_somatic_detected_280923.csv", stringsAsFactors = FALSE, header = TRUE))

NHS_cases <- NHS_cases %>% select(G_guide, original_hgvs,	hgvs_c,	vcf_hg19, filtered_sample_count, vep_consequence, omics_db_url)

Discussion_table_filtered <- Discussion_table %>% select(!c(unique_well_id, vaf, sample, exp, name,	i7_tag,	i5_tag, plasmid_label, mtr,	wtr,	depth)) %>% distinct()

George_NHS_shortlist <- left_join(Discussion_table_filtered, NHS_cases) %>% filter(filtered_sample_count > 1) %>% filter(effect != "silent") %>% distinct()

write.csv(George_NHS_shortlist, "George_NHS_shortlist.csv")
```

```{r}
#analyse follow-up prime editing screens
pegRNA_counts <- as_tibble(read.csv("PE_guide_count/pycroquet_counts_021023.csv", stringsAsFactors = FALSE, header = TRUE))
pegRNA_counts %>% filter(plasmid_reads <100)

#filter on 50 reads (no removal)

pegRNA_counts_gefit <- pegRNA_counts
pegRNA_counts_osi <- pegRNA_counts

# compute average reads
pegRNA_counts_gefit <- mutate(pegRNA_counts_gefit, control_average_reads = average(control_exp1_reads, control_exp2_reads))
pegRNA_counts_gefit <- mutate(pegRNA_counts_gefit, gefit_average_reads = average(gefit_exp1_reads, gefit_exp2_reads))
pegRNA_counts_gefit <- mutate(pegRNA_counts_gefit, edited_t7_average_reads = average(edited_t7_exp1_reads, edited_t7_exp2_reads))

pegRNA_counts_osi <- mutate(pegRNA_counts_osi, control_average_reads = average(control_exp3_reads, control_exp4_reads))
pegRNA_counts_osi <- mutate(pegRNA_counts_osi, osi_average_reads = average(osi_exp3_reads, osi_exp4_reads))
pegRNA_counts_osi <- mutate(pegRNA_counts_osi, edited_t7_average_reads = average(edited_t7_exp3_reads, edited_t7_exp4_reads))

#RPM
pegRNA_counts_gefit <- mutate_at(pegRNA_counts_gefit, vars(matches("reads")), list(RPM = RPM))
pegRNA_counts_osi <- mutate_at(pegRNA_counts_osi, vars(matches("reads")), list(RPM = RPM))

#L2FC - single exps.
pegRNA_counts_gefit <- mutate(pegRNA_counts_gefit, L2FC_PC9_control_plasmid_1 = L2FC(control_exp1_reads_RPM, plasmid_reads_RPM))
pegRNA_counts_gefit <- mutate(pegRNA_counts_gefit, L2FC_PC9_control_plasmid_2 = L2FC(control_exp2_reads_RPM, plasmid_reads_RPM))

pegRNA_counts_gefit <- mutate(pegRNA_counts_gefit, L2FC_PC9_control_t7_1 = L2FC(control_exp1_reads_RPM, edited_t7_exp1_reads_RPM))
pegRNA_counts_gefit <- mutate(pegRNA_counts_gefit, L2FC_PC9_control_t7_2 = L2FC(control_exp2_reads_RPM, edited_t7_exp2_reads_RPM))

pegRNA_counts_gefit <- mutate(pegRNA_counts_gefit, L2FC_PC9_gefit_plasmid_1 = L2FC(gefit_exp1_reads_RPM, plasmid_reads_RPM))
pegRNA_counts_gefit <- mutate(pegRNA_counts_gefit, L2FC_PC9_gefit_plasmid_2 = L2FC(gefit_exp2_reads_RPM, plasmid_reads_RPM))

pegRNA_counts_gefit <- mutate(pegRNA_counts_gefit, L2FC_PC9_gefit_control_1 = L2FC(gefit_exp1_reads_RPM, control_exp1_reads_RPM))
pegRNA_counts_gefit <- mutate(pegRNA_counts_gefit, L2FC_PC9_gefit_control_2 = L2FC(gefit_exp2_reads_RPM, control_exp2_reads_RPM))

pegRNA_counts_gefit <- mutate(pegRNA_counts_gefit, L2FC_PC9_gefit_t7_1 = L2FC(gefit_exp1_reads_RPM, edited_t7_exp1_reads_RPM))
pegRNA_counts_gefit <- mutate(pegRNA_counts_gefit, L2FC_PC9_gefit_t7_2 = L2FC(gefit_exp2_reads_RPM, edited_t7_exp2_reads_RPM))

###
pegRNA_counts_osi <- mutate(pegRNA_counts_osi, L2FC_PC9_control_plasmid_1 = L2FC(control_exp3_reads_RPM, plasmid_reads_RPM))
pegRNA_counts_osi <- mutate(pegRNA_counts_osi, L2FC_PC9_control_plasmid_2 = L2FC(control_exp4_reads_RPM, plasmid_reads_RPM))

pegRNA_counts_osi <- mutate(pegRNA_counts_osi, L2FC_PC9_control_t7_1 = L2FC(control_exp3_reads_RPM, edited_t7_exp3_reads_RPM))
pegRNA_counts_osi <- mutate(pegRNA_counts_osi, L2FC_PC9_control_t7_2 = L2FC(control_exp4_reads_RPM, edited_t7_exp4_reads_RPM))

pegRNA_counts_osi <- mutate(pegRNA_counts_osi, L2FC_PC9_osi_plasmid_1 = L2FC(osi_exp3_reads_RPM, plasmid_reads_RPM))
pegRNA_counts_osi <- mutate(pegRNA_counts_osi, L2FC_PC9_osi_plasmid_2 = L2FC(osi_exp4_reads_RPM, plasmid_reads_RPM))

pegRNA_counts_osi <- mutate(pegRNA_counts_osi, L2FC_PC9_osi_control_1 = L2FC(osi_exp3_reads_RPM, control_exp3_reads_RPM))
pegRNA_counts_osi <- mutate(pegRNA_counts_osi, L2FC_PC9_osi_control_2 = L2FC(osi_exp4_reads_RPM, control_exp4_reads_RPM))

pegRNA_counts_osi <- mutate(pegRNA_counts_osi, L2FC_PC9_osi_t7_1 = L2FC(osi_exp3_reads_RPM, edited_t7_exp3_reads_RPM))
pegRNA_counts_osi <- mutate(pegRNA_counts_osi, L2FC_PC9_osi_t7_2 = L2FC(osi_exp4_reads_RPM, edited_t7_exp4_reads_RPM))
```

```{r}
#compute L2FC from average of 2 exps from RPM values
pegRNA_counts_gefit <- mutate(pegRNA_counts_gefit, L2FC_PC9_control_plasmid_average = L2FC(control_average_reads_RPM, plasmid_reads_RPM))
pegRNA_counts_gefit <- mutate(pegRNA_counts_gefit, L2FC_PC9_control_t7_average = L2FC(control_average_reads_RPM, edited_t7_average_reads_RPM))
pegRNA_counts_gefit <- mutate(pegRNA_counts_gefit, L2FC_PC9_gefit_plasmid_average = L2FC(gefit_average_reads_RPM, plasmid_reads_RPM))
pegRNA_counts_gefit <- mutate(pegRNA_counts_gefit, L2FC_PC9_gefit_control_average = L2FC(gefit_average_reads_RPM, control_average_reads_RPM))
pegRNA_counts_gefit <- mutate(pegRNA_counts_gefit, L2FC_PC9_gefit_t7_average = L2FC(gefit_average_reads_RPM, edited_t7_average_reads_RPM))

pegRNA_counts_osi <- mutate(pegRNA_counts_osi, L2FC_PC9_control_plasmid_average = L2FC(control_average_reads_RPM, plasmid_reads_RPM))
pegRNA_counts_osi <- mutate(pegRNA_counts_osi, L2FC_PC9_control_t7_average = L2FC(control_average_reads_RPM, edited_t7_average_reads_RPM))
pegRNA_counts_osi <- mutate(pegRNA_counts_osi, L2FC_PC9_osi_plasmid_average = L2FC(osi_average_reads_RPM, plasmid_reads_RPM))
pegRNA_counts_osi <- mutate(pegRNA_counts_osi, L2FC_PC9_osi_control_average = L2FC(osi_average_reads_RPM, control_average_reads_RPM))
pegRNA_counts_osi <- mutate(pegRNA_counts_osi, L2FC_PC9_osi_t7_average = L2FC(osi_average_reads_RPM, edited_t7_average_reads_RPM))
```

```{r}
# generate z scores for pegRNAs validation library
pegRNA_counts_gefit <- mutate_at(pegRNA_counts_gefit, vars(matches("L2FC")), list(zscore = zscore))
pegRNA_counts_osi <- mutate_at(pegRNA_counts_osi, vars(matches("L2FC")), list(zscore = zscore))

pegRNA_counts_gefit
pegRNA_counts_osi
```

```{r}
#PC9 check pegRNA reads performance
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

# gefit vs plasmid
plot1 <- ggplot(pegRNA_counts_gefit, aes(y= L2FC_PC9_gefit_plasmid_average_zscore, x= L2FC_PC9_control_plasmid_average_zscore, colour = mutation_type)) +
geom_pointdensity(size=3,  position = pos) +
labs(title= "PC9 - prime editing screens", y = "gefitinib vs plasmid (z-score)", x = "control vs plasmid (z-score)", colour = "consequence") +
geom_text_repel(aes(label=ifelse(L2FC_PC9_gefit_plasmid_average_zscore > 2, paste(mutation), "")), size = 3, position = pos, max.overlaps = Inf) +
theme_bw(base_size = 15)
plot1

plot2 <- ggplot(pegRNA_counts_gefit, aes(y= L2FC_PC9_gefit_control_average_zscore, x= L2FC_PC9_control_plasmid_average_zscore, colour = mutation_type)) +
geom_pointdensity(size=3,  position = pos) +
labs(title= "PC9 - prime editing screens", y = "gefitinib vs control (z-score)", x = "control vs plasmid (z-score)", colour = "consequence") +
geom_text_repel(aes(label=ifelse(L2FC_PC9_gefit_control_average_zscore > 2, paste(mutation), "")), size = 3, position = pos, max.overlaps = Inf) +
theme_bw(base_size = 15)
plot2

plot3 <- ggplot(pegRNA_counts_gefit, aes(y= L2FC_PC9_gefit_t7_average_zscore, x= L2FC_PC9_control_plasmid_average_zscore, colour = mutation_type)) +
geom_pointdensity(size=3,  position = pos) +
labs(title= "PC9 - prime editing screens", y = "gefitinib vs t7 (z-score)", x = "control vs plasmid (z-score)", colour = "consequence") +
geom_text_repel(aes(label=ifelse(L2FC_PC9_gefit_t7_average_zscore > 2, paste(mutation), "")), size = 3, position = pos, max.overlaps = Inf) +
theme_bw(base_size = 15)
plot3

###

plot4 <- ggplot(pegRNA_counts_osi, aes(y= L2FC_PC9_osi_plasmid_average_zscore, x= L2FC_PC9_control_plasmid_average_zscore, colour = mutation_type)) +
geom_pointdensity(size=3,  position = pos) +
labs(title= "", y = "PC9 - osimertinib vs plasmid (z-score)", x = "PC9 - control vs plasmid (z-score)", colour = "consequence") +
geom_text_repel(aes(label=ifelse(L2FC_PC9_osi_plasmid_average_zscore > 1 & L2FC_PC9_osi_plasmid_1_zscore > 1 & L2FC_PC9_osi_plasmid_2_zscore > 1 | L2FC_PC9_control_plasmid_average_zscore < -1.5 | L2FC_PC9_control_plasmid_1_zscore > 1.1 & L2FC_PC9_control_plasmid_2_zscore > 1.1 | mutation %in% c("T790M"), paste(mutation), "")), size = 3, position = pos, max.overlaps = Inf) +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_vline(xintercept=c(0,0),linetype="dotted") +
  scale_color_manual( values =c("black", "salmon", "grey65")) +
theme_classic(base_size = 12)
plot4

plot5 <- ggplot(pegRNA_counts_osi, aes(y= L2FC_PC9_osi_control_average_zscore, x= L2FC_PC9_control_plasmid_average_zscore, colour = mutation_type)) +
geom_pointdensity(size=3,  position = pos) +
labs(title= "PC9 - prime editing screens", y = "osimertinib vs control (z-score)", x = "control vs plasmid (z-score)", colour = "consequence") +
geom_text_repel(aes(label=ifelse(L2FC_PC9_osi_control_average_zscore > 2, paste(mutation), "")), size = 3, position = pos, max.overlaps = Inf) +
theme_bw(base_size = 15)
plot5

plot6 <- ggplot(pegRNA_counts_osi, aes(y= L2FC_PC9_osi_t7_average_zscore, x= L2FC_PC9_control_plasmid_average_zscore, colour = mutation_type)) +
geom_pointdensity(size=3,  position = pos) +
labs(title= "PC9 - prime editing screens", y = "osimertinib vs t7 (z-score)", x = "control vs plasmid (z-score)", colour = "consequence") +
geom_text_repel(aes(label=ifelse(L2FC_PC9_osi_t7_average_zscore > 2, paste(mutation), "")), size = 3, position = pos, max.overlaps = Inf) +
theme_bw(base_size = 15)
plot6

plot7 <- ggplot(pegRNA_counts_osi, aes(y= L2FC_PC9_osi_plasmid_average_zscore, x= L2FC_PC9_control_plasmid_average_zscore, colour = mutation_type)) +
geom_pointdensity(size=3,  position = pos) +
labs(title= "", y = "PC9 - osimertinib vs plasmid (z-score)", x = "PC9 - control vs plasmid (z-score)", colour = "consequence") +
geom_text_repel(aes(label=ifelse(L2FC_PC9_osi_plasmid_average_zscore > 1 & L2FC_PC9_osi_plasmid_1_zscore > 1 & L2FC_PC9_osi_plasmid_2_zscore > 1 | L2FC_PC9_control_plasmid_average_zscore < -1.5 | mutation %in% c("T790M"), paste(mutation), "")), size = 3, position = pos, max.overlaps = Inf) +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_vline(xintercept=c(0,0),linetype="dotted") +
  scale_color_manual( values =c("black", "salmon", "grey65")) +
theme_classic(base_size = 12)
plot7
```
```{r}
plot4 <- ggplot(pegRNA_counts_osi, aes(y= L2FC_PC9_osi_plasmid_average_zscore, x= L2FC_PC9_control_plasmid_average_zscore, colour = mutation_type)) +
geom_point(size=2,  position = pos) +
labs(title= "", y = "PC9 - osimertinib vs plasmid (z-score)", x = "PC9 - control vs plasmid (z-score)", colour = "consequence") +
geom_text_repel(aes(label=ifelse(L2FC_PC9_osi_plasmid_average_zscore > 1 & L2FC_PC9_osi_plasmid_1_zscore > 1 & L2FC_PC9_osi_plasmid_2_zscore > 1 | L2FC_PC9_control_plasmid_average_zscore < -1.5 | L2FC_PC9_control_plasmid_1_zscore > 1.1 & L2FC_PC9_control_plasmid_2_zscore > 1.1 | mutation %in% c("T790M"), paste(mutation), "")), size = 3, position = pos, max.overlaps = Inf) +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_vline(xintercept=c(0,0),linetype="dotted") +
  scale_shape_manual(values =c(1)) +
  scale_color_manual( values =c("black", "salmon", "grey65")) +
theme_classic()
plot4
ggsave("Fig_prime5c.eps", width = 6, height = 4, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

```

```{r}
# replicate correlation - stats PC9 prime editing
r1 <- cor(pegRNA_counts_osi$L2FC_PC9_control_plasmid_1_zscore, pegRNA_counts_osi$L2FC_PC9_control_plasmid_2_zscore, method = "pearson")
r2 <- cor(pegRNA_counts_osi$L2FC_PC9_osi_plasmid_1_zscore, pegRNA_counts_osi$L2FC_PC9_osi_plasmid_2_zscore, method = "pearson")
```

```{r}
#replicates
plot1 <- ggplot(pegRNA_counts_gefit, aes(x= L2FC_PC9_control_plasmid_1_zscore, y= L2FC_PC9_control_plasmid_2_zscore, colour = mutation_type)) +
geom_pointdensity(size=3,  position = pos) +
labs(title= "PC9 - prime editing screens", x = "control vs plasmid exp.1 (z-score)", y = "control vs plasmid exp.2 (z-score)", colour = "consequence") +
geom_text_repel(aes(label=ifelse(L2FC_PC9_control_plasmid_1_zscore < -1 & L2FC_PC9_control_plasmid_2_zscore < -1, paste(mutation), "")), size = 3, position = pos, max.overlaps = Inf) +
theme_bw(base_size = 15)
plot1

plot2 <- ggplot(pegRNA_counts_gefit, aes(x= L2FC_PC9_gefit_control_1_zscore, y= L2FC_PC9_gefit_control_2_zscore, colour = mutation_type)) +
geom_pointdensity(size=3,  position = pos) +
labs(title= "PC9 - prime editing screens", x = "gefitinib vs control exp.1 (z-score)", y = "gefitinib vs control exp.2 (z-score)", colour = "consequence") +
geom_text_repel(aes(label=ifelse(L2FC_PC9_gefit_control_1_zscore >1 & L2FC_PC9_gefit_control_2_zscore >1, paste(mutation), "")), size = 3, position = pos, max.overlaps = Inf) +
theme_bw(base_size = 15)
plot2

plot3 <- ggplot(pegRNA_counts_gefit, aes(x= L2FC_PC9_gefit_plasmid_1_zscore, y= L2FC_PC9_gefit_plasmid_2_zscore, colour = mutation_type)) +
geom_pointdensity(size=3,  position = pos) +
labs(title= "PC9 - prime editing screens", x = "gefitinib vs plasmid exp.1 (z-score)", y = "gefitinib vs plasmid exp.2 (z-score)", colour = "consequence") +
geom_text_repel(aes(label=ifelse(L2FC_PC9_gefit_plasmid_1_zscore >1 & L2FC_PC9_gefit_plasmid_2_zscore >1, paste(mutation), "")), size = 3, position = pos, max.overlaps = Inf) +
theme_bw(base_size = 15)
plot3

plot4 <- ggplot(pegRNA_counts_gefit, aes(x= L2FC_PC9_gefit_t7_1_zscore, y= L2FC_PC9_gefit_t7_2_zscore, colour = mutation_type)) +
geom_pointdensity(size=3,  position = pos) +
labs(title= "PC9 - prime editing screens", x = "gefitinib vs t7 exp.1 (z-score)", y = "gefitinib vs t7 exp.2 (z-score)", colour = "consequence") +
geom_text_repel(aes(label=ifelse(L2FC_PC9_gefit_plasmid_1_zscore >1 & L2FC_PC9_gefit_plasmid_2_zscore >1, paste(mutation), "")), size = 3, position = pos, max.overlaps = Inf) +
theme_bw(base_size = 15)
plot4

###

plot5 <- ggplot(pegRNA_counts_osi, aes(x= L2FC_PC9_control_plasmid_1_zscore, y= L2FC_PC9_control_plasmid_2_zscore)) +
geom_pointdensity(size=3,  position = pos) +
labs(title= "", x = "control vs plasmid exp.1 (z-score)", y = "control vs plasmid exp.2 (z-score)", colour = "density") +
geom_text_repel(aes(label=ifelse(L2FC_PC9_control_plasmid_1_zscore < -1 & L2FC_PC9_control_plasmid_2_zscore < -1 | L2FC_PC9_control_plasmid_1_zscore > 1 & L2FC_PC9_control_plasmid_2_zscore > 1, paste(mutation), "")), size = 3, position = pos, max.overlaps = Inf) +
annotate("text", y = 5, x = 0, label = paste("r = ", round(r1, 2)), parse = FALSE, size = 7) +
geom_smooth(method = lm) +
theme_classic(base_size = 12)
plot5
ggsave("Fig_primeSupp5c_2.eps", width = 6, height = 4, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

plot6 <- ggplot(pegRNA_counts_osi, aes(x= L2FC_PC9_osi_control_1_zscore, y= L2FC_PC9_osi_control_2_zscore, colour = mutation_type)) +
geom_pointdensity(size=3,  position = pos) +
labs(title= "PC9 - prime editing screens", x = "osimertinib vs control exp.1 (z-score)", y = "osimertinib vs control exp.2 (z-score)", colour = "consequence") +
geom_text_repel(aes(label=ifelse(L2FC_PC9_osi_control_1_zscore >1 & L2FC_PC9_osi_control_2_zscore >1, paste(mutation), "")), size = 3, position = pos, max.overlaps = Inf) +
theme_bw(base_size = 15)
plot6

plot7 <- ggplot(pegRNA_counts_osi, aes(x= L2FC_PC9_osi_plasmid_1_zscore, y= L2FC_PC9_osi_plasmid_2_zscore)) +
geom_pointdensity(size=3,  position = pos) +
labs(title= "", x = "osimertinib vs plasmid exp.1 (z-score)", y = "osimertinib vs plasmid exp.2 (z-score)", colour = "density") +
geom_text_repel(aes(label=ifelse(L2FC_PC9_osi_plasmid_1_zscore >1 & L2FC_PC9_osi_plasmid_2_zscore >1 | L2FC_PC9_osi_plasmid_1_zscore < -1 & L2FC_PC9_osi_plasmid_2_zscore < -1, paste(mutation), "")), size = 3, position = pos, max.overlaps = Inf) +
geom_smooth(method = lm) +
annotate("text", y = 5, x = 0, label = paste("r = ", round(r2, 2)), parse = FALSE, size = 7) +
theme_classic(base_size = 12)
plot7
ggsave("Fig_primeSupp5c.eps", width = 6, height = 4, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

plot8 <- ggplot(pegRNA_counts_osi, aes(x= L2FC_PC9_osi_t7_1_zscore, y= L2FC_PC9_osi_t7_2_zscore, colour = mutation_type)) +
geom_pointdensity(size=3,  position = pos) +
labs(title= "PC9 - prime editing screens", x = "osimertinib vs t7 exp.1 (z-score)", y = "osimertinib vs t7 exp.2 (z-score)", colour = "consequence") +
geom_text_repel(aes(label=ifelse(L2FC_PC9_osi_plasmid_1_zscore >1 & L2FC_PC9_osi_plasmid_2_zscore >1, paste(mutation), "")), size = 3, position = pos, max.overlaps = Inf) +
theme_bw(base_size = 15)
plot8
```

```{r}
#prime editing - MAGeCK results
# MAGeCK
MAGeCK_PE_PC9_control <- as_tibble(read.csv("PE_guide_count/MAGeCK_plasmid_vs_control_combined.sgrna_summary.csv", stringsAsFactors = FALSE, header = TRUE))
MAGeCK_PE_PC9_osim <- as_tibble(read.csv("PE_guide_count/MAGeCK_plasmid_vs_osimertinib_combined.sgrna_summary.csv", stringsAsFactors = FALSE, header = TRUE))
MAGeCK_RRA_PE_PC9_control <- as_tibble(read.csv("PE_guide_count/RRA_plasmid_vs_control_combined.sgrna_summary.csv", stringsAsFactors = FALSE, header = TRUE))
MAGeCK_RRA_PE_PC9_osim <- as_tibble(read.csv("PE_guide_count/RRA_plasmid_vs_osimertinib_combined.sgrna_summary.csv", stringsAsFactors = FALSE, header = TRUE))

MAGeCK_PE_PC9_control <- MAGeCK_PE_PC9_control %>% 
  rename_with(~ if_else(. %in% c("sgrna"), ., paste0(., "_MAGeCK_control")), everything()) %>%
      rename(sgRNA_ID = sgrna) %>%
          mutate(p_value_MAGeCK_control = ifelse(high_in_treatment_MAGeCK_control == "True", p.high_MAGeCK_control, p.low_MAGeCK_control))

MAGeCK_PE_PC9_osim <- MAGeCK_PE_PC9_osim %>% 
  rename_with(~ if_else(. %in% c("sgrna"), ., paste0(., "_MAGeCK_osim")), everything()) %>%
      rename(sgRNA_ID = sgrna) %>%
          mutate(p_value_MAGeCK_osim = ifelse(high_in_treatment_MAGeCK_osim == "True", p.high_MAGeCK_osim, p.low_MAGeCK_osim))

MAGeCK_RRA_PE_PC9_control <- MAGeCK_RRA_PE_PC9_control %>% 
  rename_with(~ if_else(. %in% c("sgrna"), ., paste0(., "_MAGeCK_control_RRA")), everything()) %>%
      rename(sgRNA_ID = sgrna) %>%
          mutate(p_value_MAGeCK_control_RRA = ifelse(high_in_treatment_MAGeCK_control_RRA == "True", p.high_MAGeCK_control_RRA, p.low_MAGeCK_control_RRA))

MAGeCK_RRA_PE_PC9_osim <- MAGeCK_RRA_PE_PC9_osim %>% 
  rename_with(~ if_else(. %in% c("sgrna"), ., paste0(., "_MAGeCK_osim_RRA")), everything()) %>%
      rename(sgRNA_ID = sgrna) %>%
          mutate(p_value_MAGeCK_osim_RRA = ifelse(high_in_treatment_MAGeCK_osim_RRA == "True", p.high_MAGeCK_osim_RRA, p.low_MAGeCK_osim_RRA))
```

```{r}
# join MAGeCK scores to the pegRNA counts
 pegRNA_counts_osi <- pegRNA_counts_osi %>%
  rename(sgRNA_ID = id) %>%
   rename(Gene = gene)

pegRNA_counts_osi <- left_join(pegRNA_counts_osi, MAGeCK_PE_PC9_control)
pegRNA_counts_osi <- left_join(pegRNA_counts_osi, MAGeCK_PE_PC9_osim)
pegRNA_counts_osi <- left_join(pegRNA_counts_osi, MAGeCK_RRA_PE_PC9_control)
pegRNA_counts_osi <- left_join(pegRNA_counts_osi, MAGeCK_RRA_PE_PC9_osim)
```

```{r}
# plot MAGeCK results for prime editing screens
#shows the RRA and the noral MAGeCK algorthim work very similarly. 
# can adopt the RRA method to parse the control_guides

##FDR control
plot1 <- ggplot(pegRNA_counts_osi, aes(y= FDR_MAGeCK_control, x= FDR_MAGeCK_control_RRA, colour = p_value_MAGeCK_control_RRA)) +
geom_point(size=3) +
labs(title= "", y = "FDR_MAGeCK_control_RRA", x = "FDR_MAGeCK_control_RRA", colour = "p_value_MAGeCK_control_RRA") +
geom_text_repel(aes(label=ifelse(L2FC_PC9_osi_plasmid_average_zscore > 1 & L2FC_PC9_osi_plasmid_1_zscore > 1 & L2FC_PC9_osi_plasmid_2_zscore > 1 | L2FC_PC9_control_plasmid_average_zscore < -1.5 | L2FC_PC9_control_plasmid_1_zscore > 1.1 & L2FC_PC9_control_plasmid_2_zscore > 1.1 | mutation %in% c("T790M"), paste(mutation), "")), size = 3, max.overlaps = Inf) +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_vline(xintercept=c(0,0),linetype="dotted") +
theme_classic(base_size = 12)
plot1

##FDR osi
plot2 <- ggplot(pegRNA_counts_osi, aes(y= FDR_MAGeCK_osim, x= FDR_MAGeCK_osim_RRA, colour = p_value_MAGeCK_osim_RRA)) +
geom_point(size=3) +
labs(title= "", y = "FDR_MAGeCK_osim", x = "FDR_MAGeCK_osim_RRA", colour = "p_value_MAGeCK_osim_RRA") +
geom_text_repel(aes(label=ifelse(L2FC_PC9_osi_plasmid_average_zscore > 1 & L2FC_PC9_osi_plasmid_1_zscore > 1 & L2FC_PC9_osi_plasmid_2_zscore > 1 | L2FC_PC9_control_plasmid_average_zscore < -1.5 | L2FC_PC9_control_plasmid_1_zscore > 1.1 & L2FC_PC9_control_plasmid_2_zscore > 1.1 | mutation %in% c("T790M"), paste(mutation), "")), size = 3, max.overlaps = Inf) +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_vline(xintercept=c(0,0),linetype="dotted") +
theme_classic(base_size = 12)
plot2

##pvalue control
plot3 <- ggplot(pegRNA_counts_osi, aes(y= p_value_MAGeCK_control, x= p_value_MAGeCK_control_RRA, colour = FDR_MAGeCK_control_RRA)) +
geom_point(size=3) +
labs(title= "", y = "p_value_MAGeCK_control", x = "p_value_MAGeCK_control_RRA", colour = "FDR_MAGeCK_control_RRA") +
geom_text_repel(aes(label=ifelse(L2FC_PC9_osi_plasmid_average_zscore > 1 & L2FC_PC9_osi_plasmid_1_zscore > 1 & L2FC_PC9_osi_plasmid_2_zscore > 1 | L2FC_PC9_control_plasmid_average_zscore < -1.5 | L2FC_PC9_control_plasmid_1_zscore > 1.1 & L2FC_PC9_control_plasmid_2_zscore > 1.1 | mutation %in% c("T790M"), paste(mutation), "")), size = 3, max.overlaps = Inf) +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_vline(xintercept=c(0,0),linetype="dotted") +
theme_classic(base_size = 12)
plot3

##pvalue osi
plot4 <- ggplot(pegRNA_counts_osi, aes(y= p_value_MAGeCK_osim, x= p_value_MAGeCK_osim_RRA, colour = FDR_MAGeCK_osim_RRA)) +
geom_point(size=3) +
labs(title= "", y = "p_value_MAGeCK_osim", x = "p_value_MAGeCK_osim_RRA", colour = "FDR_MAGeCK_osim_RRA") +
geom_text_repel(aes(label=ifelse(L2FC_PC9_osi_plasmid_average_zscore > 1 & L2FC_PC9_osi_plasmid_1_zscore > 1 & L2FC_PC9_osi_plasmid_2_zscore > 1 | L2FC_PC9_control_plasmid_average_zscore < -1.5 | L2FC_PC9_control_plasmid_1_zscore > 1.1 & L2FC_PC9_control_plasmid_2_zscore > 1.1 | mutation %in% c("T790M"), paste(mutation), "")), size = 3, max.overlaps = Inf) +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_vline(xintercept=c(0,0),linetype="dotted") +
theme_classic(base_size = 12)
plot4
```
```{r}
# outliers identified with poor correlation between replicates 

plot1 <- ggplot(pegRNA_counts_osi, aes(y= L2FC_PC9_osi_plasmid_average_zscore, x= L2FC_PC9_control_plasmid_average_zscore, colour = FDR_MAGeCK_osim_RRA , size = p_value_MAGeCK_osim_RRA)) +
geom_pointdensity(position = pos) +
labs(title= "", y = "PC9 - osimertinib vs plasmid (z-score)", x = "PC9 - control vs plasmid (z-score)", colour = "FDR_MAGeCK_osim", size = "p_value_MAGeCK_osim") +
geom_text_repel(aes(label=ifelse(L2FC_PC9_osi_plasmid_average_zscore > 1 & L2FC_PC9_osi_plasmid_1_zscore > 1 & L2FC_PC9_osi_plasmid_2_zscore > 1 | L2FC_PC9_control_plasmid_average_zscore < -1.5 | L2FC_PC9_control_plasmid_1_zscore > 1.1 & L2FC_PC9_control_plasmid_2_zscore > 1.1 | mutation %in% c("T790M") | FDR_MAGeCK_osim > 0.1, paste(mutation), "")), position = pos, max.overlaps = Inf, size = 4) +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_vline(xintercept=c(0,0),linetype="dotted") +
#scale_size_continuous(trans = "reverse", range = c(2,4)) + 
theme_classic(base_size = 12)
plot1
```

```{r}
# make a new filtered table for prime editor screens

prime_editing_supplementary_table <- pegRNA_counts_osi %>%
  select(Gene, mutation, pridict_rank, pegRNA, length, sgRNA_ID, mutation_type, L2FC_PC9_control_plasmid_1_zscore, L2FC_PC9_control_plasmid_2_zscore, L2FC_PC9_osi_plasmid_1_zscore, L2FC_PC9_osi_plasmid_2_zscore, L2FC_PC9_control_plasmid_average_zscore, L2FC_PC9_osi_plasmid_average_zscore, plasmid_reads, control_exp3_reads, control_exp4_reads, osi_exp3_reads, osi_exp4_reads, high_in_treatment_MAGeCK_control_RRA, FDR_MAGeCK_control_RRA, p.high_MAGeCK_control_RRA, p.low_MAGeCK_control_RRA, p_value_MAGeCK_control_RRA, high_in_treatment_MAGeCK_osim_RRA, p_value_MAGeCK_osim_RRA, FDR_MAGeCK_osim_RRA, p.high_MAGeCK_osim_RRA, p.low_MAGeCK_osim_RRA) %>%
  rename(control_exp1_reads = control_exp3_reads) %>%
  rename(control_exp2_reads = control_exp4_reads) %>%
  rename(osi_exp1_reads = osi_exp3_reads) %>%
  rename(osi_exp2_reads = osi_exp4_reads) %>%
  rename(high_in_treatment_MAGeCK_control = high_in_treatment_MAGeCK_control_RRA) %>%
  rename(FDR_MAGeCK_control = FDR_MAGeCK_control_RRA) %>%
  rename(p.high_MAGeCK_control = p.high_MAGeCK_control_RRA) %>%
  rename(p.low_MAGeCK_control = p.low_MAGeCK_control_RRA) %>%
  rename(p_value_MAGeCK_control = p_value_MAGeCK_control_RRA) %>%
  rename(high_in_treatment_MAGeCK_osim = high_in_treatment_MAGeCK_osim_RRA) %>%
  rename(p_value_MAGeCK_osim = p_value_MAGeCK_osim_RRA) %>%
  rename(FDR_MAGeCK_osim = FDR_MAGeCK_osim_RRA) %>%
  rename(p.high_MAGeCK_osim = p.high_MAGeCK_osim_RRA) %>%
  rename(p.low_MAGeCK_osim = p.low_MAGeCK_osim_RRA)
  
prime_editing_supplementary_table
write.csv(prime_editing_supplementary_table, "prime_editing_supplementary_table.csv")
  
#investigate outlier - poor correlation between replicates
prime_editing_supplementary_table %>% 
  filter(FDR_MAGeCK_osim > 0.1)
```

```{r}
#prime editing amplicon seq
PE_amp_seq_gefit <- as_tibble(read.csv("PE_guide_count/PE_EGFR_3226_variant_data.csv", stringsAsFactors = FALSE, header = TRUE))
PE_amp_seq_filtered_gefit <- PE_amp_seq_gefit %>% filter(str_detect(protein, "A1013|C797|D1006|D1012|T790|V1011"))

PE_amp_seq_osi <- as_tibble(read.csv("PE_guide_count/project_3272_variant_report_egfr_jan2024.csv", stringsAsFactors = FALSE, header = TRUE))
PE_amp_seq_filtered_osi <- PE_amp_seq_osi %>% filter(str_detect(protein, "A1013|C797|D1006|D1012|T790|V1011"))
```

```{r}
# convert vaf into comparable conditions between sample using pivot_wider
pegRNA_counts_gefit
PE_amp_seq_filtered_gefit

pegRNA_counts_osi
PE_amp_seq_filtered_osi

# compare pegRNA counts with the vafs
PE_amp_seq_VAF <- as_tibble(read.csv("PE_amp_seq_pivot_171023.csv", stringsAsFactors = FALSE, header = TRUE))
PE_amp_seq_VAF
```

```{r}
#plot PE amplicon seq data
#control
plot1 <- ggplot(PE_amp_seq_VAF, aes(y= control_exp1, x= gefitinib_exp1)) +
geom_point(size=3) +
labs(title= "PC9 - prime editing screens", y = "control_exp1", x = "gefitinib_exp1") +
geom_text_repel(aes(label=protein), size = 3, max.overlaps = Inf) +
  theme_bw(base_size = 15) 
plot1

plot2 <- ggplot(PE_amp_seq_VAF, aes(y= control_exp2, x= gefitinib_exp2)) +
geom_point(size=3) +
labs(title= "PC9 - prime editing screens", y = "control_exp2", x = "gefitinib_exp2") +
geom_text_repel(aes(label=protein), size = 3, max.overlaps = Inf) +
  theme_bw(base_size = 15) 
plot2

#t7
plot3 <- ggplot(PE_amp_seq_VAF, aes(y= t7_exp1, x= gefitinib_exp1)) +
geom_point(size=3) +
labs(title= "PC9 - prime editing screens", y = "t7_exp1", x = "gefitinib_exp1") +
geom_text_repel(aes(label=protein), size = 3, max.overlaps = Inf) +
  theme_bw(base_size = 15) 
plot3

plot4 <- ggplot(PE_amp_seq_VAF, aes(y= t7_exp2, x= gefitinib_exp2)) +
geom_point(size=3) +
labs(title= "PC9 - prime editing screens", y = "t7_exp2", x = "gefitinib_exp2") +
geom_text_repel(aes(label=protein), size = 3, max.overlaps = Inf) +
  theme_bw(base_size = 15) 
plot4

plot5 <- ggplot(PE_amp_seq_VAF, aes(y= t0_exp1, x= gefitinib_exp1)) +
geom_point(size=3) +
labs(title= "PC9 - prime editing screens", y = "t0_exp1", x = "gefitinib_exp1") +
geom_text_repel(aes(label=protein), size = 3, max.overlaps = Inf) +
  theme_bw(base_size = 15) 
plot5

plot6 <- ggplot(PE_amp_seq_VAF, aes(y= t0_exp2, x= gefitinib_exp2)) +
geom_point(size=3) +
labs(title= "PC9 - prime editing screens", y = "t0_exp2", x = "gefitinib_exp2") +
geom_text_repel(aes(label=protein), size = 3, max.overlaps = Inf) +
  theme_bw(base_size = 15) 
plot6
```

```{r}
#join pegRNA_counts and VAF by protein (change)

pegRNA_counts_gefit <- pegRNA_counts_gefit %>% 
  mutate(protein = str_remove(mutation, "mut1|mut2|mut3|mut4|mut5"))

pegRNA_counts_osi <- pegRNA_counts_osi %>% 
  mutate(mutation = str_remove(mutation, "mut1|mut2|mut3|mut4|mut5"))

PE_amp_seq_VAF <- PE_amp_seq_VAF %>% 
  mutate(vaf_gefitinib_vs_control_exp1 = gefitinib_exp1/control_exp1)
PE_amp_seq_VAF <- PE_amp_seq_VAF %>% 
  mutate(vaf_gefitinib_vs_control_exp2 = gefitinib_exp2/control_exp2)

PE_amp_seq_VAF_join <- left_join(pegRNA_counts_gefit, PE_amp_seq_VAF)
PE_amp_seq_VAF_join

PE_amp_seq_filtered_osi <- PE_amp_seq_filtered_osi %>% 
  mutate(protein = str_remove(protein, "p.")) %>% 
  rename(mutation = protein) %>%
  distinct()

pegRNA_counts_osi
PE_amp_seq_filtered_osi

PE_amp_seq_VAF_join_osi <- left_join(pegRNA_counts_osi, PE_amp_seq_filtered_osi)
PE_amp_seq_VAF_join_osi
```

```{r}
#plot PE amplicon seq data
#control pegRNA vs amplicon seq

plot1 <- ggplot(PE_amp_seq_VAF_join, aes(y= L2FC_PC9_gefit_control_1_zscore, x= vaf_gefitinib_vs_control_exp1)) +
geom_point(size=3) +
labs(title= "PC9 - prime editing screens", y = "L2FC_PC9_gefit_control_1_zscore", x = "vaf_gefitinib_vs_control_exp1") +
geom_text_repel(aes(label=protein), size = 3, max.overlaps = Inf) +
  theme_bw(base_size = 15) 
plot1

plot2 <- ggplot(PE_amp_seq_VAF_join, aes(y= L2FC_PC9_gefit_control_2_zscore, x= vaf_gefitinib_vs_control_exp2)) +
geom_pointdensity(size=3,  position = pos) +
labs(title= "PC9 - prime editing screens", y = "L2FC_PC9_gefit_control_2_zscore", x = "vaf_gefitinib_vs_control_exp2", colour = "density") +
geom_text_repel(aes(label=ifelse(L2FC_PC9_gefit_control_2_zscore > 1, paste(protein), "")), size = 3, position = pos, max.overlaps = Inf) +
  theme_bw(base_size = 15) 
plot2
```

```{r}
# mutate sample names in osi datasets
PE_amp_seq_VAF_join_osi <- 
  PE_amp_seq_VAF_join_osi %>% 
  mutate(sample = ifelse(sample %in% c("PE_B1","PE_C1"), "t0_exp1", ifelse(sample %in% c("PE_B2","PE_C2"), "t7_exp1", ifelse(sample %in% c("PE_B3","PE_C3"), "control_exp1", ifelse(sample %in% c("PE_B4", "PE_C4"), "osi_exp1", ifelse(sample %in% c("PE_B5","PE_C5"), "t0_exp2", ifelse(sample %in% c("PE_B6","PE_C6"), "t7_exp2", ifelse(sample %in% c("PE_B7","PE_C7"), "control_exp2", ifelse(sample %in% c("PE_B8","PE_C8"), "osi_exp2", sample)))))))))

# pivot_wider
pegRNA_counts_osi
PE_amp_seq_filtered_osi

PE_amp_seq_filtered_osi <- PE_amp_seq_filtered_osi %>% 
  mutate(sample = ifelse(sample %in% c("PE_B1","PE_C1"), "t0_exp1", ifelse(sample %in% c("PE_B2","PE_C2"), "t7_exp1", ifelse(sample %in% c("PE_B3","PE_C3"), "control_exp1", ifelse(sample %in% c("PE_B4", "PE_C4"), "osi_exp1", ifelse(sample %in% c("PE_B5","PE_C5"), "t0_exp2", ifelse(sample %in% c("PE_B6","PE_C6"), "t7_exp2", ifelse(sample %in% c("PE_B7","PE_C7"), "control_exp2", ifelse(sample %in% c("PE_B8","PE_C8"), "osi_exp2", sample)))))))))

PE_amp_seq_filtered_osi <-
  PE_amp_seq_filtered_osi %>% 
  select(!c(mtr, wtr, depth))%>%
  pivot_wider(names_from = sample, values_from = vaf)

pegRNA_counts_osi
PE_amp_seq_filtered_osi
PE_amp_seq_VAF_join_osi <-
  left_join(pegRNA_counts_osi, PE_amp_seq_filtered_osi)

PE_amp_seq_VAF_join_osi

# make a VAF ratiometric value
PE_amp_seq_VAF_join_osi <- 
  PE_amp_seq_VAF_join_osi %>%
  mutate(vaf_osi_vs_control = ((osi_exp1+osi_exp2)/2)/((control_exp1+control_exp2)/2)) %>%
   mutate(vaf_osi_vs_t0 = ((osi_exp1+osi_exp2)/2)/((t0_exp1+t0_exp2)/2)) %>%
   mutate(vaf_osi_vs_t7 = ((osi_exp1+osi_exp2)/2)/((t7_exp1+t7_exp2)/2))

PE_amp_seq_VAF_join_osi
```

```{r}
#plot PE amplicon seq data
#control pegRNA vs amplicon seq

plot1 <- ggplot(PE_amp_seq_VAF_join_osi, aes(y= L2FC_PC9_osi_plasmid_average_zscore, x= vaf_osi_vs_control)) +
geom_pointdensity(size=3) +
labs(title= "", y = "PC9 - pegRNA sequencing - \nosimertinib vs plasmid (z-score)", x = "PC9 - endogenous EGFR sequencing \nosimertinib vs control (VAF ratio)", colour = "density") +
geom_text_repel(aes(label= ifelse( L2FC_PC9_osi_plasmid_average_zscore > 3, mutation, ""), max.overlaps = Inf)) +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_vline(xintercept=c(0,0),linetype="dotted") +
theme_classic(base_size = 12) 
plot1
ggsave("Fig_prime5d.eps", width = 6, height = 4, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

plot2 <- ggplot(PE_amp_seq_VAF_join_osi, aes(y= L2FC_PC9_osi_plasmid_average_zscore, x= vaf_osi_vs_t0)) +
geom_pointdensity(size=3) +
labs(title= "PC9 - prime editing screens", y = "L2FC_PC9_osi_plasmid_average_zscore", x = "vaf_osi_vs_t0", colour = "density") +
geom_text_repel(aes(label= mutation), size = 3, max.overlaps = Inf) +
  theme_bw(base_size = 15) 
plot2

plot3 <- ggplot(PE_amp_seq_VAF_join_osi, aes(y= L2FC_PC9_osi_plasmid_average_zscore, x= vaf_osi_vs_t7)) +
geom_pointdensity(size=3) +
labs(title= "PC9 - prime editing screens", y = "L2FC_PC9_osi_plasmid_average_zscore", x = "vaf_osi_vs_t7", colour = "density") +
geom_text_repel(aes(label= mutation), size = 3, max.overlaps = Inf) +
  theme_bw(base_size = 15) 
plot3
```

```{r}
# mek1 mek2 check gold standard variants
filtered_annotated_data %>%
  filter(Gene %in% c("MAP2K1")) %>%
  filter(str_detect(Amino_Acid_Position, "283")==TRUE) %>%
           select(Gene, Protein_Change, L2FC_HT29_Tram_plasmid_average_zscore, Amino_Acid_Position)

filtered_annotated_data %>%
  filter(Gene %in% c("MAP2K1", "MAP2K2")) %>%
  filter(L2FC_HT29_Tram_plasmid_average_zscore >3) %>%
           select(Gene, Protein_Change, L2FC_HT29_Tram_plasmid_average_zscore, Amino_Acid_Position)
```

```{r}
#outlier in RAS graph? 3'UTR mutation which is much more deleterious with adagrasib and control than sotorasib
filtered_annotated_data %>% 
  filter(L2FC_H23_Adag_plasmid_average_zscore < -4 & L2FC_H23_Sotor_plasmid_average_zscore > -0.5) %>%
  select(Gene, guide, editor, off_target_summary_NG, Amino_Acid_Position, Protein_Change, most_severe_consequence, plasmid_reads, L2FC_H23_Adag_plasmid_1_zscore, L2FC_H23_Adag_plasmid_2_zscore, L2FC_H23_Sotor_plasmid_1_zscore, L2FC_H23_Sotor_plasmid_2_zscore, L2FC_H23_Control_plasmid_average_zscore)

annotated_data %>% 
  filter(L2FC_H23_Adag_plasmid_average_zscore < -4 & L2FC_H23_Sotor_plasmid_average_zscore > -0.5) %>%
  select(sgRNA_ID, Gene, G_guide, editor, off_target_summary_NG, Amino_Acid_Position, Protein_Change, most_severe_consequence, plasmid_reads, L2FC_H23_Adag_plasmid_1_zscore, L2FC_H23_Adag_plasmid_2_zscore,L2FC_H23_Adag_plasmid_average_zscore, L2FC_H23_Sotor_plasmid_1_zscore, L2FC_H23_Sotor_plasmid_2_zscore, L2FC_H23_Sotor_plasmid_average_zscore, L2FC_H23_Control_plasmid_average_zscore, for_primer, rev_primer)

```

```{r}
#number of on-target vs off-target resistance mutations
Discussion_table
#EGFRi (5/9)
Discussion_table %>% 
  filter(gefitinib_resistance =="TRUE" | osimertinib_resistance =="TRUE") %>%
  filter(Gene =="EGFR") %>%
  select(G_guide) %>%
  distinct()

#EGFRi
Discussion_table %>% 
  filter(gefitinib_resistance =="TRUE" | osimertinib_resistance =="TRUE") %>%
  select(G_guide) %>%
  distinct()

#RASi (3/19)
Discussion_table %>% 
  filter(adagrasib_resistance =="TRUE" | sotorasib_resistance =="TRUE") %>%
  filter(Gene =="KRAS") %>%
  select(G_guide) %>%
  distinct()

#RASi
Discussion_table %>% 
  filter(adagrasib_resistance =="TRUE" | sotorasib_resistance =="TRUE") %>%
  select(G_guide) %>%
  distinct()

#PIK3CA (1/2)
Discussion_table %>% 
  filter(pictilisib_resistance =="TRUE") %>%
  filter(Gene =="PIK3CA") %>%
  select(G_guide) %>%
  distinct()

#PIK3CA
Discussion_table %>% 
  filter(pictilisib_resistance =="TRUE") %>%
  select(G_guide) %>%
  distinct()

#MEK1/2 (17/23)
Discussion_table %>% 
  filter(trametinib_resistance =="TRUE") %>%
  filter(Gene =="MAP2K1" | Gene == "MAP2K2") %>%
  select(G_guide) %>%
  distinct()

#MEK1/2
Discussion_table %>% 
  filter(trametinib_resistance =="TRUE") %>%
  select(G_guide) %>%
  distinct()

#BRAF/EGFR (5/25)
Discussion_table %>% 
  filter(debcet_resistance =="TRUE") %>%
  filter(Gene =="BRAF" | Gene == "EGFR") %>%
  select(G_guide) %>%
  distinct()

#BRAF/EGFR
Discussion_table %>% 
  filter(debcet_resistance =="TRUE") %>%
  select(G_guide) %>%
  distinct()

5+3+1+17+5
9+19+2+23+25
31/79*100
# of the instances of where a specific variant displayed a drug resistance phenotype, 39% of these were due to drug resistance variants on the drug target themselves. 
```

```{r}
# count how many variants we have tried to install with base editing
filtered_annotated_data %>% filter(is.na(Edit_Location) == TRUE) %>% select(gRNA_Target_Location, Edit_Location, G_guide, editor, Edit_in_Exon, Edit_in_CDS)

filtered_annotated_data %>%
  distinct() %>%
  filter(gene_alias %in% c("EGFR", "KRAS", "MYC", "PARP1", "PARP2", "MAP2K1", "MAP2K2", "BCL2", "BRAF", "AKT1", "PIK3CA")) %>%
  count()

filtered_annotated_data %>%
  filter(is.na(Edit_Location) == FALSE)%>%
  filter(gene_alias %in% c("EGFR", "KRAS", "MYC", "PARP1", "PARP2", "MAP2K1", "MAP2K2", "BCL2", "BRAF", "AKT1", "PIK3CA")) %>%
  select(G_guide, Edit_Location, Gene, editor) %>%
  distinct()

filtered_annotated_data %>%
  filter(is.na(Edit_Location) == FALSE)%>%
  filter(gene_alias %in% c("EGFR", "KRAS", "MYC", "PARP1", "PARP2", "MAP2K1", "MAP2K2", "BCL2", "BRAF", "AKT1", "PIK3CA")) %>%
  select(G_guide, Edit_Location, Gene, editor) %>%
  distinct() %>%
  count()

filtered_annotated_data %>%
  distinct() %>%
  filter(gene_alias %in% c("EGFR", "KRAS", "MYC", "PARP1", "PARP2", "MAP2K1", "MAP2K2", "BCL2", "BRAF", "AKT1", "PIK3CA")) %>%
    filter(is.na(Edit_Location) == TRUE)%>%
  filter(editor == "ABE") %>%
  select(editor, G_guide)
```

```{r}
# merge filtered MHHES1 data with the rest and write a new supplementary table with all data and z-scores

MAGeCK_stats <-
  filtered_annotated_data %>% select((contains("FDR") | contains ("p_value")), sgRNA_ID, editor) %>%
  distinct()

filtered_annotated_data_2 <- 
  filtered_annotated_data %>% select(-(contains("MAGeCK"))) %>%
  distinct()

filtered_annotated_data_3 <- 
  left_join(filtered_annotated_data_2, MAGeCK_stats)
filtered_annotated_data_3

MAGeCK_stats_MHHES1 <-
  filtered_test_clean_MHHES1_data %>% select((contains("FDR") | contains ("p_value")), sgRNA_ID, editor) %>%
  distinct()

filtered_test_clean_MHHES1_data_2 <- 
  filtered_test_clean_MHHES1_data %>% select(-(contains("MAGeCK"))) %>%
  distinct()

filtered_test_clean_MHHES1_data_3 <- 
  left_join(filtered_test_clean_MHHES1_data_2, MAGeCK_stats)

filtered_annotated_data_3 
filtered_test_clean_MHHES1_data_3

Supplementary_Table_X <- left_join(filtered_annotated_data_3, filtered_test_clean_MHHES1_data_3)
Supplementary_Table_X
write.csv(Supplementary_Table_X, "Supplementary_Table_X.csv")
```

```{r}
# extract interesting non-protein-coding hits that are GoF or cause drug resistance
#targets <- c("EGFR", "KRAS", "MYC", "PARP1", "PARP2", "MAP2K1", "MAP2K2", "BCL2", "BRAF", "AKT1", "PIK3CA")

filtered_annotated_data %>% 
  count(most_severe_consequence)

#KRAS drivers
filtered_annotated_data %>%
  filter(most_severe_consequence %in% c("", NA, "UTR", "intron", "synonymous")) %>%
  filter(Gene == "KRAS") %>%
  filter(L2FC_H23_Control_plasmid_average_zscore > 2) %>%
  select(Gene, G_guide, L2FC_H23_Control_plasmid_average_zscore, L2FC_H23_Control_plasmid_1_zscore, L2FC_H23_Control_plasmid_2_zscore, L2FC_H23_Adag_plasmid_average_zscore, L2FC_H23_Sotor_plasmid_average_zscore, variant_class_Adag, variant_class_Sotor, off_target_summary_NG, most_severe_consequence, for_primer, rev_primer)

#KRAS drug resistance
filtered_annotated_data %>%
  filter(most_severe_consequence %in% c("", NA, "UTR", "intron", "synonymous")) %>%
  filter(Gene == "KRAS") %>%
  filter(L2FC_H23_Control_plasmid_average_zscore > 2) %>%
  filter(L2FC_H23_Adag_plasmid_average_zscore > 1 | L2FC_H23_Sotor_plasmid_average_zscore >1) %>%
  select(Gene, G_guide, L2FC_H23_Control_plasmid_average_zscore, L2FC_H23_Control_plasmid_1_zscore, L2FC_H23_Control_plasmid_2_zscore, L2FC_H23_Adag_plasmid_average_zscore, L2FC_H23_Sotor_plasmid_average_zscore, variant_class_Adag, variant_class_Sotor, off_target_summary_NG, most_severe_consequence, for_primer, rev_primer)

#EGFR
filtered_annotated_data %>%
  filter(most_severe_consequence %in% c("", NA, "UTR", "intron", "synonymous")) %>%
  filter(Gene == "EGFR") %>%
  filter(L2FC_PC9_Control_plasmid_average_zscore > 2) %>%
  select(Gene, G_guide, L2FC_PC9_Control_plasmid_average_zscore, L2FC_PC9_Control_plasmid_1_zscore, L2FC_PC9_Control_plasmid_2_zscore, L2FC_PC9_Gefit_plasmid_average_zscore, L2FC_PC9_Osim_plasmid_average_zscore, variant_class_Gefit, variant_class_Osim, off_target_summary_NG, most_severe_consequence, for_primer, rev_primer)

#MYC
filtered_annotated_data %>%
  filter(most_severe_consequence %in% c("", NA, "UTR", "intron", "synonymous")) %>%
  filter(Gene == "MYC") %>%
  filter(L2FC_HT29_Control_plasmid_average_zscore > 2) %>%
  select(Gene, G_guide, L2FC_HT29_Control_plasmid_average_zscore, L2FC_HT29_Control_plasmid_1_zscore, L2FC_HT29_Control_plasmid_2_zscore, L2FC_HT29_Tram_plasmid_average_zscore, variant_class_Tram, off_target_summary_NG, most_severe_consequence, for_primer, rev_primer)

#BCL2
filtered_annotated_data %>%
  filter(most_severe_consequence %in% c("", NA, "UTR", "intron", "synonymous")) %>%
  filter(Gene == "BCL2") %>%
  filter(L2FC_HT29_Control_plasmid_average_zscore > 2) %>%
  select(Gene, G_guide, L2FC_HT29_Control_plasmid_average_zscore, L2FC_HT29_Control_plasmid_1_zscore, L2FC_HT29_Control_plasmid_2_zscore, L2FC_HT29_Tram_plasmid_average_zscore, variant_class_Tram, off_target_summary_NG, most_severe_consequence, for_primer, rev_primer)

#BRAF
filtered_annotated_data %>%
  filter(most_severe_consequence %in% c("", NA, "UTR", "intron", "synonymous")) %>%
  filter(Gene == "BRAF") %>%
  filter(L2FC_HT29_Control_plasmid_average_zscore > 2) %>%
  select(Gene, G_guide, L2FC_HT29_Control_plasmid_average_zscore, L2FC_HT29_Control_plasmid_1_zscore, L2FC_HT29_Control_plasmid_2_zscore, L2FC_HT29_DebCet_plasmid_average_zscore, variant_class_Debcet, off_target_summary_NG, most_severe_consequence, for_primer, rev_primer)

#AKT1
filtered_annotated_data %>%
  filter(most_severe_consequence %in% c("", NA, "UTR", "intron", "synonymous")) %>%
  filter(Gene == "AKT1") %>%
  filter(L2FC_HT29_Control_plasmid_average_zscore > 2) %>%
  select(Gene, G_guide, L2FC_HT29_Control_plasmid_average_zscore, L2FC_HT29_Control_plasmid_1_zscore, L2FC_HT29_Control_plasmid_2_zscore, L2FC_HT29_Pict_plasmid_average_zscore, variant_class_Pict, off_target_summary_NG, most_severe_consequence, for_primer, rev_primer)

#PIK3CA
filtered_annotated_data %>%
  filter(most_severe_consequence %in% c("", NA, "UTR", "intron", "synonymous")) %>%
  filter(Gene == "PIK3CA") %>%
  filter(L2FC_HT29_Control_plasmid_average_zscore > 2) %>%
  select(Gene, G_guide, L2FC_HT29_Control_plasmid_average_zscore, L2FC_HT29_Control_plasmid_1_zscore, L2FC_HT29_Control_plasmid_2_zscore, L2FC_HT29_Tram_plasmid_average_zscore, variant_class_Tram, off_target_summary_NG, most_severe_consequence, for_primer, rev_primer)

#MAP2K1
filtered_annotated_data %>%
  filter(most_severe_consequence %in% c("", NA, "UTR", "intron", "synonymous")) %>%
  filter(Gene == "MAP2K1") %>%
  filter(L2FC_HT29_Control_plasmid_average_zscore > 2) %>%
  select(Gene, G_guide, L2FC_HT29_Control_plasmid_average_zscore, L2FC_HT29_Control_plasmid_1_zscore, L2FC_HT29_Control_plasmid_2_zscore, L2FC_HT29_Tram_plasmid_average_zscore, variant_class_Tram, off_target_summary_NG, most_severe_consequence, for_primer, rev_primer)

#MAP2K2
filtered_annotated_data %>%
  filter(most_severe_consequence %in% c("", NA, "UTR", "intron", "synonymous")) %>%
  filter(Gene == "MAP2K2") %>%
  filter(L2FC_HT29_Control_plasmid_average_zscore > 1.9) %>%
  select(Gene, G_guide, L2FC_HT29_Control_plasmid_average_zscore, L2FC_HT29_Control_plasmid_1_zscore, L2FC_HT29_Control_plasmid_2_zscore, L2FC_HT29_Tram_plasmid_average_zscore, variant_class_Tram, off_target_summary_NG, most_severe_consequence, for_primer, rev_primer)
```

```{r}
# order KRAS 3'UTR interesting gRNAs

#KRAS drivers
KRAS_3UTR_drivers <- filtered_annotated_data %>%
  filter(most_severe_consequence %in% c("", NA, "UTR", "intron", "synonymous")) %>%
  filter(Gene == "KRAS") %>%
  filter(L2FC_H23_Control_plasmid_average_zscore > 2) %>%
  select(sgRNA_ID, Gene, G_guide, editor, L2FC_H23_Control_plasmid_average_zscore, L2FC_H23_Control_plasmid_1_zscore, L2FC_H23_Control_plasmid_2_zscore, L2FC_H23_Adag_plasmid_average_zscore, L2FC_H23_Sotor_plasmid_average_zscore, variant_class_Adag, variant_class_Sotor, off_target_summary_NG, most_severe_consequence, for_primer, rev_primer)

#KRAS drug resistance
KRAS_3UTR_drug_res <- filtered_annotated_data %>%
  filter(most_severe_consequence %in% c("", NA, "UTR", "intron", "synonymous")) %>%
  filter(Gene == "KRAS") %>%
  filter(L2FC_H23_Control_plasmid_average_zscore > 2) %>%
  filter(L2FC_H23_Adag_plasmid_average_zscore > 1 | L2FC_H23_Sotor_plasmid_average_zscore >1) %>%
  select(sgRNA_ID, Gene, G_guide, editor, L2FC_H23_Control_plasmid_average_zscore, L2FC_H23_Control_plasmid_1_zscore, L2FC_H23_Control_plasmid_2_zscore, L2FC_H23_Adag_plasmid_average_zscore, L2FC_H23_Sotor_plasmid_average_zscore, variant_class_Adag, variant_class_Sotor, off_target_summary_NG, most_severe_consequence, for_primer, rev_primer)

KRAS_3UTR_drivers
KRAS_3UTR_drug_res
anti_join(KRAS_3UTR_drug_res, KRAS_3UTR_drivers)
write.csv(KRAS_3UTR_drivers, "KRAS_3UTR_driver_primers_valid_151123.csv")
```

```{r}
# H23 non-coding mutations
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

# control
plot1 <- ggplot(subset(filtered_annotated_data, Gene %in% c("KRAS")), aes(x= start, y= L2FC_H23_Control_plasmid_average_zscore)) + 
geom_pointdensity(size=3,  position = pos) +
scale_colour_viridis_c() +
labs(title= "H23", x = "KRAS position", y = "L2FC_H23_Control_plasmid_average_zscore", colour = "density") +
geom_text_repel(aes(label=ifelse(L2FC_H23_Control_plasmid_average_zscore > 3 | sgRNA_ID == "44c2fe4e5a2d2fc9fb92800849cd390a92de2f87", paste(Amino_Acid_Position, most_severe_consequence), "")), size = 3, position = pos, max.overlaps = Inf) +
theme_bw(base_size = 15)
plot1

# Sotorasib
plot2 <- ggplot(subset(filtered_annotated_data, Gene %in% c("KRAS")), aes(x= start, y= L2FC_H23_Sotor_plasmid_average_zscore)) + 
geom_pointdensity(size=3,  position = pos) +
scale_colour_viridis_c() +
labs(title= "H23", x = "KRAS position", y = "L2FC_H23_Sotor_plasmid_average_zscore", colour = "density") +
geom_text_repel(aes(label=ifelse(L2FC_H23_Sotor_plasmid_average_zscore > 3 | sgRNA_ID == "44c2fe4e5a2d2fc9fb92800849cd390a92de2f87", paste(Amino_Acid_Position, most_severe_consequence), "")), size = 3, position = pos, max.overlaps = Inf) +
theme_bw(base_size = 15)
plot2

# Adagrasib
plot3 <- ggplot(subset(filtered_annotated_data, Gene %in% c("KRAS")), aes(x= start, y= L2FC_H23_Adag_plasmid_average_zscore)) + 
geom_pointdensity(size=3,  position = pos) +
labs(title= "H23", x = "KRAS position", y = "L2FC_H23_Adag_plasmid_average_zscore", colour = "density") +
geom_text_repel(aes(label=ifelse(L2FC_H23_Adag_plasmid_average_zscore > 4.5, paste(Amino_Acid_Position), "")), size = 3, position = pos, max.overlaps = Inf) +
theme_bw(base_size = 15)
plot3

# hit selection
plot4 <- ggplot(subset(filtered_annotated_data, Gene %in% c("KRAS") & most_severe_consequence %in% c("", NA, "UTR", "intron")), aes(x= L2FC_H23_Adag_plasmid_1_zscore, y= L2FC_H23_Adag_plasmid_2_zscore, colour = L2FC_H23_Sotor_plasmid_average_zscore, size = L2FC_H23_Control_plasmid_average_zscore)) + 
geom_point(position = pos) +
labs(title= "H23", x = "L2FC_H23_Adag_plasmid_1_zscore", y = "L2FC_H23_Adag_plasmid_2_zscore", colour = "sotor", size = "control") +
geom_text_repel(aes(label=ifelse(L2FC_H23_Adag_plasmid_average_zscore > 4, paste(sgRNA_ID), "")), size = 3, position = pos, max.overlaps = Inf) +
theme_bw(base_size = 15)
plot4

# hit selection - non-coding
plot5 <- ggplot(subset(filtered_annotated_data, Gene %in% c("KRAS") & most_severe_consequence %in% c("", NA, "UTR", "intron")), aes(x= L2FC_H23_Sotor_plasmid_1_zscore, y= L2FC_H23_Sotor_plasmid_2_zscore, colour = L2FC_H23_Adag_plasmid_average_zscore, size = L2FC_H23_Control_plasmid_average_zscore)) + 
geom_point(position = pos) +
labs(title= "H23", x = "L2FC_H23_Sotor_plasmid_1_zscore", y = "L2FC_H23_Sotor_plasmid_2_zscore", colour = "adag", size = "control") +
geom_text_repel(aes(label=ifelse(L2FC_H23_Sotor_plasmid_average_zscore > 3, paste(sgRNA_ID), "")), size = 3, position = pos, max.overlaps = Inf) +
theme_bw(base_size = 15)
plot5

# hit selection - all gRNAs
plot6 <- ggplot(subset(filtered_annotated_data, Gene %in% c("KRAS")), aes(x= L2FC_H23_Sotor_plasmid_1_zscore, y= L2FC_H23_Sotor_plasmid_2_zscore, colour = L2FC_H23_Adag_plasmid_average_zscore, size = L2FC_H23_Control_plasmid_average_zscore)) + 
geom_point(position = pos) +
labs(title= "H23", x = "L2FC_H23_Sotor_plasmid_1_zscore", y = "L2FC_H23_Sotor_plasmid_2_zscore", colour = "adag", size = "control") +
geom_text_repel(aes(label=ifelse(L2FC_H23_Sotor_plasmid_average_zscore > 3, paste(sgRNA_ID, Amino_Acid_Position), "")), size = 3, position = pos, max.overlaps = Inf) +
theme_bw(base_size = 15)
plot6

# hit selection - all gRNAs
plot7 <- ggplot(subset(filtered_annotated_data, Gene %in% c("KRAS")), aes(x= L2FC_H23_Adag_plasmid_1_zscore, y= L2FC_H23_Adag_plasmid_2_zscore, colour = L2FC_H23_Sotor_plasmid_average_zscore, size = L2FC_H23_Control_plasmid_average_zscore)) + 
geom_point(position = pos) +
labs(title= "H23", x = "L2FC_H23_Adag_plasmid_1_zscore", y = "L2FC_H23_Adag_plasmid_2_zscore", colour = "sotor", size = "control") +
geom_text_repel(aes(label=ifelse(L2FC_H23_Adag_plasmid_average_zscore > 3, paste(Amino_Acid_Position), "")), size = 3, position = pos, max.overlaps = Inf) +
theme_bw(base_size = 15)
plot7
```

```{r}
# extract interesting non-protein-coding hits that are GoF or cause drug resistance

#KRAS drug resistance
filtered_annotated_data %>%
  filter(most_severe_consequence %in% c("", NA, "UTR", "intron")) %>%
  filter(Gene == "KRAS") %>%
  filter(L2FC_H23_Adag_plasmid_average_zscore > 2 & L2FC_H23_Sotor_plasmid_average_zscore > 2 | L2FC_H23_Adag_plasmid_average_zscore > 5) %>%
  select(Gene, G_guide, L2FC_H23_Control_plasmid_average_zscore, L2FC_H23_Control_plasmid_1_zscore, L2FC_H23_Control_plasmid_2_zscore, L2FC_H23_Adag_plasmid_average_zscore, L2FC_H23_Adag_plasmid_1_zscore, L2FC_H23_Adag_plasmid_2_zscore, L2FC_H23_Sotor_plasmid_average_zscore, L2FC_H23_Sotor_plasmid_1_zscore, L2FC_H23_Sotor_plasmid_2_zscore, variant_class_Adag, variant_class_Sotor, off_target_summary_NG, most_severe_consequence, for_primer, rev_primer, sgRNA_ID, off_target_summary_NG, editor)

filtered_annotated_data %>%
  filter(most_severe_consequence %in% c("", NA, "UTR", "intron")) %>%
  filter(Gene == "KRAS") %>%
  filter(L2FC_H23_Adag_plasmid_average_zscore > 2 & L2FC_H23_Sotor_plasmid_average_zscore > 2 | L2FC_H23_Adag_plasmid_average_zscore > 5) %>%
  select(Gene, G_guide, L2FC_H23_Control_plasmid_average_zscore, L2FC_H23_Control_plasmid_1_zscore, L2FC_H23_Control_plasmid_2_zscore, L2FC_H23_Adag_plasmid_average_zscore, L2FC_H23_Adag_plasmid_1_zscore, L2FC_H23_Adag_plasmid_2_zscore, L2FC_H23_Sotor_plasmid_average_zscore, L2FC_H23_Sotor_plasmid_1_zscore, L2FC_H23_Sotor_plasmid_2_zscore, variant_class_Adag, variant_class_Sotor, off_target_summary_NG, most_severe_consequence, for_primer, rev_primer, sgRNA_ID, off_target_summary_NG) %>%
  write.csv("KRAS_3UTR_drug_res_030123.csv")
```

```{r}
#COSMIC patient data on drug resistance

#all samples - filter for DRUG_RESPONSE and THERAPY
COSMIC_all_samples <- as_tibble(read_tsv("COSMIC/Cosmic_Sample_v99_GRCh38.tsv"))

COSMIC_all_samples_filtered <- 
  COSMIC_all_samples %>%
  select(COSMIC_SAMPLE_ID, DRUG_RESPONSE, THERAPY, THERAPY_RELATIONSHIP) %>%
  filter(is.na(DRUG_RESPONSE) == FALSE) %>%
  filter(str_detect(DRUG_RESPONSE, "Cetuximab|Gefitinib|Dabrafenib|Osimertinib|Olaparib|Niraparib|Trametinib|Sotorasib|Adagrasib|Pictilisib")) %>%
  distinct()
COSMIC_all_samples_filtered

#all samples - filter for DRUG_RESPONSE and THERAPY (Cansu did this as it was a v large file and filtered the results)
#COSMIC_all_mutations <- as_tibble(read_tsv("COSMIC/Cosmic_GenomeScreensMutant_v99_GRCh38.tsv"))

#COSMIC_all_mutations_filtered <- 
 # COSMIC_all_mutations %>%
#  filter(COSMIC_SAMPLE_ID %in% COSMIC_all_samples_filtered$COSMIC_SAMPLE_ID) %>%
#  distinct() %>%
 # write_tsv("filtered_Cosmic_GenomeScreensMutant_v99_GRCh38_050324.tsv")

#all samples - filter for DRUG_RESPONSE and THERAPY (Cansu did this as it was a v large file and filtered the results)
COSMIC_filtered_mutations <- as_tibble(read_tsv("COSMIC/filtered_Cosmic_GenomeScreensMutant_v99_GRCh38_050324.tsv"))

#left join the COSMIC samples that match the DRUG RESPONSE fields with my screened drugs with mutation data
# filter on the Genes analysed here and non-synonymous mutations and post-treatment samples
COSMIC_resistant_samples <- left_join(COSMIC_all_samples_filtered, COSMIC_filtered_mutations) %>%
  filter(GENE_SYMBOL %in% c("AKT1", "EGFR", "KRAS", "BCL2", "MYC", "BRAF", "MAP2K1", "MAP2K2", "PIK3CA", "PARP1", "PARP2")) %>%
  filter(THERAPY_RELATIONSHIP %in% c("Pretreatment sample","Sample analysed before treatment", "Sample taken before dabrafenib plus trametinib therapy started") == FALSE) %>%
  filter(MUTATION_DESCRIPTION %in% c("inframe_deletion", "intron_variant", "synonymous_variant", NA) == FALSE) %>%
  distinct()
COSMIC_resistant_samples
```

```{r}
#COSMIC mutations that affect drug sensitivity
COSMIC_res_muts <- as_tibble(read_tsv("COSMIC/Cosmic_ResistanceMutations_v99_GRCh38.tsv"))

#filter on drug types i have screened
COSMIC_res_muts_filtered <- 
  COSMIC_res_muts %>%
  filter(DRUG_NAME %in% c("Afatinib", "Cetuximab", "Dabrafenib", "Encorafenib", "Erlotinib", "Gefitinib", "Osimertinib", "Panitumumab", "PD0325901", "Selumetinib", "Vemurafenib", "Sunitinib", "XL647")) %>%
  select(DRUG_NAME, GENE_SYMBOL, MUTATION_AA) %>%
  distinct()

COSMIC_res_muts_filtered <- 
  COSMIC_res_muts_filtered %>%
  rename(Gene = GENE_SYMBOL) %>%
  rename(protein = MUTATION_AA) %>%
  rename(DRUG_RESPONSE = DRUG_NAME) %>%
  mutate(DRUG_RESPONSE = paste(DRUG_RESPONSE, "resistance"))

COSMIC_resistant_samples <- 
  COSMIC_resistant_samples %>% 
  rename(Gene = GENE_SYMBOL) %>%
  rename(protein = MUTATION_AA) %>%
  select(COSMIC_SAMPLE_ID, DRUG_RESPONSE, THERAPY_RELATIONSHIP, Gene, protein, MUTATION_ZYGOSITY, PUBMED_PMID)

COSMIC_res_muts_filtered
COSMIC_resistant_samples

COSMIC_combined_resistance_variants <- 
  full_join(COSMIC_resistant_samples, COSMIC_res_muts_filtered)

COSMIC_combined_resistance_variants
Discussion_table_filtered

left_join(Discussion_table_filtered, COSMIC_combined_resistance_variants) %>%
  distinct()
```

```{r}
#joining the COSMIC mutations to the all mutations dataframe

COSMIC <- COSMIC_combined_resistance_variants %>% 
  mutate(Amino_Acid_Position = protein) %>%
  mutate(Amino_Acid_Position = as.numeric(str_extract(Amino_Acid_Position, "[[:digit:]]+"))) %>%
  mutate(Amino_Acid_Position =as.character(Amino_Acid_Position)) %>%
  filter(is.na(Amino_Acid_Position)==FALSE) %>%
  distinct()

# read in the citations of drug resitance variants from the clinic
citations <- as_tibble(read.csv("clinical_resistance_mutations.csv", stringsAsFactors = FALSE, header = TRUE))

citations <- 
  citations %>%
  mutate(Amino_Acid_Position = as.character(str_extract(protein, "[[:digit:]]+"))) %>%
  mutate(DRUG_RESPONSE = paste(DRUG_RESPONSE, "resistance")) %>%
  mutate(COSMIC_SAMPLE_ID = NA) %>%
  mutate(THERAPY_RELATIONSHIP = NA) %>%
  mutate(MUTATION_ZYGOSITY = NA) %>%
  distinct()
  
COSMIC <-
  COSMIC %>%
  mutate(PUBMED_PMID = as.character(PUBMED_PMID))

COSMIC2 <- 
  bind_rows(COSMIC, citations) %>% 
  distinct()

all_data <- Supplementary_Table_X %>%
  select(Amino_Acid_Position, Gene, sgRNA_ID) %>%
  separate_rows(Amino_Acid_Position, sep = ";") %>%
  distinct()
  
COSMIC_summary <- 
  left_join(all_data, COSMIC2) %>%
  group_by(sgRNA_ID) %>%
  summarise(Amino_Acid_Position = paste(unique(Amino_Acid_Position), collapse = ";"),
            COSMIC_SAMPLE_ID = paste(unique(COSMIC_SAMPLE_ID), collapse = ";"),
            DRUG_RESPONSE = paste(unique(DRUG_RESPONSE), collapse = ";"),
            THERAPY_RELATIONSHIP = paste(unique(THERAPY_RELATIONSHIP), collapse = ";"),
            Gene = paste(unique(Gene), collapse = ";"),
            PUBMED_PMID = paste(unique(PUBMED_PMID), collapse = ";"), 
            MUTATION_ZYGOSITY = paste(unique(MUTATION_ZYGOSITY), collapse = ";"), 
            protein = paste(unique(protein), collapse = ";")) %>%
  select(!(Amino_Acid_Position)) %>%
  rename(COSMIC_mutation = protein) %>%
  rename(COSMIC_PUBMED_PMID = PUBMED_PMID) %>%
  rename(COSMIC_MUTATION_ZYGOSITY = MUTATION_ZYGOSITY)

COSMIC_summary
```

```{r}
#write a new supplementary table with COSMIC mutations

Supplementary_Table_Y <- 
  Supplementary_Table_X %>% 
  select(!c(plasmid_reads, plasmid_reads_RPM, cosmic_freq, sc_valid_plasmid_reads, sc_valid_plasmid_reads_RPM)) %>%
  relocate(variant_class_Olap, .after = variant_class_Osim) %>%
  relocate(variant_class_Nirap, .after = variant_class_Olap)

Supplementary_Table_Z <- 
  left_join(Supplementary_Table_Y, COSMIC_summary)

Supplementary_Table_Z
```

```{r}
# read in COSMIC counts

COSMIC_counts <- as_tibble(read.csv("COSMIC_11_230822.csv", stringsAsFactors = FALSE, header = TRUE))
COSMIC_counts <- rename(COSMIC_counts, Amino_Acid_Position = Position)
COSMIC_counts <- rename(COSMIC_counts, COSMIC_count = Count)
COSMIC_counts <- rename(COSMIC_counts, Amino_Acid_Change_COSMIC = AA.Mutation)
COSMIC_counts <- rename(COSMIC_counts, CDS_mutation_COSMIC = CDS.Mutation)
COSMIC_counts <- mutate(COSMIC_counts, Amino_Acid_Position = as.character(Amino_Acid_Position))

COSMIC_counts2 <- select(COSMIC_counts, Gene, Amino_Acid_Position, COSMIC_count) %>%
  distinct()

COSMIC_counts3 <- COSMIC_counts2 %>%
  group_by(Gene, Amino_Acid_Position) %>%
  summarise(COSMIC_count = sum(COSMIC_count)) %>% distinct()

COSMIC_counts4 <- COSMIC_counts3 %>% 
  mutate(COSMIC_count = ifelse(is.na(COSMIC_count) == TRUE, "0", COSMIC_count)) %>%
  mutate(COSMIC_count = as.numeric(COSMIC_count))

COSMIC_counts4
```

```{r}
# make a tabel of COSMIC counts that you can join with the base editing screens

all_data <- Supplementary_Table_X %>%
  select(Amino_Acid_Position, Gene, sgRNA_ID) %>%
  separate_rows(Amino_Acid_Position, sep = ";") %>%
  distinct()
  
COSMIC_counts4

COSMIC_counts_summary <- 
  left_join(all_data, COSMIC_counts4) %>%
  mutate(COSMIC_count = ifelse(is.na(COSMIC_count) == TRUE, 0, COSMIC_count)) %>%
  group_by(sgRNA_ID) %>%
  summarise(Amino_Acid_Position = paste(unique(Amino_Acid_Position), collapse = ";"),
            COSMIC_count_sum = sum(unique(COSMIC_count)),
            COSMIC_count = paste(unique(COSMIC_count), collapse =";"),
            Gene = paste(unique(Gene), collapse = ";")) %>%
            select(!(Amino_Acid_Position))

COSMIC_counts_summary
```

```{r}
# join the COSMIC counts with all data
# generate Supplementary table final

Supplementary_Table_final <- left_join(Supplementary_Table_Z, COSMIC_counts_summary)
Supplementary_Table_final
Supplementary_Table_final %>% 
  select(!c(COSMIC_count_sum, COSMIC_count, is_stop, is_synonymous)) %>%
  write.csv("Supplementary_Table_final.csv")

COSMIC2 %>% 
  filter(Gene %in% targets) %>%
write.csv("COSMIC_table.csv")

COSMIC2

Supplementary_Table_revisions <- mutate(Supplementary_Table_final, gefitinib_resistance = ifelse(L2FC_PC9_Gefit_plasmid_1_zscore > 1 & L2FC_PC9_Gefit_plasmid_2_zscore > 1 & L2FC_PC9_Gefit_plasmid_average_zscore > 2 & L2FC_PC9_Gefit_plasmid_scres1valid_zscore > 2, TRUE, FALSE))
Supplementary_Table_revisions <- mutate(Supplementary_Table_revisions, osimertinib_resistance = ifelse(L2FC_PC9_Osim_plasmid_1_zscore > 1 & L2FC_PC9_Osim_plasmid_2_zscore > 1 &  L2FC_PC9_Osim_plasmid_average_zscore > 2 & L2FC_PC9_Osim_plasmid_scres1valid_zscore > 2, TRUE, FALSE))
Supplementary_Table_revisions <- mutate(Supplementary_Table_revisions, trametinib_resistance = ifelse(L2FC_HT29_Tram_plasmid_1_zscore > 1 & L2FC_HT29_Tram_plasmid_2_zscore > 1 & L2FC_HT29_Tram_plasmid_average_zscore > 2, TRUE, FALSE))
Supplementary_Table_revisions <- mutate(Supplementary_Table_revisions, pictilisib_resistance = ifelse(L2FC_HT29_Pict_plasmid_1_zscore > 1 & L2FC_HT29_Pict_plasmid_2_zscore > 1 & L2FC_HT29_Pict_plasmid_average_zscore > 2, TRUE, FALSE))
Supplementary_Table_revisions <- mutate(Supplementary_Table_revisions, debcet_resistance = ifelse(L2FC_HT29_DebCet_plasmid_1_zscore > 1 & L2FC_HT29_DebCet_plasmid_2_zscore > 1 & L2FC_HT29_DebCet_plasmid_average_zscore > 2, TRUE, FALSE))
Supplementary_Table_revisions <- mutate(Supplementary_Table_revisions, adagrasib_resistance = ifelse(L2FC_H23_Adag_plasmid_1_zscore > 1 & L2FC_H23_Adag_plasmid_2_zscore > 1 & L2FC_H23_Adag_plasmid_average_zscore > 2, TRUE, FALSE))
Supplementary_Table_revisions <- mutate(Supplementary_Table_revisions, sotorasib_resistance = ifelse(L2FC_H23_Sotor_plasmid_1_zscore > 1 & L2FC_H23_Sotor_plasmid_2_zscore > 1 & L2FC_H23_Sotor_plasmid_average_zscore > 2, TRUE, FALSE))
Supplementary_Table_revisions <- mutate(Supplementary_Table_revisions, olaparib_resistance = ifelse(L2FC_MHHES1_Olap_plasmid_1_zscore > 1 & L2FC_MHHES1_Olap_plasmid_2_zscore > 1 & L2FC_MHHES1_Olap_plasmid_average_zscore > 2, TRUE, FALSE))
Supplementary_Table_revisions <- mutate(Supplementary_Table_revisions, niraparib_resistance = ifelse(L2FC_MHHES1_Nirap_plasmid_1_zscore > 1 & L2FC_MHHES1_Nirap_plasmid_2_zscore > 1 & L2FC_MHHES1_Nirap_plasmid_average_zscore > 2, TRUE, FALSE))

Supplementary_Table_revisions %>%
  filter(gefitinib_resistance == TRUE | osimertinib_resistance == TRUE | adagrasib_resistance == TRUE | sotorasib_resistance == TRUE | trametinib_resistance == TRUE | debcet_resistance == TRUE | pictilisib_resistance == TRUE | olaparib_resistance == TRUE | niraparib_resistance == TRUE) %>%
  distinct()
```

```{r}
# how many COSMIC resistance mutations did we find, and how many did we miss

#distinct mutations associated with resistance
COSMIC2 %>% 
  select(Gene, Amino_Acid_Position) %>%
  filter(Gene %in% targets) %>%
  filter(is.na(Amino_Acid_Position) == FALSE) %>%
  filter(Amino_Acid_Position != "") %>%
  distinct()
# 89 Amino acid resistant positions in the COSMIC database  

Supplementary_Table_revisions %>%
  select(Gene, Amino_Acid_Position) %>%
  filter(Gene %in% targets) %>%
  separate_rows(Amino_Acid_Position, sep = ";") %>%
  filter(is.na(Amino_Acid_Position) == FALSE) %>%
  filter(Amino_Acid_Position != "") %>%
  distinct()
# 6246 amino acid positions assessed in the gene targets

# find through anti-join the variants private to each dataset
cosmic_positions_in_targets <-
  COSMIC2 %>% 
  select(Gene, Amino_Acid_Position) %>%
  filter(Gene %in% targets) %>%
  filter(is.na(Amino_Acid_Position) == FALSE) %>%
  filter(Amino_Acid_Position != "") %>%
  distinct()

amino_acid_positions_screened <- 
  Supplementary_Table_revisions %>%
  select(Gene, Amino_Acid_Position) %>%
    filter(Gene %in% targets) %>%
  separate_rows(Amino_Acid_Position, sep = ";") %>%
  filter(is.na(Amino_Acid_Position) == FALSE) %>%
  filter(Amino_Acid_Position != "") %>%
  distinct()

cosmic_positions_in_targets
amino_acid_positions_screened

anti_join(cosmic_positions_in_targets, amino_acid_positions_screened)

#of 89 Amino acid positions we screened all but 3

# overlapping with COSMIC - 43 amino acid positions are resistance variants for at least one drug
Supplementary_Table_revisions %>%
  filter(is.na(DRUG_RESPONSE) == FALSE) %>%
  filter(DRUG_RESPONSE != "" & DRUG_RESPONSE != "NA") %>%
    filter(Gene %in% targets) %>%
    filter(is.na(Amino_Acid_Position) == FALSE) %>%
    filter(Amino_Acid_Position != "" & Amino_Acid_Position != "NA") %>%
  select(Amino_Acid_Position, Gene, DRUG_RESPONSE, variant_class_Adag, variant_class_Osim, variant_class_Debcet, variant_class_Pict, variant_class_Sotor, variant_class_Gefit, variant_class_Olap, variant_class_Nirap, variant_class_Tram, gefitinib_resistance, osimertinib_resistance, adagrasib_resistance, sotorasib_resistance, trametinib_resistance, debcet_resistance, pictilisib_resistance , olaparib_resistance, niraparib_resistance) %>%
  filter(gefitinib_resistance == TRUE | osimertinib_resistance == TRUE | adagrasib_resistance == TRUE | sotorasib_resistance == TRUE | trametinib_resistance == TRUE | debcet_resistance == TRUE | pictilisib_resistance == TRUE | olaparib_resistance == TRUE | niraparib_resistance == TRUE) %>%
  distinct()

# not overlapping with COSMIC - 262
Supplementary_Table_revisions %>%
  filter(DRUG_RESPONSE == "" | DRUG_RESPONSE == "NA" | is.na(DRUG_RESPONSE)==TRUE) %>%
    filter(Gene %in% targets) %>%
    filter(is.na(Amino_Acid_Position) == FALSE) %>%
    filter(Amino_Acid_Position != "" & Amino_Acid_Position != "NA") %>%
 select(sgRNA_ID, DRUG_RESPONSE, Gene, Amino_Acid_Position, Protein_Change, most_severe_consequence, gefitinib_resistance, osimertinib_resistance, adagrasib_resistance, sotorasib_resistance, trametinib_resistance, debcet_resistance, pictilisib_resistance , olaparib_resistance, niraparib_resistance, variant_class_Adag, variant_class_Osim, variant_class_Debcet, variant_class_Pict, variant_class_Sotor, variant_class_Gefit, variant_class_Olap, variant_class_Nirap, variant_class_Tram) %>%
  filter(gefitinib_resistance == TRUE | osimertinib_resistance == TRUE | adagrasib_resistance == TRUE | sotorasib_resistance == TRUE | trametinib_resistance == TRUE | debcet_resistance == TRUE | pictilisib_resistance == TRUE | olaparib_resistance == TRUE | niraparib_resistance == TRUE) %>%
  distinct()

#  overlapping with COSMIC - 151
Supplementary_Table_revisions %>%
filter(DRUG_RESPONSE != "" & DRUG_RESPONSE != "NA" & is.na(DRUG_RESPONSE)==FALSE) %>%
  filter(Gene %in% targets) %>%
    filter(is.na(Amino_Acid_Position) == FALSE) %>%
    filter(Amino_Acid_Position != "" & Amino_Acid_Position != "NA") %>%
 select(Amino_Acid_Position, Gene, DRUG_RESPONSE, variant_class_Adag, variant_class_Osim, variant_class_Debcet, variant_class_Pict, variant_class_Sotor, variant_class_Gefit, variant_class_Olap, variant_class_Nirap, variant_class_Tram, gefitinib_resistance, osimertinib_resistance, adagrasib_resistance, sotorasib_resistance, trametinib_resistance, debcet_resistance, pictilisib_resistance , olaparib_resistance, niraparib_resistance) %>%
  filter(gefitinib_resistance == FALSE & osimertinib_resistance == FALSE & adagrasib_resistance == FALSE & sotorasib_resistance == FALSE & trametinib_resistance == FALSE & debcet_resistance == FALSE & pictilisib_resistance == FALSE & olaparib_resistance == FALSE & niraparib_resistance == FALSE) %>%
  distinct()

# all coding variants in targets with COSMIC DRUG_RESPONSE - 360
Supplementary_Table_revisions %>%
  filter(DRUG_RESPONSE != "" & DRUG_RESPONSE != "NA" & is.na(DRUG_RESPONSE)==FALSE) %>%
    filter(Gene %in% targets) %>%
    filter(is.na(Amino_Acid_Position) == FALSE) %>%
    filter(Amino_Acid_Position != "" & Amino_Acid_Position != "NA") %>%
  select(Amino_Acid_Position, Gene, DRUG_RESPONSE, variant_class_Adag, variant_class_Osim, variant_class_Debcet, variant_class_Pict, variant_class_Sotor, variant_class_Gefit, variant_class_Olap, variant_class_Nirap, variant_class_Tram, gefitinib_resistance, osimertinib_resistance, adagrasib_resistance, sotorasib_resistance, trametinib_resistance, debcet_resistance, pictilisib_resistance , olaparib_resistance, niraparib_resistance) %>%
  distinct()

# total coding drug class variants in targets including drug-sensitising - 518
Supplementary_Table_revisions %>%
    filter(Gene %in% targets) %>%
    filter(is.na(Amino_Acid_Position) == FALSE) %>%
    filter(Amino_Acid_Position != "" & Amino_Acid_Position != "NA") %>%
  filter(variant_class_Adag != "control" | variant_class_Osim != "control" | variant_class_Debcet!= "control" | variant_class_Pict!= "control" | variant_class_Sotor!= "control" | variant_class_Gefit!= "control" | variant_class_Olap!= "control" | variant_class_Nirap!= "control" | variant_class_Tram!= "control") %>% 
  distinct()

# total coding drug resistance variants in targets - 316
Supplementary_Table_revisions %>%
  filter(gefitinib_resistance == TRUE | osimertinib_resistance == TRUE | adagrasib_resistance == TRUE | sotorasib_resistance == TRUE | trametinib_resistance == TRUE | debcet_resistance == TRUE | pictilisib_resistance == TRUE | olaparib_resistance == TRUE | niraparib_resistance == TRUE) %>%
  filter(Gene %in% targets) %>%
    filter(is.na(Amino_Acid_Position) == FALSE) %>%
    filter(Amino_Acid_Position != "" & Amino_Acid_Position != "NA") %>%
  distinct()

#of 89 amino acid positions with COSMIC-annotated drug response data in target proteins, 43/89 (48.31 %) were found in our variant dataset to be associated with a drug resistance phenotype for at least one drug, and 31/43 (72.09 %) were correctly associated with resistance to the specific drug used clinically.
```

```{r}

test1 <- Supplementary_Table_revisions %>%
  select(sgRNA_ID, editor, Gene, Amino_Acid_Position, trametinib_resistance, pictilisib_resistance, debcet_resistance, sotorasib_resistance, adagrasib_resistance, gefitinib_resistance, osimertinib_resistance, olaparib_resistance, niraparib_resistance) %>%
    filter(Gene %in% targets) %>%
  separate_rows(Amino_Acid_Position, sep = ";") %>%
  filter(is.na(Amino_Acid_Position) == FALSE) %>%
  filter(Amino_Acid_Position != "") %>%
  #filter(gefitinib_resistance == TRUE | osimertinib_resistance == TRUE | adagrasib_resistance == TRUE | sotorasib_resistance == TRUE | trametinib_resistance == TRUE | debcet_resistance == TRUE | pictilisib_resistance == TRUE | olaparib_resistance == TRUE | niraparib_resistance == TRUE) %>%
  distinct()
  
test2 <- COSMIC2 %>%
  group_by(Gene, Amino_Acid_Position) %>%
  summarise(protein = paste(unique(protein), collapse = ";"),
            MUTATION_ZYGOSITY = paste(unique(MUTATION_ZYGOSITY), collapse = ";"),
            PUBMED_PMID = paste(unique(PUBMED_PMID), collapse = ";"),
            THERAPY_RELATIONSHIP = paste(unique(THERAPY_RELATIONSHIP), collapse = ";"),
            DRUG_RESPONSE = paste(unique(DRUG_RESPONSE), collapse = ";"),
             COSMIC_SAMPLE_ID = paste(unique(COSMIC_SAMPLE_ID), collapse = ";")) %>%
  filter(Gene %in% targets) %>%
  filter(is.na(Amino_Acid_Position) == FALSE) %>%
  filter(Amino_Acid_Position != "") %>%
  distinct()

test1
test2

left_join(test2, test1) %>%
  distinct() %>% 
  write.csv("test.csv")
```

```{r}
# dot plots _ proliferation - HT-29 COSMIC graph

targets <- c("EGFR", "KRAS", "MYC", "PARP1", "PARP2", "MAP2K1", "MAP2K2", "BCL2", "BRAF", "AKT1", "PIK3CA")

pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

plot1 <- ggplot(subset(Supplementary_Table_revisions, Gene == "AKT1"), aes(x=Amino_Acid_Position_simple, y= L2FC_HT29_Control_plasmid_average_zscore, size = COSMIC_count_sum)) + 
geom_point(position = pos, shape = 1, alpha = 1) +
geom_segment(aes(xend = Amino_Acid_Position_simple, yend = 0), position = pos, color = "grey50", linewidth = 0.3) +
labs(title = "", x="AKT1 amino acid position", y = "HT-29 control vs plasmid (z-score)", size = "COSMIC count") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_HT29_Control_plasmid_average_zscore > 2 & (is.na(Amino_Acid_Position) == FALSE), paste(Amino_Acid_Position),  "")), position = pos, size = 4) +
scale_size(range = c(3, 10)) +
theme_classic()
plot1
ggsave("Fig1e.eps", width = 6, height = 4.1, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

plot2 <- ggplot(subset(Supplementary_Table_revisions, Gene == "PIK3CA"), aes(x=Amino_Acid_Position_simple, y= L2FC_HT29_Control_plasmid_average_zscore, size = COSMIC_count_sum)) + 
geom_point(position = pos, shape = 1, alpha = 1) +
geom_segment(aes(xend = Amino_Acid_Position_simple, yend = 0), position = pos, color = "grey50", linewidth = 0.3) +
labs(title = "", x="PIK3CA amino acid position", y = "HT-29 control vs plasmid (z-score)", size = "COSMIC count") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_HT29_Control_plasmid_average_zscore > 2 & (is.na(Amino_Acid_Position) == FALSE), paste(Amino_Acid_Position),  "")), position = pos, size = 4) +
scale_size(range = c(3, 10)) +
theme_classic()
plot2
ggsave("Fig1f.eps", width = 6, height = 4.1, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")
```

```{r}
# dot plots _ correlation between z-score and COSMIC count

targets <- c("EGFR", "KRAS", "MYC", "PARP1", "PARP2", "MAP2K1", "MAP2K2", "BCL2", "BRAF", "AKT1", "PIK3CA")

pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

plot1 <- ggplot(subset(Supplementary_Table_revisions, Gene == "AKT1"), aes(x=COSMIC_count_sum, y= L2FC_HT29_Control_plasmid_average_zscore)) + 
geom_point(position = pos, shape = 1, alpha = 1, size = 3) +
labs(title = "", y = "HT-29 control vs plasmid (z-score)", x = "COSMIC count") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_HT29_Control_plasmid_average_zscore > 2  & (is.na(Amino_Acid_Position) == FALSE) | COSMIC_count_sum > 200 & (is.na(Amino_Acid_Position) == FALSE), paste(Amino_Acid_Position),  "")), position = pos, size = 4, max.overlaps = Inf) +
theme_classic()
plot1

plot2 <- ggplot(subset(Supplementary_Table_revisions, Gene == "PIK3CA"), aes(x=COSMIC_count_sum, y= L2FC_HT29_Control_plasmid_average_zscore)) + 
geom_point(position = pos, shape = 1, alpha = 1, size = 3) +
labs(title = "", y = "HT-29 control vs plasmid (z-score)", x = "COSMIC count") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_HT29_Control_plasmid_average_zscore > 2  & (is.na(Amino_Acid_Position) == FALSE) | COSMIC_count_sum > 200 & (is.na(Amino_Acid_Position) == FALSE), paste(Amino_Acid_Position),  "")), position = pos, size = 4, max.overlaps = Inf) +
theme_classic()
plot2
```

```{r}
# number of hits that are controls - Sotor

#variant class
#Sotor control hits
Supplementary_Table_revisions %>%
  select(gene_alias, Amino_Acid_Position, variant_class_Sotor, L2FC_H23_Sotor_plasmid_average_zscore, FDR_MAGeCK_H23_Sotor, p_value_MAGeCK_H23_Sotor, L2FC_H23_Control_plasmid_average_zscore, FDR_MAGeCK_H23_control, p_value_MAGeCK_H23_control) %>%
  filter(variant_class_Sotor != "control") %>%
  filter(gene_alias %in% c("essential", "non-targeting", "non-essential", "intergenic")) %>%
  distinct()

#Sotor hits
Supplementary_Table_revisions %>%
  select(gene_alias, Amino_Acid_Position, variant_class_Sotor, L2FC_H23_Sotor_plasmid_average_zscore, FDR_MAGeCK_H23_Sotor, p_value_MAGeCK_H23_Sotor, L2FC_H23_Control_plasmid_average_zscore, FDR_MAGeCK_H23_control, p_value_MAGeCK_H23_control) %>%
  filter(variant_class_Sotor != "control") %>%
  filter(gene_alias %in% targets) %>%
  distinct()

#Sotor hits driver not significant
Supplementary_Table_revisions %>%
  select(gene_alias, Amino_Acid_Position, variant_class_Sotor, L2FC_H23_Sotor_plasmid_average_zscore, FDR_MAGeCK_H23_Sotor, p_value_MAGeCK_H23_Sotor, L2FC_H23_Control_plasmid_average_zscore, FDR_MAGeCK_H23_control, p_value_MAGeCK_H23_control) %>%
  filter(variant_class_Sotor %in% c("driver")) %>%
  filter(FDR_MAGeCK_H23_control > 0.1 | p_value_MAGeCK_H23_control > 0.05) %>%
  distinct()

#Sotor hits drug resistance not significant
Supplementary_Table_revisions %>%
  select(gene_alias, Amino_Acid_Position, variant_class_Sotor, L2FC_H23_Sotor_plasmid_average_zscore, FDR_MAGeCK_H23_Sotor, p_value_MAGeCK_H23_Sotor, L2FC_H23_Control_plasmid_average_zscore, FDR_MAGeCK_H23_control, p_value_MAGeCK_H23_control) %>%
  filter(variant_class_Sotor %in% c("drug addiction", "canonical drug resistance", "drug-sensitising")) %>%
  filter(FDR_MAGeCK_H23_Sotor > 0.1 | p_value_MAGeCK_H23_Sotor > 0.05) %>%
  distinct()

#drug resistance
#Sotor control hits
Supplementary_Table_revisions %>%
  select(gene_alias, Amino_Acid_Position, variant_class_Sotor, L2FC_H23_Sotor_plasmid_average_zscore, FDR_MAGeCK_H23_Sotor, p_value_MAGeCK_H23_Sotor, L2FC_H23_Control_plasmid_average_zscore, FDR_MAGeCK_H23_control, p_value_MAGeCK_H23_control, sotorasib_resistance) %>%
  filter(sotorasib_resistance == TRUE) %>%
  filter(gene_alias %in% c("essential", "non-targeting", "non-essential", "intergenic")) %>%
  distinct()

#Sotor hits
Supplementary_Table_revisions %>%
  select(gene_alias, Amino_Acid_Position, variant_class_Sotor, L2FC_H23_Sotor_plasmid_average_zscore, FDR_MAGeCK_H23_Sotor, p_value_MAGeCK_H23_Sotor, L2FC_H23_Control_plasmid_average_zscore, FDR_MAGeCK_H23_control, p_value_MAGeCK_H23_control, sotorasib_resistance) %>%
    filter(sotorasib_resistance == TRUE) %>%
  filter(gene_alias %in% targets) %>%
  distinct()

#Sotor hits not significant
Supplementary_Table_revisions %>%
  select(gene_alias, Amino_Acid_Position, variant_class_Sotor, L2FC_H23_Sotor_plasmid_average_zscore, FDR_MAGeCK_H23_Sotor, p_value_MAGeCK_H23_Sotor, L2FC_H23_Control_plasmid_average_zscore, FDR_MAGeCK_H23_control, p_value_MAGeCK_H23_control, sotorasib_resistance) %>%
  filter(sotorasib_resistance == TRUE) %>%
  filter(FDR_MAGeCK_H23_Sotor > 0.1 | p_value_MAGeCK_H23_Sotor > 0.05) %>%
  distinct()
```

```{r}
# number of hits that are controls - Adag

#Adag control hits
Supplementary_Table_revisions %>%
  select(gene_alias, Amino_Acid_Position, variant_class_Adag, L2FC_H23_Adag_plasmid_average_zscore, FDR_MAGeCK_H23_Adag, p_value_MAGeCK_H23_Adag, L2FC_H23_Control_plasmid_average_zscore, FDR_MAGeCK_H23_control, p_value_MAGeCK_H23_control) %>%
  filter(variant_class_Adag != "control") %>%
  filter(gene_alias %in% c("essential", "non-targeting", "non-essential", "intergenic")) %>%
  distinct()

#Adag hits
Supplementary_Table_revisions %>%
  select(gene_alias, Amino_Acid_Position, variant_class_Adag, L2FC_H23_Adag_plasmid_average_zscore, FDR_MAGeCK_H23_Adag, p_value_MAGeCK_H23_Adag, L2FC_H23_Control_plasmid_average_zscore, FDR_MAGeCK_H23_control, p_value_MAGeCK_H23_control) %>%
  filter(variant_class_Adag != "control") %>%
  filter(gene_alias %in% targets) %>%
  distinct()

#Adag hits
Supplementary_Table_revisions %>%
  select(gene_alias, Amino_Acid_Position, variant_class_Adag, L2FC_H23_Adag_plasmid_average_zscore, FDR_MAGeCK_H23_Adag, p_value_MAGeCK_H23_Adag, L2FC_H23_Control_plasmid_average_zscore, FDR_MAGeCK_H23_control, p_value_MAGeCK_H23_control) %>%
  filter(variant_class_Adag != "control") %>%
  filter(gene_alias %in% targets) %>%
  count(variant_class_Adag) %>%
  distinct()

#Adag hits driver not significant
Supplementary_Table_revisions %>%
  select(gene_alias, Amino_Acid_Position, variant_class_Adag, L2FC_H23_Adag_plasmid_average_zscore, FDR_MAGeCK_H23_Adag, p_value_MAGeCK_H23_Adag, L2FC_H23_Control_plasmid_average_zscore, FDR_MAGeCK_H23_control, p_value_MAGeCK_H23_control) %>%
  filter(variant_class_Adag %in% c("driver")) %>%
  filter(FDR_MAGeCK_H23_control > 0.1 | p_value_MAGeCK_H23_control > 0.05) %>%
  distinct()

#drug resistance
#Adag control hits
Supplementary_Table_revisions %>%
  select(gene_alias, Amino_Acid_Position, variant_class_Adag, L2FC_H23_Adag_plasmid_average_zscore, FDR_MAGeCK_H23_Adag, p_value_MAGeCK_H23_Adag, L2FC_H23_Control_plasmid_average_zscore, FDR_MAGeCK_H23_control, p_value_MAGeCK_H23_control, adagrasib_resistance) %>%
  filter(adagrasib_resistance == TRUE) %>%
  filter(gene_alias %in% c("essential", "non-targeting", "non-essential", "intergenic")) %>%
  distinct()

#Adag hits
Supplementary_Table_revisions %>%
  select(gene_alias, Amino_Acid_Position, variant_class_Adag, L2FC_H23_Adag_plasmid_average_zscore, FDR_MAGeCK_H23_Adag, p_value_MAGeCK_H23_Adag, L2FC_H23_Control_plasmid_average_zscore, FDR_MAGeCK_H23_control, p_value_MAGeCK_H23_control, adagrasib_resistance) %>%
    filter(adagrasib_resistance == TRUE) %>%
  filter(gene_alias %in% targets) %>%
  distinct()

#Adag hits not significant
Supplementary_Table_revisions %>%
  select(gene_alias, Amino_Acid_Position, variant_class_Adag, L2FC_H23_Adag_plasmid_average_zscore, FDR_MAGeCK_H23_Adag, p_value_MAGeCK_H23_Adag, L2FC_H23_Control_plasmid_average_zscore, FDR_MAGeCK_H23_control, p_value_MAGeCK_H23_control, adagrasib_resistance) %>%
  filter(adagrasib_resistance == TRUE) %>%
  filter(FDR_MAGeCK_H23_Adag > 0.1 | p_value_MAGeCK_H23_Adag > 0.05) %>%
  distinct()
```

```{r}
# number of hits that are controls - Tram

#Tram control hits
Supplementary_Table_revisions %>%
  select(gene_alias, Amino_Acid_Position, variant_class_Tram, L2FC_HT29_Tram_plasmid_average_zscore, FDR_MAGeCK_HT29_Tram, p_value_MAGeCK_HT29_Tram, L2FC_HT29_Control_plasmid_average_zscore, FDR_MAGeCK_HT29_control, p_value_MAGeCK_HT29_control) %>%
  filter(variant_class_Tram != "control") %>%
  filter(gene_alias %in% c("essential", "non-targeting", "non-essential", "intergenic")) %>%
  distinct()

#Tram hits
Supplementary_Table_revisions %>%
  select(gene_alias, Amino_Acid_Position, variant_class_Tram, L2FC_HT29_Tram_plasmid_average_zscore, FDR_MAGeCK_HT29_Tram, p_value_MAGeCK_HT29_Tram, L2FC_HT29_Control_plasmid_average_zscore, FDR_MAGeCK_HT29_control, p_value_MAGeCK_HT29_control) %>%
  filter(variant_class_Tram != "control") %>%
  filter(gene_alias %in% targets) %>%
  distinct()

#Tram hits class
Supplementary_Table_revisions %>%
  select(gene_alias, Amino_Acid_Position, variant_class_Tram, L2FC_HT29_Tram_plasmid_average_zscore, FDR_MAGeCK_HT29_Tram, p_value_MAGeCK_HT29_Tram, L2FC_HT29_Control_plasmid_average_zscore, FDR_MAGeCK_HT29_control, p_value_MAGeCK_HT29_control) %>%
  filter(variant_class_Tram != "control") %>%
  filter(gene_alias %in% targets) %>%
  count(variant_class_Tram) %>%
  distinct()

#Tram hits driver not significant
Supplementary_Table_revisions %>%
  select(gene_alias, Amino_Acid_Position, variant_class_Tram, L2FC_HT29_Tram_plasmid_average_zscore, FDR_MAGeCK_HT29_Tram, p_value_MAGeCK_HT29_Tram, L2FC_HT29_Control_plasmid_average_zscore, FDR_MAGeCK_HT29_control, p_value_MAGeCK_HT29_control) %>%
  filter(variant_class_Tram %in% c("driver")) %>%
  filter(FDR_MAGeCK_HT29_control > 0.1 | p_value_MAGeCK_HT29_control > 0.05) %>%
  distinct()

#Tram hits drug resistance not significant
Supplementary_Table_revisions %>%
  select(gene_alias, Amino_Acid_Position, variant_class_Tram, L2FC_HT29_Tram_plasmid_average_zscore, FDR_MAGeCK_HT29_Tram, p_value_MAGeCK_HT29_Tram, L2FC_HT29_Control_plasmid_average_zscore, FDR_MAGeCK_HT29_control, p_value_MAGeCK_HT29_control) %>%
  filter(variant_class_Tram %in% c("drug addiction", "canonical drug resistance", "drug-sensitising")) %>%
  filter(FDR_MAGeCK_HT29_Tram > 0.1 | p_value_MAGeCK_HT29_Tram > 0.05) %>%
  distinct()

#drug resistance
#Tram control hits
Supplementary_Table_revisions %>%
  select(gene_alias, Amino_Acid_Position, variant_class_Tram, L2FC_HT29_Tram_plasmid_average_zscore, FDR_MAGeCK_HT29_Tram, p_value_MAGeCK_HT29_Tram, L2FC_HT29_Control_plasmid_average_zscore, FDR_MAGeCK_HT29_control, p_value_MAGeCK_HT29_control, trametinib_resistance) %>%
  filter(trametinib_resistance == TRUE) %>%
  filter(gene_alias %in% c("essential", "non-targeting", "non-essential", "intergenic")) %>%
  distinct()

#Tram hits
Supplementary_Table_revisions %>%
  select(gene_alias, Amino_Acid_Position, variant_class_Tram, L2FC_HT29_Tram_plasmid_average_zscore, FDR_MAGeCK_HT29_Tram, p_value_MAGeCK_HT29_Tram, L2FC_HT29_Control_plasmid_average_zscore, FDR_MAGeCK_HT29_control, p_value_MAGeCK_HT29_control, trametinib_resistance) %>%
    filter(trametinib_resistance == TRUE) %>%
  filter(gene_alias %in% targets) %>%
  distinct()

#Tram hits not significant
Supplementary_Table_revisions %>%
  select(gene_alias, Amino_Acid_Position, variant_class_Tram, L2FC_HT29_Tram_plasmid_average_zscore, FDR_MAGeCK_HT29_Tram, p_value_MAGeCK_HT29_Tram, L2FC_HT29_Control_plasmid_average_zscore, FDR_MAGeCK_HT29_control, p_value_MAGeCK_HT29_control, trametinib_resistance) %>%
  filter(trametinib_resistance == TRUE) %>%
  filter(FDR_MAGeCK_HT29_Tram > 0.1 | p_value_MAGeCK_HT29_Tram > 0.05) %>%
  distinct()
```

```{r}
# number of hits that are controls - Pict

#Pict control hits
Supplementary_Table_revisions %>%
  select(gene_alias, Amino_Acid_Position, variant_class_Pict, L2FC_HT29_Pict_plasmid_average_zscore, FDR_MAGeCK_HT29_Pict, p_value_MAGeCK_HT29_Pict, L2FC_HT29_Control_plasmid_average_zscore, FDR_MAGeCK_HT29_control, p_value_MAGeCK_HT29_control) %>%
  filter(variant_class_Pict != "control") %>%
  filter(gene_alias %in% c("essential", "non-targeting", "non-essential", "intergenic")) %>%
  distinct()

#Pict hits
Supplementary_Table_revisions %>%
  select(gene_alias, Amino_Acid_Position, variant_class_Pict, L2FC_HT29_Pict_plasmid_average_zscore, FDR_MAGeCK_HT29_Pict, p_value_MAGeCK_HT29_Pict, L2FC_HT29_Control_plasmid_average_zscore, FDR_MAGeCK_HT29_control, p_value_MAGeCK_HT29_control) %>%
  filter(variant_class_Pict != "control") %>%
  filter(gene_alias %in% targets) %>%
  distinct()

#Pict hits
Supplementary_Table_revisions %>%
  select(gene_alias, Amino_Acid_Position, variant_class_Pict, L2FC_HT29_Pict_plasmid_average_zscore, FDR_MAGeCK_HT29_Pict, p_value_MAGeCK_HT29_Pict, L2FC_HT29_Control_plasmid_average_zscore, FDR_MAGeCK_HT29_control, p_value_MAGeCK_HT29_control) %>%
  filter(variant_class_Pict != "control") %>%
  filter(gene_alias %in% targets) %>%
  count(variant_class_Pict) %>%
  distinct()

#Pict hits driver not significant
Supplementary_Table_revisions %>%
  select(gene_alias, Amino_Acid_Position, variant_class_Pict, L2FC_HT29_Pict_plasmid_average_zscore, FDR_MAGeCK_HT29_Pict, p_value_MAGeCK_HT29_Pict, L2FC_HT29_Control_plasmid_average_zscore, FDR_MAGeCK_HT29_control, p_value_MAGeCK_HT29_control) %>%
  filter(variant_class_Pict %in% c("driver")) %>%
  filter(FDR_MAGeCK_HT29_control > 0.1 | p_value_MAGeCK_HT29_control > 0.05) %>%
  distinct()

#Pict hits drug resistance not significant
Supplementary_Table_revisions %>%
  select(gene_alias, Amino_Acid_Position, variant_class_Pict, L2FC_HT29_Pict_plasmid_average_zscore, FDR_MAGeCK_HT29_Pict, p_value_MAGeCK_HT29_Pict, L2FC_HT29_Control_plasmid_average_zscore, FDR_MAGeCK_HT29_control, p_value_MAGeCK_HT29_control) %>%
  filter(variant_class_Pict %in% c("drug addiction", "canonical drug resistance", "drug-sensitising")) %>%
  filter(FDR_MAGeCK_HT29_Pict > 0.1 | p_value_MAGeCK_HT29_Pict > 0.05) %>%
  distinct()
```

```{r}
# number of hits that are controls - Debcet

#DebCet control hits
Supplementary_Table_revisions %>%
  select(gene_alias, Amino_Acid_Position, variant_class_Debcet, L2FC_HT29_DebCet_plasmid_average_zscore, FDR_MAGeCK_HT29_DebCet, p_value_MAGeCK_HT29_DebCet, L2FC_HT29_Control_plasmid_average_zscore, FDR_MAGeCK_HT29_control, p_value_MAGeCK_HT29_control) %>%
  filter(variant_class_Debcet != "control") %>%
  filter(gene_alias %in% c("essential", "non-targeting", "non-essential", "intergenic")) %>%
  distinct()

#DebCet hits
Supplementary_Table_revisions %>%
  select(gene_alias, Amino_Acid_Position, variant_class_Debcet, L2FC_HT29_DebCet_plasmid_average_zscore, FDR_MAGeCK_HT29_DebCet, p_value_MAGeCK_HT29_DebCet, L2FC_HT29_Control_plasmid_average_zscore, FDR_MAGeCK_HT29_control, p_value_MAGeCK_HT29_control) %>%
  filter(variant_class_Debcet != "control") %>%
  filter(gene_alias %in% targets) %>%
  distinct()

#DebCet hits classes
Supplementary_Table_revisions %>%
  select(gene_alias, Amino_Acid_Position, variant_class_Debcet, L2FC_HT29_DebCet_plasmid_average_zscore, FDR_MAGeCK_HT29_DebCet, p_value_MAGeCK_HT29_DebCet, L2FC_HT29_Control_plasmid_average_zscore, FDR_MAGeCK_HT29_control, p_value_MAGeCK_HT29_control) %>%
  filter(variant_class_Debcet != "control") %>%
  filter(gene_alias %in% targets) %>%
  count(variant_class_Debcet) %>%
  distinct()

#DebCet hits driver not significant
Supplementary_Table_revisions %>%
  select(gene_alias, Amino_Acid_Position, variant_class_Debcet, L2FC_HT29_DebCet_plasmid_average_zscore, FDR_MAGeCK_HT29_DebCet, p_value_MAGeCK_HT29_DebCet, L2FC_HT29_Control_plasmid_average_zscore, FDR_MAGeCK_HT29_control, p_value_MAGeCK_HT29_control) %>%
  filter(variant_class_Debcet %in% c("driver")) %>%
  filter(FDR_MAGeCK_HT29_control > 0.1 | p_value_MAGeCK_HT29_control > 0.05) %>%
  distinct()

#DebCet hits drug resistance not significant
Supplementary_Table_revisions %>%
  select(gene_alias, Amino_Acid_Position, variant_class_Debcet, L2FC_HT29_DebCet_plasmid_average_zscore, FDR_MAGeCK_HT29_DebCet, p_value_MAGeCK_HT29_DebCet, L2FC_HT29_Control_plasmid_average_zscore, FDR_MAGeCK_HT29_control, p_value_MAGeCK_HT29_control) %>%
  filter(variant_class_Debcet %in% c("drug addiction", "canonical drug resistance", "drug-sensitising")) %>%
  filter(FDR_MAGeCK_HT29_DebCet > 0.1 | p_value_MAGeCK_HT29_DebCet > 0.05) %>%
  distinct()
```

```{r}
# number of hits that are controls - Gefit

#Gefit control hits
Supplementary_Table_revisions %>%
  select(gene_alias, Amino_Acid_Position, variant_class_Gefit, L2FC_PC9_Gefit_plasmid_average_zscore, FDR_MAGeCK_PC9_Gefit, p_value_MAGeCK_PC9_Gefit, L2FC_PC9_Control_plasmid_average_zscore, FDR_MAGeCK_PC9_control, p_value_MAGeCK_PC9_control) %>%
  filter(variant_class_Gefit != "control") %>%
  filter(gene_alias %in% c("essential", "non-targeting", "non-essential", "intergenic")) %>%
  distinct()

#Gefit hits
Supplementary_Table_revisions %>%
  select(gene_alias, Amino_Acid_Position, variant_class_Gefit, L2FC_PC9_Gefit_plasmid_average_zscore, FDR_MAGeCK_PC9_Gefit, p_value_MAGeCK_PC9_Gefit, L2FC_PC9_Control_plasmid_average_zscore, FDR_MAGeCK_PC9_control, p_value_MAGeCK_PC9_control) %>%
  filter(variant_class_Gefit != "control") %>%
  filter(gene_alias %in% targets) %>%
  distinct()

#Gefit hit classes
Supplementary_Table_revisions %>%
  select(gene_alias, Amino_Acid_Position, variant_class_Gefit, L2FC_PC9_Gefit_plasmid_average_zscore, FDR_MAGeCK_PC9_Gefit, p_value_MAGeCK_PC9_Gefit, L2FC_PC9_Control_plasmid_average_zscore, FDR_MAGeCK_PC9_control, p_value_MAGeCK_PC9_control) %>%
  filter(variant_class_Gefit != "control") %>%
  filter(gene_alias %in% targets) %>%
  count(variant_class_Gefit) %>%
  distinct()

#Gefit hits driver not significant
Supplementary_Table_revisions %>%
  select(gene_alias, Amino_Acid_Position, variant_class_Gefit, L2FC_PC9_Gefit_plasmid_average_zscore, FDR_MAGeCK_PC9_Gefit, p_value_MAGeCK_PC9_Gefit, L2FC_PC9_Control_plasmid_average_zscore, FDR_MAGeCK_PC9_control, p_value_MAGeCK_PC9_control) %>%
  filter(variant_class_Gefit %in% c("driver")) %>%
  filter(FDR_MAGeCK_PC9_control > 0.1 | p_value_MAGeCK_PC9_control > 0.05) %>%
  distinct()

#Gefit hits drug resistance not significant
Supplementary_Table_revisions %>%
  select(gene_alias, Amino_Acid_Position, variant_class_Gefit, L2FC_PC9_Gefit_plasmid_average_zscore, FDR_MAGeCK_PC9_Gefit, p_value_MAGeCK_PC9_Gefit, L2FC_PC9_Control_plasmid_average_zscore, FDR_MAGeCK_PC9_control, p_value_MAGeCK_PC9_control) %>%
  filter(variant_class_Gefit %in% c("drug addiction", "canonical drug resistance", "drug-sensitising")) %>%
  filter(FDR_MAGeCK_PC9_Gefit > 0.1 | p_value_MAGeCK_PC9_Gefit > 0.05) %>%
  distinct()

#Gefit hits drug resistance not significant
Supplementary_Table_revisions %>%
  select(gene_alias, Amino_Acid_Position, variant_class_Gefit, L2FC_PC9_Gefit_plasmid_average_zscore, FDR_MAGeCK_PC9_Gefit, p_value_MAGeCK_PC9_Gefit, L2FC_PC9_Control_plasmid_average_zscore, FDR_MAGeCK_PC9_control, p_value_MAGeCK_PC9_control, gefitinib_resistance) %>%
  filter(gefitinib_resistance == TRUE) %>%
  filter(FDR_MAGeCK_PC9_Gefit > 0.1 | p_value_MAGeCK_PC9_Gefit > 0.05) %>%
  distinct()
```

```{r}
# number of hits that are controls - Osim

#Osim control hits
Supplementary_Table_revisions %>%
  select(gene_alias, Amino_Acid_Position, variant_class_Osim, L2FC_PC9_Osim_plasmid_average_zscore, FDR_MAGeCK_PC9_Osim, p_value_MAGeCK_PC9_Osim, L2FC_PC9_Control_plasmid_average_zscore, FDR_MAGeCK_PC9_control, p_value_MAGeCK_PC9_control) %>%
  filter(variant_class_Osim != "control") %>%
  filter(gene_alias %in% c("essential", "non-targeting", "non-essential", "intergenic")) %>%
  distinct()

#Osim hits
Supplementary_Table_revisions %>%
  select(gene_alias, Amino_Acid_Position, variant_class_Osim, L2FC_PC9_Osim_plasmid_average_zscore, FDR_MAGeCK_PC9_Osim, p_value_MAGeCK_PC9_Osim, L2FC_PC9_Control_plasmid_average_zscore, FDR_MAGeCK_PC9_control, p_value_MAGeCK_PC9_control) %>%
  filter(variant_class_Osim != "control") %>%
  filter(gene_alias %in% targets) %>%
  distinct()

#Osim hit classes
Supplementary_Table_revisions %>%
  select(gene_alias, Amino_Acid_Position, variant_class_Osim, L2FC_PC9_Osim_plasmid_average_zscore, FDR_MAGeCK_PC9_Osim, p_value_MAGeCK_PC9_Osim, L2FC_PC9_Control_plasmid_average_zscore, FDR_MAGeCK_PC9_control, p_value_MAGeCK_PC9_control) %>%
  filter(variant_class_Osim != "control") %>%
  filter(gene_alias %in% targets) %>%
  count(variant_class_Osim) %>%
  distinct()

#Osim hits driver not significant
Supplementary_Table_revisions %>%
  select(gene_alias, Amino_Acid_Position, variant_class_Osim, L2FC_PC9_Osim_plasmid_average_zscore, FDR_MAGeCK_PC9_Osim, p_value_MAGeCK_PC9_Osim, L2FC_PC9_Control_plasmid_average_zscore, FDR_MAGeCK_PC9_control, p_value_MAGeCK_PC9_control) %>%
  filter(variant_class_Osim %in% c("driver")) %>%
  filter(FDR_MAGeCK_PC9_control > 0.1 | p_value_MAGeCK_PC9_control > 0.05) %>%
  distinct()

#Osim hits drug resistance not significant
Supplementary_Table_revisions %>%
  select(gene_alias, Amino_Acid_Position, variant_class_Osim, L2FC_PC9_Osim_plasmid_average_zscore, FDR_MAGeCK_PC9_Osim, p_value_MAGeCK_PC9_Osim, L2FC_PC9_Control_plasmid_average_zscore, FDR_MAGeCK_PC9_control, p_value_MAGeCK_PC9_control) %>%
  filter(variant_class_Osim %in% c("drug addiction", "canonical drug resistance", "drug-sensitising")) %>%
  filter(FDR_MAGeCK_PC9_Osim > 0.1 | p_value_MAGeCK_PC9_Osim > 0.05) %>%
  distinct()
```

```{r}
# number of hits that are controls - Olap

#Olap control hits
Supplementary_Table_revisions %>%
  select(gene_alias, Amino_Acid_Position, variant_class_Olap, L2FC_MHHES1_Olap_plasmid_average_zscore, FDR_MAGeCK_MHHES1_Olap, p_value_MAGeCK_MHHES1_Olap, L2FC_MHHES1_Control_plasmid_average_zscore, FDR_MAGeCK_MHHES1_control, p_value_MAGeCK_MHHES1_control) %>%
  filter(variant_class_Olap != "control") %>%
  filter(gene_alias %in% c("essential", "non-targeting", "non-essential", "intergenic")) %>%
  distinct()

#Olap hits
Supplementary_Table_revisions %>%
  select(gene_alias, Amino_Acid_Position, variant_class_Olap, L2FC_MHHES1_Olap_plasmid_average_zscore, FDR_MAGeCK_MHHES1_Olap, p_value_MAGeCK_MHHES1_Olap, L2FC_MHHES1_Control_plasmid_average_zscore, FDR_MAGeCK_MHHES1_control, p_value_MAGeCK_MHHES1_control) %>%
  filter(variant_class_Olap != "control") %>%
  filter(gene_alias %in% targets) %>%
  distinct()

#Olap hits classes
Supplementary_Table_revisions %>%
  select(gene_alias, Amino_Acid_Position, variant_class_Olap, L2FC_MHHES1_Olap_plasmid_average_zscore, FDR_MAGeCK_MHHES1_Olap, p_value_MAGeCK_MHHES1_Olap, L2FC_MHHES1_Control_plasmid_average_zscore, FDR_MAGeCK_MHHES1_control, p_value_MAGeCK_MHHES1_control) %>%
  filter(variant_class_Olap != "control") %>%
  filter(gene_alias %in% targets) %>%
  count(variant_class_Olap) %>%
  distinct()

#Olap hits driver not significant
Supplementary_Table_revisions %>%
  select(gene_alias, Amino_Acid_Position, variant_class_Olap, L2FC_MHHES1_Olap_plasmid_average_zscore, FDR_MAGeCK_MHHES1_Olap, p_value_MAGeCK_MHHES1_Olap, L2FC_MHHES1_Control_plasmid_average_zscore, FDR_MAGeCK_MHHES1_control, p_value_MAGeCK_MHHES1_control) %>%
  filter(variant_class_Olap %in% c("driver")) %>%
  filter(FDR_MAGeCK_MHHES1_control > 0.1 | p_value_MAGeCK_MHHES1_control > 0.05) %>%
  distinct()

#Olap hits drug resistance not significant
Supplementary_Table_revisions %>%
  select(gene_alias, Amino_Acid_Position, variant_class_Olap, L2FC_MHHES1_Olap_plasmid_average_zscore, FDR_MAGeCK_MHHES1_Olap, p_value_MAGeCK_MHHES1_Olap, L2FC_MHHES1_Control_plasmid_average_zscore, FDR_MAGeCK_MHHES1_control, p_value_MAGeCK_MHHES1_control) %>%
  filter(variant_class_Olap %in% c("drug addiction", "canonical drug resistance", "drug-sensitising")) %>%
  filter(FDR_MAGeCK_MHHES1_Olap > 0.1 | p_value_MAGeCK_MHHES1_Olap > 0.05) %>%
  distinct()

#Olap hits drug resistance not significant
Supplementary_Table_revisions %>%
  select(gene_alias, Amino_Acid_Position, variant_class_Olap, L2FC_MHHES1_Olap_plasmid_average_zscore, FDR_MAGeCK_MHHES1_Olap, p_value_MAGeCK_MHHES1_Olap, L2FC_MHHES1_Control_plasmid_average_zscore, FDR_MAGeCK_MHHES1_control, p_value_MAGeCK_MHHES1_control, olaparib_resistance) %>%
  filter(olaparib_resistance ==TRUE) %>%
  filter(FDR_MAGeCK_MHHES1_Olap > 0.1 | p_value_MAGeCK_MHHES1_Olap > 0.05) %>%
  distinct()
```

```{r}
# number of hits that are controls - Nirap

#Nirap control hits
Supplementary_Table_revisions %>%
  select(gene_alias, Amino_Acid_Position, variant_class_Nirap, L2FC_MHHES1_Nirap_plasmid_average_zscore, FDR_MAGeCK_MHHES1_Nirap, p_value_MAGeCK_MHHES1_Nirap, L2FC_MHHES1_Control_plasmid_average_zscore, FDR_MAGeCK_MHHES1_control, p_value_MAGeCK_MHHES1_control) %>%
  filter(variant_class_Nirap != "control") %>%
  filter(gene_alias %in% c("essential", "non-targeting", "non-essential", "intergenic")) %>%
  distinct()

#Nirap hits
Supplementary_Table_revisions %>%
  select(gene_alias, Amino_Acid_Position, variant_class_Nirap, L2FC_MHHES1_Nirap_plasmid_average_zscore, FDR_MAGeCK_MHHES1_Nirap, p_value_MAGeCK_MHHES1_Nirap, L2FC_MHHES1_Control_plasmid_average_zscore, FDR_MAGeCK_MHHES1_control, p_value_MAGeCK_MHHES1_control) %>%
  filter(variant_class_Nirap != "control") %>%
  filter(gene_alias %in% targets) %>%
  distinct()

#Nirap hits classes
Supplementary_Table_revisions %>%
  select(gene_alias, Amino_Acid_Position, variant_class_Nirap, L2FC_MHHES1_Nirap_plasmid_average_zscore, FDR_MAGeCK_MHHES1_Nirap, p_value_MAGeCK_MHHES1_Nirap, L2FC_MHHES1_Control_plasmid_average_zscore, FDR_MAGeCK_MHHES1_control, p_value_MAGeCK_MHHES1_control) %>%
  filter(variant_class_Nirap != "control") %>%
  filter(gene_alias %in% targets) %>%
  count(variant_class_Nirap) %>%
  distinct()

#Nirap hits driver not significant
Supplementary_Table_revisions %>%
  select(gene_alias, Amino_Acid_Position, variant_class_Nirap, L2FC_MHHES1_Nirap_plasmid_average_zscore, FDR_MAGeCK_MHHES1_Nirap, p_value_MAGeCK_MHHES1_Nirap, L2FC_MHHES1_Control_plasmid_average_zscore, FDR_MAGeCK_MHHES1_control, p_value_MAGeCK_MHHES1_control) %>%
  filter(variant_class_Nirap %in% c("driver")) %>%
  filter(FDR_MAGeCK_MHHES1_control > 0.1 | p_value_MAGeCK_MHHES1_control > 0.05) %>%
  distinct()

#Nirap hits drug resistance not significant
Supplementary_Table_revisions %>%
  select(gene_alias, Amino_Acid_Position, variant_class_Nirap, L2FC_MHHES1_Nirap_plasmid_average_zscore, FDR_MAGeCK_MHHES1_Nirap, p_value_MAGeCK_MHHES1_Nirap, L2FC_MHHES1_Control_plasmid_average_zscore, FDR_MAGeCK_MHHES1_control, p_value_MAGeCK_MHHES1_control) %>%
  filter(variant_class_Nirap %in% c("drug addiction", "canonical drug resistance", "drug-sensitising")) %>%
  filter(FDR_MAGeCK_MHHES1_Nirap > 0.1 | p_value_MAGeCK_MHHES1_Nirap > 0.05) %>%
  distinct()

#Nirap hits drug resistance not significant
Supplementary_Table_revisions %>%
  select(gene_alias, Amino_Acid_Position, variant_class_Nirap, L2FC_MHHES1_Nirap_plasmid_average_zscore, FDR_MAGeCK_MHHES1_Nirap, p_value_MAGeCK_MHHES1_Nirap, L2FC_MHHES1_Control_plasmid_average_zscore, FDR_MAGeCK_MHHES1_control, p_value_MAGeCK_MHHES1_control, niraparib_resistance) %>%
  filter(niraparib_resistance == TRUE) %>%
  filter(FDR_MAGeCK_MHHES1_Nirap > 0.1 | p_value_MAGeCK_MHHES1_Nirap > 0.05) %>%
  distinct()
```

```{r}
# number of gRNAs that were hits and had a score in the peturb seq screens

#energy_distances <- as_tibble(read.csv("energy_distances_100823.csv", stringsAsFactors = FALSE, header = TRUE))
#energy_distances <- energy_distances %>%
#  select(!c(variant_class))

energy_distances <- as_tibble(read.csv("energy_distances_080524.csv", stringsAsFactors = FALSE, header = TRUE))

Supplementary_Table_sc <-
  Supplementary_Table_revisions %>%
  mutate(ID = paste0(Gene, "_", substr(sgRNA_ID, 1, 6)))

Supplementary_Table_sc <-
  left_join(Supplementary_Table_sc, energy_distances)
```

```{r}
# join supplementary table to the genotyping data

mutations_amp_seq <- Discussion_table %>%
  select(sgRNA_ID, edit, effect, Gene, editor, vaf, HT29_inferred_ploidy, HT29_existing_SNVs) %>%
  group_by(sgRNA_ID) %>%
  summarise(edit = paste(unique(edit), collapse = ";"),
            editor = paste(unique(editor), collapse = ";"),
            effect = paste(unique(effect), collapse = ";"),
            HT29_inferred_ploidy = paste(unique(HT29_inferred_ploidy), collapse = ";"),
            HT29_existing_SNVs = paste(unique(HT29_existing_SNVs), collapse = ";"),
            vaf = (mean(vaf))) %>%
  distinct()

NG_revisions_data <- 
  left_join(Supplementary_Table_sc, mutations_amp_seq) %>% 
  rename(mean_vaf = vaf) %>%
  rename(amplicon_seq_edit = edit) %>%
  rename(energy_distance = ed) %>%
  rename(scRNAseq_ID = ID) %>%
  rename(consequence_amp_seq = effect) %>%
  rename(variant_class_DabCet = variant_class_Debcet) %>%
  distinct()

NG_revisions_data %>% 
  filter(is.na(mean_vaf)==FALSE) %>%
  distinct()

NG_revisions_data %>% 
  filter(is.na(mean_vaf)==FALSE) %>%
  filter(consequence_amp_seq != "silent") %>%
  distinct()
```

```{r}
# plot the gRNAs that score in the perturb-seq screens

plot1 <- ggplot(NG_revisions_data, aes(x= energy_distance, y= L2FC_HT29_DebCet_plasmid_average_zscore, colour = variant_class_DabCet)) +
geom_point(size=3, ) +
labs(title= "", x = "perturb-seq energy distance", y = "HT-29 dabrafenib + cetuximab vs plasmid (z-score)", colour = "variant class") +
scale_color_manual(values =c("royalblue", "grey70", "orange", "brown", "red1")) +
geom_text_repel(aes(label=ifelse(scRNAseq_ID %in% c("KRAS_33f23d", "KRAS_d32ed9"), paste(Gene, Amino_Acid_Position), "")), size = 4, max.overlaps = Inf) +
theme_classic(base_size = 12)
plot1
ggsave("FigS_perturb_hits2.eps", width = 6, height = 4, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")
```

```{r}
#how many hits have a significant impact in the scRNA seq?

plot1 <- ggplot(NG_revisions_data, aes(x= energy_distance, y= L2FC_HT29_DebCet_plasmid_average_zscore, colour = sign_transcriptional.impact)) +
geom_point(size=3, ) +
labs(title= "", x = "perturb-seq energy distance", y = "HT-29 dabrafenib + cetuximab vs plasmid (z-score)", significant = "variant class") +
scale_color_manual(values =c("royalblue", "orange", "grey75")) +
geom_text_repel(aes(label=ifelse(scRNAseq_ID %in% c("KRAS_33f23d", "KRAS_d32ed9"), paste(Gene, Amino_Acid_Position), "")), size = 4, max.overlaps = Inf) +
theme_classic(base_size = 12)
plot1

#RESISTANCE
#all drug resistance variants tested = 35
NG_revisions_data %>%
  filter(debcet_resistance == TRUE) %>%
  filter(is.na(energy_distance) == FALSE) %>%
  filter(is.na(sign_transcriptional.impact) == FALSE) %>%
  distinct()

#all drug resistance variants tested with significant transcriptional impact = 22
NG_revisions_data %>%
  filter(debcet_resistance == TRUE) %>%
  filter(is.na(energy_distance) == FALSE) %>%
  filter(sign_transcriptional.impact == TRUE) %>%
  distinct()

#all drug resistance variants tested with significant transcriptional impact = 13
NG_revisions_data %>%
  filter(debcet_resistance == TRUE) %>%
  filter(is.na(energy_distance) == FALSE) %>%
  filter(sign_transcriptional.impact == FALSE) %>%
  distinct()

###

# variant class!= control
#all drug resistance variants tested = 34
NG_revisions_data %>%
  filter(variant_class_DabCet != "control") %>%
  filter(is.na(energy_distance) == FALSE) %>%
  filter(is.na(sign_transcriptional.impact) == FALSE) %>%
  distinct()

#all drug resistance variants tested with significant transcriptional impact = 22
NG_revisions_data %>%
 filter(variant_class_DabCet != "control") %>%
  filter(is.na(energy_distance) == FALSE) %>%
  filter(sign_transcriptional.impact == TRUE) %>%
  distinct()

#all drug resistance variants tested with significant transcriptional impact = 12
NG_revisions_data %>%
  filter(variant_class_DabCet != "control") %>%
  filter(is.na(energy_distance) == FALSE) %>%
  filter(sign_transcriptional.impact == FALSE) %>%
  distinct()
```

```{r}
#validated gRNAs, other than the genotyped by amplicon seq.

# PARP1
#sgRNA_ID

#28cef786a635e5792cffe26aacec59497ad0ace8
#080e6b389754e5abdb84902b9cb6d53bd6093680
#b71d243cc9ac999e64e9dddecaa6d1eb8b47af21
#e35500339bc9159a2e52d80427fdea8027c8b6f6
#36a4f147a614eb55f29189e75a429ee8056482bf
#0b1031ddef17cf74156fbd5c2ecbdda4af9d7313
#98ccd1464d222bc9c7d58d4fc24bb9583a5127d7

#EGFR
#G_guide

#GCACCAGGCACTGGGAGGAA
#GCAAGCCACTCACCAGGCAC
#GAAGCCACTCACCAGGCACT

#Western/incucyte experimental validation
#GCCTTCAAAAATGAAGTAGG
#GTGGAGGGCTTGACATCTGC
#GCTTCTATGGTGCGTTCTAC
#GTGAATTAGCTGGAGAAAGG
#GCCCAGAAGCAGAAGGTGGG
#GCAGGTCAAGAGGAGTACAG
#GTAAATGTGATTTGCCTTCT
#GGCCCCGTAGAAGCCCACGA

#GCTGCAGCTCCCTTATGATC
#GTGCAGCTCCCTTATGATCT
#GATTTCAGGAAACAAAAATT
#GTGTACTCCTCTTGACCTGC

#add validated gRNAs to the data
NG_revisions_data_validation <- 
  NG_revisions_data %>%
  mutate(experimentally_validated_gRNA = ifelse(G_guide %in% c("GCACCAGGCACTGGGAGGAA", "GCAAGCCACTCACCAGGCAC", "GAAGCCACTCACCAGGCACT", "GCCTTCAAAAATGAAGTAGG", "GTGGAGGGCTTGACATCTGC", "GCTTCTATGGTGCGTTCTAC", "GTGAATTAGCTGGAGAAAGG", "GCCCAGAAGCAGAAGGTGGG", "GCAGGTCAAGAGGAGTACAG", "GTAAATGTGATTTGCCTTCT", "GGCCCCGTAGAAGCCCACGA", "GGCCCCGTAGAAGCCCACGA", "GCTGCAGCTCCCTTATGATC", "GTGCAGCTCCCTTATGATCT", "GATTTCAGGAAACAAAAATT", "GTGTACTCCTCTTGACCTGC", "GCAGGTCAAGAGGAGTACAG") | sgRNA_ID %in% c("28cef786a635e5792cffe26aacec59497ad0ace8", "080e6b389754e5abdb84902b9cb6d53bd6093680", "b71d243cc9ac999e64e9dddecaa6d1eb8b47af21", "e35500339bc9159a2e52d80427fdea8027c8b6f6", "36a4f147a614eb55f29189e75a429ee8056482bf", "0b1031ddef17cf74156fbd5c2ecbdda4af9d7313", "98ccd1464d222bc9c7d58d4fc24bb9583a5127d7"), TRUE, FALSE)) %>%
mutate(experimentally_validated_gRNA = ifelse(gefitinib_resistance == TRUE | osimertinib_resistance == TRUE | adagrasib_resistance == TRUE | sotorasib_resistance == TRUE | trametinib_resistance == TRUE | debcet_resistance == TRUE | pictilisib_resistance == TRUE | olaparib_resistance == TRUE | niraparib_resistance == TRUE | variant_class_Adag != "control" | variant_class_Osim != "control" | variant_class_DabCet!= "control" | variant_class_Pict!= "control" | variant_class_Sotor!= "control" | variant_class_Gefit!= "control" | variant_class_Olap!= "control" | variant_class_Nirap!= "control" | variant_class_Tram!= "control", experimentally_validated_gRNA, FALSE))

NG_revisions_data_validation %>%
  filter(sgRNA_ID == "28cef786a635e5792cffe26aacec59497ad0ace8")

# count 20 experimentally validated gRNAs
NG_revisions_data_validation %>%
  select(sgRNA_ID, experimentally_validated_gRNA, editor, Gene, Amino_Acid_Position) %>%
  filter(experimentally_validated_gRNA == "TRUE") %>%
  distinct()

# count 20 experimentally validated gRNAs, 11 have been genotyped by amplicon sequencing and 7 have Sanger seq. 
NG_revisions_data_validation %>%
    select(sgRNA_ID, experimentally_validated_gRNA, gene_alias, Amino_Acid_Position, amplicon_seq_edit, mean_vaf, editor) %>%
  filter(experimentally_validated_gRNA == "TRUE") %>%
  filter(is.na(mean_vaf)==FALSE) %>%
  distinct()

NG_revisions_data_validation %>%
    select(sgRNA_ID, experimentally_validated_gRNA, gene_alias, amplicon_seq_edit, mean_vaf, Amino_Acid_Position) %>%
  filter(experimentally_validated_gRNA == "TRUE") %>%
  filter(is.na(mean_vaf)==TRUE) %>%
  distinct()

NG_revisions_data_validation <- 
  NG_revisions_data_validation %>%
  select(!c(is_stop, is_synonymous))

NG_revisions_data_validation %>% 
  write.csv("Revised_Supplementary_Table_2.csv")
```

```{r}
# novel or not overlapping with COSMIC - 252 gRNA species (non-synonymous coding in targets)

NG_revisions_data_validation %>%
  filter(DRUG_RESPONSE == "" | DRUG_RESPONSE == "NA" | is.na(DRUG_RESPONSE)==TRUE) %>%
    filter(Gene %in% targets) %>%
    filter(is.na(Amino_Acid_Position) == FALSE) %>%
    filter(Amino_Acid_Position != "" & Amino_Acid_Position != "NA") %>%
 select(sgRNA_ID, editor, DRUG_RESPONSE, Gene, Amino_Acid_Position, Protein_Change, most_severe_consequence, gefitinib_resistance, osimertinib_resistance, adagrasib_resistance, sotorasib_resistance, trametinib_resistance, debcet_resistance, pictilisib_resistance , olaparib_resistance, niraparib_resistance, variant_class_Adag, variant_class_Osim, variant_class_DabCet, variant_class_Pict, variant_class_Sotor, variant_class_Gefit, variant_class_Olap, variant_class_Nirap, variant_class_Tram, experimentally_validated_gRNA, consequence_amp_seq, amplicon_seq_edit, mean_vaf) %>%
  filter(gefitinib_resistance == TRUE | osimertinib_resistance == TRUE | adagrasib_resistance == TRUE | sotorasib_resistance == TRUE | trametinib_resistance == TRUE | debcet_resistance == TRUE | pictilisib_resistance == TRUE | olaparib_resistance == TRUE | niraparib_resistance == TRUE) %>%
  filter(most_severe_consequence != "synonymous") %>%
   distinct()

NG_revisions_data_validation %>%
  filter(DRUG_RESPONSE == "" | DRUG_RESPONSE == "NA" | is.na(DRUG_RESPONSE)==TRUE) %>%
    filter(Gene %in% targets) %>%
    filter(is.na(Amino_Acid_Position) == FALSE) %>%
    filter(Amino_Acid_Position != "" & Amino_Acid_Position != "NA") %>%
 select(sgRNA_ID, editor, DRUG_RESPONSE, Gene, Amino_Acid_Position, Protein_Change, most_severe_consequence, gefitinib_resistance, osimertinib_resistance, adagrasib_resistance, sotorasib_resistance, trametinib_resistance, debcet_resistance, pictilisib_resistance , olaparib_resistance, niraparib_resistance, variant_class_Adag, variant_class_Osim, variant_class_DabCet, variant_class_Pict, variant_class_Sotor, variant_class_Gefit, variant_class_Olap, variant_class_Nirap, variant_class_Tram, experimentally_validated_gRNA, consequence_amp_seq, amplicon_seq_edit, mean_vaf) %>%
  filter(gefitinib_resistance == TRUE | osimertinib_resistance == TRUE | adagrasib_resistance == TRUE | sotorasib_resistance == TRUE | trametinib_resistance == TRUE | debcet_resistance == TRUE | pictilisib_resistance == TRUE | olaparib_resistance == TRUE | niraparib_resistance == TRUE) %>%
  filter(most_severe_consequence != "synonymous") %>%
  distinct() %>%
  write.csv("novel_coding_drug_resistance_mutations.csv")
```

```{r}
#calculate numbers of gRNAs that we have experimentally validated in the study 
#DabCet
NG_revisions_data_validation %>%
  filter(variant_class_DabCet == "drug addiction") %>%
  distinct()

NG_revisions_data_validation %>%
  filter(variant_class_DabCet == "drug addiction") %>%
  filter(experimentally_validated_gRNA == "TRUE") %>%
  distinct()

NG_revisions_data_validation %>%
  filter(variant_class_DabCet == "canonical drug resistance") %>%
  distinct()

NG_revisions_data_validation %>%
  filter(variant_class_DabCet == "canonical drug resistance") %>%
  filter(experimentally_validated_gRNA == "TRUE") %>%
  distinct()

###

#Tram
NG_revisions_data_validation %>%
  filter(variant_class_Tram == "drug addiction") %>%
  distinct()

NG_revisions_data_validation %>%
  filter(variant_class_Tram == "drug addiction") %>%
  filter(experimentally_validated_gRNA == "TRUE") %>%
  distinct()

NG_revisions_data_validation %>%
  filter(variant_class_Tram == "canonical drug resistance") %>%
  distinct()

NG_revisions_data_validation %>%
  filter(variant_class_Tram == "canonical drug resistance") %>%
  filter(experimentally_validated_gRNA == "TRUE") %>%
  distinct()

NG_revisions_data_validation %>%
  filter(variant_class_Osim == "drug-sensitising") %>%
  distinct()

NG_revisions_data_validation %>%
  filter(variant_class_Osim == "drug-sensitising") %>%
  filter(experimentally_validated_gRNA == "TRUE") %>%
  distinct()

NG_revisions_data_validation %>%
  filter(variant_class_Gefit == "drug-sensitising") %>%
  distinct()

NG_revisions_data_validation %>%
  filter(variant_class_Gefit == "drug-sensitising") %>%
  filter(experimentally_validated_gRNA == "TRUE") %>%
  distinct()
```

```{r}
#check MAGeCK FDR rates across screens
NG_revisions_data_validation %>%
  distinct()

NG_revisions_data_validation %>%
  filter(FDR_MAGeCK_HT29_control <0.01) %>%
  distinct()

NG_revisions_data_validation %>%
  filter(FDR_MAGeCK_HT29_Tram <0.01) %>%
  distinct()

NG_revisions_data_validation %>%
  filter(FDR_MAGeCK_HT29_DebCet <0.01) %>%
  distinct()

NG_revisions_data_validation %>%
  filter(FDR_MAGeCK_HT29_Pict <0.01) %>%
  distinct()

NG_revisions_data_validation %>%
  filter(FDR_MAGeCK_PC9_control <0.01) %>%
  distinct()

NG_revisions_data_validation %>%
  filter(FDR_MAGeCK_PC9_Gefit <0.01) %>%
  distinct()

NG_revisions_data_validation %>%
  filter(FDR_MAGeCK_PC9_Osim <0.01) %>%
  distinct()

NG_revisions_data_validation %>%
  filter(FDR_MAGeCK_H23_control <0.01) %>%
  distinct()

NG_revisions_data_validation %>%
  filter(FDR_MAGeCK_H23_Sotor <0.01) %>%
  distinct()

NG_revisions_data_validation %>%
  filter(FDR_MAGeCK_H23_Adag <0.01) %>%
  distinct()

NG_revisions_data_validation %>%
  filter(FDR_MAGeCK_MHHES1_control <0.01) %>%
  distinct()

NG_revisions_data_validation %>%
  filter(FDR_MAGeCK_MHHES1_Olap <0.01) %>%
  distinct()

NG_revisions_data_validation %>%
  filter(FDR_MAGeCK_MHHES1_Nirap <0.01) %>%
  distinct()
```

```{r}
# mave db deposit
NG_revisions_data_validation
# select only single changes with missense for predicted
# select amplicon seq changes (validated)
# change single letter code to three letter code
#export these as csv files

NG_revisions_data_validation_MAVE <-
  NG_revisions_data_validation %>%
  mutate(hgvs_pro_predicted = paste0("p.", Edited_AA, Amino_Acid_Position, New_AA))

NG_revisions_data_validation_MAVE

NG_revisions_data_validation_MAVE %>%
  filter(is.na(Amino_Acid_Position) == FALSE & Amino_Acid_Position != "") %>%
  filter(str_detect(hgvs_pro_predicted, ";") == FALSE) %>%
  filter(most_severe_consequence == "missense") %>%
  select(Gene, hgvs_pro_predicted, most_severe_consequence, L2FC_HT29_DebCet_plasmid_average_zscore, L2FC_HT29_Pict_plasmid_average_zscore, L2FC_HT29_Tram_plasmid_average_zscore, L2FC_PC9_Gefit_plasmid_average_zscore, L2FC_PC9_Osim_plasmid_average_zscore, L2FC_H23_Sotor_plasmid_average_zscore, L2FC_H23_Adag_plasmid_average_zscore, L2FC_MHHES1_Olap_plasmid_average_zscore, L2FC_MHHES1_Nirap_plasmid_average_zscore) %>%
  mutate(hgvs_pro_predicted = str_replace_all(string = hgvs_pro_predicted, 
                c("A" = "Ala",
                  "R" = "Arg",
                  "N" = "Asn",
                  "D" = "Asp",
                  "C" = "Cys",
                  "G" = "Gly",
                  "E" = "Glu",
                  "Q" = "Gln",
                  "H" = "His",
                  "I" = "Ile",
                  "L" = "Leu",
                  "K" = "Lys",
                  "M" = "Met",
                  "P" = "Pro",
                  "F" = "Phe",
                  "S" = "Ser",
                  "T" = "Thr",
                  "W" = "Trp",
                  "Y" = "Tyr",
                  "V" = "Val",
                  "\\*" = "Ter"))) %>%
  write.csv("predicted_MAVEdb.csv")

NG_revisions_data_validation_MAVE %>%
  filter(is.na(amplicon_seq_edit) == FALSE & amplicon_seq_edit != "") %>%
  mutate(amplicon_seq_edit = str_remove_all(amplicon_seq_edit, "EGFR")) %>%
  mutate(amplicon_seq_edit = str_remove_all(amplicon_seq_edit, "MAP2K1")) %>%
  mutate(amplicon_seq_edit = str_remove_all(amplicon_seq_edit, "MAP2K2")) %>%
  mutate(amplicon_seq_edit = str_remove_all(amplicon_seq_edit, "KRAS")) %>%
  mutate(amplicon_seq_edit = str_remove_all(amplicon_seq_edit, "AKT1")) %>%
  mutate(amplicon_seq_edit = str_remove_all(amplicon_seq_edit, "PIK3CA")) %>%
  mutate(amplicon_seq_edit = str_remove_all(amplicon_seq_edit, "BRAF")) %>%
  rename(hgvs_pro_validated = amplicon_seq_edit) %>%
  select(Gene, hgvs_pro_validated, consequence_amp_seq, L2FC_HT29_DebCet_plasmid_average_zscore, L2FC_HT29_Pict_plasmid_average_zscore, L2FC_HT29_Tram_plasmid_average_zscore, L2FC_PC9_Gefit_plasmid_average_zscore, L2FC_PC9_Osim_plasmid_average_zscore, L2FC_H23_Sotor_plasmid_average_zscore, L2FC_H23_Adag_plasmid_average_zscore, L2FC_MHHES1_Olap_plasmid_average_zscore, L2FC_MHHES1_Nirap_plasmid_average_zscore) %>%
  mutate(hgvs_pro_validated = str_replace_all(string = hgvs_pro_validated, 
                c("A" = "Ala",
                  "R" = "Arg",
                  "N" = "Asn",
                  "D" = "Asp",
                  "C" = "Cys",
                  "G" = "Gly",
                  "E" = "Glu",
                  "Q" = "Gln",
                  "H" = "His",
                  "I" = "Ile",
                  "L" = "Leu",
                  "K" = "Lys",
                  "M" = "Met",
                  "P" = "Pro",
                  "F" = "Phe",
                  "S" = "Ser",
                  "T" = "Thr",
                  "W" = "Trp",
                  "Y" = "Tyr",
                  "V" = "Val",
                  "\\*" = "Ter"))) %>%
  write.csv("validated_MAVEdb.csv")

```

```{r}
# make protein_change_simplified into three letter code for compatibility
amplicon_seq_data <- mutate(amplicon_seq_data, protein_change_simplified = ifelse(protein_change_simplified != c("intronic", "splice variant"), str_replace_all(string = protein_change_simplified, 
                c("A" = "Ala",
                  "R" = "Arg",
                  "N" = "Asn",
                  "D" = "Asp",
                  "C" = "Cys",
                  "G" = "Gly",
                  "E" = "Glu",
                  "Q" = "Gln",
                  "H" = "His",
                  "I" = "Ile",
                  "L" = "Leu",
                  "K" = "Lys",
                  "M" = "Met",
                  "P" = "Pro",
                  "F" = "Phe",
                  "S" = "Ser",
                  "T" = "Thr",
                  "W" = "Trp",
                  "Y" = "Tyr",
                  "V" = "Val",
                  "\\*" = "Ter")), protein_change_simplified))
```