---
title: "Res_1_screens_070422"
author: "Matt Coelho"
date: "07.04.2022"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
# load packages
library(tidyverse)
library(reshape2)
library(ggrepel)
library(stringr)
library(digest)
library(ggpointdensity)

#set max.overlaps of labelling graphs to infinity
options(geom_text_repel.max.overlaps = Inf)
```

```{r}
# read in gRNA library
guides <- as_tibble(read.csv("Oncogene_BE_ABE_NGN_TWIST_LIB_041121.csv", stringsAsFactors = FALSE, header = TRUE))
guides

# read in off-target NGN analysis from Joel GPU
off_targets <- as_tibble(read.csv("Oncogene_BE_ABE_NGN_TWIST_LIB_092021.offtargets.csv", stringsAsFactors = FALSE, header = TRUE))
off_targets %>% filter(is_dupe == TRUE)

off_targets <- off_targets %>% select(ID, guide, off_target_metric_NG, off_target_summary_NG, polyT, Direction, is_dupe)
  
# join the off targets to the tibble
lib <- left_join(guides, off_targets)

# count guides
guides
lib
lib %>% filter(is_dupe == TRUE)

#  unique guides
lib <- distinct(lib)
lib <- lib %>%
  select(!c(is_dupe, strand, off_target_summary))

# infer strand from gRNA "Direction"
lib <- mutate(lib, strand = ifelse(Direction == "right", "+", ifelse(Direction == "left", "-", "NA")))
lib

# off_target_reports
lib <- separate(lib, off_target_summary_NG, remove = FALSE, into = c("perfect_matches", "one_mismatch", "two_mismatches", "three_mismatches", "four_mismatches", "extra_"), sep = c(","), extra = "merge")
lib <- lib %>% 
  select(!"extra_")

lib <- mutate(lib, perfect_matches = str_remove_all(perfect_matches, "\\{"))
lib <- mutate(lib, four_mismatches = str_remove_all(four_mismatches, "\\}"))

lib <- mutate(lib, perfect_matches = as.numeric(substr(perfect_matches, 4, 50)))
lib <- mutate(lib, one_mismatch = as.numeric(substr(one_mismatch, 4, 50)))
lib <- mutate(lib, two_mismatches = as.numeric(substr(two_mismatches, 4, 50)))
lib <- mutate(lib, three_mismatches = as.numeric(substr(three_mismatches, 4, 50)))
lib <- mutate(lib, four_mismatches = as.numeric(substr(four_mismatches, 4, 50)))

# rename ID to sgRNA_ID
lib <- rename(lib, sgRNA_ID = ID)
lib <- rename(lib, start = start_BEstimate)
lib <- rename(lib, end = end_BEstimate)
lib <- rename(lib, chr = chr_BEstimate)

lib %>% 
  arrange(desc(perfect_matches)) %>% select(Gene, off_target_summary_NG)

lib %>% filter(perfect_matches >1 | is.na(perfect_matches)) 
lib %>% filter(perfect_matches >1) %>% count(Gene)
lib %>% count(Gene)
lib %>% filter(perfect_matches >1) %>% select(Gene, off_target_summary_NG)
lib
```

```{r}
# read in raw gRNA reads counts
raw_reads <- as_tibble(read.csv("raw_read_counts_120422.csv", stringsAsFactors = FALSE, header = TRUE))
raw_reads <- mutate(raw_reads, sgRNA_ID = as.character(sgRNA_ID))

# compute RPM normalised values
# define functions
RPM <- function(x) ((x/sum(x) *1000000) + 1)
average <- function(x,y) ((x + y) /2)
L2FC <- function(x,y) (log2(x/y))
```

```{r}
# compute average reads
# HT-29 average
#BE3.9max NGN
raw_reads <- mutate(raw_reads, HT29_CBE_T0_average_reads = average(HT29_CBE_T0_exp1_reads, HT29_CBE_T0_exp2_reads))
raw_reads <- mutate(raw_reads, HT29_CBE_Control_average_reads = average(HT29_CBE_Control_exp1_reads, HT29_CBE_Control_exp2_reads))
raw_reads <- mutate(raw_reads, HT29_CBE_Tram_average_reads = average(HT29_CBE_Tram_exp1_reads, HT29_CBE_Tram_exp2_reads))
raw_reads <- mutate(raw_reads, HT29_CBE_Pict_average_reads = average(HT29_CBE_Pict_exp1_reads, HT29_CBE_Pict_exp2_reads))
raw_reads <- mutate(raw_reads, HT29_CBE_DebCet_average_reads = average(HT29_CBE_DebCet_exp1_reads, HT29_CBE_DebCet_exp2_reads))

#ABE8e NGN
raw_reads <- mutate(raw_reads, HT29_ABE_T0_average_reads = average(HT29_ABE_T0_exp1_reads, HT29_ABE_T0_exp2_reads))
raw_reads <- mutate(raw_reads, HT29_ABE_Control_average_reads = average(HT29_ABE_Control_exp1_reads, HT29_ABE_Control_exp2_reads))
raw_reads <- mutate(raw_reads, HT29_ABE_Tram_average_reads = average(HT29_ABE_Tram_exp1_reads, HT29_ABE_Tram_exp2_reads))
raw_reads <- mutate(raw_reads, HT29_ABE_Pict_average_reads = average(HT29_ABE_Pict_exp1_reads, HT29_ABE_Pict_exp2_reads))
raw_reads <- mutate(raw_reads, HT29_ABE_DebCet_average_reads = average(HT29_ABE_DebCet_exp1_reads, HT29_ABE_DebCet_exp2_reads))

# H23 average
#BE3.9max NGN
raw_reads <- mutate(raw_reads, H23_CBE_T0_average_reads = average(H23_CBE_T0_exp1_reads, H23_CBE_T0_exp2_reads))
raw_reads <- mutate(raw_reads, H23_CBE_Control_average_reads = average(H23_CBE_Control_exp1_reads, H23_CBE_Control_exp2_reads))
raw_reads <- mutate(raw_reads, H23_CBE_Sotor_average_reads = average(H23_CBE_Sotor_exp1_reads, H23_CBE_Sotor_exp2_reads))
raw_reads <- mutate(raw_reads, H23_CBE_Adag_average_reads = average(H23_CBE_Adag_exp1_reads, H23_CBE_Adag_exp2_reads))

#ABE8e NGN
raw_reads <- mutate(raw_reads, H23_ABE_T0_average_reads = average(H23_ABE_T0_exp1_reads, H23_ABE_T0_exp2_reads))
raw_reads <- mutate(raw_reads, H23_ABE_Control_average_reads = average(H23_ABE_Control_exp1_reads, H23_ABE_Control_exp2_reads))
raw_reads <- mutate(raw_reads, H23_ABE_Sotor_average_reads = average(H23_ABE_Sotor_exp1_reads, H23_ABE_Sotor_exp2_reads))
raw_reads <- mutate(raw_reads, H23_ABE_Adag_average_reads = average(H23_ABE_Adag_exp1_reads, H23_ABE_Adag_exp2_reads))


# PC9 average
#BE3.9max NGN
raw_reads <- mutate(raw_reads, PC9_CBE_T0_average_reads = average(PC9_CBE_T0_exp1_reads, PC9_CBE_T0_exp2_reads))
raw_reads <- mutate(raw_reads, PC9_CBE_Control_average_reads = average(PC9_CBE_Control_exp1_reads, PC9_CBE_Control_exp2_reads))
raw_reads <- mutate(raw_reads, PC9_CBE_Gefit_average_reads = average(PC9_CBE_Gefit_exp1_reads, PC9_CBE_Gefit_exp2_reads))
raw_reads <- mutate(raw_reads, PC9_CBE_Osim_average_reads = average(PC9_CBE_Osim_exp1_reads, PC9_CBE_Osim_exp2_reads))

#ABE8e NGN
raw_reads <- mutate(raw_reads, PC9_ABE_T0_average_reads = average(PC9_ABE_T0_exp1_reads, PC9_ABE_T0_exp2_reads))
raw_reads <- mutate(raw_reads, PC9_ABE_Control_average_reads = average(PC9_ABE_Control_exp1_reads, PC9_ABE_Control_exp2_reads))
raw_reads <- mutate(raw_reads, PC9_ABE_Gefit_average_reads = average(PC9_ABE_Gefit_exp1_reads, PC9_ABE_Gefit_exp2_reads))
raw_reads <- mutate(raw_reads, PC9_ABE_Osim_average_reads = average(PC9_ABE_Osim_exp1_reads, PC9_ABE_Osim_exp2_reads))

# MHH-ES-1 average

#ABE8e NGN
raw_reads <- mutate(raw_reads, MHHES1_ABE_T0_average_reads = average(MHHES1_ABE_T0_exp1_reads, MHHES1_ABE_T0_exp2_reads))
raw_reads <- mutate(raw_reads, MHHES1_ABE_Control_average_reads = average(MHHES1_ABE_Control_exp1_reads, MHHES1_ABE_Control_exp2_reads))
raw_reads <- mutate(raw_reads, MHHES1_ABE_Olap_average_reads = average(MHHES1_ABE_Olap_exp1_reads, MHHES1_ABE_Olap_exp2_reads))
raw_reads <- mutate(raw_reads, MHHES1_ABE_Nirap_average_reads = average(MHHES1_ABE_Nirap_exp1_reads, MHHES1_ABE_Nirap_exp2_reads))
```

```{r}
#compute RPM values
raw_reads <- mutate_at(raw_reads, vars(matches("reads")), list(RPM = RPM))
raw_reads
```

```{r}
# compute L2FC from normalised gRNA counts - single experiments
# HT-29 exp 1
#BE3.9max NGN
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_Control_plasmid_1 = L2FC(HT29_CBE_Control_exp1_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_Control_T0_1 = L2FC(HT29_CBE_Control_exp1_reads_RPM, HT29_CBE_T0_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_Tram_Control_1 = L2FC(HT29_CBE_Tram_exp1_reads_RPM, HT29_CBE_Control_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_Pict_Control_1 = L2FC(HT29_CBE_Pict_exp1_reads_RPM, HT29_CBE_Control_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_DebCet_Control_1 = L2FC(HT29_CBE_DebCet_exp1_reads_RPM, HT29_CBE_Control_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_Tram_plasmid_1 = L2FC(HT29_CBE_Tram_exp1_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_Pict_plasmid_1 = L2FC(HT29_CBE_Pict_exp1_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_DebCet_plasmid_1 = L2FC(HT29_CBE_DebCet_exp1_reads_RPM, plasmid_reads_RPM))

#ABE8e NGN
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_Control_plasmid_1 = L2FC(HT29_ABE_Control_exp1_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_Control_T0_1 = L2FC(HT29_ABE_Control_exp1_reads_RPM, HT29_ABE_T0_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_Tram_Control_1 = L2FC(HT29_ABE_Tram_exp1_reads_RPM, HT29_ABE_Control_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_Pict_Control_1 = L2FC(HT29_ABE_Pict_exp1_reads_RPM, HT29_ABE_Control_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_DebCet_Control_1 = L2FC(HT29_ABE_DebCet_exp1_reads_RPM, HT29_ABE_Control_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_Tram_plasmid_1 = L2FC(HT29_ABE_Tram_exp1_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_Pict_plasmid_1 = L2FC(HT29_ABE_Pict_exp1_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_DebCet_plasmid_1 = L2FC(HT29_ABE_DebCet_exp1_reads_RPM, plasmid_reads_RPM))

# HT-29 exp 2
#BE3.9max NGN
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_Control_plasmid_2 = L2FC(HT29_CBE_Control_exp2_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_Control_T0_2 = L2FC(HT29_CBE_Control_exp2_reads_RPM, HT29_CBE_T0_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_Tram_Control_2 = L2FC(HT29_CBE_Tram_exp2_reads_RPM, HT29_CBE_Control_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_Pict_Control_2 = L2FC(HT29_CBE_Pict_exp2_reads_RPM, HT29_CBE_Control_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_DebCet_Control_2 = L2FC(HT29_CBE_DebCet_exp2_reads_RPM, HT29_CBE_Control_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_Tram_plasmid_2 = L2FC(HT29_CBE_Tram_exp2_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_Pict_plasmid_2 = L2FC(HT29_CBE_Pict_exp2_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_DebCet_plasmid_2 = L2FC(HT29_CBE_DebCet_exp2_reads_RPM, plasmid_reads_RPM))

#ABE8e NGN
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_Control_plasmid_2 = L2FC(HT29_ABE_Control_exp2_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_Control_T0_2 = L2FC(HT29_ABE_Control_exp2_reads_RPM, HT29_ABE_T0_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_Tram_Control_2 = L2FC(HT29_ABE_Tram_exp2_reads_RPM, HT29_ABE_Control_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_Pict_Control_2 = L2FC(HT29_ABE_Pict_exp2_reads_RPM, HT29_ABE_Control_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_DebCet_Control_2 = L2FC(HT29_ABE_DebCet_exp2_reads_RPM, HT29_ABE_Control_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_Tram_plasmid_2 = L2FC(HT29_ABE_Tram_exp2_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_Pict_plasmid_2 = L2FC(HT29_ABE_Pict_exp2_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_DebCet_plasmid_2 = L2FC(HT29_ABE_DebCet_exp2_reads_RPM, plasmid_reads_RPM))


# H23 exp 1
#BE3.9max NGN
raw_reads <- mutate(raw_reads, L2FC_H23_CBE_Control_plasmid_1 = L2FC(H23_CBE_Control_exp1_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_CBE_Control_T0_1 = L2FC(H23_CBE_Control_exp1_reads_RPM, H23_CBE_T0_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_CBE_Sotor_Control_1 = L2FC(H23_CBE_Sotor_exp1_reads_RPM, H23_CBE_Control_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_CBE_Adag_Control_1 = L2FC(H23_CBE_Adag_exp1_reads_RPM, H23_CBE_Control_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_CBE_Sotor_plasmid_1 = L2FC(H23_CBE_Sotor_exp1_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_CBE_Adag_plasmid_1 = L2FC(H23_CBE_Adag_exp1_reads_RPM, plasmid_reads_RPM))

#ABE8e NGN
raw_reads <- mutate(raw_reads, L2FC_H23_ABE_Control_plasmid_1 = L2FC(H23_ABE_Control_exp1_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_ABE_Control_T0_1 = L2FC(H23_ABE_Control_exp1_reads_RPM, H23_ABE_T0_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_ABE_Sotor_Control_1 = L2FC(H23_ABE_Sotor_exp1_reads_RPM, H23_ABE_Control_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_ABE_Adag_Control_1 = L2FC(H23_ABE_Adag_exp1_reads_RPM, H23_ABE_Control_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_ABE_Sotor_plasmid_1 = L2FC(H23_ABE_Sotor_exp1_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_ABE_Adag_plasmid_1 = L2FC(H23_ABE_Adag_exp1_reads_RPM, plasmid_reads_RPM))

# H23 exp 2
#BE3.9max NGN
raw_reads <- mutate(raw_reads, L2FC_H23_CBE_Control_plasmid_2 = L2FC(H23_CBE_Control_exp2_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_CBE_Control_T0_2 = L2FC(H23_CBE_Control_exp2_reads_RPM, H23_CBE_T0_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_CBE_Sotor_Control_2 = L2FC(H23_CBE_Sotor_exp2_reads_RPM, H23_CBE_Control_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_CBE_Adag_Control_2 = L2FC(H23_CBE_Adag_exp2_reads_RPM, H23_CBE_Control_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_CBE_Sotor_plasmid_2 = L2FC(H23_CBE_Sotor_exp2_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_CBE_Adag_plasmid_2 = L2FC(H23_CBE_Adag_exp2_reads_RPM, plasmid_reads_RPM))

#ABE8e NGN
raw_reads <- mutate(raw_reads, L2FC_H23_ABE_Control_plasmid_2 = L2FC(H23_ABE_Control_exp2_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_ABE_Control_T0_2 = L2FC(H23_ABE_Control_exp2_reads_RPM, H23_ABE_T0_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_ABE_Sotor_Control_2 = L2FC(H23_ABE_Sotor_exp2_reads_RPM, H23_ABE_Control_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_ABE_Adag_Control_2 = L2FC(H23_ABE_Adag_exp2_reads_RPM, H23_ABE_Control_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_ABE_Sotor_plasmid_2 = L2FC(H23_ABE_Sotor_exp2_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_ABE_Adag_plasmid_2 = L2FC(H23_ABE_Adag_exp2_reads_RPM, plasmid_reads_RPM))

# PC9 exp 1
#BE3.9max NGN
raw_reads <- mutate(raw_reads, L2FC_PC9_CBE_Control_plasmid_1 = L2FC(PC9_CBE_Control_exp1_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_CBE_Control_T0_1 = L2FC(PC9_CBE_Control_exp1_reads_RPM, PC9_CBE_T0_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_CBE_Gefit_Control_1 = L2FC(PC9_CBE_Gefit_exp1_reads_RPM, PC9_CBE_Control_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_CBE_Osim_Control_1 = L2FC(PC9_CBE_Osim_exp1_reads_RPM, PC9_CBE_Control_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_CBE_Gefit_plasmid_1 = L2FC(PC9_CBE_Gefit_exp1_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_CBE_Osim_plasmid_1 = L2FC(PC9_CBE_Osim_exp1_reads_RPM, plasmid_reads_RPM))

#ABE8e NGN
raw_reads <- mutate(raw_reads, L2FC_PC9_ABE_Control_plasmid_1 = L2FC(PC9_ABE_Control_exp1_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_ABE_Control_T0_1 = L2FC(PC9_ABE_Control_exp1_reads_RPM, PC9_ABE_T0_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_ABE_Gefit_Control_1 = L2FC(PC9_ABE_Gefit_exp1_reads_RPM, PC9_ABE_Control_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_ABE_Osim_Control_1 = L2FC(PC9_ABE_Osim_exp1_reads_RPM, PC9_ABE_Control_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_ABE_Gefit_plasmid_1 = L2FC(PC9_ABE_Gefit_exp1_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_ABE_Osim_plasmid_1 = L2FC(PC9_ABE_Osim_exp1_reads_RPM, plasmid_reads_RPM))

# PC9 exp 2
#BE3.9max NGN
raw_reads <- mutate(raw_reads, L2FC_PC9_CBE_Control_plasmid_2 = L2FC(PC9_CBE_Control_exp2_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_CBE_Control_T0_2 = L2FC(PC9_CBE_Control_exp2_reads_RPM, PC9_CBE_T0_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_CBE_Gefit_Control_2 = L2FC(PC9_CBE_Gefit_exp2_reads_RPM, PC9_CBE_Control_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_CBE_Osim_Control_2 = L2FC(PC9_CBE_Osim_exp2_reads_RPM, PC9_CBE_Control_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_CBE_Gefit_plasmid_2 = L2FC(PC9_CBE_Gefit_exp2_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_CBE_Osim_plasmid_2 = L2FC(PC9_CBE_Osim_exp2_reads_RPM, plasmid_reads_RPM))

#ABE8e NGN
raw_reads <- mutate(raw_reads, L2FC_PC9_ABE_Control_plasmid_2 = L2FC(PC9_ABE_Control_exp2_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_ABE_Control_T0_2 = L2FC(PC9_ABE_Control_exp2_reads_RPM, PC9_ABE_T0_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_ABE_Gefit_Control_2 = L2FC(PC9_ABE_Gefit_exp2_reads_RPM, PC9_ABE_Control_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_ABE_Osim_Control_2 = L2FC(PC9_ABE_Osim_exp2_reads_RPM, PC9_ABE_Control_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_ABE_Gefit_plasmid_2 = L2FC(PC9_ABE_Gefit_exp2_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_ABE_Osim_plasmid_2 = L2FC(PC9_ABE_Osim_exp2_reads_RPM, plasmid_reads_RPM))

# PC9 exp 3
#BE3.9max NGN
raw_reads <- mutate(raw_reads, L2FC_PC9_CBE_Gefit_plasmid_3 = L2FC(PC9_CBE_Gefit_exp3_reads_RPM, plasmid_reads_RPM))

# MHH-ES-1 exp 1
#ABE8e NGN
raw_reads <- mutate(raw_reads, L2FC_MHHES1_ABE_Control_plasmid_1 = L2FC(MHHES1_ABE_Control_exp1_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_MHHES1_ABE_Control_T0_1 = L2FC(MHHES1_ABE_Control_exp1_reads_RPM, MHHES1_ABE_T0_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_MHHES1_ABE_Olap_Control_1 = L2FC(MHHES1_ABE_Olap_exp1_reads_RPM, MHHES1_ABE_Control_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_MHHES1_ABE_Nirap_Control_1 = L2FC(MHHES1_ABE_Nirap_exp1_reads_RPM, MHHES1_ABE_Control_exp1_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_MHHES1_ABE_Olap_plasmid_1 = L2FC(MHHES1_ABE_Olap_exp1_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_MHHES1_ABE_Nirap_plasmid_1 = L2FC(MHHES1_ABE_Nirap_exp1_reads_RPM, plasmid_reads_RPM))

# MHH-ES-1 exp 2
#ABE8e NGN
raw_reads <- mutate(raw_reads, L2FC_MHHES1_ABE_Control_plasmid_2 = L2FC(MHHES1_ABE_Control_exp2_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_MHHES1_ABE_Control_T0_2 = L2FC(MHHES1_ABE_Control_exp2_reads_RPM, MHHES1_ABE_T0_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_MHHES1_ABE_Olap_Control_2 = L2FC(MHHES1_ABE_Olap_exp2_reads_RPM, MHHES1_ABE_Control_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_MHHES1_ABE_Nirap_Control_2 = L2FC(MHHES1_ABE_Nirap_exp2_reads_RPM, MHHES1_ABE_Control_exp2_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_MHHES1_ABE_Olap_plasmid_2 = L2FC(MHHES1_ABE_Olap_exp2_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_MHHES1_ABE_Nirap_plasmid_2 = L2FC(MHHES1_ABE_Nirap_exp2_reads_RPM, plasmid_reads_RPM))
```

```{r}
#compute L2FC from average of 2 exps from RPM values
# HT-29 average
#BE3.9max NGN
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_Control_plasmid_average = L2FC(HT29_CBE_Control_average_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_Control_T0_average = L2FC(HT29_CBE_Control_average_reads_RPM, HT29_CBE_T0_average_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_Tram_Control_average = L2FC(HT29_CBE_Tram_average_reads_RPM, HT29_CBE_Control_average_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_Pict_Control_average = L2FC(HT29_CBE_Pict_average_reads_RPM, HT29_CBE_Control_average_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_DebCet_Control_average = L2FC(HT29_CBE_DebCet_average_reads_RPM, HT29_CBE_Control_average_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_Tram_plasmid_average = L2FC(HT29_CBE_Tram_average_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_Pict_plasmid_average = L2FC(HT29_CBE_Pict_average_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_CBE_DebCet_plasmid_average = L2FC(HT29_CBE_DebCet_average_reads_RPM, plasmid_reads_RPM))

#ABE8e NGN
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_Control_plasmid_average = L2FC(HT29_ABE_Control_average_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_Control_T0_average = L2FC(HT29_ABE_Control_average_reads_RPM, HT29_ABE_T0_average_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_Tram_Control_average = L2FC(HT29_ABE_Tram_average_reads_RPM, HT29_ABE_Control_average_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_Pict_Control_average = L2FC(HT29_ABE_Pict_average_reads_RPM, HT29_ABE_Control_average_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_DebCet_Control_average = L2FC(HT29_ABE_DebCet_average_reads_RPM, HT29_ABE_Control_average_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_Tram_plasmid_average = L2FC(HT29_ABE_Tram_average_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_Pict_plasmid_average = L2FC(HT29_ABE_Pict_average_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_HT29_ABE_DebCet_plasmid_average = L2FC(HT29_ABE_DebCet_average_reads_RPM, plasmid_reads_RPM))

# H23 average
#BE3.9max NGN
raw_reads <- mutate(raw_reads, L2FC_H23_CBE_Control_plasmid_average = L2FC(H23_CBE_Control_average_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_CBE_Control_T0_average = L2FC(H23_CBE_Control_average_reads_RPM, H23_CBE_T0_average_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_CBE_Sotor_Control_average = L2FC(H23_CBE_Sotor_average_reads_RPM, H23_CBE_Control_average_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_CBE_Adag_Control_average = L2FC(H23_CBE_Adag_average_reads_RPM, H23_CBE_Control_average_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_CBE_Sotor_plasmid_average = L2FC(H23_CBE_Sotor_average_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_CBE_Adag_plasmid_average = L2FC(H23_CBE_Adag_average_reads_RPM, plasmid_reads_RPM))

#ABE8e NGN
raw_reads <- mutate(raw_reads, L2FC_H23_ABE_Control_plasmid_average = L2FC(H23_ABE_Control_average_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_ABE_Control_T0_average = L2FC(H23_ABE_Control_average_reads_RPM, H23_ABE_T0_average_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_ABE_Sotor_Control_average = L2FC(H23_ABE_Sotor_average_reads_RPM, H23_ABE_Control_average_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_ABE_Adag_Control_average = L2FC(H23_ABE_Adag_average_reads_RPM, H23_ABE_Control_average_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_ABE_Sotor_plasmid_average = L2FC(H23_ABE_Sotor_average_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_H23_ABE_Adag_plasmid_average = L2FC(H23_ABE_Adag_average_reads_RPM, plasmid_reads_RPM))

# PC9 average
#BE3.9max NGN
raw_reads <- mutate(raw_reads, L2FC_PC9_CBE_Control_plasmid_average = L2FC(PC9_CBE_Control_average_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_CBE_Control_T0_average = L2FC(PC9_CBE_Control_average_reads_RPM, PC9_CBE_T0_average_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_CBE_Gefit_Control_average = L2FC(PC9_CBE_Gefit_average_reads_RPM, PC9_CBE_Control_average_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_CBE_Osim_Control_average = L2FC(PC9_CBE_Osim_average_reads_RPM, PC9_CBE_Control_average_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_CBE_Gefit_plasmid_average = L2FC(PC9_CBE_Gefit_average_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_CBE_Osim_plasmid_average = L2FC(PC9_CBE_Osim_average_reads_RPM, plasmid_reads_RPM))

#ABE8e NGN
raw_reads <- mutate(raw_reads, L2FC_PC9_ABE_Control_plasmid_average = L2FC(PC9_ABE_Control_average_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_ABE_Control_T0_average = L2FC(PC9_ABE_Control_average_reads_RPM, PC9_ABE_T0_average_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_ABE_Gefit_Control_average = L2FC(PC9_ABE_Gefit_average_reads_RPM, PC9_ABE_Control_average_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_ABE_Osim_Control_average = L2FC(PC9_ABE_Osim_average_reads_RPM, PC9_ABE_Control_average_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_ABE_Gefit_plasmid_average = L2FC(PC9_ABE_Gefit_average_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_PC9_ABE_Osim_plasmid_average = L2FC(PC9_ABE_Osim_average_reads_RPM, plasmid_reads_RPM))

# MHH-ES-1 average
#ABE8e NGN
raw_reads <- mutate(raw_reads, L2FC_MHHES1_ABE_Control_plasmid_average = L2FC(MHHES1_ABE_Control_average_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_MHHES1_ABE_Control_T0_average = L2FC(MHHES1_ABE_Control_average_reads_RPM, MHHES1_ABE_T0_average_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_MHHES1_ABE_Olap_Control_average = L2FC(MHHES1_ABE_Olap_average_reads_RPM, MHHES1_ABE_Control_average_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_MHHES1_ABE_Nirap_Control_average = L2FC(MHHES1_ABE_Nirap_average_reads_RPM, MHHES1_ABE_Control_average_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_MHHES1_ABE_Olap_plasmid_average = L2FC(MHHES1_ABE_Olap_average_reads_RPM, plasmid_reads_RPM))
raw_reads <- mutate(raw_reads, L2FC_MHHES1_ABE_Nirap_plasmid_average = L2FC(MHHES1_ABE_Nirap_average_reads_RPM, plasmid_reads_RPM))
```

```{r}
# from L2FC, define z-scores
# define function
zscore <- function(x) ((x-mean(x))/sd(x))
raw_reads <- mutate_at(raw_reads, vars(matches("L2FC")), list(zscore = zscore))
```

```{r}
# left join gRNA library and read counts
lib <- mutate(lib, Gene = ifelse(sgRNA_type == "intergenic", "N/A", Gene))
lib %>% filter(sgRNA_type =="intergenic") %>% select(Gene, sgRNA_type)
data <- left_join(lib, raw_reads)
data %>% count(sgRNA_type)
```

```{r}
#HT-29
# off-targets pre-filter
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

# CBE control vs T0
plot1 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_HT29_CBE_Control_plasmid_average_zscore, y= perfect_matches, colour=Gene)) + geom_point(size=3, position = pos) +
labs(title= "HT-29 CBE: exonic gRNAs", x = "z-score - control vs plasmid", y = "number of off-targets", colour = "gene") +
scale_color_manual(values =c("black", "springgreen4", "orange", "red", "blue", "grey55", "brown", "purple", "pink", "salmon", "royalblue")) +
theme_classic()
plot1
ggsave("FigS1a.eps", width = 6, height = 4, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")


# ABE control vs T0
plot2 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_HT29_ABE_Control_plasmid_average_zscore, y= perfect_matches, colour=Gene)) + geom_point(size=3, position = pos) +
labs(title= "HT-29 ABE: exonic gRNAs", x = "zscore - control vs plasmid", y = "number of off-targets", colour = "gene") +
scale_color_manual(values =c("black", "springgreen4", "orange", "red", "blue", "grey55", "brown", "purple", "pink", "salmon", "royalblue")) +
theme_classic()
plot2
ggsave("FigS1b.eps", width = 6, height = 4, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")
```

```{r}
#filter out more than 2 perfect mismatches
data %>% filter(perfect_matches >2 | is.na(perfect_matches))
data %>% filter(perfect_matches >500 | is.na(perfect_matches))
data <- data %>% filter(perfect_matches < 3 | is.na(perfect_matches))
```

```{r}
# quality control plots
# HT-29
plot1 <- ggplot(data, aes(y= plasmid_reads)) + 
geom_histogram() +
labs(title= "plasmid Reads")
plot1

plot2 <- ggplot(data, aes(y= HT29_CBE_T0_exp1_reads)) + 
geom_histogram() +
labs(title= "HT29 CBE T0 reads")
plot2

plot3 <- ggplot(data, aes(y= HT29_ABE_T0_exp1_reads)) + 
geom_histogram() +
labs(title= "HT29 ABE T0 reads")
plot3

plot4 <- ggplot(data, aes(x= L2FC_HT29_CBE_Control_T0_1_zscore, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "zscore_CBE_Control_T0_1") +
theme_classic()
plot4

plot5 <- ggplot(data, aes(x= L2FC_HT29_ABE_Control_T0_1_zscore, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "zscore_ABE_Control_T0_1") +
theme_classic()
plot5

plot6 <- ggplot(data, aes(x= L2FC_HT29_CBE_Control_plasmid_1_zscore, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "zscore_CBE_Control_plasmid_1") +
theme_classic()
plot6

plot7 <- ggplot(data, aes(x= L2FC_HT29_ABE_Control_plasmid_1_zscore, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "zscore_ABE_Control_plasmid_1") +
theme_classic()
plot7

plot8 <- ggplot(data, aes(x= L2FC_HT29_CBE_Control_plasmid_2_zscore, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "zscore_CBE_Control_plasmid_2") +
theme_classic()
plot8

plot9 <- ggplot(data, aes(x= L2FC_HT29_ABE_Control_plasmid_2_zscore, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "zscore_ABE_Control_plasmid_2") +
theme_classic()
plot9
```

```{r}
# quality control plots
# H23
plot1 <- ggplot(data, aes(y= H23_CBE_T0_exp1_reads)) + 
geom_histogram() +
labs(title= "H23 CBE T0 reads")
plot1

plot2 <- ggplot(data, aes(y= H23_ABE_T0_exp1_reads)) + 
geom_histogram() +
labs(title= "H23 ABE T0 reads")
plot2

plot3 <- ggplot(data, aes(x= L2FC_H23_CBE_Control_T0_1_zscore, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "zscore_CBE_Control_T0_1") +
theme_classic()
plot3

plot4 <- ggplot(data, aes(x= L2FC_H23_ABE_Control_T0_1_zscore, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "zscore_ABE_Control_T0_1") +
theme_classic()
plot4

plot5 <- ggplot(data, aes(x= L2FC_H23_CBE_Control_plasmid_1_zscore, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "zscore_CBE_Control_plasmid_1") +
theme_classic()
plot5

plot6 <- ggplot(data, aes(x= L2FC_H23_ABE_Control_plasmid_1_zscore, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "zscore_ABE_Control_plasmid_1") +
theme_classic()
plot6

plot7 <- ggplot(data, aes(x= L2FC_H23_CBE_Control_plasmid_2_zscore, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "zscore_CBE_Control_plasmid_2") +
theme_classic()
plot7

plot8 <- ggplot(data, aes(x= L2FC_H23_ABE_Control_plasmid_2_zscore, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "zscore_ABE_Control_plasmid_2") +
theme_classic()
plot8
```

```{r}
# quality control plots
# PC9
plot1 <- ggplot(data, aes(y= PC9_CBE_T0_exp1_reads)) + 
geom_histogram() +
labs(title= "PC9 CBE T0 reads")
plot1

plot2 <- ggplot(data, aes(y= PC9_ABE_T0_exp1_reads)) + 
geom_histogram() +
labs(title= "PC9 ABE T0 reads")
plot2

plot3 <- ggplot(data, aes(x= L2FC_PC9_CBE_Control_T0_1_zscore, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "zscore_CBE_Control_T0_1") +
theme_classic()
plot3

plot4 <- ggplot(data, aes(x= L2FC_PC9_ABE_Control_T0_1_zscore, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "zscore_ABE_Control_T0_1") +
theme_classic()
plot4

plot5 <- ggplot(data, aes(x= L2FC_PC9_CBE_Control_plasmid_1_zscore, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "zscore_CBE_Control_plasmid_1") +
theme_classic()
plot5

plot6 <- ggplot(data, aes(x= L2FC_PC9_ABE_Control_plasmid_1_zscore, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "zscore_ABE_Control_plasmid_1") +
theme_classic()
plot6

plot8 <- ggplot(data, aes(x= L2FC_PC9_CBE_Control_plasmid_2_zscore, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "zscore_CBE_Control_plasmid_2") +
theme_classic()
plot8

plot9 <- ggplot(data, aes(x= L2FC_PC9_ABE_Control_plasmid_2_zscore, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "zscore_ABE_Control_plasmid_2") +
theme_classic()
plot9
```

```{r}
# quality control plots
# MHH-ES-1
plot1 <- ggplot(data, aes(y= MHHES1_ABE_T0_exp1_reads)) + 
geom_histogram() +
labs(title= "MHHES1 ABE T0 reads")
plot1

plot2 <- ggplot(data, aes(x= L2FC_MHHES1_ABE_Control_T0_1_zscore, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "zscore_ABE_Control_T0_1") +
theme_classic()
plot2

plot3 <- ggplot(data, aes(x= L2FC_MHHES1_ABE_Control_plasmid_1_zscore, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "zscore_ABE_Control_plasmid_1") +
theme_classic()
plot3

plot4 <- ggplot(data, aes(x= L2FC_MHHES1_ABE_Control_plasmid_2_zscore, colour = sgRNA_type, fill = sgRNA_type)) + 
geom_density(alpha = 0.1) +
labs(title= "", x = "zscore_ABE_Control_plasmid_2") +
theme_classic()
plot4
```

```{r}
# check for low read numbers and filter out <100 reads
data %>% filter(plasmid_reads <100)
data %>% filter(HT29_CBE_T0_exp1_reads <100)
data %>% filter(H23_CBE_T0_exp1_reads <100)
data %>% filter(PC9_CBE_T0_exp1_reads <100)

data %>% filter(HT29_ABE_T0_exp1_reads <100)
data %>% filter(H23_ABE_T0_exp1_reads <100)
data %>% filter(PC9_ABE_T0_exp1_reads <100)
data %>% filter(MHHES1_ABE_T0_exp1_reads <100)

data %>% filter(plasmid_reads <100 | HT29_CBE_T0_exp1_reads <100 | H23_CBE_T0_exp1_reads <100 | PC9_CBE_T0_exp1_reads <100 | HT29_ABE_T0_exp1_reads <100 | H23_ABE_T0_exp1_reads <100 | PC9_ABE_T0_exp1_reads <100 |MHHES1_ABE_T0_exp1_reads <100)

data <- filter(data, plasmid_reads >100 & HT29_CBE_T0_exp1_reads >100 & H23_CBE_T0_exp1_reads >100 & PC9_CBE_T0_exp1_reads >100 & HT29_ABE_T0_exp1_reads >100 & H23_ABE_T0_exp1_reads >100 & PC9_ABE_T0_exp1_reads >100 & MHHES1_ABE_T0_exp1_reads >100)
```

```{r}
#HT-29
# check L2FC PLASMID vs Gene
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

# CBE control vs plasmid
plot1 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_HT29_CBE_Control_plasmid_1_zscore)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "L2FC_CBE_plasmid_T0_1_zscore") +
theme_classic()
plot1

# CBE Tram vs plasmid
plot2 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_HT29_CBE_Tram_plasmid_1_zscore)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "L2FC_CBE_Tram_plasmid_1_zscore") +
theme_classic()
plot2

# CBE Pict vs plasmid
plot3 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_HT29_CBE_Pict_plasmid_1_zscore)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "L2FC_CBE_Pict_plasmid_1_zscore") +
theme_classic()
plot3

# CBE DebCet vs plasmid
plot4 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_HT29_CBE_DebCet_plasmid_1_zscore)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "L2FC_CBE_DebCet_plasmid_1_zscore") +
theme_classic()
plot4


# ABE plasmid vs T0
plot5 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_HT29_ABE_Control_plasmid_1_zscore)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "L2FC_ABE_plasmid_T0_1_zscore") +
theme_classic()
plot5

# ABE Tram vs plasmid
plot6 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_HT29_ABE_Tram_plasmid_1_zscore)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "L2FC_ABE_Tram_plasmid_1_zscore") +
theme_classic()
plot6

# ABE Pict vs plasmid
plot7 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_HT29_ABE_Pict_plasmid_1_zscore)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "L2FC_ABE_Pict_plasmid_1_zscore") +
theme_classic()
plot7

# ABE DebCet vs plasmid
plot8 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_HT29_ABE_DebCet_plasmid_1_zscore)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "L2FC_ABE_DebCet_plasmid_1_zscore") +
theme_classic()
plot8
```

```{r}
# replicate correlation - stats HT-29
r1 <- cor(data$L2FC_HT29_CBE_Control_plasmid_1_zscore, data$L2FC_HT29_CBE_Control_plasmid_2, method = "pearson")
r2 <- cor(data$L2FC_HT29_CBE_Tram_plasmid_1_zscore, data$L2FC_HT29_CBE_Tram_plasmid_2_zscore, method = "pearson")
r3 <- cor(data$L2FC_HT29_CBE_Pict_plasmid_1_zscore, data$L2FC_HT29_CBE_Pict_plasmid_2_zscore, method = "pearson")
r4 <- cor(data$L2FC_HT29_CBE_DebCet_plasmid_1_zscore, data$L2FC_HT29_CBE_DebCet_plasmid_2_zscore, method = "pearson")
r5 <- cor(data$L2FC_HT29_ABE_Control_plasmid_1_zscore, data$L2FC_HT29_ABE_Control_plasmid_2_zscore, method = "pearson")
r6 <- cor(data$L2FC_HT29_ABE_Tram_plasmid_1_zscore, data$L2FC_HT29_ABE_Tram_plasmid_2_zscore, method = "pearson")
r7 <- cor(data$L2FC_HT29_ABE_Pict_plasmid_1_zscore, data$L2FC_HT29_ABE_Pict_plasmid_2_zscore, method = "pearson")
r8 <- cor(data$L2FC_HT29_ABE_DebCet_plasmid_1_zscore, data$L2FC_HT29_ABE_DebCet_plasmid_2_zscore, method = "pearson")
```

```{r}
#HT-29 check correlation between replicates
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

# CBE control vs plasmid
plot1 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_HT29_CBE_Control_plasmid_1_zscore, y= L2FC_HT29_CBE_Control_plasmid_2_zscore)) + 
geom_pointdensity(size=3,  position = pos) +
labs(title= "HT-29 - replicate correlation", x = "zscore CBE control vs plasmid exp.1", y = "zscore CBE control vs plasmid exp.2", colour = "density") +
geom_smooth(method = lm) +
annotate("text", y = 5, x = 0, label = paste("R = ", round(r1, 2)), parse = FALSE, size = 7) +
theme_bw(base_size = 15)
plot1
ggsave("FigS2_1.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

# CBE Tram vs plasmid
plot2 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_HT29_CBE_Tram_plasmid_1_zscore, y= L2FC_HT29_CBE_Tram_plasmid_2_zscore)) + geom_pointdensity(size=3,  position = pos) +
labs(title= "HT-29 - replicate correlation", x = "zscore CBE Trametinib vs plasmid exp.1", y = "zscore CBE Trametinib vs plasmid exp.2", colour = "density") +
geom_smooth(method = lm) +
annotate("text", y = 5, x = 0, label = paste("R = ", round(r2, 2)), parse = FALSE, size = 7) +
theme_bw(base_size = 15)
plot2
ggsave("FigS2_2.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

# CBE Pict vs plasmid
plot3 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_HT29_CBE_Pict_plasmid_1_zscore, y= L2FC_HT29_CBE_Pict_plasmid_2_zscore)) + geom_pointdensity(size=3,  position = pos) +
labs(title= "HT-29 - replicate correlation", x = "zscore CBE Pictilisib vs plasmid exp.1", y = "zscore CBE Pictilisib vs plasmid exp.2", colour = "density") +
geom_smooth(method = lm) +
annotate("text", y = 5, x = 0, label = paste("R = ", round(r3, 2)), parse = FALSE, size = 7) +
theme_bw(base_size = 15)
plot3
ggsave("FigS2_3.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

# CBE DebCet vs plasmid
plot4 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_HT29_CBE_DebCet_plasmid_1_zscore, y= L2FC_HT29_CBE_DebCet_plasmid_2_zscore)) + 
geom_pointdensity(size=3,  position = pos) +
labs(title= "HT-29 - replicate correlation", x = "zscore CBE Dabrafenib+Cetuximab vs plasmid exp.1", y = "zscore CBE Dabrafenib+Cetuximab vs plasmid exp.2", colour = "density") +
geom_smooth(method = lm) +
annotate("text", y = 5, x = 0, label = paste("R = ", round(r4, 2)), parse = FALSE, size = 7) +
theme_bw(base_size = 15)
plot4
ggsave("FigS2_4.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

# ABE control vs plasmid
plot5 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_HT29_ABE_Control_plasmid_1_zscore, y= L2FC_HT29_ABE_Control_plasmid_2_zscore)) +
geom_pointdensity(size=3,  position = pos) +
scale_color_viridis_c() +
labs(title= "HT-29 - replicate correlation", x = "zscore ABE control vs plasmid exp.1", y = "zscore ABE control vs plasmid exp.2", colour = "density") +
geom_smooth(method = lm) +
annotate("text", y = 5, x = 0, label = paste("R = ", round(r5, 2)), parse = FALSE, size = 7) +
theme_bw(base_size = 15)
plot5
ggsave("FigS2_5.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

# ABE Tram vs plasmid
plot6 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_HT29_ABE_Tram_plasmid_1_zscore, y= L2FC_HT29_ABE_Tram_plasmid_2_zscore)) + geom_pointdensity(size=3,  position = pos) +
scale_color_viridis_c() +
labs(title= "HT-29 - replicate correlation", x = "zscore ABE Trametinib vs plasmid exp.1", y = "zscore ABE Trametinib vs plasmid exp.2", colour = "density") +
geom_smooth(method = lm) +
annotate("text", y = 5, x = 0, label = paste("R = ", round(r6, 2)), parse = FALSE, size = 7) +
theme_bw(base_size = 15)
plot6
ggsave("FigS2_6.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

# ABE Pict vs plasmid
plot7 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_HT29_ABE_Pict_plasmid_1_zscore, y= L2FC_HT29_ABE_Pict_plasmid_2_zscore)) + geom_pointdensity(size=3,  position = pos) +
scale_color_viridis_c() +
labs(title= "HT-29 - replicate correlation", x = "zscore ABE Pictilisb vs plasmid exp.1", y = "zscore ABE Pictilisb vs plasmid exp.2", colour = "density") +
geom_smooth(method = lm) +
annotate("text", y = 5, x = 0, label = paste("R = ", round(r7, 2)), parse = FALSE, size = 7) +
theme_bw(base_size = 15)
plot7
ggsave("FigS2_7.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

# ABE DebCet vs plasmid
plot8 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_HT29_ABE_DebCet_plasmid_1_zscore, y= L2FC_HT29_ABE_DebCet_plasmid_2_zscore)) + 
geom_pointdensity(size=3,  position = pos) +
scale_color_viridis_c() +
labs(title= "HT-29 - replicate correlation", x = "zscore ABE Dabrafenib+Cetuximab vs plasmid exp.1", y = "zscore ABE Dabrafenib+Cetuximab vs plasmid exp.2", colour = "density") +
geom_smooth(method = lm) +
annotate("text", y = 5, x = 0, label = paste("R = ", round(r8, 2)), parse = FALSE, size = 7) +
theme_bw(base_size = 15)
plot8
ggsave("FigS2_8.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")
```

```{r}
#H23
# check L2FC PLASMID vs Gene
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

# CBE control vs plasmid
plot1 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_H23_CBE_Control_plasmid_1_zscore)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "zscore_CBE_control_plasmid_1") +
theme_classic()
plot1

# CBE Sotor vs plasmid
plot2 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_H23_CBE_Sotor_plasmid_1_zscore)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "zscore_CBE_Sotor_plasmid_1") +
theme_classic()
plot2

# CBE Adag vs plasmid
plot3 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_H23_CBE_Adag_plasmid_1_zscore)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "zscore_CBE_Adag_plasmid_1") +
theme_classic()
plot3


# ABE plasmid vs T0
plot5 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_H23_ABE_Control_plasmid_1_zscore)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "zscore_ABE_plasmid_T0_1") +
theme_classic()
plot5

# ABE Sotor vs plasmid
plot6 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_H23_ABE_Sotor_plasmid_1_zscore)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "zscore_ABE_Sotor_plasmid_1") +
theme_classic()
plot6

# ABE Adag vs plasmid
plot7 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_H23_ABE_Adag_plasmid_1_zscore)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "zscore_ABE_Adag_plasmid_1") +
scale_color_continuous() +
theme_classic()
plot7
```

```{r}
# replicate correlation - stats H23
r9 <- cor(data$L2FC_H23_CBE_Control_plasmid_1_zscore, data$L2FC_H23_CBE_Control_plasmid_2_zscore, method = "pearson")
r10 <- cor(data$L2FC_H23_CBE_Sotor_plasmid_1_zscore, data$L2FC_H23_CBE_Sotor_plasmid_2_zscore, method = "pearson")
r11 <- cor(data$L2FC_H23_CBE_Adag_plasmid_1_zscore, data$L2FC_H23_CBE_Adag_plasmid_2_zscore, method = "pearson")
r12 <- cor(data$L2FC_H23_ABE_Control_plasmid_1_zscore, data$L2FC_H23_ABE_Control_plasmid_2_zscore, method = "pearson")
r13 <- cor(data$L2FC_H23_ABE_Sotor_plasmid_1_zscore, data$L2FC_H23_ABE_Sotor_plasmid_2_zscore, method = "pearson")
r14 <- cor(data$L2FC_H23_ABE_Adag_plasmid_1_zscore, data$L2FC_H23_ABE_Adag_plasmid_2_zscore, method = "pearson")
```

```{r}
#H23 check correlation between replicates
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

# CBE control vs plasmid
plot1 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_H23_CBE_Control_plasmid_1_zscore, y= L2FC_H23_CBE_Control_plasmid_2_zscore)) +
geom_pointdensity(size=3,  position = pos) +
labs(title= "H23 - replicate correlation", x = "zscore CBE control vs plasmid exp.1", y = "zscore CBE control vs plasmid exp.2", colour = "density") +
geom_smooth(method = lm) +
annotate("text", y = 5, x = 0, label = paste("R = ", round(r9, 2)), parse = FALSE, size = 7) +
theme_bw(base_size = 15)
plot1
ggsave("FigS2_9.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

# CBE Sotor vs plasmid
plot2 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_H23_CBE_Sotor_plasmid_1_zscore, y= L2FC_H23_CBE_Sotor_plasmid_2_zscore)) + geom_pointdensity(size=3,  position = pos) +
labs(title= "H23 - replicate correlation", x = "zscore CBE Sotorasib vs plasmid exp.1", y = "zscore CBE Sotorasib vs plasmid exp.2", colour = "density") +
geom_smooth(method = lm) +
annotate("text", y = 5, x = 0, label = paste("R = ", round(r10, 2)), parse = FALSE, size = 7) +
theme_bw(base_size = 15)
plot2
ggsave("FigS2_10.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

# CBE Adag vs plasmid
plot3 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_H23_CBE_Adag_plasmid_1_zscore, y= L2FC_H23_CBE_Adag_plasmid_2_zscore)) + geom_pointdensity(size=3,  position = pos) +
labs(title= "H23 - replicate correlation", x = "zscore CBE Adagrasib vs plasmid exp.1", y = "zscore CBE Adagrasib vs plasmid exp.2", colour = "density") +
geom_smooth(method = lm) +
annotate("text", y = 5, x = 0, label = paste("R = ", round(r11, 2)), parse = FALSE, size = 7) +
theme_bw(base_size = 15)
plot3
ggsave("FigS2_11.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")


# ABE control vs plasmid
plot4 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_H23_ABE_Control_plasmid_1_zscore, y= L2FC_H23_ABE_Control_plasmid_2_zscore)) +
geom_pointdensity(size=3,  position = pos) +
scale_color_viridis_c() +
labs(title= "H23 - replicate correlation", x = "zscore ABE control vs plasmid exp.1", y = "zscore ABE control vs plasmid exp.2", colour = "density") +
geom_smooth(method = lm) +
annotate("text", y = 7, x = 0, label = paste("R = ", round(r12, 2)), parse = FALSE, size = 7) +
theme_bw(base_size = 15)
plot4
ggsave("FigS2_12.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

# ABE Sotor vs plasmid
plot5 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_H23_ABE_Sotor_plasmid_1_zscore, y= L2FC_H23_ABE_Sotor_plasmid_2_zscore)) + geom_pointdensity(size=3,  position = pos) +
scale_color_viridis_c() +
labs(title= "H23 - replicate correlation", x = "zscore ABE Sotorasib vs plasmid exp.1", y = "zscore ABE Sotorasib vs plasmid exp.2", colour = "density") +
geom_smooth(method = lm) +
annotate("text", y = 7, x = 0, label = paste("R = ", round(r13, 2)), parse = FALSE, size = 7) +
theme_bw(base_size = 15)
plot5
ggsave("FigS2_13.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

# ABE Adag vs plasmid
plot6 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_H23_ABE_Adag_plasmid_1_zscore, y= L2FC_H23_ABE_Adag_plasmid_2_zscore)) + geom_pointdensity(size=3,  position = pos) +
scale_color_viridis_c() +
labs(title= "H23 - replicate correlation", x = "zscore ABE Adagrasib vs plasmid exp.1", y = "zscore ABE Adagrasib vs plasmid exp.2", colour = "density") +
geom_smooth(method = lm) +
annotate("text", y = 7, x = 0, label = paste("R = ", round(r14, 2)), parse = FALSE, size = 7) +
theme_bw(base_size = 15)
plot6
ggsave("FigS2_14.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")
```

```{r}
#PC9
# check L2FC PLASMID vs Gene
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

# CBE control vs plasmid
plot1 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_PC9_CBE_Control_plasmid_1_zscore)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "zscore_CBE_plasmid_T0_1") +
theme_classic()
plot1

# CBE Gefit vs plasmid
plot2 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_PC9_CBE_Gefit_plasmid_1_zscore)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "zscore_CBE_Gefit_plasmid_1") +
theme_classic()
plot2

# CBE Adag vs plasmid
plot3 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_PC9_CBE_Osim_plasmid_1_zscore)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "zscore_CBE_Osim_plasmid_1") +
theme_classic()
plot3


# ABE plasmid vs T0
plot5 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_PC9_ABE_Control_plasmid_1_zscore)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "zscore_ABE_plasmid_T0_1") +
theme_classic()
plot5

# ABE Gefit vs plasmid
plot6 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_PC9_ABE_Gefit_plasmid_1_zscore)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "zscore_ABE_Gefit_plasmid_1") +
theme_classic()
plot6

# ABE Osim vs plasmid
plot7 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_PC9_ABE_Osim_plasmid_1_zscore)) + geom_jitter(size=3, alpha=0.8, position = pos) +
labs(title= "all gRNAs", x = "Gene", y = "zscore_ABE_Osim_plasmid_1") +
theme_classic()
plot7
```

```{r}
# replicate correlation - stats PC9
r15 <- cor(data$L2FC_PC9_CBE_Control_plasmid_1_zscore, data$L2FC_PC9_CBE_Control_plasmid_2_zscore, method = "pearson")
r16 <- cor(data$L2FC_PC9_CBE_Gefit_plasmid_1_zscore, data$L2FC_PC9_CBE_Gefit_plasmid_2_zscore, method = "pearson")
r17 <- cor(data$L2FC_PC9_CBE_Gefit_plasmid_1_zscore, data$L2FC_PC9_CBE_Gefit_plasmid_3_zscore, method = "pearson")
r18 <- cor(data$L2FC_PC9_CBE_Gefit_plasmid_2_zscore, data$L2FC_PC9_CBE_Gefit_plasmid_3_zscore, method = "pearson")
r19 <- cor(data$L2FC_PC9_CBE_Osim_plasmid_1_zscore, data$L2FC_PC9_CBE_Osim_plasmid_2_zscore, method = "pearson")
r20 <- cor(data$L2FC_PC9_ABE_Control_plasmid_1_zscore, data$L2FC_PC9_ABE_Control_plasmid_2_zscore, method = "pearson")
r21 <- cor(data$L2FC_PC9_ABE_Gefit_plasmid_1_zscore, data$L2FC_PC9_ABE_Gefit_plasmid_2_zscore, method = "pearson")
r22 <- cor(data$L2FC_PC9_ABE_Osim_plasmid_1_zscore, data$L2FC_PC9_ABE_Osim_plasmid_2_zscore, method = "pearson")
```

```{r}
#PC9 check correlation between replicates
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

# CBE control vs plasmid
plot1 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_PC9_CBE_Control_plasmid_1_zscore, y= L2FC_PC9_CBE_Control_plasmid_2_zscore)) + 
geom_pointdensity(size=3,  position = pos) +
labs(title= "PC9 - replicate correlation", x = "zscore CBE control vs plasmid exp.1", y = "zscore CBE control vs plasmid exp.2", colour = "density") +
geom_smooth(method = lm) +
annotate("text", y = 5, x = 0, label = paste("R = ", round(r15, 2)), parse = FALSE, size = 7) +
theme_bw(base_size = 15)
plot1
ggsave("FigS2_15.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

# CBE exp 1/2 Gefit vs plasmid
plot2 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_PC9_CBE_Gefit_plasmid_1_zscore, y= L2FC_PC9_CBE_Gefit_plasmid_2_zscore)) + geom_pointdensity(size=3,  position = pos) +
labs(title= "PC9 - replicate correlation", x = "zscore CBE Gefitinib vs plasmid exp.1", y = "zscore CBE Gefitinib vs plasmid exp.2", colour = "density") +
geom_smooth(method = lm) +
annotate("text", y = 12, x = 0, label = paste("R = ", round(r16, 2)), parse = FALSE, size = 7) +
theme_bw(base_size = 15)
plot2
ggsave("FigS2_16.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

# CBE exp 1/3 Gefit vs plasmid
plot3 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_PC9_CBE_Gefit_plasmid_1_zscore, y= L2FC_PC9_CBE_Gefit_plasmid_3_zscore)) + geom_pointdensity(size=3,  position = pos) +
labs(title= "PC9 - replicate correlation", x = "zscore CBE Gefitinib vs plasmid exp.1", y = "zscore CBE Gefitinib vs plasmid exp.3", colour = "density") +
geom_smooth(method = lm) +
annotate("text", y = 12, x = 0, label = paste("R = ", round(r17, 2)), parse = FALSE, size = 7) +
theme_bw(base_size = 15)
plot3
ggsave("FigS2_XXX.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

# CBE exp 2/3 Gefit vs plasmid
plot4 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_PC9_CBE_Gefit_plasmid_2_zscore, y= L2FC_PC9_CBE_Gefit_plasmid_3_zscore)) + geom_pointdensity(size=3,  position = pos) +
labs(title= "PC9 - replicate correlation", x = "zscore CBE Gefitinib vs plasmid exp.2", y = "zscore CBE Gefitinib vs plasmid exp.3", colour = "density") +
geom_smooth(method = lm) +
annotate("text", y = 10, x = 0, label = paste("R = ", round(r18, 2)), parse = FALSE, size = 7) +
theme_bw(base_size = 15)
plot4

# CBE Osim vs plasmid
plot5 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_PC9_CBE_Osim_plasmid_1_zscore, y= L2FC_PC9_CBE_Osim_plasmid_2_zscore)) + geom_pointdensity(size=3,  position = pos) +
labs(title= "PC9 - replicate correlation", x = "zscore CBE Osimertinib vs plasmid exp.1", y = "zscore CBE Osimertinib vs plasmid exp.2", colour = "density") +
geom_smooth(method = lm) +
annotate("text", y = 5, x = 0, label = paste("R = ", round(r19, 2)), parse = FALSE, size = 7) +
theme_bw(base_size = 15)
plot5
ggsave("FigS2_17.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

# ABE control vs plasmid
plot6 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_PC9_ABE_Control_plasmid_1_zscore, y= L2FC_PC9_ABE_Control_plasmid_2_zscore)) +
geom_pointdensity(size=3,  position = pos) +
scale_color_viridis_c() +
labs(title= "PC9 - replicate correlation", x = "zscore ABE control vs plasmid exp.1", y = "zscore ABE control vs plasmid exp.2", colour = "density") +
geom_smooth(method = lm) +
annotate("text", y = 5, x = 0, label = paste("R = ", round(r20, 2)), parse = FALSE, size = 7) +
theme_bw(base_size = 15)
plot6
ggsave("FigS2_18.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

# ABE Gefit vs plasmid
plot7 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_PC9_ABE_Gefit_plasmid_1_zscore, y= L2FC_PC9_ABE_Gefit_plasmid_2_zscore)) +
geom_pointdensity(size=3,  position = pos) +
scale_color_viridis_c() +
labs(title= "PC9 - replicate correlation", x = "zscore ABE Gefitinib vs plasmid exp.1", y = "zscore ABE Gefitinib vs plasmid exp.2", colour = "density") +
geom_smooth(method = lm) +
annotate("text", y = 5, x = 0, label = paste("R = ", round(r21, 2)), parse = FALSE, size = 7) +
theme_bw(base_size = 15)
plot7
ggsave("FigS2_19.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

# ABE Osim vs plasmid
plot8 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_PC9_ABE_Osim_plasmid_1_zscore, y= L2FC_PC9_ABE_Osim_plasmid_2_zscore)) +
geom_pointdensity(size=3,  position = pos) +
scale_color_viridis_c() +
labs(title= "PC9 - replicate correlation", x = "zscore ABE Osimertinib vs plasmid exp.1", y = "zscore ABE Osimertinib vs plasmid exp.2", colour = "density") +
geom_smooth(method = lm) +
annotate("text", y = 5, x = 0, label = paste("R = ", round(r22, 2)), parse = FALSE, size = 7) +
theme_bw(base_size = 15)
plot8
ggsave("FigS2_20.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")
```
```{r}
# MHHES1
# check L2FC PLASMID vs Gene
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

# ABE plasmid vs T0
plot1 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_MHHES1_ABE_Control_plasmid_1_zscore)) + geom_jitter(size=3, alpha=1, position = pos, colour = "black") +
labs(title= "all gRNAs", x = "gene", y = "control vs plasmid (z-score)") +
theme_classic()
plot1

# ABE Olap vs plasmid
plot2 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_MHHES1_ABE_Olap_plasmid_1_zscore)) + geom_jitter(size=3, alpha=1, position = pos, colour = "blue") +
labs(title= "all gRNAs", x = "gene", y = "olaparib vs plasmid (z-score)") +
theme_classic()
plot2

# ABE Nirap vs plasmid
plot3 <- ggplot(subset(data, sgRNA_type %in% c("exonic", "non_targeting")), aes(x= Gene, y= L2FC_MHHES1_ABE_Nirap_plasmid_1_zscore)) + geom_jitter(size=3, alpha=1, position = pos, colour = "red") +
labs(title= "all gRNAs", x = "gene", y = "niraparib vs plasmid (z-score)") +
theme_classic()
plot3
```

```{r}
# replicate correlation - stats MHH-ES-1
r23 <- cor(data$L2FC_MHHES1_ABE_Control_plasmid_1_zscore, data$L2FC_MHHES1_ABE_Control_plasmid_2_zscore, method = "pearson")
r24 <- cor(data$L2FC_MHHES1_ABE_Olap_plasmid_1_zscore, data$L2FC_MHHES1_ABE_Olap_plasmid_2_zscore, method = "pearson")
r25 <- cor(data$L2FC_MHHES1_ABE_Nirap_plasmid_1_zscore, data$L2FC_MHHES1_ABE_Nirap_plasmid_2_zscore, method = "pearson")
```

```{r}
# MHH-ES-1 check correlation between replicates
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

# ABE control vs plasmid
plot1 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_MHHES1_ABE_Control_plasmid_1_zscore, y= L2FC_MHHES1_ABE_Control_plasmid_2_zscore)) +
geom_pointdensity(size=3,  position = pos) +
scale_color_viridis_c() +
labs(title= "MHH-ES-1 - replicate correlation", x = "zscore ABE control vs plasmid exp.1", y = "zscore ABE control vs plasmid exp.2", colour = "density") +
geom_smooth(method = lm) +
annotate("text", y = 5, x = 0, label = paste("R = ", round(r23, 2)), parse = FALSE, size = 7) +
theme_bw(base_size = 15)
plot1
ggsave("FigS2_21.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

# ABE Olap vs plasmid
plot6 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_MHHES1_ABE_Olap_plasmid_1_zscore, y= L2FC_MHHES1_ABE_Olap_plasmid_2_zscore)) + 
geom_pointdensity(size=3,  position = pos) +
scale_color_viridis_c() +
labs(title= "MHH-ES-1 - replicate correlation", x = "zscore ABE Olaparib vs plasmid exp.1", y = "zscore ABE Olaparib vs plasmid exp.2", colour = "density") +
geom_smooth(method = lm) +
annotate("text", y = 5, x = 0, label = paste("R = ", round(r24, 2)), parse = FALSE, size = 7) +
theme_bw(base_size = 15)
plot6
ggsave("FigS2_22.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

# ABE Nirap vs plasmid
plot7 <- ggplot(subset(data, sgRNA_type %in% c("exonic")), aes(x= L2FC_MHHES1_ABE_Nirap_plasmid_1_zscore, y= L2FC_MHHES1_ABE_Nirap_plasmid_2_zscore)) + geom_pointdensity(size=3,  position = pos) +
scale_color_viridis_c() +
labs(title= "MHH-ES-1 - replicate correlation", x = "zscore ABE Niraparib vs plasmid exp.1", y = "zscore ABE Niraparib vs plasmid exp.2", colour = "density") +
geom_smooth(method = lm) +
annotate("text", y = 5, x = 0, label = paste("R = ", round(r25, 2)), parse = FALSE, size = 7) +
theme_bw(base_size = 15)
plot7
ggsave("FigS2_23.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")
```

```{r}
#BEstimate annotations for newest Bestimate tool with 3-9 editing window for CBE and ABE

#read in BEstimate sumamry dfs
AKT1_CBE <- as_tibble(read.csv("BEstimate_protein_050423/CBE_NGN_AKT1_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
MYC_CBE <- as_tibble(read.csv("BEstimate_protein_050423/CBE_NGN_MYC_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
PIK3CA_CBE <- as_tibble(read.csv("BEstimate_protein_050423/CBE_NGN_PIK3CA_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
KRAS_CBE <- as_tibble(read.csv("BEstimate_protein_050423/CBE_NGN_KRAS_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
BCL2_CBE <- as_tibble(read.csv("BEstimate_protein_050423/CBE_NGN_BCL2_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
BRAF_CBE <- as_tibble(read.csv("BEstimate_protein_050423/CBE_NGN_BRAF_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
PARP1_CBE <- as_tibble(read.csv("BEstimate_protein_050423/CBE_NGN_PARP1_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
PARP2_CBE <- as_tibble(read.csv("BEstimate_protein_050423/CBE_NGN_PARP2_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
EGFR_CBE <- as_tibble(read.csv("BEstimate_protein_050423/CBE_NGN_EGFR_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
MAP2K1_CBE <- as_tibble(read.csv("BEstimate_protein_050423/CBE_NGN_MAP2K1_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
MAP2K2_CBE <- as_tibble(read.csv("BEstimate_protein_050423/CBE_NGN_MAP2K2_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))

AKT1_ABE <- as_tibble(read.csv("BEstimate_protein_050423/ABE_NGN_AKT1_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
MYC_ABE <- as_tibble(read.csv("BEstimate_protein_050423/ABE_NGN_MYC_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
PIK3CA_ABE <- as_tibble(read.csv("BEstimate_protein_050423/ABE_NGN_PIK3CA_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
KRAS_ABE <- as_tibble(read.csv("BEstimate_protein_050423/ABE_NGN_KRAS_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
BCL2_ABE <- as_tibble(read.csv("BEstimate_protein_050423/ABE_NGN_BCL2_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
BRAF_ABE <- as_tibble(read.csv("BEstimate_protein_050423/ABE_NGN_BRAF_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
PARP1_ABE <- as_tibble(read.csv("BEstimate_protein_050423/ABE_NGN_PARP1_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
PARP2_ABE <- as_tibble(read.csv("BEstimate_protein_050423/ABE_NGN_PARP2_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
EGFR_ABE <- as_tibble(read.csv("BEstimate_protein_050423/ABE_NGN_EGFR_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
MAP2K1_ABE <- as_tibble(read.csv("BEstimate_protein_050423/ABE_NGN_MAP2K1_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))
MAP2K2_ABE <- as_tibble(read.csv("BEstimate_protein_050423/ABE_NGN_MAP2K2_39_summary_df.csv", stringsAsFactors = FALSE, header = TRUE))

#bind rows of all BEsimate dfs
CBE_BEstimate <- bind_rows(AKT1_CBE, MYC_CBE, BCL2_CBE, BRAF_CBE, MAP2K1_CBE, MAP2K2_CBE, PARP1_CBE, PARP2_CBE, PIK3CA_CBE, EGFR_CBE, KRAS_CBE)
ABE_BEstimate <- bind_rows(AKT1_ABE, MYC_ABE, BCL2_ABE, BRAF_ABE, MAP2K1_ABE, MAP2K2_ABE, PARP1_ABE, PARP2_ABE, PIK3CA_ABE, EGFR_ABE, KRAS_ABE)

# remove data columns that mess up the left join
data <- data %>% select(!c(Transcript_ID, Exon_ID, Direction))

#rename guide and Gene columns
CBE_BEstimate<- rename(CBE_BEstimate, Gene = Hugo_Symbol)
CBE_BEstimate<- rename(CBE_BEstimate, Amino_Acid_Position = Protein_Position)
CBE_BEstimate<- rename(CBE_BEstimate, guide = gRNA_Target_Sequence)

ABE_BEstimate<- rename(ABE_BEstimate, Gene = Hugo_Symbol)
ABE_BEstimate<- rename(ABE_BEstimate, Amino_Acid_Position = Protein_Position)
ABE_BEstimate<- rename(ABE_BEstimate, guide = gRNA_Target_Sequence)

#left join the library and read counts with the BEstimate predictions
data_CBE <- left_join(data, CBE_BEstimate)
data_CBE <- mutate(data_CBE, editor = "CBE")

data_ABE <- left_join(data, ABE_BEstimate)
data_ABE <- mutate(data_ABE, editor = "ABE")

annotated_data <- bind_rows(data_CBE, data_ABE)
annotated_data
```
#```{r}
# read in z-scores from DrugZ

#HT-29
#CBE
HT29_zscore_CBE_control_plasmid <- as_tibble(read.delim("drugZ_zscores_110423/HT29_REP1_REP2_AVERAGE/Control_CBE_vs_plasmid_foldchange", stringsAsFactors = FALSE, header = TRUE, sep ="\t"))
HT29_zscore_CBE_control_plasmid <- HT29_zscore_CBE_control_plasmid %>% rename(c(sgRNA_ID = X, Gene = GENE, HT29_zscore_control_plasmid = zscore)) %>% select(Gene, sgRNA_ID, HT29_zscore_control_plasmid) %>% mutate(editor = "CBE")

HT29_zscore_CBE_Tram_plasmid <- as_tibble(read.delim("drugZ_zscores_110423/HT29_REP1_REP2_AVERAGE/Tram_CBE_vs_plasmid_foldchange", stringsAsFactors = FALSE, header = TRUE, sep ="\t"))
HT29_zscore_CBE_Tram_plasmid <- HT29_zscore_CBE_Tram_plasmid %>% rename(c(sgRNA_ID = X, Gene = GENE, HT29_zscore_Tram_plasmid = zscore)) %>% select(Gene, sgRNA_ID, HT29_zscore_Tram_plasmid) %>% mutate(editor = "CBE")

HT29_zscore_CBE_Pict_plasmid <- as_tibble(read.delim("drugZ_zscores_110423/HT29_REP1_REP2_AVERAGE/Pict_CBE_vs_plasmid_foldchange", stringsAsFactors = FALSE, header = TRUE, sep ="\t"))
HT29_zscore_CBE_Pict_plasmid <- HT29_zscore_CBE_Pict_plasmid %>% rename(c(sgRNA_ID = X, Gene = GENE, HT29_zscore_Pict_plasmid = zscore)) %>% select(Gene, sgRNA_ID, HT29_zscore_Pict_plasmid) %>% mutate(editor = "CBE")

HT29_zscore_CBE_DebCet_plasmid <- as_tibble(read.delim("drugZ_zscores_110423/HT29_REP1_REP2_AVERAGE/DebCet_CBE_vs_plasmid_foldchange", stringsAsFactors = FALSE, header = TRUE, sep ="\t"))
HT29_zscore_CBE_DebCet_plasmid <- HT29_zscore_CBE_DebCet_plasmid %>% rename(c(sgRNA_ID = X, Gene = GENE, HT29_zscore_DebCet_plasmid = zscore)) %>% select(Gene, sgRNA_ID, HT29_zscore_DebCet_plasmid) %>% mutate(editor = "CBE")

#ABE
HT29_zscore_ABE_control_plasmid <- as_tibble(read.delim("drugZ_zscores_110423/HT29_REP1_REP2_AVERAGE/Control_ABE_vs_plasmid_foldchange", stringsAsFactors = FALSE, header = TRUE, sep ="\t"))
HT29_zscore_ABE_control_plasmid <- HT29_zscore_ABE_control_plasmid %>% rename(c(sgRNA_ID = X, Gene = GENE, HT29_zscore_control_plasmid = zscore)) %>% select(Gene, sgRNA_ID, HT29_zscore_control_plasmid) %>% mutate(editor = "ABE")

HT29_zscore_ABE_Tram_plasmid <- as_tibble(read.delim("drugZ_zscores_110423/HT29_REP1_REP2_AVERAGE/Tram_ABE_vs_plasmid_foldchange", stringsAsFactors = FALSE, header = TRUE, sep ="\t"))
HT29_zscore_ABE_Tram_plasmid <- HT29_zscore_ABE_Tram_plasmid %>% rename(c(sgRNA_ID = X, Gene = GENE, HT29_zscore_Tram_plasmid = zscore)) %>% select(Gene, sgRNA_ID, HT29_zscore_Tram_plasmid) %>% mutate(editor = "ABE")

HT29_zscore_ABE_Pict_plasmid <- as_tibble(read.delim("drugZ_zscores_110423/HT29_REP1_REP2_AVERAGE/Pict_ABE_vs_plasmid_foldchange", stringsAsFactors = FALSE, header = TRUE, sep ="\t"))
HT29_zscore_ABE_Pict_plasmid <- HT29_zscore_ABE_Pict_plasmid %>% rename(c(sgRNA_ID = X, Gene = GENE, HT29_zscore_Pict_plasmid = zscore)) %>% select(Gene, sgRNA_ID, HT29_zscore_Pict_plasmid) %>% mutate(editor = "ABE")

HT29_zscore_ABE_DebCet_plasmid <- as_tibble(read.delim("drugZ_zscores_110423/HT29_REP1_REP2_AVERAGE/DebCet_ABE_vs_plasmid_foldchange", stringsAsFactors = FALSE, header = TRUE, sep ="\t"))
HT29_zscore_ABE_DebCet_plasmid <- HT29_zscore_ABE_DebCet_plasmid %>% rename(c(sgRNA_ID = X, Gene = GENE, HT29_zscore_DebCet_plasmid = zscore)) %>% select(Gene, sgRNA_ID, HT29_zscore_DebCet_plasmid) %>% mutate(editor = "ABE")

#H23
#CBE
H23_zscore_CBE_control_plasmid <- as_tibble(read.delim("drugZ_zscores_110423/H23_REP1_REP2_AVERAGE/Control_CBE_vs_plasmid_foldchange", stringsAsFactors = FALSE, header = TRUE, sep ="\t"))
H23_zscore_CBE_control_plasmid <- H23_zscore_CBE_control_plasmid %>% rename(c(sgRNA_ID = X, Gene = GENE, H23_zscore_control_plasmid = zscore)) %>% select(Gene, sgRNA_ID, H23_zscore_control_plasmid) %>% mutate(editor = "CBE")

H23_zscore_CBE_Adag_plasmid <- as_tibble(read.delim("drugZ_zscores_110423/H23_REP1_REP2_AVERAGE/Adag_CBE_vs_plasmid_foldchange", stringsAsFactors = FALSE, header = TRUE, sep ="\t"))
H23_zscore_CBE_Adag_plasmid <- H23_zscore_CBE_Adag_plasmid %>% rename(c(sgRNA_ID = X, Gene = GENE, H23_zscore_Adag_plasmid = zscore)) %>% select(Gene, sgRNA_ID, H23_zscore_Adag_plasmid) %>% mutate(editor = "CBE")

H23_zscore_CBE_Sotor_plasmid <- as_tibble(read.delim("drugZ_zscores_110423/H23_REP1_REP2_AVERAGE/Sotor_CBE_vs_plasmid_foldchange", stringsAsFactors = FALSE, header = TRUE, sep ="\t"))
H23_zscore_CBE_Sotor_plasmid <- H23_zscore_CBE_Sotor_plasmid %>% rename(c(sgRNA_ID = X, Gene = GENE, H23_zscore_Sotor_plasmid = zscore)) %>% select(Gene, sgRNA_ID, H23_zscore_Sotor_plasmid) %>% mutate(editor = "CBE")

#ABE
H23_zscore_ABE_control_plasmid <- as_tibble(read.delim("drugZ_zscores_110423/H23_REP1_REP2_AVERAGE/Control_ABE_vs_plasmid_foldchange", stringsAsFactors = FALSE, header = TRUE, sep ="\t"))
H23_zscore_ABE_control_plasmid <- H23_zscore_ABE_control_plasmid %>% rename(c(sgRNA_ID = X, Gene = GENE, H23_zscore_control_plasmid = zscore)) %>% select(Gene, sgRNA_ID, H23_zscore_control_plasmid) %>% mutate(editor = "ABE")

H23_zscore_ABE_Adag_plasmid <- as_tibble(read.delim("drugZ_zscores_110423/H23_REP1_REP2_AVERAGE/Adag_ABE_vs_plasmid_foldchange", stringsAsFactors = FALSE, header = TRUE, sep ="\t"))
H23_zscore_ABE_Adag_plasmid <- H23_zscore_ABE_Adag_plasmid %>% rename(c(sgRNA_ID = X, Gene = GENE, H23_zscore_Adag_plasmid = zscore)) %>% select(Gene, sgRNA_ID, H23_zscore_Adag_plasmid) %>% mutate(editor = "ABE")

H23_zscore_ABE_Sotor_plasmid <- as_tibble(read.delim("drugZ_zscores_110423/H23_REP1_REP2_AVERAGE/Sotor_ABE_vs_plasmid_foldchange", stringsAsFactors = FALSE, header = TRUE, sep ="\t"))
H23_zscore_ABE_Sotor_plasmid <- H23_zscore_ABE_Sotor_plasmid %>% rename(c(sgRNA_ID = X, Gene = GENE, H23_zscore_Sotor_plasmid = zscore)) %>% select(Gene, sgRNA_ID, H23_zscore_Sotor_plasmid) %>% mutate(editor = "ABE")

#PC9
#CBE
PC9_zscore_CBE_control_plasmid <- as_tibble(read.delim("drugZ_zscores_110423/PC9_REP1_REP2_AVERAGE/Control_CBE_vs_plasmid_foldchange", stringsAsFactors = FALSE, header = TRUE, sep ="\t"))
PC9_zscore_CBE_control_plasmid <- PC9_zscore_CBE_control_plasmid %>% rename(c(sgRNA_ID = X, Gene = GENE, PC9_zscore_control_plasmid = zscore)) %>% select(Gene, sgRNA_ID, PC9_zscore_control_plasmid) %>% mutate(editor = "CBE")

PC9_zscore_CBE_Gefit_plasmid <- as_tibble(read.delim("drugZ_zscores_110423/PC9_REP1_REP2_AVERAGE/Gefit_CBE_vs_plasmid_foldchange", stringsAsFactors = FALSE, header = TRUE, sep ="\t"))
PC9_zscore_CBE_Gefit_plasmid <- PC9_zscore_CBE_Gefit_plasmid %>% rename(c(sgRNA_ID = X, Gene = GENE, PC9_zscore_Gefit_plasmid = zscore)) %>% select(Gene, sgRNA_ID, PC9_zscore_Gefit_plasmid) %>% mutate(editor = "CBE")

PC9_zscore_CBE_Osim_plasmid <- as_tibble(read.delim("drugZ_zscores_110423/PC9_REP1_REP2_AVERAGE/Osim_CBE_vs_plasmid_foldchange", stringsAsFactors = FALSE, header = TRUE, sep ="\t"))
PC9_zscore_CBE_Osim_plasmid <- PC9_zscore_CBE_Osim_plasmid %>% rename(c(sgRNA_ID = X, Gene = GENE, PC9_zscore_Osim_plasmid = zscore)) %>% select(Gene, sgRNA_ID, PC9_zscore_Osim_plasmid) %>% mutate(editor = "CBE")

#ABE
PC9_zscore_ABE_control_plasmid <- as_tibble(read.delim("drugZ_zscores_110423/PC9_REP1_REP2_AVERAGE/Control_ABE_vs_plasmid_foldchange", stringsAsFactors = FALSE, header = TRUE, sep ="\t"))
PC9_zscore_ABE_control_plasmid <- PC9_zscore_ABE_control_plasmid %>% rename(c(sgRNA_ID = X, Gene = GENE, PC9_zscore_control_plasmid = zscore)) %>% select(Gene, sgRNA_ID, PC9_zscore_control_plasmid) %>% mutate(editor = "ABE")

PC9_zscore_ABE_Gefit_plasmid <- as_tibble(read.delim("drugZ_zscores_110423/PC9_REP1_REP2_AVERAGE/Gefit_ABE_vs_plasmid_foldchange", stringsAsFactors = FALSE, header = TRUE, sep ="\t"))
PC9_zscore_ABE_Gefit_plasmid <- PC9_zscore_ABE_Gefit_plasmid %>% rename(c(sgRNA_ID = X, Gene = GENE, PC9_zscore_Gefit_plasmid = zscore)) %>% select(Gene, sgRNA_ID, PC9_zscore_Gefit_plasmid) %>% mutate(editor = "ABE")

PC9_zscore_ABE_Osim_plasmid <- as_tibble(read.delim("drugZ_zscores_110423/PC9_REP1_REP2_AVERAGE/Osim_ABE_vs_plasmid_foldchange", stringsAsFactors = FALSE, header = TRUE, sep ="\t"))
PC9_zscore_ABE_Osim_plasmid <- PC9_zscore_ABE_Osim_plasmid %>% rename(c(sgRNA_ID = X, Gene = GENE, PC9_zscore_Osim_plasmid = zscore)) %>% select(Gene, sgRNA_ID, PC9_zscore_Osim_plasmid) %>% mutate(editor = "ABE")


CBE_zscores <- left_join(HT29_zscore_CBE_control_plasmid, HT29_zscore_CBE_Tram_plasmid)
CBE_zscores <- left_join(CBE_zscores, HT29_zscore_CBE_Pict_plasmid)
CBE_zscores <- left_join(CBE_zscores, HT29_zscore_CBE_DebCet_plasmid)
CBE_zscores <- left_join(CBE_zscores, H23_zscore_CBE_control_plasmid)
CBE_zscores <- left_join(CBE_zscores, H23_zscore_CBE_Sotor_plasmid)
CBE_zscores <- left_join(CBE_zscores, H23_zscore_CBE_Adag_plasmid)
CBE_zscores <- left_join(CBE_zscores, PC9_zscore_CBE_control_plasmid)
CBE_zscores <- left_join(CBE_zscores, PC9_zscore_CBE_Gefit_plasmid)
CBE_zscores <- left_join(CBE_zscores, PC9_zscore_CBE_Osim_plasmid)

ABE_zscores <- left_join(HT29_zscore_ABE_control_plasmid, HT29_zscore_ABE_Tram_plasmid)
ABE_zscores <- left_join(ABE_zscores, HT29_zscore_ABE_Pict_plasmid)
ABE_zscores <- left_join(ABE_zscores, HT29_zscore_ABE_DebCet_plasmid)
ABE_zscores <- left_join(ABE_zscores, H23_zscore_ABE_control_plasmid)
ABE_zscores <- left_join(ABE_zscores, H23_zscore_ABE_Sotor_plasmid)
ABE_zscores <- left_join(ABE_zscores, H23_zscore_ABE_Adag_plasmid)
ABE_zscores <- left_join(ABE_zscores, PC9_zscore_ABE_control_plasmid)
ABE_zscores <- left_join(ABE_zscores, PC9_zscore_ABE_Gefit_plasmid)
ABE_zscores <- left_join(ABE_zscores, PC9_zscore_ABE_Osim_plasmid)

zscores <- bind_rows(CBE_zscores, ABE_zscores)
zscores
#```

#```{r}
#left join the dataframe with the drugZ zscores
annotated_data <- left_join(annotated_data, zscores)
#```

#```{r}
#drugz and L2FC correlation and manual zscore correlation
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

# HT29 vs plasmid
plot1 <- ggplot(annotated_data, aes(x= L2FC_HT29_CBE_Control_plasmid_average, y= HT29_zscore_control_plasmid, colour = sgRNA_type, shape = editor)) + geom_jitter(size=3,  position = pos) +
labs(title= "HT29 - zscore L2FC correlation", x = "L2FC_HT29_CBE_Control_plasmid_average", y = "HT29_zscore_control_plasmid") +
geom_smooth(method = lm) +
theme_bw()
plot1

plot2 <- ggplot(annotated_data, aes(x= L2FC_HT29_ABE_Control_plasmid_average, y= HT29_zscore_control_plasmid, colour = sgRNA_type, shape = editor)) + geom_jitter(size=3,  position = pos) +
labs(title= "HT29 - zscore L2FC correlation", x = "L2FC_HT29_ABE_Control_plasmid_average", y = "HT29_zscore_control_plasmid") +
geom_smooth(method = lm) +
theme_bw()
plot2

plot3 <- ggplot(annotated_data, aes(x= L2FC_HT29_CBE_Control_plasmid_average, y= L2FC_HT29_CBE_Control_plasmid_average_zscore, colour = sgRNA_type, shape = editor)) + geom_jitter(size=3,  position = pos) +
labs(title= "HT29 - zscore L2FC correlation", x = "L2FC_HT29_CBE_Control_plasmid_average", y = "L2FC_HT29_CBE_Control_plasmid_average_zscore") +
geom_smooth(method = lm) +
theme_bw()
plot3

plot4 <- ggplot(annotated_data, aes(x= L2FC_HT29_ABE_Control_plasmid_average, y= L2FC_HT29_ABE_Control_plasmid_average_zscore, colour = sgRNA_type, shape = editor)) + geom_jitter(size=3,  position = pos) +
labs(title= "HT29 - zscore L2FC correlation", x = "L2FC_HT29_CBE_Control_plasmid_average", y = "L2FC_HT29_ABE_Control_plasmid_average_zscore") +
geom_smooth(method = lm) +
theme_bw()
plot4
#```

```{r}
# consolidate gRNA scores for gRNAs based on editor

annotated_data <- mutate(annotated_data, L2FC_HT29_DebCet_plasmid_1_zscore = as.numeric(ifelse(editor == "CBE", L2FC_HT29_CBE_DebCet_plasmid_1_zscore, ifelse(editor == "ABE", L2FC_HT29_ABE_DebCet_plasmid_1_zscore, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_HT29_Tram_plasmid_1_zscore = as.numeric(ifelse(editor == "CBE", L2FC_HT29_CBE_Tram_plasmid_1_zscore, ifelse(editor == "ABE", L2FC_HT29_ABE_Tram_plasmid_1_zscore, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_HT29_Pict_plasmid_1_zscore = as.numeric(ifelse(editor == "CBE", L2FC_HT29_CBE_Pict_plasmid_1_zscore, ifelse(editor == "ABE", L2FC_HT29_ABE_Pict_plasmid_1_zscore, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_HT29_Control_plasmid_1_zscore = as.numeric(ifelse(editor == "CBE", L2FC_HT29_CBE_Control_plasmid_1_zscore, ifelse(editor == "ABE", L2FC_HT29_ABE_Control_plasmid_1_zscore, "ERROR"))))

annotated_data <- mutate(annotated_data, L2FC_HT29_DebCet_plasmid_2_zscore = as.numeric(ifelse(editor == "CBE", L2FC_HT29_CBE_DebCet_plasmid_2_zscore, ifelse(editor == "ABE", L2FC_HT29_ABE_DebCet_plasmid_2_zscore, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_HT29_Tram_plasmid_2_zscore = as.numeric(ifelse(editor == "CBE", L2FC_HT29_CBE_Tram_plasmid_2_zscore, ifelse(editor == "ABE", L2FC_HT29_ABE_Tram_plasmid_2_zscore, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_HT29_Pict_plasmid_2_zscore = as.numeric(ifelse(editor == "CBE", L2FC_HT29_CBE_Pict_plasmid_2_zscore, ifelse(editor == "ABE", L2FC_HT29_ABE_Pict_plasmid_2_zscore, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_HT29_Control_plasmid_2_zscore = as.numeric(ifelse(editor == "CBE", L2FC_HT29_CBE_Control_plasmid_2_zscore, ifelse(editor == "ABE", L2FC_HT29_ABE_Control_plasmid_2_zscore, "ERROR"))))

annotated_data <- mutate(annotated_data, L2FC_HT29_DebCet_plasmid_average_zscore = as.numeric(ifelse(editor == "CBE", L2FC_HT29_CBE_DebCet_plasmid_average_zscore, ifelse(editor == "ABE", L2FC_HT29_ABE_DebCet_plasmid_average_zscore, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_HT29_Tram_plasmid_average_zscore = as.numeric(ifelse(editor == "CBE", L2FC_HT29_CBE_Tram_plasmid_average_zscore, ifelse(editor == "ABE", L2FC_HT29_ABE_Tram_plasmid_average_zscore, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_HT29_Pict_plasmid_average_zscore = as.numeric(ifelse(editor == "CBE", L2FC_HT29_CBE_Pict_plasmid_average_zscore, ifelse(editor == "ABE", L2FC_HT29_ABE_Pict_plasmid_average_zscore, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_HT29_Control_plasmid_average_zscore = as.numeric(ifelse(editor == "CBE", L2FC_HT29_CBE_Control_plasmid_average_zscore, ifelse(editor == "ABE", L2FC_HT29_ABE_Control_plasmid_average_zscore, "ERROR"))))

annotated_data <- mutate(annotated_data, L2FC_H23_Adag_plasmid_1_zscore = as.numeric(ifelse(editor == "CBE", L2FC_H23_CBE_Adag_plasmid_1_zscore, ifelse(editor == "ABE", L2FC_H23_ABE_Adag_plasmid_1_zscore, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_H23_Sotor_plasmid_1_zscore = as.numeric(ifelse(editor == "CBE", L2FC_H23_CBE_Sotor_plasmid_1_zscore, ifelse(editor == "ABE", L2FC_H23_ABE_Sotor_plasmid_1_zscore, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_H23_Control_plasmid_1_zscore = as.numeric(ifelse(editor == "CBE", L2FC_H23_CBE_Control_plasmid_1_zscore, ifelse(editor == "ABE", L2FC_H23_ABE_Control_plasmid_1_zscore, "ERROR"))))

annotated_data <- mutate(annotated_data, L2FC_H23_Adag_plasmid_2_zscore = as.numeric(ifelse(editor == "CBE", L2FC_H23_CBE_Adag_plasmid_2_zscore, ifelse(editor == "ABE", L2FC_H23_ABE_Adag_plasmid_2_zscore, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_H23_Sotor_plasmid_2_zscore = as.numeric(ifelse(editor == "CBE", L2FC_H23_CBE_Sotor_plasmid_2_zscore, ifelse(editor == "ABE", L2FC_H23_ABE_Sotor_plasmid_2_zscore, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_H23_Control_plasmid_2 = as.numeric(ifelse(editor == "CBE", L2FC_H23_CBE_Control_plasmid_2, ifelse(editor == "ABE", L2FC_H23_ABE_Control_plasmid_2, "ERROR"))))

annotated_data <- mutate(annotated_data, L2FC_H23_Adag_plasmid_average_zscore = as.numeric(ifelse(editor == "CBE", L2FC_H23_CBE_Adag_plasmid_average_zscore, ifelse(editor == "ABE", L2FC_H23_ABE_Adag_plasmid_average_zscore, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_H23_Sotor_plasmid_average_zscore = as.numeric(ifelse(editor == "CBE", L2FC_H23_CBE_Sotor_plasmid_average_zscore, ifelse(editor == "ABE", L2FC_H23_ABE_Sotor_plasmid_average_zscore, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_H23_Control_plasmid_average_zscore = as.numeric(ifelse(editor == "CBE", L2FC_H23_CBE_Control_plasmid_average_zscore, ifelse(editor == "ABE", L2FC_H23_ABE_Control_plasmid_average_zscore, "ERROR"))))

annotated_data <- mutate(annotated_data, L2FC_PC9_Control_plasmid_1_zscore = as.numeric(ifelse(editor == "CBE", L2FC_PC9_CBE_Control_plasmid_1_zscore, ifelse(editor == "ABE", L2FC_PC9_ABE_Control_plasmid_1_zscore, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_PC9_Osim_plasmid_1_zscore = as.numeric(ifelse(editor == "CBE", L2FC_PC9_CBE_Osim_plasmid_1_zscore, ifelse(editor == "ABE", L2FC_PC9_ABE_Osim_plasmid_1_zscore, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_PC9_Gefit_plasmid_1_zscore = as.numeric(ifelse(editor == "CBE", L2FC_PC9_CBE_Gefit_plasmid_1_zscore, ifelse(editor == "ABE", L2FC_PC9_ABE_Gefit_plasmid_1_zscore, "ERROR"))))

annotated_data <- mutate(annotated_data, L2FC_PC9_Control_plasmid_2_zscore = as.numeric(ifelse(editor == "CBE", L2FC_PC9_CBE_Control_plasmid_2_zscore, ifelse(editor == "ABE", L2FC_PC9_ABE_Control_plasmid_2_zscore, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_PC9_Osim_plasmid_2_zscore = as.numeric(ifelse(editor == "CBE", L2FC_PC9_CBE_Osim_plasmid_2_zscore, ifelse(editor == "ABE", L2FC_PC9_ABE_Osim_plasmid_2_zscore, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_PC9_Gefit_plasmid_2_zscore = as.numeric(ifelse(editor == "CBE", L2FC_PC9_CBE_Gefit_plasmid_2_zscore, ifelse(editor == "ABE", L2FC_PC9_ABE_Gefit_plasmid_2_zscore, "ERROR"))))

annotated_data <- mutate(annotated_data, L2FC_PC9_Control_plasmid_average_zscore = as.numeric(ifelse(editor == "CBE", L2FC_PC9_CBE_Control_plasmid_average_zscore, ifelse(editor == "ABE", L2FC_PC9_ABE_Control_plasmid_average_zscore, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_PC9_Osim_plasmid_average_zscore = as.numeric(ifelse(editor == "CBE", L2FC_PC9_CBE_Osim_plasmid_average_zscore, ifelse(editor == "ABE", L2FC_PC9_ABE_Osim_plasmid_average_zscore, "ERROR"))))
annotated_data <- mutate(annotated_data, L2FC_PC9_Gefit_plasmid_average_zscore = as.numeric(ifelse(editor == "CBE", L2FC_PC9_CBE_Gefit_plasmid_average_zscore, ifelse(editor == "ABE", L2FC_PC9_ABE_Gefit_plasmid_average_zscore, "ERROR"))))

annotated_data <- mutate(annotated_data, L2FC_MHHES1_Olap_plasmid_1_zscore = as.numeric(ifelse(editor == "ABE", L2FC_MHHES1_ABE_Olap_plasmid_1_zscore, "ERROR")))
annotated_data <- mutate(annotated_data, L2FC_MHHES1_Nirap_plasmid_1_zscore = as.numeric(ifelse(editor == "ABE", L2FC_MHHES1_ABE_Nirap_plasmid_1_zscore, "ERROR")))
annotated_data <- mutate(annotated_data, L2FC_MHHES1_Control_plasmid_1_zscore = as.numeric(ifelse(editor == "ABE", L2FC_MHHES1_ABE_Control_plasmid_1_zscore, "ERROR")))

annotated_data <- mutate(annotated_data, L2FC_MHHES1_Olap_plasmid_2_zscore = as.numeric(ifelse(editor == "ABE", L2FC_MHHES1_ABE_Olap_plasmid_2_zscore, "ERROR")))
annotated_data <- mutate(annotated_data, L2FC_MHHES1_Nirap_plasmid_2_zscore = as.numeric(ifelse(editor == "ABE", L2FC_MHHES1_ABE_Nirap_plasmid_2_zscore, "ERROR")))
annotated_data <- mutate(annotated_data, L2FC_MHHES1_Control_plasmid_2_zscore = as.numeric(ifelse(editor == "ABE", L2FC_MHHES1_ABE_Control_plasmid_2_zscore, "ERROR")))

annotated_data <- mutate(annotated_data, L2FC_MHHES1_Olap_plasmid_average_zscore = as.numeric(ifelse(editor == "ABE", L2FC_MHHES1_ABE_Olap_plasmid_average_zscore, "ERROR")))
annotated_data <- mutate(annotated_data, L2FC_MHHES1_Nirap_plasmid_average_zscore = as.numeric(ifelse(editor == "ABE", L2FC_MHHES1_ABE_Nirap_plasmid_average_zscore, "ERROR")))
annotated_data <- mutate(annotated_data, L2FC_MHHES1_Control_plasmid_average_zscore = as.numeric(ifelse(editor == "ABE", L2FC_MHHES1_ABE_Control_plasmid_average_zscore, "ERROR")))

#remove all the old score fields based on CBE or ABE
annotated_data <- annotated_data %>%
  select(!c(L2FC_HT29_CBE_DebCet_plasmid_1, L2FC_HT29_CBE_Tram_plasmid_1, L2FC_HT29_CBE_Control_plasmid_1, L2FC_HT29_CBE_Pict_plasmid_1, L2FC_HT29_ABE_DebCet_plasmid_1, L2FC_HT29_ABE_Tram_plasmid_1, L2FC_HT29_ABE_Control_plasmid_1, L2FC_HT29_ABE_Pict_plasmid_1, L2FC_HT29_CBE_DebCet_plasmid_2, L2FC_HT29_CBE_Tram_plasmid_2, L2FC_HT29_CBE_Control_plasmid_2, L2FC_HT29_CBE_Pict_plasmid_2, L2FC_HT29_ABE_DebCet_plasmid_2, L2FC_HT29_ABE_Tram_plasmid_2, L2FC_HT29_ABE_Control_plasmid_2, L2FC_HT29_ABE_Pict_plasmid_2, L2FC_HT29_CBE_DebCet_plasmid_average, L2FC_HT29_CBE_Tram_plasmid_average, L2FC_HT29_CBE_Control_plasmid_average, L2FC_HT29_CBE_Pict_plasmid_average, L2FC_HT29_ABE_DebCet_plasmid_average, L2FC_HT29_ABE_Tram_plasmid_average, L2FC_HT29_ABE_Control_plasmid_average, L2FC_HT29_ABE_Pict_plasmid_average, L2FC_H23_CBE_Adag_plasmid_1, L2FC_H23_CBE_Sotor_plasmid_1, L2FC_H23_CBE_Control_plasmid_1, L2FC_H23_ABE_Adag_plasmid_1, L2FC_H23_ABE_Sotor_plasmid_1, L2FC_H23_ABE_Control_plasmid_1, L2FC_H23_CBE_Adag_plasmid_2, L2FC_H23_CBE_Sotor_plasmid_2, L2FC_H23_CBE_Control_plasmid_2, L2FC_H23_ABE_Adag_plasmid_2, L2FC_H23_ABE_Sotor_plasmid_2, L2FC_H23_ABE_Control_plasmid_2, L2FC_H23_CBE_Adag_plasmid_average, L2FC_H23_CBE_Sotor_plasmid_average, L2FC_H23_CBE_Control_plasmid_average, L2FC_H23_ABE_Adag_plasmid_average, L2FC_H23_ABE_Sotor_plasmid_average, L2FC_H23_ABE_Control_plasmid_average, L2FC_PC9_CBE_Control_plasmid_1, L2FC_PC9_CBE_Gefit_plasmid_1, L2FC_PC9_CBE_Osim_plasmid_1, L2FC_PC9_ABE_Control_plasmid_1, L2FC_PC9_ABE_Gefit_plasmid_1, L2FC_PC9_ABE_Osim_plasmid_1, L2FC_PC9_CBE_Control_plasmid_2, L2FC_PC9_CBE_Gefit_plasmid_2, L2FC_PC9_CBE_Osim_plasmid_2, L2FC_PC9_ABE_Control_plasmid_2, L2FC_PC9_ABE_Gefit_plasmid_2, L2FC_PC9_ABE_Osim_plasmid_2, L2FC_PC9_CBE_Control_plasmid_average, L2FC_PC9_CBE_Gefit_plasmid_average, L2FC_PC9_CBE_Osim_plasmid_average, L2FC_PC9_ABE_Control_plasmid_average, L2FC_PC9_ABE_Gefit_plasmid_average, L2FC_PC9_ABE_Osim_plasmid_average)) %>%
  distinct()
```

```{r} 
# get annotated_data to classify all non-essential and essential genes as one by making gene_alias
annotated_data %>% count(sgRNA_type)
annotated_data <- mutate(annotated_data, gene_alias = ifelse(sgRNA_type == "non_targeting", "non-targeting", ifelse(sgRNA_type == "splice_essential", "essential", ifelse(sgRNA_type == "splice_nonessential", "non-essential", ifelse(sgRNA_type == "intergenic", "intergenic", Gene)))))
```

```{r}
# Make a simplified amino acid position such that we can plot on a numerical scale
annotated_data <- mutate(annotated_data, Amino_Acid_Position_simple = Amino_Acid_Position)
annotated_data <- mutate(annotated_data, Amino_Acid_Position_simple = as.numeric(str_extract(Amino_Acid_Position_simple, "[[:digit:]]+")))
annotated_data %>% select(Amino_Acid_Position, Amino_Acid_Position_simple)
```

```{r}
# dot plots _ proliferation - HT-29

targets <- c("EGFR", "KRAS", "MYC", "PARP1", "PARP2", "MAP2K1", "MAP2K2", "BCL2", "BRAF", "AKT1", "PIK3CA")

pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

plot1 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=L2FC_HT29_Control_plasmid_average_zscore, y= L2FC_HT29_Tram_plasmid_average_zscore, colour = Gene, shape = editor)) + 
geom_point(position = pos, alpha = 1, size =3) +
labs(title= "", x="HT-29 control vs plasmid (z-score)", y = "HT-29 trametinib vs plasmid (z-score)", colour = "gene") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_vline(xintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_HT29_Tram_plasmid_average_zscore > 4 & (is.na(Amino_Acid_Position) == FALSE) | L2FC_HT29_Control_plasmid_average_zscore > 3 & (is.na(Amino_Acid_Position) == FALSE), paste(Amino_Acid_Position), ifelse( L2FC_HT29_Tram_plasmid_average_zscore > 4 & (is.na(Amino_Acid_Position) == TRUE) & (is.na(most_severe_consequence) == FALSE),  paste(most_severe_consequence), ifelse(L2FC_HT29_Tram_plasmid_average_zscore < -3.5 & L2FC_HT29_Control_plasmid_average_zscore > - 0.25, paste(most_severe_consequence), ifelse(L2FC_HT29_Tram_plasmid_average_zscore > 3 & L2FC_HT29_Control_plasmid_average_zscore < -3 & (is.na(Amino_Acid_Position) == FALSE), paste(Amino_Acid_Position), ""))))), size = 3, position = pos, max.overlaps = Inf) +
scale_color_manual( values =c("royalblue", "turquoise3", "orange", "red", "springgreen4", "grey55", "brown", "purple", "salmon", "pink", "black")) +
scale_shape_manual(values =c(19, 1)) +
theme_classic()
plot1
ggsave("Fig4a.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

plot2 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=L2FC_HT29_Control_plasmid_average_zscore, y= L2FC_HT29_Pict_plasmid_average_zscore, colour = Gene, shape = editor)) + 
geom_point(position = pos, alpha = 1, size =3) +
labs(title= "", x="HT-29 control vs plasmid (z-score)", y = "HT-29 pictilisib vs plasmid (z-score)", colour = "gene") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_vline(xintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_HT29_Pict_plasmid_average_zscore > 4 & (is.na(Amino_Acid_Position) == FALSE) | L2FC_HT29_Control_plasmid_average_zscore > 4 & (is.na(Amino_Acid_Position) == FALSE), paste(Amino_Acid_Position), "")), size = 3, position = pos, max.overlaps = Inf) +
scale_color_manual(values =c("royalblue", "turquoise3", "orange", "red", "springgreen4", "grey55", "brown", "purple", "salmon", "pink", "black")) +
scale_shape_manual(values =c(19, 1)) +
theme_classic()
plot2
ggsave("Fig2a.eps", width = 6, height = 4, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

plot3 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=L2FC_HT29_Control_plasmid_average_zscore, y= L2FC_HT29_DebCet_plasmid_average_zscore, colour = Gene, shape = editor)) + 
geom_point(position = pos, alpha = 1, size =3) +
labs(title= "", x="HT-29 control vs plasmid (z-score)", y = "HT-29 dabrafenib + cetuximab vs plasmid (z-score)", colour = "gene") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_vline(xintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_HT29_DebCet_plasmid_average_zscore > 5 & (is.na(Amino_Acid_Position) == FALSE) | L2FC_HT29_Control_plasmid_average_zscore > 3 & (is.na(Amino_Acid_Position) == FALSE), paste(Amino_Acid_Position), ifelse(L2FC_HT29_DebCet_plasmid_average_zscore > 3 & (is.na(Amino_Acid_Position) == TRUE) & (is.na(most_severe_consequence) == FALSE) | L2FC_HT29_Control_plasmid_average_zscore > 3 & (is.na(Amino_Acid_Position) == TRUE) & (is.na(most_severe_consequence) == FALSE), paste(most_severe_consequence), ""))), size = 3, position = pos, max.overlaps = Inf) +
scale_color_manual(values =c("royalblue", "turquoise3", "orange", "red", "springgreen4", "grey55", "brown", "purple", "salmon", "pink", "black")) +
scale_shape_manual(values =c(19, 1)) +
theme_classic()
plot3
ggsave("Fig4b.eps", width = 6, height = 4.3, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")
```

```{r}
# dot plots _ proliferation - H23

targets <- c("EGFR", "KRAS", "MYC", "PARP1", "PARP2", "MAP2K1", "MAP2K2", "BCL2", "BRAF", "AKT1", "PIK3CA")

pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

plot1 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=L2FC_H23_Control_plasmid_average_zscore, y= L2FC_H23_Sotor_plasmid_average_zscore, colour = Gene, shape = editor)) + 
geom_point(position = pos, alpha = 0.80, size =3) +
labs(title= "", x="L2FC_H23_Control_plasmid_1", y = "L2FC_H23_Sotor_plasmid_1") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_H23_Sotor_plasmid_average_zscore > 2 , paste(Amino_Acid_Position, ",", Gene), "")), size = 3, position = pos) +
#facet_wrap(~Gene, scales ="free") +
scale_color_manual(values =c("black", "springgreen4", "orange", "red", "blue", "grey55", "brown", "purple", "pink", "salmon", "royalblue")) +
scale_shape_manual(values =c(19, 1)) +
theme_classic()
plot1

plot2 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=L2FC_H23_Control_plasmid_average_zscore, y= L2FC_H23_Adag_plasmid_average_zscore, colour = Gene, shape = editor)) + 
geom_point(position = pos, alpha = 0.80, size =3) +
labs(title= "", x="L2FC_H23_Control_plasmid_1", y = "L2FC_H23_Adag_plasmid_1") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_H23_Adag_plasmid_average_zscore > 2.5 , paste(Amino_Acid_Position, ",", Gene), "")), size = 3, position = pos) +
scale_color_manual(values =c("black", "springgreen4", "orange", "red", "blue", "grey55", "brown", "purple", "pink", "salmon", "royalblue")) +
scale_shape_manual(values =c(19, 1)) +
theme_classic()
plot2
```

```{r}
# dot plots _ proliferation - H23 Adag vs Sotor

targets <- c("EGFR", "KRAS", "MYC", "PARP1", "PARP2", "MAP2K1", "MAP2K2", "BCL2", "BRAF", "AKT1", "PIK3CA")

pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

plot1 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=L2FC_H23_Adag_plasmid_1_zscore, y= L2FC_H23_Sotor_plasmid_1_zscore, colour = Gene, shape = editor)) + 
geom_point(position = pos, alpha = 0.90, size = 3) +
labs(title= "H23 - Sotorasib vs Adagrasib", x="L2FC Adagrasib vs plasmid", y = "L2FC Sotorasib vs plasmid", colour = "gene") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_vline(xintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_H23_Adag_plasmid_1_zscore > 2.2 & (is.na(Amino_Acid_Position) == FALSE)| L2FC_H23_Sotor_plasmid_1_zscore > 2 & (is.na(Amino_Acid_Position) == FALSE), paste(Amino_Acid_Position, Gene), ifelse(L2FC_H23_Adag_plasmid_1_zscore > 2.2 & (is.na(Amino_Acid_Position) == TRUE)| L2FC_H23_Sotor_plasmid_1_zscore > 2 & (is.na(Amino_Acid_Position) == TRUE), paste(most_severe_consequence, Gene), ""))), size = 3, position = pos) +
#facet_wrap(~Gene, scales ="free") +
scale_color_manual(values =c("black", "springgreen4", "orange", "red", "blue", "grey55", "brown", "purple", "pink", "salmon", "royalblue")) +
scale_shape_manual(values =c(19, 1)) +
theme_classic()
plot1

plot2 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=L2FC_H23_Adag_plasmid_2_zscore, y= L2FC_H23_Sotor_plasmid_2_zscore, colour = Gene, shape = editor)) + 
geom_point(position = pos, alpha = 0.90, size = 3) +
labs(title= "H23 - Sotorasib vs Adagrasib", x="L2FC Adagrasib vs plasmid", y = "L2FC Sotorasib vs plasmid", colour = "gene") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_vline(xintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_H23_Adag_plasmid_2_zscore > 2.2 & (is.na(Amino_Acid_Position) == FALSE)| L2FC_H23_Sotor_plasmid_2_zscore > 2 & (is.na(Amino_Acid_Position) == FALSE), paste(Amino_Acid_Position, Gene), ifelse(L2FC_H23_Adag_plasmid_2_zscore > 2.2 & (is.na(Amino_Acid_Position) == TRUE)| L2FC_H23_Sotor_plasmid_2_zscore > 2 & (is.na(Amino_Acid_Position) == TRUE), paste(most_severe_consequence, Gene), ""))), size = 3, position = pos) +
#facet_wrap(~Gene, scales ="free") +
scale_color_manual(values =c("black", "springgreen4", "orange", "red", "blue", "grey55", "brown", "purple", "pink", "salmon", "royalblue")) +
scale_shape_manual(values =c(19, 1)) +
theme_classic()
plot2

plot3 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=L2FC_H23_Control_plasmid_average_zscore, y= L2FC_H23_Sotor_plasmid_average_zscore, colour = Gene, shape = editor)) + 
geom_point(position = pos, alpha = 0.90, size = 3) +
labs(title= "H23 - Sotorasib vs plasmid", x="L2FC Control vs plasmid", y = "L2FC Sotorasib vs plasmid", colour = "gene") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_vline(xintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_H23_Sotor_plasmid_average_zscore > 2 & (is.na(Amino_Acid_Position) == FALSE)| L2FC_H23_Sotor_plasmid_average_zscore > 2 & (is.na(Amino_Acid_Position) == FALSE), paste(Amino_Acid_Position, Gene), ifelse(guide == "TGTCAGCAGGACCACCACAG", paste(Amino_Acid_Position, Gene), ""))), size = 3, position = pos) +
#facet_wrap(~Gene, scales ="free") +
scale_color_manual(values =c("black", "springgreen4", "orange", "red", "blue", "grey55", "brown", "purple", "pink", "salmon", "royalblue")) +
scale_shape_manual(values =c(19, 1)) +
theme_classic()
plot3

plot4 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=L2FC_H23_Control_plasmid_average_zscore, y= L2FC_H23_Adag_plasmid_average_zscore, colour = Gene, shape = editor)) + 
geom_point(position = pos, alpha = 0.90, size = 3) +
labs(title= "H23 - Adagrasib vs plasmid", x="L2FC Control vs plasmid", y = "L2FC Adagrasib vs plasmid", colour = "gene") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_vline(xintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_H23_Adag_plasmid_average_zscore > 2 & (is.na(Amino_Acid_Position) == FALSE)| L2FC_H23_Adag_plasmid_average_zscore > 2 & (is.na(Amino_Acid_Position) == FALSE), paste(Amino_Acid_Position, Gene), ifelse(guide == "TGTCAGCAGGACCACCACAG", paste(Amino_Acid_Position, Gene), ""))), size = 3, position = pos) +
#facet_wrap(~Gene, scales ="free") +
scale_color_manual(values =c("black", "springgreen4", "orange", "red", "blue", "grey55", "brown", "purple", "pink", "salmon", "royalblue")) +
scale_shape_manual(values =c(19, 1)) +
theme_classic()
plot4

plot5 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=L2FC_H23_Adag_plasmid_average_zscore, y= L2FC_H23_Sotor_plasmid_average_zscore, colour = Gene, shape = editor)) + 
geom_point(position = pos, alpha = 1, size = 3) +
labs(title= "", x="H23 adagrasib vs plasmid (z-score)", y = "H23 sotorasib vs plasmid (z-score)", colour = "gene") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_vline(xintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_H23_Adag_plasmid_average_zscore > 4.2 & (is.na(Amino_Acid_Position) == FALSE)| L2FC_H23_Sotor_plasmid_average_zscore > 4.2 & (is.na(Amino_Acid_Position) == FALSE), paste(Amino_Acid_Position), ifelse(L2FC_H23_Adag_plasmid_average_zscore > 4.2 & (is.na(Amino_Acid_Position) == TRUE)| L2FC_H23_Sotor_plasmid_average_zscore > 4 & (is.na(Amino_Acid_Position) == TRUE), paste(most_severe_consequence),""))), size = 3, position = pos, max.overlaps = Inf) +
scale_color_manual( values =c("royalblue", "turquoise3", "orange", "red", "springgreen4", "grey55", "brown", "purple", "salmon", "pink", "black")) +
scale_shape_manual(values =c(19, 1)) +
theme_classic()
plot5
ggsave("Fig2b.eps", width = 6, height = 4, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")
```

```{r}
# check RAS outlier data point

annotated_data %>% 
  filter(L2FC_H23_Adag_plasmid_average_zscore < -2 & L2FC_H23_Sotor_plasmid_average_zscore > -0.1) %>%
  select(Gene, guide, editor, Amino_Acid_Position, Protein_Change, most_severe_consequence, plasmid_reads, H23_ABE_Control_exp1_reads, H23_ABE_Control_exp2_reads, H23_ABE_Adag_exp1_reads, H23_ABE_Adag_exp2_reads, H23_ABE_Sotor_exp1_reads, H23_ABE_Sotor_exp2_reads, L2FC_H23_Adag_plasmid_1_zscore, L2FC_H23_Adag_plasmid_2_zscore, L2FC_H23_Sotor_plasmid_1_zscore, L2FC_H23_Sotor_plasmid_2_zscore)

```

```{r}
# dot plots _ proliferation - PC9 average

targets <- c("EGFR", "KRAS", "MYC", "PARP1", "PARP2", "MAP2K1", "MAP2K2", "BCL2", "BRAF", "AKT1", "PIK3CA")

pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

plot1 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=L2FC_PC9_Control_plasmid_average_zscore, y= L2FC_PC9_Gefit_plasmid_average_zscore, colour = Gene, shape = editor)) + 
geom_point(position = pos, alpha = 1, size =3) +
labs(title= " ", x="PC9 control vs plasmid (z-score)", y = "PC9 gefitinib vs plasmid (z-score)", colour = "gene") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_vline(xintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_PC9_Gefit_plasmid_average_zscore > 12 & (is.na(Amino_Acid_Position) == FALSE), paste(Amino_Acid_Position), ifelse(L2FC_PC9_Gefit_plasmid_average_zscore < -3.5 & L2FC_PC9_Control_plasmid_average_zscore > -0.5 & (is.na(Amino_Acid_Position) == FALSE), paste(Amino_Acid_Position, most_severe_consequence), ifelse(L2FC_PC9_Gefit_plasmid_average_zscore > 12 & (is.na(Amino_Acid_Position) == TRUE) | L2FC_PC9_Gefit_plasmid_average_zscore < -3.5 & L2FC_PC9_Control_plasmid_average_zscore > -0.5 & (is.na(Amino_Acid_Position) == TRUE), paste(most_severe_consequence), "")))), size = 3, position = pos, max.overlaps = Inf) +
scale_color_manual( values =c("royalblue", "turquoise3", "orange", "red", "springgreen4", "grey55", "brown", "purple", "salmon", "pink", "black")) +
scale_shape_manual(values =c(19, 1)) +
theme_classic()
plot1
ggsave("Fig2c.eps", width = 6, height = 4, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

plot2 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=L2FC_PC9_Control_plasmid_average_zscore, y= L2FC_PC9_Osim_plasmid_average_zscore, colour = Gene, shape = editor)) + 
geom_point(position = pos, alpha = 1, size =3) +
labs(title= " ", x="PC9 control vs plasmid (z-score)", y = "PC9 osimertinib vs plasmid (z-score)", colour = "gene") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_vline(xintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_PC9_Osim_plasmid_average_zscore > 5.5 & (is.na(Amino_Acid_Position) == FALSE), paste(Amino_Acid_Position), ifelse(L2FC_PC9_Osim_plasmid_average_zscore < -5 & L2FC_PC9_Control_plasmid_average_zscore > -1 & (is.na(Amino_Acid_Position) == FALSE), paste(Amino_Acid_Position, most_severe_consequence), ifelse(L2FC_PC9_Osim_plasmid_average_zscore > 5.5 & (is.na(Amino_Acid_Position) == TRUE) | L2FC_PC9_Osim_plasmid_average_zscore < -5 & L2FC_PC9_Control_plasmid_average_zscore > -1 & (is.na(Amino_Acid_Position) == TRUE), paste(most_severe_consequence), "")))), size = 3, position = pos, max.overlaps = Inf) +
scale_color_manual( values =c("royalblue", "turquoise3", "orange", "red", "springgreen4", "grey55", "brown", "purple", "salmon", "pink", "black")) +
scale_shape_manual(values =c(19, 1)) +
theme_classic()
plot2
ggsave("Fig2d.eps", width = 6, height = 4, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")
```

```{r}
# investigate BCL2 driver mutation outlier - does not correlate between replicates
# impose a 10 fold change difference between reads like in Cancer Cell paper - like a 50 fold difference

annotated_data %>% 
  filter(L2FC_MHHES1_Control_plasmid_average_zscore >3) %>%
  select(Gene, guide, MHHES1_ABE_Control_exp1_reads, MHHES1_ABE_Control_exp2_reads)

annotated_data %>% 
  filter(MHHES1_ABE_Control_exp1_reads > 10*MHHES1_ABE_Control_exp2_reads & MHHES1_ABE_Control_exp2_reads != 0 | MHHES1_ABE_Control_exp2_reads > 10*MHHES1_ABE_Control_exp1_reads & MHHES1_ABE_Control_exp1_reads != 0) %>%
  select(Gene, guide, MHHES1_ABE_Control_exp1_reads, MHHES1_ABE_Control_exp2_reads, MHHES1_ABE_T0_exp1_reads, MHHES1_ABE_T0_exp2_reads) 

outliers <- annotated_data %>% 
  filter(MHHES1_ABE_Control_exp1_reads > 10*MHHES1_ABE_Control_exp2_reads & MHHES1_ABE_Control_exp2_reads != 0 | MHHES1_ABE_Control_exp2_reads > 10*MHHES1_ABE_Control_exp1_reads & MHHES1_ABE_Control_exp1_reads != 0) %>%
  select(sgRNA_ID, Gene, guide, MHHES1_ABE_Control_exp1_reads, MHHES1_ABE_Control_exp2_reads, MHHES1_ABE_T0_exp1_reads, MHHES1_ABE_T0_exp2_reads) %>%
  select(sgRNA_ID)

test_clean_MHHES1_data <- annotated_data %>%
  filter(sgRNA_ID %in% c(outliers$sgRNA_ID) == FALSE)
  
annotated_data %>% 
  filter(MHHES1_ABE_Olap_exp1_reads > 10*MHHES1_ABE_Olap_exp2_reads & MHHES1_ABE_Olap_exp2_reads != 0 | MHHES1_ABE_Control_exp2_reads > 10*MHHES1_ABE_Control_exp1_reads & MHHES1_ABE_Control_exp1_reads != 0) %>%
  select(Gene, guide, MHHES1_ABE_Olap_exp1_reads, MHHES1_ABE_Olap_exp2_reads, MHHES1_ABE_T0_exp1_reads, MHHES1_ABE_T0_exp2_reads)

test_clean_MHHES1_data %>% 
  filter(MHHES1_ABE_Olap_exp1_reads > 10*MHHES1_ABE_Olap_exp2_reads & MHHES1_ABE_Olap_exp2_reads != 0 | MHHES1_ABE_Control_exp2_reads > 10*MHHES1_ABE_Control_exp1_reads & MHHES1_ABE_Control_exp1_reads != 0) %>%
  select(Gene, guide, MHHES1_ABE_Olap_exp1_reads, MHHES1_ABE_Olap_exp2_reads, MHHES1_ABE_T0_exp1_reads, MHHES1_ABE_T0_exp2_reads)

annotated_data %>% 
  filter(MHHES1_ABE_Nirap_exp1_reads > 10*MHHES1_ABE_Nirap_exp2_reads & MHHES1_ABE_Nirap_exp2_reads != 0 | MHHES1_ABE_Nirap_exp2_reads > 10*MHHES1_ABE_Control_exp1_reads & MHHES1_ABE_Control_exp1_reads != 0) %>%
  select(Gene, guide, MHHES1_ABE_Nirap_exp1_reads, MHHES1_ABE_Nirap_exp2_reads, MHHES1_ABE_T0_exp1_reads, MHHES1_ABE_T0_exp2_reads)

test_clean_MHHES1_data %>% 
  filter(MHHES1_ABE_Nirap_exp1_reads > 10*MHHES1_ABE_Nirap_exp2_reads & MHHES1_ABE_Nirap_exp2_reads != 0 | MHHES1_ABE_Nirap_exp2_reads > 10*MHHES1_ABE_Control_exp1_reads & MHHES1_ABE_Control_exp1_reads != 0) %>%
  select(Gene, guide, MHHES1_ABE_Nirap_exp1_reads, MHHES1_ABE_Nirap_exp2_reads, MHHES1_ABE_T0_exp1_reads, MHHES1_ABE_T0_exp2_reads)

#check hit read counts
test_clean_MHHES1_data %>% 
  filter(L2FC_MHHES1_Control_plasmid_average_zscore >3 | L2FC_MHHES1_Olap_plasmid_average_zscore > 3 | L2FC_MHHES1_Nirap_plasmid_average_zscore > 3) %>%
  select(sgRNA_ID, Gene, guide, MHHES1_ABE_Control_exp1_reads, MHHES1_ABE_Control_exp2_reads, MHHES1_ABE_Olap_exp1_reads, MHHES1_ABE_Olap_exp2_reads, MHHES1_ABE_Nirap_exp1_reads, MHHES1_ABE_Nirap_exp2_reads)
```
```{r}
# investigate PARP1/2 hits

test_clean_MHHES1_data %>% filter(L2FC_MHHES1_Control_plasmid_average_zscore >3) %>%
  select(Gene, G_guide, most_severe_consequence, Amino_Acid_Position, editor, Edit_Location)

test_clean_MHHES1_data %>% filter(L2FC_MHHES1_Olap_plasmid_average_zscore >3) %>%
  select(Gene, G_guide, most_severe_consequence, Amino_Acid_Position, editor, Protein_Change)

test_clean_MHHES1_data %>% filter(L2FC_MHHES1_Nirap_plasmid_average_zscore >3) %>%
  select(Gene, G_guide, most_severe_consequence, Amino_Acid_Position, editor, Protein_Change)

test_clean_MHHES1_data %>% filter(L2FC_MHHES1_Olap_plasmid_average_zscore < -3 & L2FC_MHHES1_Control_plasmid_average_zscore > -1) %>%
  select(Gene, G_guide, most_severe_consequence, Amino_Acid_Position, editor, Protein_Change)

test_clean_MHHES1_data %>% filter(L2FC_MHHES1_Nirap_plasmid_average_zscore < -3 & L2FC_MHHES1_Control_plasmid_average_zscore > -1) %>%
  select(Gene, G_guide, most_severe_consequence, Amino_Acid_Position, editor, Protein_Change)

test_clean_MHHES1_data %>% filter(L2FC_MHHES1_Olap_plasmid_average_zscore < -3 & L2FC_MHHES1_Nirap_plasmid_average_zscore < -3 & L2FC_MHHES1_Control_plasmid_average_zscore > -0.8) %>%
  select(Gene, Amino_Acid_Position, editor, sgRNA_ID, L2FC_MHHES1_Olap_plasmid_average_zscore, L2FC_MHHES1_Nirap_plasmid_average_zscore, L2FC_MHHES1_Control_plasmid_average_zscore)

PARP_sensitors <- test_clean_MHHES1_data %>% filter(L2FC_MHHES1_Olap_plasmid_average_zscore < -3 & L2FC_MHHES1_Nirap_plasmid_average_zscore < -3 & L2FC_MHHES1_Control_plasmid_average_zscore > -1) %>% 
  select(sgRNA_ID)

PARP_resistors <- test_clean_MHHES1_data %>% filter(L2FC_MHHES1_Olap_plasmid_average_zscore > 3 & L2FC_MHHES1_Olap_plasmid_1_zscore >2 & L2FC_MHHES1_Olap_plasmid_2_zscore > 2 | L2FC_MHHES1_Nirap_plasmid_average_zscore > 3 & L2FC_MHHES1_Nirap_plasmid_1_zscore >2 & L2FC_MHHES1_Nirap_plasmid_2_zscore >2) %>%
  select(sgRNA_ID)

test_clean_MHHES1_data %>% filter(L2FC_MHHES1_Olap_plasmid_average_zscore > 3 | L2FC_MHHES1_Nirap_plasmid_average_zscore > 3) %>%
  select(Gene, most_severe_consequence, Amino_Acid_Position, Protein_Change)

```


```{r}
# dot plots _ proliferation - MHH-ES-1

targets <- c("EGFR", "KRAS", "MYC", "PARP1", "PARP2", "MAP2K1", "MAP2K2", "BCL2", "BRAF", "AKT1", "PIK3CA")

pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

plot1 <- ggplot(subset(test_clean_MHHES1_data, Gene %in% c(targets)), aes(x=L2FC_MHHES1_Control_plasmid_average_zscore, y= L2FC_MHHES1_Olap_plasmid_average_zscore, colour = Gene, shape = editor)) + 
geom_point(position = pos, alpha = 1, size =3) +
labs(title= "", x="MHH-ES-1 control vs plasmid (z-score)", y = "MHH-ES-1 olaparib vs plasmid (z-score)", colour = "gene") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_vline(xintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_MHHES1_Olap_plasmid_average_zscore > 3.1 & (is.na(Amino_Acid_Position) == FALSE), paste(Amino_Acid_Position), ifelse(L2FC_MHHES1_Olap_plasmid_1_zscore > 3.1 & (is.na(Amino_Acid_Position) == TRUE),  paste(most_severe_consequence, Gene), ifelse(L2FC_MHHES1_Control_plasmid_average_zscore > 3.5, paste(most_severe_consequence, Gene), ifelse(sgRNA_ID %in% PARP_sensitors$sgRNA_ID, paste(Gene, Amino_Acid_Position), ""))))), size = 3, position = pos, max.overlaps = Inf) +
scale_color_manual( values =c("royalblue", "turquoise3", "orange", "red", "springgreen4", "grey55", "brown", "purple", "salmon", "pink", "black")) +
scale_shape_manual(values =c(19, 1)) +
theme_classic() 
plot1
ggsave("Fig2c1.eps", width = 6, height = 4, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

plot2 <- ggplot(subset(test_clean_MHHES1_data, Gene %in% c(targets)), aes(x=L2FC_MHHES1_Control_plasmid_average_zscore, y= L2FC_MHHES1_Nirap_plasmid_average_zscore, colour = Gene, shape = editor)) + 
geom_point(position = pos, alpha = 1, size =3) +
labs(title= "", x="MHH-ES-1 control vs plasmid (z-score)", y = "MHH-ES-1 niraparib vs plasmid (z-score)", colour = "gene") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_vline(xintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_MHHES1_Nirap_plasmid_average_zscore > 4 & (is.na(Amino_Acid_Position) == FALSE), paste(Amino_Acid_Position), ifelse( L2FC_MHHES1_Nirap_plasmid_1_zscore > 4 & (is.na(Amino_Acid_Position) == TRUE),  paste(most_severe_consequence, Gene), ifelse(L2FC_MHHES1_Control_plasmid_average_zscore > 4, paste(most_severe_consequence, Gene), ifelse(sgRNA_ID %in% PARP_sensitors$sgRNA_ID, paste(Gene, Amino_Acid_Position), ""))))), size = 3, position = pos, max.overlaps = Inf) +
scale_color_manual( values =c("royalblue", "turquoise3", "orange", "red", "springgreen4", "grey55", "brown", "purple", "salmon", "pink", "black")) +
scale_shape_manual(values =c(19, 1)) +
theme_classic()
plot2
ggsave("Fig2c2.eps", width = 6, height = 4, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")
```

```{r}
# dot plots _ proliferation

targets <- c("EGFR", "KRAS", "MYC", "PARP1", "PARP2", "MAP2K1", "MAP2K2", "BCL2", "BRAF", "AKT1", "PIK3CA")

pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

plot1 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=Amino_Acid_Position_simple, y= L2FC_HT29_Control_plasmid_average_zscore, colour = most_severe_consequence)) + 
geom_point(position = pos, shape = 16, alpha = 0.8, size =2) +
labs(title= "", x="amino acid position", y = "L2FC_HT29_Control_plasmid_average") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
facet_wrap(~Gene, scales = "free_x") +
scale_color_manual(values =c("grey65", "black", "black", "orange", "blue", "red", "purple", "grey65", "grey65", "grey65", "grey65")) +
theme_classic()
plot1

plot2 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=Amino_Acid_Position_simple, y= L2FC_H23_Control_plasmid_1_zscore, colour = most_severe_consequence)) + 
geom_point(position = pos, shape = 16, alpha = 0.8, size =2) +
labs(title= "", x="amino acid position", y = "L2FC_H23_Control_plasmid_1") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
facet_wrap(~Gene, scales = "free_x") +
scale_color_manual(values =c("grey65", "black", "black", "orange", "blue", "red", "purple", "grey65", "grey65", "grey65", "grey65")) +
theme_classic()
plot2

plot3 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=Amino_Acid_Position_simple, y= L2FC_PC9_Control_plasmid_1_zscore, colour = most_severe_consequence)) + 
geom_point(position = pos, shape = 16, alpha = 0.8, size =2) +
labs(title= "", x="amino acid position", y = "L2FC_PC9_Control_plasmid_1") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
facet_wrap(~Gene, scales = "free_x") +
scale_color_manual(values =c("grey65", "black", "black", "orange", "blue", "red", "purple", "grey65", "grey65", "grey65", "grey65")) +
theme_classic()
plot3

plot4 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=Amino_Acid_Position_simple, y= L2FC_MHHES1_Control_plasmid_1_zscore, colour = most_severe_consequence)) + 
geom_point(position = pos, shape = 16, alpha = 0.8, size =2) +
labs(title= "", x="amino acid position", y = "L2FC_MHHES1_Control_plasmid_1") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
facet_wrap(~Gene, scales = "free_x") +
scale_color_manual(values =c("grey65", "black", "black", "orange", "blue", "red", "purple", "grey65", "grey65", "grey65", "grey65")) +
theme_classic()
plot4
```

```{r}
# dot plots _ proliferation - gene essentially

targets <- c("KRAS", "MYC", "BRAF")

pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

plot1 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=L2FC_HT29_Control_plasmid_average_zscore, y= L2FC_H23_Control_plasmid_average_zscore, colour = Gene)) + 
geom_point(position = pos, shape = 16, alpha = 0.8, size =3) +
labs(title= "", x="L2FC_HT29_Control_plasmid_average", y = "L2FC_H23_Control_plasmid_average") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
scale_color_manual(values =c("red", "orange", "black")) +
theme_bw()
plot1

targets <- c("EGFR", "MYC", "BRAF")

plot2 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=L2FC_HT29_Control_plasmid_average_zscore, y= L2FC_PC9_Control_plasmid_average_zscore, colour = Gene)) + 
geom_point(position = pos, shape = 16, alpha = 1, size =3) +
labs(title= "", x="HT-29 control vs plasmid (z-score)", y = "PC9 control vs plasmid (z-score)", colour = "gene") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_vline(xintercept=c(0,0),linetype="dotted") +
scale_color_manual(values =c("red", "orange", "black")) +
theme_classic()
plot2
ggsave("Fig1c.eps", width = 5, height = 4, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

targets <- c("EGFR", "MYC", "KRAS")

plot3 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=L2FC_PC9_Control_plasmid_average_zscore, y= L2FC_H23_Control_plasmid_average_zscore, colour = Gene)) + 
geom_point(position = pos, shape = 16, alpha = 0.8, size =3) +
labs(title= "", x="L2FC_PC9_Control_plasmid_average", y = "L2FC_H23_Control_plasmid_average") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
scale_color_manual(values =c("red", "orange", "black")) +
theme_bw()
plot3

targets <- c("BRAF", "MYC", "PARP1")

plot4 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=L2FC_HT29_Control_plasmid_average_zscore, y= L2FC_MHHES1_Control_plasmid_average_zscore, colour = Gene)) + 
geom_point(position = pos, shape = 16, alpha = 0.8, size =3) +
labs(title= "", x="L2FC_HT29_Control_plasmid_average", y = "L2FC_MHHES1_Control_plasmid_average") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
scale_color_manual(values =c("red", "orange", "black")) +
theme_bw()
plot4
```

```{r}
# read in COSMIC counts

COSMIC <- as_tibble(read.csv("COSMIC_11_230822.csv", stringsAsFactors = FALSE, header = TRUE))
COSMIC <- rename(COSMIC, Amino_Acid_Position = Position)
COSMIC <- rename(COSMIC, COSMIC_count = Count)
COSMIC <- rename(COSMIC, Amino_Acid_Change_COSMIC = AA.Mutation)
COSMIC <- rename(COSMIC, CDS_mutation_COSMIC = CDS.Mutation)

COSMIC <- mutate(COSMIC, Amino_Acid_Position = as.character(Amino_Acid_Position))

COSMIC2 <- select(COSMIC, Gene, Amino_Acid_Position, COSMIC_count)
COSMIC2 <- COSMIC2 %>%
  group_by(Gene, Amino_Acid_Position) %>%
  summarise(COSMIC_count = sum(COSMIC_count)) %>% distinct()
COSMIC2
annotated_data_cosmic <- left_join(annotated_data, COSMIC2) %>% distinct()
annotated_data_cosmic %>% filter(Gene == "AKT1" & Amino_Acid_Position =="17")
annotated_data_cosmic %>% filter(COSMIC_count > 100) %>% select(Gene, Amino_Acid_Position, COSMIC_count)
annotated_data_cosmic <- annotated_data_cosmic %>% 
  mutate(COSMIC_count = ifelse(is.na(COSMIC_count) == TRUE, "0", COSMIC_count)) %>%
  mutate(COSMIC_count = as.numeric(COSMIC_count))
```

```{r}
# dot plots _ proliferation - HT-29 COSMIC graph

targets <- c("EGFR", "KRAS", "MYC", "PARP1", "PARP2", "MAP2K1", "MAP2K2", "BCL2", "BRAF", "AKT1", "PIK3CA")

pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

plot1 <- ggplot(subset(annotated_data_cosmic, Gene == "AKT1"), aes(x=Amino_Acid_Position_simple, y= L2FC_HT29_Control_plasmid_average_zscore, size = COSMIC_count)) + 
geom_point(position = pos, shape = 1, alpha = 1) +
labs(title = "", x="AKT1 amino acid position", y = "HT-29 control vs plasmid (z-score)", size = "COSMIC count") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_HT29_Control_plasmid_average_zscore > 2 & (is.na(Amino_Acid_Position) == FALSE), paste(Amino_Acid_Position),  "")), position = pos, size = 4) +
scale_size(range = c(3, 10)) +
theme_classic()
plot1
ggsave("Fig1e.eps", width = 6, height = 4.1, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

plot2 <- ggplot(subset(annotated_data_cosmic, Gene == "PIK3CA"), aes(x=Amino_Acid_Position_simple, y= L2FC_HT29_Control_plasmid_average_zscore, size = COSMIC_count)) + 
geom_point(position = pos, shape = 1, alpha = 1) +
labs(title = "", x="PIK3CA amino acid position", y = "HT-29 control vs plasmid (z-score)", size = "COSMIC count") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_HT29_Control_plasmid_average_zscore > 2 & (is.na(Amino_Acid_Position) == FALSE), paste(Amino_Acid_Position),  "")), position = pos, size = 4) +
scale_size(range = c(3, 10)) +
theme_classic()
plot2
ggsave("Fig1f.eps", width = 6, height = 4.1, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")
```

```{r}
# mutate BRAF curated_Domain field to simplify for figure
annotated_data %>% filter(Gene == "BRAF") %>%
  select(Gene, curated_Domain) %>%
  count(curated_Domain)

annotated_data <- mutate(annotated_data, curated_Domain = ifelse(curated_Domain == "ATP binding site;Protein kinase" & Gene == "BRAF", "ATP binding site", ifelse(curated_Domain == "Protein kinase;ATP binding site", "ATP binding site", ifelse(curated_Domain == "Breakpoint for translocation to form KIAA1549-BRAF fusion protein;Disordered", "Disordered", ifelse(curated_Domain == "Breakpoint for translocation to form KIAA1549-BRAF fusion protein", "", curated_Domain)))))

annotated_data %>% filter(Gene == "BRAF") %>%
  select(Gene, curated_Domain) %>%
  count(curated_Domain)
```

```{r}
# dot plots _ proliferation - BRAF
targets <- c("BRAF")
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

plot1 <- ggplot(subset(annotated_data, Gene %in% c(targets) & (is.na(Amino_Acid_Position_simple) ==FALSE)), aes(x=Amino_Acid_Position_simple, y= L2FC_HT29_Control_plasmid_average_zscore, shape = most_severe_consequence, colour = curated_Domain)) + 
geom_point(position = pos, alpha = 1, size = 3) +
labs(title= "", x="BRAF amino acid position", y = "HT-29 control vs plasmid (z-score)", colour = "domain", shape = "consequence") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_HT29_Control_plasmid_average_zscore < -3 & most_severe_consequence == "missense", paste(PTM), ifelse(L2FC_HT29_Control_plasmid_average_zscore > 2, paste(Amino_Acid_Position), ""))), size = 4, position = pos, max.overlaps = Inf) +
scale_color_manual(values =c("grey55", "purple", "black", "royalblue", "orange", "red")) +
scale_shape_manual(values=c(19, 2, 11, 7, 3)) +
theme_classic()
plot1
ggsave("Fig1d.eps", width = 6, height = 4, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")
```

```{r}
# BRAF mutation investigate

annotated_data %>% filter(L2FC_HT29_Control_plasmid_average_zscore > 2 & Gene =="BRAF") %>% select(Amino_Acid_Position, Protein_Change, most_severe_consequence, L2FC_HT29_Control_plasmid_average_zscore, Edit_Location)

```

```{r}
# figures for domains in BRAF with LoF 

annotated_data %>%
  filter(Gene == "BRAF") %>%
  filter(most_severe_consequence =="missense") %>%
  filter(L2FC_HT29_Control_plasmid_average_zscore < -2) %>%
  select(Gene, most_severe_consequence, L2FC_HT29_Control_plasmid_average_zscore, curated_Domain) %>%
  distinct()

total<-annotated_data %>%
  filter(Gene == "BRAF") %>%
  filter(most_severe_consequence =="missense") %>%
  filter(L2FC_HT29_Control_plasmid_average_zscore < -2) %>%
  select(Gene, most_severe_consequence, L2FC_HT29_Control_plasmid_average_zscore, curated_Domain) %>%
  distinct() %>%
  count()

RBD<-annotated_data %>%
  filter(Gene == "BRAF") %>%
  filter(most_severe_consequence =="missense") %>%
  filter(L2FC_HT29_Control_plasmid_average_zscore < -2) %>%
  select(Gene, most_severe_consequence, L2FC_HT29_Control_plasmid_average_zscore, curated_Domain) %>%
  filter(curated_Domain =="RBD") %>%
  distinct() %>% 
  count()

kinase<-annotated_data %>%
  filter(Gene == "BRAF") %>%
  filter(most_severe_consequence =="missense") %>%
  filter(L2FC_HT29_Control_plasmid_average_zscore < -2) %>%
  select(Gene, most_severe_consequence, L2FC_HT29_Control_plasmid_average_zscore, curated_Domain) %>%
  filter(curated_Domain =="Protein kinase") %>%
  distinct() %>% 
  count()

#RBD
RBD/total*100
#kinase
kinase/total*100
```

```{r}
# mutate PARP1/2 curated_Domain field to simplify for figure
test_clean_MHHES1_data %>% filter(Gene %in% c("PARP1", "PARP2")) %>%
  select(Gene, curated_Domain) %>%
  count(curated_Domain)

test_clean_MHHES1_data <- mutate(test_clean_MHHES1_data, curated_Domain = ifelse(curated_Domain == "Automodification domain;Disordered" & Gene %in% c("PARP1", "PARP2"), "Automodification domain", ifelse(curated_Domain %in% c("Cleavage; by caspase-3 and caspase", "Cleavage; by caspase", "Cleavage; by caspase-8", "Disordered;Cleavage; by caspase-3 and caspase", "Cleavage; by caspase-3 and caspase-7") & Gene %in% c("PARP1", "PARP2"), "Caspase cleavage", ifelse(curated_Domain %in% c("Disordered;Nuclear localization signal", "Nuclear localization signal;Disordered	", "Nuclear localization signal;Disordered") & Gene %in% c("PARP1", "PARP2"), "Nuclear localization signal", ifelse(curated_Domain %in% c("Disordered;PARP-type", "Disordered;PARP-type 2") & Gene %in% c("PARP1", "PARP2"), "Disordered",ifelse(curated_Domain %in% c("Zinc ribbon;PADR1 zinc-binding", "PADR1 zinc-binding;Zinc ribbon") & Gene %in% c("PARP1", "PARP2"), "Zinc ribbon", ifelse(curated_Domain %in% c("Disordered;PARP-type", "Disordered;PARP-type 2") & Gene %in% c("PARP1", "PARP2"), "Disordered",ifelse(curated_Domain %in% c("PARP catalytic;NAD(+) binding site", "PNAD(+) binding site;PARP catalytic", "NAD(+) binding site;PARP catalytic") & Gene %in% c("PARP1", "PARP2"), "PARP catalytic", curated_Domain))))))))

test_clean_MHHES1_data %>% filter(Gene == "PARP2") %>%
  select(Gene, curated_Domain) %>%
  count(curated_Domain)
```

```{r}
# dot plots _ proliferation - PARP1/2

targets <- c("PARP1", "PARP2")
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

plot1 <- ggplot(subset(test_clean_MHHES1_data, Gene %in% c(targets) & (is.na(Amino_Acid_Position_simple) ==FALSE)), aes(x=Amino_Acid_Position_simple, y= L2FC_MHHES1_Control_plasmid_average_zscore, shape = most_severe_consequence, colour = curated_Domain)) + 
geom_point(position = pos, alpha = 1, size = 3) +
labs(title= "", x="PARP1/2 amino acid position", y = "MHH-ES-1 control vs plasmid (z-score)", colour = "domain", shape = "consequence") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
#geom_text_repel(aes(label=ifelse(L2FC_MHHES1_Control_plasmid_average_zscore > -4, paste(PTM), "")), size = 3, position = pos, max.overlaps = Inf) +
scale_color_manual(values =c("grey55", "purple", "black", "royalblue", "orange", "red", "salmon", "red3", "blue3", "pink3", "yellow4", "grey35", "grey75", "orange3", "green3")) +
scale_shape_manual(values=c(1, 19, 2, 11, 7, 3)) +
theme_classic() +
facet_wrap(~Gene, scales = "free_x", nrow = 2)
plot1
ggsave("FigS6a.eps", width = 6, height = 4, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

plot2 <- ggplot(subset(test_clean_MHHES1_data, Gene %in% c(targets) & (is.na(Amino_Acid_Position_simple) ==FALSE)), aes(x=Amino_Acid_Position_simple, y= L2FC_MHHES1_Olap_plasmid_average_zscore, shape = most_severe_consequence, colour = curated_Domain)) + 
geom_point(position = pos, alpha = 1, size = 3) +
labs(title= "", x="PARP1/2 amino acid position", y = "MHH-ES-1 olaparib vs plasmid (z-score)", colour = "domain", shape = "consequence") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_MHHES1_Olap_plasmid_average_zscore > 3, paste(Amino_Acid_Position), ifelse(sgRNA_ID %in% PARP_sensitors$sgRNA_ID, paste(Amino_Acid_Position), ""))), size = 3, position = pos, max.overlaps = Inf) +
scale_color_manual(values =c("grey55", "purple", "black", "royalblue", "orange", "red", "salmon", "red3", "blue3", "pink3", "yellow4", "grey35", "grey75", "orange3", "green3")) +
scale_shape_manual(values=c(1, 19, 2, 11, 7, 3)) +
theme_classic() +
facet_wrap(~Gene, scales = "free_x", nrow = 2)
plot2
ggsave("FigS6b.eps", width = 6, height = 4, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

plot3 <- ggplot(subset(test_clean_MHHES1_data, Gene %in% c(targets) & (is.na(Amino_Acid_Position_simple) ==FALSE)), aes(x=Amino_Acid_Position_simple, y= L2FC_MHHES1_Nirap_plasmid_average_zscore, shape = most_severe_consequence, colour = curated_Domain)) + 
geom_point(position = pos, alpha = 1, size = 3) +
labs(title= "", x="PARP1/2 amino acid position", y = "MHH-ES-1 niraparib vs plasmid (z-score)", colour = "domain", shape = "consequence") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(L2FC_MHHES1_Nirap_plasmid_average_zscore > 3, paste(Amino_Acid_Position), ifelse(sgRNA_ID %in% PARP_sensitors$sgRNA_ID, paste(Amino_Acid_Position), ""))), size = 3, position = pos, max.overlaps = Inf) +
scale_color_manual(values =c("grey55", "purple", "black", "royalblue", "orange", "red", "salmon", "red3", "blue3", "pink3", "yellow4", "grey35", "grey75", "orange3", "green3")) +
scale_shape_manual(values=c(1, 19, 2, 11, 7, 3)) +
theme_classic() +
facet_wrap(~Gene, scales = "free_x", nrow = 2)
plot3
ggsave("FigS6c.eps", width = 6, height = 4, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")
```

```{r}
# dot plots _ proliferation - PARP1/2
targets <- c("PARP1", "PARP2")
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

plot1 <- ggplot(subset(test_clean_MHHES1_data, Gene %in% c(targets) & (is.na(Amino_Acid_Position_simple) ==FALSE)), aes(x=Amino_Acid_Position_simple, y= L2FC_MHHES1_Control_plasmid_average_zscore, colour = Gene)) + 
geom_point(position = pos, alpha = 1, size = 3) +
labs(title= "", x="PARP1/2 amino acid position", y = "MHH-ES-1 control vs plasmid (z-score)", colour = "gene") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
scale_color_manual(values =c("salmon", "pink2")) +
scale_shape_manual(values=c(19, 2, 11, 7, 3)) +
theme_classic() +
facet_wrap(~Gene, scales = "free_x", ncol = 2) +
scale_y_continuous(limits = c(-6, 5))
plot1
ggsave("Fig3d_1.eps", width = 11, height = 4, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

plot2 <- ggplot(subset(test_clean_MHHES1_data, Gene %in% c(targets) & (is.na(Amino_Acid_Position_simple) ==FALSE)), aes(x=Amino_Acid_Position_simple, y= L2FC_MHHES1_Olap_plasmid_average_zscore, colour = Gene)) + 
geom_point(position = pos, alpha = 1, size = 3) +
labs(title= "", x="PARP1/2 amino acid position", y = "MHH-ES-1 olaparib vs plasmid (z-score)", colour = "gene") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(sgRNA_ID %in% PARP_sensitors$sgRNA_ID, paste(Amino_Acid_Position), "")), size = 4.5, position = pos, max.overlaps = Inf) +
scale_color_manual(values =c("salmon", "pink2")) +
scale_shape_manual(values=c(19, 2, 11, 7, 3)) +
theme_classic() +
facet_wrap(~Gene, scales = "free_x", ncol = 2)+
scale_y_continuous(limits = c(-6, 5))
plot2
ggsave("Fig3d_2.eps", width = 11, height = 4, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

plot3 <- ggplot(subset(test_clean_MHHES1_data, Gene %in% c(targets) & (is.na(Amino_Acid_Position_simple) ==FALSE)), aes(x=Amino_Acid_Position_simple, y= L2FC_MHHES1_Nirap_plasmid_average_zscore, colour = Gene)) + 
geom_point(position = pos, alpha = 1, size =3) +
labs(title= "", x="PARP1/2 amino acid position", y = "MHH-ES-1 niraparib vs plasmid (z-score)", colour = "gene") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
geom_text_repel(aes(label=ifelse(sgRNA_ID %in% PARP_sensitors$sgRNA_ID, paste(Amino_Acid_Position), ifelse(sgRNA_ID %in% PARP_resistors$sgRNA_ID & L2FC_MHHES1_Nirap_plasmid_average_zscore >4 , paste(Amino_Acid_Position), ifelse(sgRNA_ID %in% PARP_resistors$sgRNA_ID & Amino_Acid_Position == "1" , paste(Amino_Acid_Position, most_severe_consequence), "")))), size = 4.5, position = pos, max.overlaps = Inf) +
scale_color_manual(values =c("salmon", "pink2")) +
scale_shape_manual(values=c(19, 2, 11, 7, 3)) +
theme_classic() +
facet_wrap(~Gene, scales = "free_x", ncol = 2) +
scale_y_continuous(limits = c(-6, 5))
plot3
ggsave("Fig3d_3.eps", width = 11, height = 4, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")
```

```{r}
# primers for Golden Gate arrayed validation studies - PARP validation
# forward_primer
# add a G for the expression from U6 promoter by replacing the base at pos. 1 if, it is not a G
test_clean_MHHES1_data <- mutate(test_clean_MHHES1_data, G_guide = ifelse(substr(guide, 1, 1) == "G", guide, paste0("G", substr(guide, 2, 20))))

test_clean_MHHES1_data <- mutate(test_clean_MHHES1_data, for_primer = paste0("CACC", G_guide))

# reverse_primer
test_clean_MHHES1_data <- mutate(test_clean_MHHES1_data, rev_primer = G_guide)
test_clean_MHHES1_data <- mutate(test_clean_MHHES1_data, rev_primer = str_replace_all(rev_primer, "A", "t"))
test_clean_MHHES1_data <- mutate(test_clean_MHHES1_data, rev_primer = str_replace_all(rev_primer, "T", "a"))
test_clean_MHHES1_data <- mutate(test_clean_MHHES1_data, rev_primer = str_replace_all(rev_primer, "C", "g"))
test_clean_MHHES1_data <- mutate(test_clean_MHHES1_data, rev_primer = str_replace_all(rev_primer, "G", "c"))
test_clean_MHHES1_data <- mutate(test_clean_MHHES1_data, rev_primer = str_to_upper(rev_primer))

test_clean_MHHES1_data <- mutate(test_clean_MHHES1_data, rev_primer = paste0(substr(rev_primer, 20, 20), substr(rev_primer, 19, 19),substr(rev_primer, 18, 18),substr(rev_primer, 17, 17),substr(rev_primer, 16, 16),substr(rev_primer, 15, 15), substr(rev_primer, 14, 14), substr(rev_primer, 13, 13), substr(rev_primer, 12, 12), substr(rev_primer, 11, 11), substr(rev_primer, 10, 10), substr(rev_primer, 9, 9), substr(rev_primer, 8, 8), substr(rev_primer, 7, 7), substr(rev_primer, 6, 6), substr(rev_primer, 5, 5), substr(rev_primer, 4, 4), substr(rev_primer, 3, 3), substr(rev_primer, 2, 2), substr(rev_primer, 1, 1)))

test_clean_MHHES1_data <- mutate(test_clean_MHHES1_data, rev_primer = paste0("AAAC", rev_primer))

test_clean_MHHES1_data %>% filter(L2FC_MHHES1_Olap_plasmid_average_zscore > 3 & L2FC_MHHES1_Olap_plasmid_1_zscore >2 & L2FC_MHHES1_Olap_plasmid_2_zscore > 2 | L2FC_MHHES1_Nirap_plasmid_average_zscore > 3 & L2FC_MHHES1_Nirap_plasmid_1_zscore >2 & L2FC_MHHES1_Nirap_plasmid_2_zscore >2) %>%
  select(sgRNA_ID, Gene, G_guide, most_severe_consequence, Amino_Acid_Position, Protein_Change, L2FC_MHHES1_Control_plasmid_average_zscore, L2FC_MHHES1_Control_plasmid_1_zscore, L2FC_MHHES1_Control_plasmid_2_zscore, L2FC_MHHES1_Olap_plasmid_average_zscore, L2FC_MHHES1_Olap_plasmid_1_zscore, L2FC_MHHES1_Olap_plasmid_2_zscore, L2FC_MHHES1_Nirap_plasmid_average_zscore, L2FC_MHHES1_Nirap_plasmid_1_zscore, L2FC_MHHES1_Nirap_plasmid_2_zscore, curated_Domain, MHHES1_ABE_Control_exp1_reads, MHHES1_ABE_Control_exp2_reads, MHHES1_ABE_Olap_exp1_reads, MHHES1_ABE_Olap_exp2_reads, MHHES1_ABE_Nirap_exp1_reads, MHHES1_ABE_Nirap_exp2_reads, for_primer, rev_primer) %>% 
  write.csv("PARP_resistance.csv")

test_clean_MHHES1_data %>% filter(L2FC_MHHES1_Olap_plasmid_average_zscore < -3 & L2FC_MHHES1_Nirap_plasmid_average_zscore < -3 & L2FC_MHHES1_Control_plasmid_average_zscore > -1) %>%
  select(sgRNA_ID, Gene, G_guide, most_severe_consequence, Amino_Acid_Position, Protein_Change, L2FC_MHHES1_Control_plasmid_average_zscore, L2FC_MHHES1_Control_plasmid_1_zscore, L2FC_MHHES1_Control_plasmid_2_zscore, L2FC_MHHES1_Olap_plasmid_average_zscore, L2FC_MHHES1_Olap_plasmid_1_zscore, L2FC_MHHES1_Olap_plasmid_2_zscore, L2FC_MHHES1_Nirap_plasmid_average_zscore, L2FC_MHHES1_Nirap_plasmid_1_zscore, L2FC_MHHES1_Nirap_plasmid_2_zscore, curated_Domain, MHHES1_ABE_Control_exp1_reads, MHHES1_ABE_Control_exp2_reads, MHHES1_ABE_Olap_exp1_reads, MHHES1_ABE_Olap_exp2_reads, MHHES1_ABE_Nirap_exp1_reads, MHHES1_ABE_Nirap_exp2_reads, for_primer, rev_primer) %>% 
  write.csv("PARP_sensitisors.csv")
```

```{r}
# dot plots _ proliferation

targets <- c("EGFR", "KRAS", "MYC", "PARP1", "PARP2", "MAP2K1", "MAP2K2", "BCL2", "BRAF", "AKT1", "PIK3CA")

pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

plot1 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=L2FC_HT29_Control_plasmid_1_zscore, y= L2FC_H23_Control_plasmid_1_zscore, colour = most_severe_consequence)) + 
geom_point(position = pos, alpha = 0.80, size =3) +
labs(title= "", x="L2FC_HT29_Control_plasmid_1", y = "L2FC_H23_Control_plasmid_1") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
#geom_text_repel(aes(label=ifelse(L2FC_PC9_Gefit_plasmid_1 > 4 | L2FC_PC9_Gefit_plasmid_1 < -2.5 & L2FC_PC9_Control_plasmid_1 > -0.4, paste(Amino_Acid_Position, ",", Gene, ",", Consequence), "")), size = 3, position = pos) +
scale_color_manual(values =c("black", "springgreen4", "orange", "red", "blue", "grey55", "brown", "purple", "pink", "salmon", "royalblue")) +
#scale_shape_manual(values =c(19, 1)) +
facet_wrap(~Gene, scales = "free_x") +
theme_classic()
plot1

plot2 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=L2FC_HT29_Control_plasmid_1_zscore, y= L2FC_PC9_Control_plasmid_1_zscore, colour = most_severe_consequence)) + 
geom_point(position = pos, alpha = 0.80, size =3) +
labs(title= "", x="L2FC_HT29_Control_plasmid_1", y = "L2FC_PC9_Control_plasmid_1") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
#geom_text_repel(aes(label=ifelse(L2FC_PC9_Gefit_plasmid_1 > 4 | L2FC_PC9_Gefit_plasmid_1 < -2.5 & L2FC_PC9_Control_plasmid_1 > -0.4, paste(Amino_Acid_Position, ",", Gene, ",", Consequence), "")), size = 3, position = pos) +
scale_color_manual(values =c("black", "springgreen4", "orange", "red", "blue", "grey55", "brown", "purple", "pink", "salmon", "royalblue")) +
#scale_shape_manual(values =c(19, 1)) +
facet_wrap(~Gene, scales = "free_x") +
theme_classic()
plot2

plot3 <- ggplot(subset(annotated_data, Gene %in% c(targets)), aes(x=L2FC_PC9_Control_plasmid_1_zscore, y= L2FC_H23_Control_plasmid_1_zscore, colour = most_severe_consequence)) + 
geom_point(position = pos, alpha = 0.80, size =3) +
labs(title= "", x="L2FC_PC9_Control_plasmid_1", y = "L2FC_H23_Control_plasmid_1") +
geom_hline(yintercept=c(0,0),linetype="dotted") +
#geom_text_repel(aes(label=ifelse(L2FC_PC9_Gefit_plasmid_1 > 4 | L2FC_PC9_Gefit_plasmid_1 < -2.5 & L2FC_PC9_Control_plasmid_1 > -0.4, paste(Amino_Acid_Position, ",", Gene, ",", Consequence), "")), size = 3, position = pos) +
scale_color_manual(values =c("black", "springgreen4", "orange", "red", "blue", "grey55", "brown", "purple", "pink", "salmon", "royalblue")) +
#scale_shape_manual(values =c(19, 1)) +
facet_wrap(~Gene, scales = "free_x") +
theme_classic()
plot3
```

```{r}
#stats for violin plots - Student's t-test

#HT29
NT_HT29 <- annotated_data %>%
  filter(gene_alias == "non-targeting") %>%
  select(L2FC_HT29_Control_plasmid_average_zscore)

essential_HT29 <- annotated_data %>%
  filter(gene_alias == "essential") %>%
  select(L2FC_HT29_Control_plasmid_average_zscore)

t_test_1 <- t.test(NT_HT29, essential_HT29, var.equal = TRUE)
t_test_1

#H23
NT_H23 <- annotated_data %>%
  filter(gene_alias == "non-targeting") %>%
  select(L2FC_H23_Control_plasmid_average_zscore)

essential_H23 <- annotated_data %>%
  filter(gene_alias == "essential") %>%
  select(L2FC_H23_Control_plasmid_average_zscore)

t_test_2 <- t.test(NT_H23, essential_H23, var.equal = TRUE)
t_test_2

#PC9
NT_PC9 <- annotated_data %>%
  filter(gene_alias == "non-targeting") %>%
  select(L2FC_PC9_Control_plasmid_average_zscore)

essential_PC9 <- annotated_data %>%
  filter(gene_alias == "essential") %>%
  select(L2FC_PC9_Control_plasmid_average_zscore)

t_test_3 <- t.test(NT_PC9, essential_PC9, var.equal = TRUE)
t_test_3

# MHH-ES-1
NT_MHHES1 <- test_clean_MHHES1_data %>%
  filter(gene_alias == "non-targeting") %>%
  select(L2FC_MHHES1_Control_plasmid_average_zscore)

essential_MHHES1 <- test_clean_MHHES1_data %>%
  filter(gene_alias == "essential") %>%
  select(L2FC_MHHES1_Control_plasmid_average_zscore)

t_test_4 <- t.test(NT_MHHES1, essential_MHHES1, var.equal = TRUE)
t_test_4
```

```{r} 
# Violin overview plots Figure 1
# control vs plasmid
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

#HT-29
plot1 <- ggplot(annotated_data, aes(y = forcats::fct_relevel(gene_alias, "PARP1", "PARP2", "PIK3CA", "AKT1", "BCL2", "MAP2K1", "MAP2K2", "BRAF", "KRAS","EGFR","MYC", "essential", "non-essential", "intergenic", "non-targeting") , x = L2FC_HT29_Control_plasmid_average_zscore)) +
labs(title= "", y = "", x = "HT-29 control vs plasmid (z-score)") +
geom_point(size=3, alpha=1, position = pos, shape = 1) +
geom_violin(colour = "salmon", alpha = 1) +
annotate("text", y = 12.2, x = 3, label = paste("p =", formatC(t_test_1$p.value, digits = 2)), parse = FALSE, size = 4) +
theme_classic()
plot1
ggsave("Fig1b.eps", width = 6, height = 5, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

#H23
plot3 <- ggplot(annotated_data, aes(y = forcats::fct_relevel(gene_alias, "PARP1", "PARP2", "PIK3CA", "AKT1", "BCL2", "MAP2K1", "MAP2K2", "BRAF", "KRAS","EGFR","MYC", "essential", "non-essential", "non-targeting") , x = L2FC_H23_Control_plasmid_average_zscore)) +
labs(title= "", y = "", x = "H23 control vs plasmid (z-score)") +
geom_point(size=3, alpha=1, position = pos, shape = 1) +
geom_violin(colour = "salmon", alpha = 1) +
annotate("text", y = 12.2, x = 2.5, label = paste("p =", formatC(t_test_2$p.value, digits = 2)), parse = FALSE, size = 4) +
theme_classic()
plot3
ggsave("FigS1b_1.eps", width = 6, height = 5, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

#PC9
plot2 <- ggplot(annotated_data, aes(y = forcats::fct_relevel(gene_alias, "PARP1", "PARP2", "PIK3CA", "AKT1", "BCL2", "MAP2K1", "MAP2K2", "BRAF", "KRAS","EGFR","MYC", "essential", "non-essential", "non-targeting") , x = L2FC_PC9_Control_plasmid_average_zscore)) +
labs(title= "", y = "", x = "PC9 control vs plasmid (z-score)") +
geom_point(size=3, alpha=1, position = pos, shape = 1) +
geom_violin(colour = "salmon", alpha = 1) +
annotate("text", y = 12.2, x = 2.3, label = paste("p =", formatC(t_test_3$p.value, digits = 2)), parse = FALSE, size = 4) +
theme_classic()
plot2
ggsave("FigS1b_2.eps", width = 6, height = 5, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

# MHH-ES-1
plot3 <- ggplot(test_clean_MHHES1_data, aes(y = forcats::fct_relevel(gene_alias, "PARP1", "PARP2", "PIK3CA", "AKT1", "BCL2", "MAP2K1", "MAP2K2", "BRAF", "KRAS","EGFR","MYC", "essential", "non-essential", "non-targeting") , x = L2FC_MHHES1_Control_plasmid_average_zscore)) +
labs(title= "", y = "", x = "MHH-ES-1 control vs plasmid (z-score)") +
geom_point(size=3, alpha=1, position = pos, shape = 1) +
geom_violin(colour = "salmon", alpha = 1) +
annotate("text", y = 12.2, x = 2.3, label = paste("p =", formatC(t_test_4$p.value, digits = 2)), parse = FALSE, size = 4) +
theme_classic()
plot3
ggsave("FigS1b_3.eps", width = 6, height = 5, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")
```

```{r}
# primers for Golden Gate arrayed validation studies
# forward_primer
# add a G for the expression from U6 promoter by replacing the base at pos. 1 if, it is not a G
annotated_data <- mutate(annotated_data, G_guide = ifelse(substr(guide, 1, 1) == "G", guide, paste0("G", substr(guide, 2, 20))))

annotated_data <- mutate(annotated_data, for_primer = paste0("CACC", G_guide))

# reverse_primer
annotated_data <- mutate(annotated_data, rev_primer = G_guide)
annotated_data <- mutate(annotated_data, rev_primer = str_replace_all(rev_primer, "A", "t"))
annotated_data <- mutate(annotated_data, rev_primer = str_replace_all(rev_primer, "T", "a"))
annotated_data <- mutate(annotated_data, rev_primer = str_replace_all(rev_primer, "C", "g"))
annotated_data <- mutate(annotated_data, rev_primer = str_replace_all(rev_primer, "G", "c"))
annotated_data <- mutate(annotated_data, rev_primer = str_to_upper(rev_primer))

annotated_data <- mutate(annotated_data, rev_primer = paste0(substr(rev_primer, 20, 20), substr(rev_primer, 19, 19),substr(rev_primer, 18, 18),substr(rev_primer, 17, 17),substr(rev_primer, 16, 16),substr(rev_primer, 15, 15), substr(rev_primer, 14, 14), substr(rev_primer, 13, 13), substr(rev_primer, 12, 12), substr(rev_primer, 11, 11), substr(rev_primer, 10, 10), substr(rev_primer, 9, 9), substr(rev_primer, 8, 8), substr(rev_primer, 7, 7), substr(rev_primer, 6, 6), substr(rev_primer, 5, 5), substr(rev_primer, 4, 4), substr(rev_primer, 3, 3), substr(rev_primer, 2, 2), substr(rev_primer, 1, 1)))

annotated_data <- mutate(annotated_data, rev_primer = paste0("AAAC", rev_primer))
```

```{r} 
# investigate BRAF hits
#annotated_data %>% 
# filter(Gene == "BRAF") %>%
#  filter(L2FC_HT29_Control_plasmid_average > 1.6 | L2FC_PC9_Control_plasmid_average > 1.6 | L2FC_H23_Control_plasmid_average > 1.6) %>%
#  select(Gene, guide, most_severe_consequence, Amino_Acid_Position, editor, L2FC_HT29_Control_plasmid_average, #L2FC_PC9_Control_plasmid_average, L2FC_H23_Control_plasmid_average, sgRNA_type, for_primer, rev_primer)

#annotated_data %>% 
#  filter(Gene == "BRAF") %>%
#  filter(L2FC_HT29_Control_plasmid_average > 1.6 | L2FC_PC9_Control_plasmid_average > 1.6 | L2FC_H23_Control_plasmid_average > 1.6) %>%
#  select(Gene, guide, most_severe_consequence, Amino_Acid_Position, editor, L2FC_HT29_Control_plasmid_average, L2FC_PC9_Control_plasmid_average, L2FC_H23_Control_plasmid_average, sgRNA_type, for_primer, rev_primer) %>% write.csv("BRAF_valid.csv")
```

```{r}
# read in raw gRNA reads counts from sc validation library
raw_reads_scvalid <- as_tibble(read.csv("scvRes1_readcounts_270423.csv", stringsAsFactors = FALSE, header = TRUE))
raw_reads_scvalid <- mutate(raw_reads_scvalid, sgRNA_ID = as.character(sgRNA_ID))

# compute RPM normalised values
raw_reads_scvalid <- mutate_at(raw_reads_scvalid, vars(matches("reads")), list(RPM = RPM))
raw_reads_scvalid
```

```{r}
# compute L2FC from normalised gRNA counts - single experiments. -scResvalid library small validation screen

# PC9
#BE3.9max NGN
raw_reads_scvalid <- mutate(raw_reads_scvalid, L2FC_PC9_CBE_Control_plasmid_scres1valid = L2FC(PC9_CBE_cntrl_scres1valid_reads_RPM, sc_valid_plasmid_reads_RPM))
raw_reads_scvalid <- mutate(raw_reads_scvalid, L2FC_PC9_CBE_Control_T0_scres1valid = L2FC(PC9_CBE_cntrl_scres1valid_reads_RPM, PC9_CBE_T0_scres1valid_reads))
raw_reads_scvalid <- mutate(raw_reads_scvalid, L2FC_PC9_CBE_Gefit_Control_scres1valid = L2FC(PC9_CBE_gefit_scres1valid_reads_RPM, PC9_CBE_cntrl_scres1valid_reads_RPM))
raw_reads_scvalid <- mutate(raw_reads_scvalid, L2FC_PC9_CBE_Osim_Control_scres1valid = L2FC(PC9_CBE_osim_scres1valid_reads_RPM, PC9_CBE_cntrl_scres1valid_reads_RPM))
raw_reads_scvalid <- mutate(raw_reads_scvalid, L2FC_PC9_CBE_Gefit_plasmid_scres1valid = L2FC(PC9_CBE_gefit_scres1valid_reads_RPM, sc_valid_plasmid_reads_RPM))
raw_reads_scvalid <- mutate(raw_reads_scvalid, L2FC_PC9_CBE_Osim_plasmid_scres1valid = L2FC(PC9_CBE_osim_scres1valid_reads_RPM, sc_valid_plasmid_reads_RPM))

#ABE8e NGN
raw_reads_scvalid <- mutate(raw_reads_scvalid, L2FC_PC9_ABE_Control_plasmid_scres1valid = L2FC(PC9_ABE_cntrl_scres1valid_reads_RPM, sc_valid_plasmid_reads_RPM))
raw_reads_scvalid <- mutate(raw_reads_scvalid, L2FC_PC9_ABE_Control_T0_scres1valid = L2FC(PC9_ABE_cntrl_scres1valid_reads_RPM, PC9_ABE_T0_scres1valid_reads))
raw_reads_scvalid <- mutate(raw_reads_scvalid, L2FC_PC9_ABE_Gefit_Control_scres1valid = L2FC(PC9_ABE_gefit_scres1valid_reads_RPM, PC9_ABE_cntrl_scres1valid_reads_RPM))
raw_reads_scvalid <- mutate(raw_reads_scvalid, L2FC_PC9_ABE_Osim_Control_scres1valid = L2FC(PC9_ABE_osim_scres1valid_reads_RPM, PC9_ABE_cntrl_scres1valid_reads_RPM))
raw_reads_scvalid <- mutate(raw_reads_scvalid, L2FC_PC9_ABE_Gefit_plasmid_scres1valid = L2FC(PC9_ABE_gefit_scres1valid_reads_RPM, sc_valid_plasmid_reads_RPM))
raw_reads_scvalid <- mutate(raw_reads_scvalid, L2FC_PC9_ABE_Osim_plasmid_scres1valid = L2FC(PC9_ABE_osim_scres1valid_reads_RPM, sc_valid_plasmid_reads_RPM))

raw_reads_scvalid
```

```{r}
# generate z scores for sc validation library
raw_reads_scvalid <- mutate_at(raw_reads_scvalid, vars(matches("L2FC")), list(zscore = zscore))
raw_reads_scvalid
```

```{r}
#check zscore correlation with L2FC

pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

# CBE gefit vs plasmid
plot1 <- ggplot(raw_reads_scvalid, aes(x= L2FC_PC9_CBE_Gefit_plasmid_scres1valid, y= L2FC_PC9_CBE_Gefit_plasmid_scres1valid_zscore)) + geom_jitter(size=3,  position = pos, colour = "grey35") +
labs(title= "PC9 - zscore L2FC correlation", x = "L2FC_PC9_CBE_Gefit_plasmid_scres1valid", y = "L2FC_PC9_CBE_Gefit_plasmid_scres1valid_zscore") +
geom_smooth(method = lm) +
theme_bw()
plot1

# ABE osim vs plasmid
plot2 <- ggplot(raw_reads_scvalid, aes(x= L2FC_PC9_ABE_Osim_plasmid_scres1valid, y= L2FC_PC9_ABE_Osim_plasmid_scres1valid_zscore)) + geom_jitter(size=3,  position = pos, colour = "grey35") +
labs(title= "PC9 - zscore L2FC correlation", x = "L2FC_PC9_ABE_Osim_plasmid_scres1valid", y = "L2FC_PC9_ABE_Osim_plasmid_scres1valid_zscore") +
geom_smooth(method = lm) +
theme_bw()
plot2
```

```{r}
#left_join the sc validation library results
annotated_data <- left_join(annotated_data, raw_reads_scvalid)
```

```{r}
#PC9 check correlation between large screens and validation library
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

# CBE gefit vs plasmid
plot1 <- ggplot(annotated_data, aes(x= L2FC_PC9_Gefit_plasmid_1_zscore, y= L2FC_PC9_CBE_Gefit_plasmid_scres1valid_zscore)) + geom_jitter(size=3,  position = pos, colour = "grey35") +
labs(title= "PC9 - replicate correlation", x = "L2FC_PC9_Gefit_plasmid_1", y = "L2FC_PC9_CBE_Gefit_plasmid_scres1valid") +
geom_smooth(method = lm) +
theme_bw()
plot1

plot2 <- ggplot(annotated_data, aes(x= L2FC_PC9_Gefit_plasmid_2_zscore, y= L2FC_PC9_CBE_Gefit_plasmid_scres1valid_zscore)) + geom_jitter(size=3,  position = pos, colour = "grey35") +
labs(title= "PC9 - replicate correlation", x = "L2FC_PC9_Gefit_plasmid_2", y = "L2FC_PC9_CBE_Gefit_plasmid_scres1valid") +
geom_smooth(method = lm) +
theme_bw()
plot2

plot3 <- ggplot(annotated_data, aes(x= L2FC_PC9_Gefit_plasmid_average_zscore, y= L2FC_PC9_CBE_Gefit_plasmid_scres1valid_zscore)) + geom_jitter(size=3,  position = pos, colour = "grey35") +
labs(title= "PC9 - replicate correlation", x = "L2FC_PC9_Gefit_plasmid_average", y = "L2FC_PC9_CBE_Gefit_plasmid_scres1valid") +
geom_smooth(method = lm) +
theme_bw()
plot3

# CBE osim vs plasmid
plot4 <- ggplot(annotated_data, aes(x= L2FC_PC9_Osim_plasmid_average_zscore, y= L2FC_PC9_CBE_Osim_plasmid_scres1valid_zscore)) + geom_jitter(size=3,  position = pos, colour = "grey35") +
labs(title= "PC9 - replicate correlation", x = "L2FC_PC9_Osim_plasmid_average", y = "L2FC_PC9_CBE_Osim_plasmid_scres1valid") +
geom_smooth(method = lm) +
theme_bw()
plot4

# ABE gefit vs plasmid
plot5 <- ggplot(annotated_data, aes(x= L2FC_PC9_Gefit_plasmid_1_zscore, y= L2FC_PC9_ABE_Gefit_plasmid_scres1valid_zscore)) + geom_jitter(size=3,  position = pos, colour = "grey35") +
labs(title= "PC9 - replicate correlation", x = "L2FC_PC9_Gefit_plasmid_1", y = "L2FC_PC9_ABE_Gefit_plasmid_scres1valid") +
geom_smooth(method = lm) +
theme_bw()
plot5

plot6 <- ggplot(annotated_data, aes(x= L2FC_PC9_Gefit_plasmid_2_zscore, y= L2FC_PC9_ABE_Gefit_plasmid_scres1valid_zscore)) + geom_jitter(size=3,  position = pos, colour = "grey35") +
labs(title= "PC9 - replicate correlation", x = "L2FC_PC9_Gefit_plasmid_2", y = "L2FC_PC9_ABE_Gefit_plasmid_scres1valid") +
geom_smooth(method = lm) +
theme_bw()
plot6

plot7 <- ggplot(annotated_data, aes(x= L2FC_PC9_Gefit_plasmid_average_zscore, y= L2FC_PC9_ABE_Gefit_plasmid_scres1valid_zscore)) + geom_jitter(size=3,  position = pos, colour = "grey35") +
labs(title= "PC9 - replicate correlation", x = "L2FC_PC9_Gefit_plasmid_average", y = "L2FC_PC9_ABE_Gefit_plasmid_scres1valid") +
geom_smooth(method = lm) +
theme_bw()
plot7

# ABE osim vs plasmid
plot8 <- ggplot(annotated_data, aes(x= L2FC_PC9_Osim_plasmid_average_zscore, y= L2FC_PC9_ABE_Osim_plasmid_scres1valid_zscore)) + geom_jitter(size=3,  position = pos, colour = "grey35") +
labs(title= "PC9 - replicate correlation", x = "L2FC_PC9_Osim_plasmid_average", y = "L2FC_PC9_ABE_Osim_plasmid_scres1valid") +
geom_smooth(method = lm) +
theme_bw()
plot8
```

```{r}
# consolidate gRNA scores for gRNAs based on editor from sc valid library

annotated_data <- mutate(annotated_data, L2FC_PC9_Control_plasmid_scres1valid_zscore = as.numeric(ifelse(editor == "CBE", L2FC_PC9_CBE_Control_plasmid_scres1valid_zscore, ifelse(editor == "ABE", L2FC_PC9_ABE_Control_plasmid_scres1valid_zscore, "ERROR"))))

annotated_data <- mutate(annotated_data, L2FC_PC9_Gefit_plasmid_scres1valid_zscore = as.numeric(ifelse(editor == "CBE", L2FC_PC9_CBE_Gefit_plasmid_scres1valid_zscore, ifelse(editor == "ABE", L2FC_PC9_ABE_Gefit_plasmid_scres1valid_zscore, "ERROR"))))

annotated_data <- mutate(annotated_data, L2FC_PC9_Osim_plasmid_scres1valid_zscore = as.numeric(ifelse(editor == "CBE", L2FC_PC9_CBE_Osim_plasmid_scres1valid_zscore, ifelse(editor == "ABE", L2FC_PC9_ABE_Osim_plasmid_scres1valid_zscore, "ERROR"))))


#remove all the old score fields based on CBE or ABE
annotated_data <- annotated_data %>%
  select(!c(L2FC_PC9_CBE_Gefit_plasmid_scres1valid_zscore, L2FC_PC9_ABE_Gefit_plasmid_scres1valid_zscore, L2FC_PC9_CBE_Osim_plasmid_scres1valid_zscore, L2FC_PC9_ABE_Osim_plasmid_scres1valid_zscore)) %>%
  distinct()
annotated_data
```

```{r}
# replicate correlation - stats PC9 big screen vs validation screen
overlapping_data <- annotated_data %>% filter(is.na(L2FC_PC9_Gefit_plasmid_scres1valid_zscore) == FALSE & is.na(L2FC_PC9_Osim_plasmid_scres1valid_zscore) == FALSE)
overlapping_data
r1 <- cor(overlapping_data$L2FC_PC9_Gefit_plasmid_average_zscore, overlapping_data$L2FC_PC9_Gefit_plasmid_scres1valid_zscore, method = "pearson")
r2 <- cor(overlapping_data$L2FC_PC9_Osim_plasmid_average_zscore, overlapping_data$L2FC_PC9_Osim_plasmid_scres1valid_zscore, method = "pearson")
```

```{r}
#hits overlapping with validation screen
#gefit
gefit_hits <- annotated_data %>% 
  filter(L2FC_PC9_Gefit_plasmid_average_zscore >3 & L2FC_PC9_Gefit_plasmid_scres1valid_zscore >3 | L2FC_PC9_Gefit_plasmid_average_zscore < -2.5 & L2FC_PC9_Gefit_plasmid_scres1valid_zscore < -2.5 & sgRNA_type == "exonic") %>%
  select(guide, Gene, most_severe_consequence, Amino_Acid_Position)

#osim
osim_hits <- annotated_data %>% 
  filter(L2FC_PC9_Osim_plasmid_average_zscore >3 & L2FC_PC9_Osim_plasmid_scres1valid_zscore >3 | L2FC_PC9_Osim_plasmid_average_zscore < -3 & L2FC_PC9_Osim_plasmid_scres1valid_zscore < -3 & sgRNA_type == "exonic") %>%
  select(guide, Gene, most_severe_consequence, Amino_Acid_Position)
```

```{r}
#PC9 check correlation between large screens and validation library
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

# gefit vs plasmid
plot1 <- ggplot(annotated_data, aes(x= L2FC_PC9_Gefit_plasmid_average_zscore, y= L2FC_PC9_Gefit_plasmid_scres1valid_zscore)) + 
geom_pointdensity(size=3,  position = pos) +
scale_colour_viridis_c() +
labs(title= "PC9 - validation screens", x = "original screens: \nPC9 gefitinib vs plasmid (z-score)", y = "validation screens: \nPC9 gefitinib vs plasmid (z-score)", colour = "density") +
geom_smooth(method = lm) +
annotate("text", y = 7, x = 0, label = paste("R = ", round(r1, 2)), parse = FALSE, size = 7) +
geom_text_repel(aes(label=ifelse(guide %in% c(gefit_hits$guide), paste0(Gene, ",", Amino_Acid_Position, ",", most_severe_consequence), "")), size = 3, position = pos, max.overlaps = Inf) +
theme_bw(base_size = 15)
plot1
ggsave("FigS5a_1.eps", width = 6, height = 5, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

# osim vs plasmid
plot2 <- ggplot(annotated_data, aes(x= L2FC_PC9_Osim_plasmid_average_zscore, y= L2FC_PC9_Osim_plasmid_scres1valid_zscore)) + 
geom_pointdensity(size=3,  position = pos) +
scale_colour_viridis_c() +
labs(title= "PC9 - validation screens",  x = "original screens: \nPC9 osimertinib vs plasmid (z-score)", y = "validation screens: \nPC9 osimertinib vs plasmid (z-score)", colour = "density") +
geom_smooth(method = lm) +
annotate("text", y = 5, x = 0, label = paste("R = ", round(r2, 2)), parse = FALSE, size = 7) +
geom_text_repel(aes(label=ifelse(guide %in% c(osim_hits$guide), paste0(Gene, ",", Amino_Acid_Position, ",", most_severe_consequence), "")), size = 3, position = pos, max.overlaps = Inf) +
theme_bw(base_size = 15)
plot2
ggsave("FigS5a_2.eps", width = 6, height = 5, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")
```

```{r}
#prime editing validation scope
annotated_data %>% 
filter(L2FC_PC9_Gefit_plasmid_average_zscore > 2 & L2FC_PC9_Gefit_plasmid_scres1valid_zscore > 2| L2FC_PC9_Osim_plasmid_average_zscore > 2 & L2FC_PC9_Osim_plasmid_scres1valid_zscore > 2) %>%
  select(Gene, most_severe_consequence, Amino_Acid_Position, guide, L2FC_PC9_Gefit_plasmid_average_zscore, L2FC_PC9_Gefit_plasmid_scres1valid_zscore, L2FC_PC9_Osim_plasmid_average_zscore, L2FC_PC9_Osim_plasmid_scres1valid_zscore, off_target_summary_NG, editor) %>%
  write.csv("Marie_pridict_test_input.csv")
```

```{r}
# Ryan AZ list 08.05.23
annotated_data %>% 
  mutate(gefitinib_resistance = ifelse(L2FC_PC9_Gefit_plasmid_average_zscore > 1.5 & L2FC_PC9_Gefit_plasmid_scres1valid_zscore > 1.5, TRUE, FALSE)) %>%
  mutate(osimertinib_resistance = ifelse(L2FC_PC9_Osim_plasmid_average_zscore > 1.5 & L2FC_PC9_Osim_plasmid_scres1valid_zscore > 1.5, TRUE, FALSE)) %>%
filter(L2FC_PC9_Gefit_plasmid_average_zscore > 1.5 & L2FC_PC9_Gefit_plasmid_scres1valid_zscore > 1 | L2FC_PC9_Osim_plasmid_average_zscore > 1 & L2FC_PC9_Osim_plasmid_scres1valid_zscore > 1.5) %>%
  select(Gene, most_severe_consequence, Amino_Acid_Position, Protein_Change, Transcript_ID, guide, L2FC_PC9_Gefit_plasmid_average_zscore, L2FC_PC9_Gefit_plasmid_scres1valid_zscore, L2FC_PC9_Osim_plasmid_average_zscore, L2FC_PC9_Osim_plasmid_scres1valid_zscore, off_target_summary_NG, editor, gefitinib_resistance, osimertinib_resistance) %>%
  write.csv("AZ_res_mutations.csv")
```

```{r}
#all hits for Figure 6

validation_set_HT29 <- annotated_data %>% 
  filter(L2FC_HT29_Control_plasmid_average_zscore >1.5 | L2FC_HT29_Tram_plasmid_average_zscore >1.5 | L2FC_HT29_Pict_plasmid_average_zscore >1.5 | L2FC_HT29_DebCet_plasmid_average_zscore >1.5) %>%
  filter(sgRNA_type == "exonic")

validation_set_H23 <- annotated_data %>% 
  filter(L2FC_H23_Control_plasmid_average_zscore >1.5 | L2FC_H23_Adag_plasmid_average_zscore >1.5 | L2FC_H23_Sotor_plasmid_average_zscore >1.5) %>%
  filter(sgRNA_type == "exonic")

validation_set_PC9 <- annotated_data %>% 
  filter(L2FC_PC9_Control_plasmid_average_zscore >1.5 | L2FC_PC9_Gefit_plasmid_average_zscore >1.5 & L2FC_PC9_Gefit_plasmid_scres1valid_zscore > 1.5 | L2FC_PC9_Osim_plasmid_average_zscore >1.5 & L2FC_PC9_Osim_plasmid_scres1valid_zscore > 1.5 | sgRNA_ID %in% c("156dcfe646185f7d34b04987763174c762f065c9", "53b63e6975f3b2790703e4fc6191d23902a5b9e8", "6034219e040dd7b9c2e6224f5d36039f1bdd9e07", "34131bd6ecb323bf952225a891399241b58b3131")) %>%
  filter(sgRNA_type == "exonic")

resistance_hits <- bind_rows(validation_set_HT29, validation_set_H23, validation_set_PC9) %>%
  filter(is.na(Amino_Acid_Position) == FALSE) %>%
  distinct()

resistance_hits <- select(resistance_hits, c(Gene, guide, editor, Amino_Acid_Position, Protein_Change, sgRNA_type, most_severe_consequence, off_target_summary_NG, sgRNA_ID, L2FC_HT29_Control_plasmid_average_zscore, L2FC_HT29_Tram_plasmid_average_zscore, L2FC_HT29_Pict_plasmid_average_zscore, L2FC_HT29_DebCet_plasmid_average_zscore, L2FC_H23_Control_plasmid_average_zscore, L2FC_H23_Adag_plasmid_average_zscore, L2FC_H23_Sotor_plasmid_average_zscore, L2FC_PC9_Control_plasmid_average_zscore, L2FC_PC9_Gefit_plasmid_average_zscore, L2FC_PC9_Gefit_plasmid_scres1valid_zscore, L2FC_PC9_Osim_plasmid_average_zscore, L2FC_PC9_Osim_plasmid_scres1valid_zscore))
```

```{r}
# previous annotations for Magda analysis - change to use zscore and check consistency between replicates

sc_HT29_debcet <- annotated_data %>% filter(is.na(L2FC_PC9_Gefit_plasmid_scres1valid_zscore) == FALSE & is.na(L2FC_PC9_Osim_plasmid_scres1valid_zscore) == FALSE) %>% select(Gene, sgRNA_ID, sgRNA_type, chr, start, end, guide, off_target_summary_NG, Amino_Acid_Position, most_severe_consequence, Amino_Acid_Position, Amino_Acid_Position_simple, editor, G_guide, L2FC_HT29_DebCet_plasmid_1_zscore, L2FC_HT29_DebCet_plasmid_2_zscore, L2FC_HT29_DebCet_plasmid_average_zscore, L2FC_HT29_Control_plasmid_1_zscore, L2FC_HT29_Control_plasmid_2_zscore, L2FC_HT29_Control_plasmid_average_zscore)

#rename Gene N/A to non-targeting
sc_HT29_debcet <- mutate(sc_HT29_debcet, Gene = ifelse(Gene =="N/A", "non_targeting", Gene))

#add binary resistant classifier
sc_HT29_debcet <- mutate(sc_HT29_debcet, resistance_mutation = ifelse(L2FC_HT29_DebCet_plasmid_1_zscore >1 & L2FC_HT29_DebCet_plasmid_2_zscore >1 & L2FC_HT29_DebCet_plasmid_average_zscore > 2, TRUE, FALSE))

#add a shorter sgRNA_ID with gene name
sc_HT29_debcet <- mutate(sc_HT29_debcet, ID = paste0(Gene, "_", substr(sgRNA_ID, 0,6)))

#add variant class
sc_HT29_debcet <- mutate(sc_HT29_debcet, variant_class = ifelse(L2FC_HT29_DebCet_plasmid_1_zscore > 1 & L2FC_HT29_DebCet_plasmid_2_zscore > 1  & L2FC_HT29_DebCet_plasmid_average_zscore > 2 & L2FC_HT29_Control_plasmid_1_zscore < -1 & L2FC_HT29_Control_plasmid_2_zscore < -1 & L2FC_HT29_Control_plasmid_average_zscore < -2, "drug addiction", ifelse(L2FC_HT29_DebCet_plasmid_1_zscore > 1 & L2FC_HT29_DebCet_plasmid_2_zscore > 1 & L2FC_HT29_DebCet_plasmid_average_zscore > 2 & L2FC_HT29_Control_plasmid_1_zscore > -1 & L2FC_HT29_Control_plasmid_2_zscore > -1 & L2FC_HT29_Control_plasmid_average_zscore > -2 & L2FC_HT29_Control_plasmid_1_zscore < 1 & L2FC_HT29_Control_plasmid_2_zscore < 1 & L2FC_HT29_Control_plasmid_average_zscore < 2, "canonical drug resistance", ifelse(L2FC_HT29_Control_plasmid_1_zscore > 1 & L2FC_HT29_Control_plasmid_2_zscore > 1 & L2FC_HT29_Control_plasmid_average_zscore > 2, "driver", "control") )))

write.csv(sc_HT29_debcet, "sc_HT29_debcet_annotated_270623.csv")
```

```{r}
# annotate resistance hits

resistance_hits <- mutate(resistance_hits, gefitinib_resistance = ifelse(L2FC_PC9_Gefit_plasmid_average_zscore > 1.5 & L2FC_PC9_Gefit_plasmid_scres1valid_zscore > 1.5, TRUE, FALSE))
resistance_hits <- mutate(resistance_hits, osimertinib_resistance = ifelse(L2FC_PC9_Osim_plasmid_average_zscore > 1.5 & L2FC_PC9_Osim_plasmid_scres1valid_zscore > 1.5, TRUE, FALSE))
resistance_hits <- mutate(resistance_hits, trametinib_resistance = ifelse(L2FC_HT29_Tram_plasmid_average_zscore > 1.5, TRUE, FALSE))
resistance_hits <- mutate(resistance_hits, pictilisib_resistance = ifelse(L2FC_HT29_Pict_plasmid_average_zscore > 1.5, TRUE, FALSE))
resistance_hits <- mutate(resistance_hits, debcet_resistance = ifelse(L2FC_HT29_DebCet_plasmid_average_zscore > 1.5, TRUE, FALSE))
resistance_hits <- mutate(resistance_hits, adagrasib_resistance = ifelse(L2FC_H23_Adag_plasmid_average_zscore > 1.5, TRUE, FALSE))
resistance_hits <- mutate(resistance_hits, sotorasib_resistance = ifelse(L2FC_H23_Sotor_plasmid_average_zscore > 1.5, TRUE, FALSE))

#add variant class
resistance_hits <- mutate(resistance_hits, variant_class = ifelse(L2FC_HT29_DebCet_plasmid_average_zscore > 1 & L2FC_HT29_Control_plasmid_average_zscore < -1, "drug addiction", ifelse(L2FC_HT29_DebCet_plasmid_average_zscore > 1 & L2FC_HT29_Control_plasmid_average_zscore > -1  & L2FC_HT29_Control_plasmid_average_zscore < 1, "canonical drug resistance", ifelse(L2FC_HT29_Control_plasmid_average_zscore > 1, "driver", "") )))

resistance_hits <- mutate(resistance_hits, variant_class = ifelse(L2FC_HT29_Tram_plasmid_average_zscore > 1 & L2FC_HT29_Control_plasmid_average_zscore < -1, "drug addiction", ifelse(L2FC_HT29_Tram_plasmid_average_zscore > 1 & L2FC_HT29_Control_plasmid_average_zscore > -1  & L2FC_HT29_Control_plasmid_average_zscore < 1, "canonical drug resistance", ifelse(L2FC_HT29_Control_plasmid_average_zscore > 1, "driver", "") )))

resistance_hits %>% write.csv("resistance_hits.csv")
resistance_hits
```

```{r}
# read in resistance regions for validation arrayed amplicons seq

amplicons <- as_tibble(read.csv("Res1_amplicon_seq/res1_ampliconseq_hits.csv", stringsAsFactors = FALSE, header = TRUE))

amplicon_valid <- left_join(amplicons, annotated_data)

amplicon_valid <- select(amplicon_valid, c(Gene, G_guide, guide, sgRNA_ID, for_primer, rev_primer, chr, start, end, Location, editor, strand, Amino_Acid_Position, Protein_Change))
amplicon_valid %>% write.csv("Res1_amplicon_seq/ramplicon_valid.csv")
```

```{r}
# count how many variants we have tried to install with base editing
annotated_data

annotated_data %>%
  distinct() %>%
  filter(gene_alias %in% c("EGFR", "KRAS", "MYC", "PARP1", "PARP2", "MAP2K1", "MAP2K2", "BCL2", "BRAF", "AKT1", "PIK3CA")) %>%
  count()

annotated_data %>%
  filter(is.na(Edit_Location) == FALSE)%>%
  filter(gene_alias %in% c("EGFR", "KRAS", "MYC", "PARP1", "PARP2", "MAP2K1", "MAP2K2", "BCL2", "BRAF", "AKT1", "PIK3CA")) %>%
  select(G_guide, Edit_Location, Gene, editor) %>%
  distinct()

annotated_data %>%
  filter(is.na(Edit_Location) == FALSE)%>%
  filter(gene_alias %in% c("EGFR", "KRAS", "MYC", "PARP1", "PARP2", "MAP2K1", "MAP2K2", "BCL2", "BRAF", "AKT1", "PIK3CA")) %>%
  select(G_guide, Edit_Location, Gene, editor) %>%
  distinct() %>%
  count()
```

```{r}
# write a table of resistance hits that i have amplicon sequencing validation for

primer_metadata <- as_tibble(read.csv("amp_primer_metadata.csv", stringsAsFactors = FALSE, header = TRUE))
amplicon_to_gRNA <- as_tibble(read.csv("010823_amplicon_to_gRNA.csv", stringsAsFactors = FALSE, header = TRUE))
energy_disances <- as_tibble(read.csv("energy_distances_100823.csv", stringsAsFactors = FALSE, header = TRUE))
energy_disances
amplicon_to_gRNA
primer_metadata
resistance_hits
amplicon_valid <- amplicon_valid %>% filter(sgRNA_ID != "82dc01c1370b9615c1c25c4711aef34db87a3e7a")
amplicon_valid

Supplementary_Table_2 <- left_join(amplicon_valid, resistance_hits)
Supplementary_Table_2 <- left_join(Supplementary_Table_2, amplicon_to_gRNA)
Supplementary_Table_2 <- left_join(Supplementary_Table_2, primer_metadata)

Supplementary_Table_2 <- mutate(Supplementary_Table_2, ID = paste0(Gene, "_", substr(sgRNA_ID, 1,6)))
Supplementary_Table_2 <- left_join(Supplementary_Table_2, energy_disances)
Supplementary_Table_2

write.csv(Supplementary_Table_2, "Supplementary_Table_2.csv")
```

```{r}
# analyse the amplicon sequencing data
# filter on >1000 mut reads, depth>1000 reads and, >0.1 vaf
#exclude edit == MAP2K2 p.F213C p. V64V, as pre-existing in HT-29

amp_seq_reference <- as_tibble(read.csv("amp_seq_referencing_220823.csv", stringsAsFactors = FALSE, header = TRUE))
amplicon_seq_data <- as_tibble(read.csv("3189_amplicon_var_report_22_08_2023.csv", stringsAsFactors = FALSE, header = TRUE))

amplicon_seq_data <- amplicon_seq_data %>% filter(depth>1000) %>% distinct()

amp_seq_results <- left_join(amplicon_seq_data, amp_seq_reference)
amp_seq_results <- amp_seq_results %>% mutate(edit_position = ifelse(strand == "-", end-pos_amp, pos_amp-start))
amp_seq_results <- amp_seq_results %>% mutate(edit = paste(Gene, protein))
amp_seq_results <- amp_seq_results %>% mutate(ref_alt = paste0(ref, ">", alt))

#exclude edit == MAP2K2 p.F213C, as pre-existing in HT-29
amp_seq_results <- amp_seq_results %>% filter(protein %in% c("p.F213C") == FALSE)
amp_seq_results <- amp_seq_results %>% filter(protein %in% c("p.V64V") == FALSE )
amp_seq_results <- amp_seq_results %>% filter(protein %in% c("p.Q787Q") == FALSE)
```

```{r}
#count gRNAs in amp seq
amp_seq_results <- amp_seq_results %>% mutate(guide = substr(G_guide, 2,20))
amp_seq_results %>% select(guide) %>% distinct() %>% count()
amp_seq_results %>% select(guide) %>% distinct()
# 45 gRNAs but actually 43 gRNAs with an edit with more than 10 % VAF

# filter vaf by 0.1
amp_seq_results <- amp_seq_results %>% filter(vaf>0.1) %>% distinct()
amp_seq_results

# ampseq analysis
#number of variants
amp_seq_results_CBE <- filter(amp_seq_results, editor == "CBE")
amp_seq_results_ABE <- filter(amp_seq_results, editor == "ABE")
```

```{r}
# calculate editing efficiency in the window and out the window
amp_seq_results %>% summarise(mean = mean(vaf), sd = sd(vaf))
amp_seq_results %>% filter(edit_position >=4 & edit_position <=9) %>% summarise(median = median(vaf), sd = sd(vaf))
amp_seq_results %>% filter(edit_position <4 | edit_position >9) %>% summarise(median = median(vaf), sd = sd(vaf))

# calculate number of unexpectged base edits
amp_seq_results %>% select(dna, ref_alt, editor) %>% distinct() %>% count()
amp_seq_results %>% select(dna, ref_alt, editor) %>% filter(editor == "CBE") %>% filter(ref_alt %in% c("C>T", "G>A")) %>% distinct() %>% count()
amp_seq_results %>% select(dna, ref_alt, editor) %>% filter(editor == "ABE") %>% filter(ref_alt %in% c("A>G", "T>C")) %>% distinct() %>% count()
139-40-93
```

```{r}
#write supp table3
write.csv(amp_seq_results, "Supplementary_Table_3.csv")
amp_seq_results %>% filter(effect %in% c("silent") == FALSE) %>% write.csv("non_syn_Supplementary_Table_3.csv")
```

```{r}
# make graphs of variants from amplicon seq of VAF colour against the edit position
plot1 <- ggplot(subset(amp_seq_results, is.na(editor)==FALSE), aes(x= edit_position, y= edit, fill = vaf)) +
  geom_tile() +
theme_classic()+
  labs(y="", x ="edit position", fill = "VAF") +
  geom_vline(xintercept = 4, linetype="dotted", colour = "grey65") +
  geom_vline(xintercept = 9, linetype="dotted", colour = "grey65") +
    scale_fill_gradient(low = "grey80", high = "red") +
  facet_wrap(~editor)
plot1

plot2 <- ggplot(subset(amp_seq_results, is.na(editor)==FALSE), aes(x= edit_position, y= edit, colour = vaf, shape = ref_alt)) +
  geom_point() +
theme_classic()+
  theme(legend.position = "none") +
  labs(y="", x ="edit position", colour = "VAF", shape = "") +
  geom_vline(xintercept = 4, linetype="dotted", colour = "grey65") +
  geom_vline(xintercept = 9, linetype="dotted", colour = "grey65") +
    scale_color_continuous(low = "grey80", high = "red") +
    scale_shape_manual(values =c(19, 15, 15, 8, 19)) +
  facet_wrap(~editor)
plot2
ggsave("FigS7a.eps", width = 4.5, height = 16, path ="/Users/mc32/Documents/Matt_Coelho/Bioinformatics/Res1/figures_export")

plot3 <- ggplot(subset(amp_seq_results, is.na(editor)==FALSE & effect %in% c("missense", "ess_splice")), aes(x= edit_position, y= edit, fill = vaf)) +
  geom_tile() +
theme_classic()+
  labs(y="", x ="edit position", fill = "VAF") +
  geom_vline(xintercept = 4, linetype="dotted", colour = "grey65") +
  geom_vline(xintercept = 9, linetype="dotted", colour = "grey65") +
    scale_fill_gradient(low = "grey80", high = "red") +
  facet_wrap(~editor)
plot3

plot4 <- ggplot(subset(amp_seq_results_ABE, is.na(editor)==FALSE), aes(x= edit_position, y= guide, fill = ref_alt, alpha = vaf)) +
  geom_tile() +
theme_classic()+
  labs(y="", x ="edit position", fill = "", alpha = "VAF") +
  geom_vline(xintercept = 4, linetype="dotted", colour = "grey65") +
  geom_vline(xintercept = 9, linetype="dotted", colour = "grey65") +
  facet_wrap(~Gene)
plot4

plot5 <- ggplot(subset(amp_seq_results_CBE, is.na(editor)==FALSE), aes(x= edit_position, y= guide, fill = ref_alt, alpha = vaf)) +
  geom_tile() +
theme_classic()+
  labs(y="", x ="edit position", fill = "", alpha = "VAF") +
  geom_vline(xintercept = 4, linetype="dotted", colour = "grey65") +
  geom_vline(xintercept = 9, linetype="dotted", colour = "grey65") +
  facet_wrap(~Gene)
plot5
```

```{r}
# join amp_seq_results and Supplementary_Table_2

amp_seq_results %>% head
Supplementary_Table_2_join <-  Supplementary_Table_2 %>% select(!c(Amino_Acid_Position, chr, start, end, strand, guide, amplicon))
left_join(amp_seq_results, Supplementary_Table_2_join)

Discussion_table <- left_join(amp_seq_results, Supplementary_Table_2_join)

write.csv(Discussion_table, "Discussion_table.csv")
```

```{r}
# make hit list for NHS George Burghel

NHS_cases <- as_tibble(read.csv("NHS_in_house_sanger_somatic_detected_280923.csv", stringsAsFactors = FALSE, header = TRUE))

NHS_cases <- NHS_cases %>% select(G_guide, original_hgvs,	hgvs_c,	vcf_hg19, filtered_sample_count, vep_consequence, omics_db_url)

Discussion_table_filtered <- Discussion_table %>% select(!c(unique_well_id, vaf, sample, exp, name,	i7_tag,	i5_tag, plasmid_label, mtr,	wtr,	depth)) %>% distinct()

George_NHS_shortlist <- left_join(Discussion_table_filtered, NHS_cases) %>% filter(filtered_sample_count > 1) %>% filter(effect != "silent") %>% distinct()

write.csv(George_NHS_shortlist, "George_NHS_shortlist.csv")
```
```{r}
#analyse follow-up prime editing screens
pegRNA_counts <- as_tibble(read.csv("PE_guide_count/pycroquet_counts_021023.csv", stringsAsFactors = FALSE, header = TRUE))

#filter on 50 reads (no removal)
pegRNA_counts <- pegRNA_counts %>% filter(plasmid_reads >50 & edited_t7_exp1_reads >50 & edited_t7_exp2_reads >50 & osi_exp1_reads >50 & osi_exp2_reads >50)
pegRNA_counts

# compute average reads
pegRNA_counts <- mutate(pegRNA_counts, control_average_reads = average(control_exp1_reads, control_exp2_reads))
pegRNA_counts <- mutate(pegRNA_counts, gefit_average_reads = average(gefit_exp1_reads, gefit_exp2_reads))
pegRNA_counts <- mutate(pegRNA_counts, osi_average_reads = average(osi_exp1_reads, osi_exp2_reads))
pegRNA_counts <- mutate(pegRNA_counts, edited_t7_average_reads = average(edited_t7_exp1_reads, edited_t7_exp2_reads))

#RPM
pegRNA_counts <- mutate_at(pegRNA_counts, vars(matches("reads")), list(RPM = RPM))
pegRNA_counts

#L2FC - single exps.
pegRNA_counts <- mutate(pegRNA_counts, L2FC_PC9_control_plasmid_1 = L2FC(control_exp1_reads_RPM, plasmid_reads_RPM))
pegRNA_counts <- mutate(pegRNA_counts, L2FC_PC9_control_plasmid_2 = L2FC(control_exp2_reads_RPM, plasmid_reads_RPM))

pegRNA_counts <- mutate(pegRNA_counts, L2FC_PC9_control_t7_1 = L2FC(control_exp1_reads_RPM, edited_t7_exp1_reads_RPM))
pegRNA_counts <- mutate(pegRNA_counts, L2FC_PC9_control_t7_2 = L2FC(control_exp2_reads_RPM, edited_t7_exp2_reads_RPM))

pegRNA_counts <- mutate(pegRNA_counts, L2FC_PC9_gefit_plasmid_1 = L2FC(gefit_exp1_reads_RPM, plasmid_reads_RPM))
pegRNA_counts <- mutate(pegRNA_counts, L2FC_PC9_gefit_plasmid_2 = L2FC(gefit_exp2_reads_RPM, plasmid_reads_RPM))

pegRNA_counts <- mutate(pegRNA_counts, L2FC_PC9_osi_plasmid_1 = L2FC(osi_exp1_reads_RPM, plasmid_reads_RPM))
pegRNA_counts <- mutate(pegRNA_counts, L2FC_PC9_osi_plasmid_2 = L2FC(osi_exp2_reads_RPM, plasmid_reads_RPM))

pegRNA_counts <- mutate(pegRNA_counts, L2FC_PC9_gefit_control_1 = L2FC(gefit_exp1_reads_RPM, control_exp1_reads_RPM))
pegRNA_counts <- mutate(pegRNA_counts, L2FC_PC9_gefit_control_2 = L2FC(gefit_exp2_reads_RPM, control_exp2_reads_RPM))

pegRNA_counts <- mutate(pegRNA_counts, L2FC_PC9_osi_control_1 = L2FC(osi_exp1_reads_RPM, control_exp1_reads_RPM))
pegRNA_counts <- mutate(pegRNA_counts, L2FC_PC9_osi_control_2 = L2FC(osi_exp2_reads_RPM, control_exp2_reads_RPM))

pegRNA_counts <- mutate(pegRNA_counts, L2FC_PC9_gefit_t7_1 = L2FC(gefit_exp1_reads_RPM, edited_t7_exp1_reads_RPM))
pegRNA_counts <- mutate(pegRNA_counts, L2FC_PC9_gefit_t7_2 = L2FC(gefit_exp2_reads_RPM, edited_t7_exp2_reads_RPM))

pegRNA_counts <- mutate(pegRNA_counts, L2FC_PC9_osi_t7_1 = L2FC(osi_exp1_reads_RPM, edited_t7_exp1_reads_RPM))
pegRNA_counts <- mutate(pegRNA_counts, L2FC_PC9_osi_t7_2 = L2FC(osi_exp2_reads_RPM, edited_t7_exp2_reads_RPM))
```
```{r}
#compute L2FC from average of 2 exps from RPM values
pegRNA_counts <- mutate(pegRNA_counts, L2FC_PC9_control_plasmid_average = L2FC(control_average_reads_RPM, plasmid_reads_RPM))
pegRNA_counts <- mutate(pegRNA_counts, L2FC_PC9_control_t7_average = L2FC(control_average_reads_RPM, edited_t7_average_reads_RPM))
pegRNA_counts <- mutate(pegRNA_counts, L2FC_PC9_gefit_plasmid_average = L2FC(gefit_average_reads_RPM, plasmid_reads_RPM))
pegRNA_counts <- mutate(pegRNA_counts, L2FC_PC9_osi_plasmid_average = L2FC(osi_average_reads_RPM, plasmid_reads_RPM))
pegRNA_counts <- mutate(pegRNA_counts, L2FC_PC9_gefit_control_average = L2FC(gefit_average_reads_RPM, control_average_reads_RPM))
pegRNA_counts <- mutate(pegRNA_counts, L2FC_PC9_osi_control_average = L2FC(osi_average_reads_RPM, control_average_reads_RPM))
pegRNA_counts <- mutate(pegRNA_counts, L2FC_PC9_gefit_t7_average = L2FC(gefit_average_reads_RPM, edited_t7_average_reads_RPM))
pegRNA_counts <- mutate(pegRNA_counts, L2FC_PC9_osi_t7_average = L2FC(osi_average_reads_RPM, edited_t7_average_reads_RPM))
```

```{r}
# generate z scores for sc validation library
pegRNA_counts <- mutate_at(pegRNA_counts, vars(matches("L2FC")), list(zscore = zscore))
pegRNA_counts
```
```{r}
#PC9 check correlation between large screens and validation library
pos <- position_jitter(seed = 2, width = 0.2, height = 0.2)

# gefit vs plasmid
plot1 <- ggplot(pegRNA_counts, aes(y= L2FC_PC9_gefit_plasmid_average_zscore, x= L2FC_PC9_control_plasmid_average_zscore, colour = mutation_type)) +
geom_pointdensity(size=3,  position = pos) +
labs(title= "PC9 - prime editing screens", y = "gefitinib vs plasmid (z-score)", x = "control vs plasmid (z-score)", colour = "consequence") +
geom_text_repel(aes(label=ifelse(L2FC_PC9_gefit_plasmid_average_zscore > 2, paste(mutation), "")), size = 3, position = pos, max.overlaps = Inf) +
theme_bw(base_size = 15)
plot1

plot2 <- ggplot(pegRNA_counts, aes(y= L2FC_PC9_osi_plasmid_average_zscore, x= L2FC_PC9_control_plasmid_average_zscore, colour = mutation_type)) +
geom_pointdensity(size=3,  position = pos) +
labs(title= "PC9 - prime editing screens", y = "osimertinib vs plasmid (z-score)", x = "control vs plasmid (z-score)", colour = "consequence") +
geom_text_repel(aes(label=ifelse(L2FC_PC9_osi_plasmid_average_zscore > 2, paste(mutation), "")), size = 3, position = pos, max.overlaps = Inf) +
theme_bw(base_size = 15)
plot2

plot3 <- ggplot(pegRNA_counts, aes(y= L2FC_PC9_gefit_control_average_zscore, x= L2FC_PC9_control_plasmid_average_zscore, colour = mutation_type)) +
geom_pointdensity(size=3,  position = pos) +
labs(title= "PC9 - prime editing screens", y = "gefitinib vs control (z-score)", x = "control vs plasmid (z-score)", colour = "consequence") +
geom_text_repel(aes(label=ifelse(L2FC_PC9_gefit_control_average_zscore > 2, paste(mutation), "")), size = 3, position = pos, max.overlaps = Inf) +
theme_bw(base_size = 15)
plot3

plot4 <- ggplot(pegRNA_counts, aes(y= L2FC_PC9_osi_control_average_zscore, x= L2FC_PC9_control_plasmid_average_zscore, colour = mutation_type)) +
geom_pointdensity(size=3,  position = pos) +
labs(title= "PC9 - prime editing screens", y = "osimertinib vs control (z-score)", x = "control vs plasmid (z-score)", colour = "consequence") +
geom_text_repel(aes(label=ifelse(L2FC_PC9_osi_control_average_zscore > 2, paste(mutation), "")), size = 3, position = pos, max.overlaps = Inf) +
theme_bw(base_size = 15)
plot4

plot5 <- ggplot(pegRNA_counts, aes(y= L2FC_PC9_gefit_t7_average_zscore, x= L2FC_PC9_control_plasmid_average_zscore, colour = mutation_type)) +
geom_pointdensity(size=3,  position = pos) +
labs(title= "PC9 - prime editing screens", y = "gefitinib vs t7 (z-score)", x = "control vs plasmid (z-score)", colour = "consequence") +
geom_text_repel(aes(label=ifelse(L2FC_PC9_gefit_t7_average_zscore > 2, paste(mutation), "")), size = 3, position = pos, max.overlaps = Inf) +
theme_bw(base_size = 15)
plot5

plot6 <- ggplot(pegRNA_counts, aes(y= L2FC_PC9_osi_t7_average_zscore, x= L2FC_PC9_control_plasmid_average_zscore, colour = mutation_type)) +
geom_pointdensity(size=3,  position = pos) +
labs(title= "PC9 - prime editing screens", y = "osimertinib vs t7 (z-score)", x = "control vs plasmid (z-score)", colour = "consequence") +
geom_text_repel(aes(label=ifelse(L2FC_PC9_osi_t7_average_zscore > 2, paste(mutation), "")), size = 3, position = pos, max.overlaps = Inf) +
theme_bw(base_size = 15)
plot6

plot7 <- ggplot(pegRNA_counts, aes(y= L2FC_PC9_gefit_plasmid_average_zscore, x= L2FC_PC9_osi_plasmid_average_zscore, colour = mutation_type)) +
geom_pointdensity(size=3,  position = pos) +
labs(title= "PC9 - prime editing screens", y = "gefitinib vs plasmid (z-score)", x = "osimertinib vs plasmid (z-score)", colour = "consequence") +
geom_text_repel(aes(label=ifelse(L2FC_PC9_gefit_plasmid_average_zscore > 2 | L2FC_PC9_osi_plasmid_average_zscore >1, paste(mutation), "")), size = 3, position = pos, max.overlaps = Inf) +
theme_bw(base_size = 15)
plot7

plot8 <- ggplot(pegRNA_counts, aes(y= L2FC_PC9_gefit_control_average_zscore, x= L2FC_PC9_osi_control_average_zscore, colour = mutation_type)) +
geom_pointdensity(size=3,  position = pos) +
labs(title= "PC9 - prime editing screens", y = "gefitinib vs control (z-score)", x = "osimertinib vs control (z-score)", colour = "consequence") +
geom_text_repel(aes(label=ifelse(L2FC_PC9_gefit_control_average_zscore > 2 | L2FC_PC9_osi_control_average_zscore >1, paste(mutation), "")), size = 3, position = pos, max.overlaps = Inf) +
theme_bw(base_size = 15)
plot8

plot9 <- ggplot(pegRNA_counts, aes(y= L2FC_PC9_gefit_t7_average_zscore, x= L2FC_PC9_osi_t7_average_zscore, colour = mutation_type)) +
geom_pointdensity(size=3,  position = pos) +
labs(title= "PC9 - prime editing screens", y = "gefitinib vs t7 (z-score)", x = "osimertinib vs t7 (z-score)", colour = "consequence") +
geom_text_repel(aes(label=ifelse(L2FC_PC9_gefit_t7_average_zscore > 2 | L2FC_PC9_osi_t7_average_zscore >1, paste(mutation), "")), size = 3, position = pos, max.overlaps = Inf) +
theme_bw(base_size = 15)
plot9
```

```{r}
#replicates
plot1 <- ggplot(pegRNA_counts, aes(x= L2FC_PC9_gefit_control_1_zscore, y= L2FC_PC9_gefit_control_2_zscore, colour = mutation_type)) +
geom_pointdensity(size=3,  position = pos) +
labs(title= "PC9 - prime editing screens", x = "gefitinib vs control exp.1 (z-score)", y = "gefitinib vs control exp.2 (z-score)", colour = "consequence") +
geom_text_repel(aes(label=ifelse(L2FC_PC9_gefit_control_1_zscore >1 & L2FC_PC9_gefit_control_2_zscore >1, paste(mutation), "")), size = 3, position = pos, max.overlaps = Inf) +
theme_bw(base_size = 15)
plot1

plot2 <- ggplot(pegRNA_counts, aes(x= L2FC_PC9_osi_control_1_zscore, y= L2FC_PC9_osi_control_2_zscore, colour = mutation_type)) +
geom_pointdensity(size=3,  position = pos) +
labs(title= "PC9 - prime editing screens", x = "osimertinib vs control exp.1 (z-score)", y = "osimertinib vs control exp.2 (z-score)", colour = "consequence") +
geom_text_repel(aes(label=ifelse(L2FC_PC9_osi_control_1_zscore >1 & L2FC_PC9_osi_control_2_zscore >1, paste(mutation), "")), size = 3, position = pos, max.overlaps = Inf) +
theme_bw(base_size = 15)
plot2

plot3 <- ggplot(pegRNA_counts, aes(x= L2FC_PC9_gefit_plasmid_1_zscore, y= L2FC_PC9_gefit_plasmid_2_zscore, colour = mutation_type)) +
geom_pointdensity(size=3,  position = pos) +
labs(title= "PC9 - prime editing screens", x = "gefitinib vs plasmid exp.1 (z-score)", y = "gefitinib vs plasmid exp.2 (z-score)", colour = "consequence") +
geom_text_repel(aes(label=ifelse(L2FC_PC9_gefit_plasmid_1_zscore >1 & L2FC_PC9_gefit_plasmid_2_zscore >1, paste(mutation), "")), size = 3, position = pos, max.overlaps = Inf) +
theme_bw(base_size = 15)
plot3

plot4 <- ggplot(pegRNA_counts, aes(x= L2FC_PC9_osi_plasmid_1_zscore, y= L2FC_PC9_osi_plasmid_2_zscore, colour = mutation_type)) +
geom_pointdensity(size=3,  position = pos) +
labs(title= "PC9 - prime editing screens", x = "osimertinib vs plasmid exp.1 (z-score)", y = "osimertinib vs plasmid exp.2 (z-score)", colour = "consequence") +
geom_text_repel(aes(label=ifelse(L2FC_PC9_osi_plasmid_1_zscore >1 & L2FC_PC9_osi_plasmid_2_zscore >1, paste(mutation), "")), size = 3, position = pos, max.overlaps = Inf) +
theme_bw(base_size = 15)
plot4

plot5 <- ggplot(pegRNA_counts, aes(x= L2FC_PC9_gefit_t7_1_zscore, y= L2FC_PC9_gefit_t7_2_zscore, colour = mutation_type)) +
geom_pointdensity(size=3,  position = pos) +
labs(title= "PC9 - prime editing screens", x = "gefitinib vs t7 exp.1 (z-score)", y = "gefitinib vs t7 exp.2 (z-score)", colour = "consequence") +
geom_text_repel(aes(label=ifelse(L2FC_PC9_gefit_t7_1_zscore >1 & L2FC_PC9_gefit_t7_2_zscore >1, paste(mutation), "")), size = 3, position = pos, max.overlaps = Inf) +
theme_bw(base_size = 15)
plot5

plot6 <- ggplot(pegRNA_counts, aes(x= L2FC_PC9_osi_t7_1_zscore, y= L2FC_PC9_osi_t7_2_zscore, colour = mutation_type)) +
geom_pointdensity(size=3,  position = pos) +
labs(title= "PC9 - prime editing screens", x = "osimertinib vs t7 exp.1 (z-score)", y = "osimertinib vs t7 exp.2 (z-score)", colour = "consequence") +
geom_text_repel(aes(label=ifelse(L2FC_PC9_osi_t7_1_zscore >1 & L2FC_PC9_osi_t7_2_zscore >1, paste(mutation), "")), size = 3, position = pos, max.overlaps = Inf) +
theme_bw(base_size = 15)
plot6

plot7 <- ggplot(pegRNA_counts, aes(x= L2FC_PC9_control_plasmid_1_zscore, y= L2FC_PC9_control_plasmid_2_zscore, colour = mutation_type)) +
geom_pointdensity(size=3,  position = pos) +
labs(title= "PC9 - prime editing screens", x = "control vs plasmid exp.1 (z-score)", y = "control vs plasmid exp.2 (z-score)", colour = "consequence") +
geom_text_repel(aes(label=ifelse(L2FC_PC9_control_plasmid_1_zscore < -1 & L2FC_PC9_control_plasmid_2_zscore < -1, paste(mutation), "")), size = 3, position = pos, max.overlaps = Inf) +
theme_bw(base_size = 15)
plot7
```

```{r}
#histograms
plot1 <- ggplot(pegRNA_counts, aes(x= L2FC_PC9_osi_t7_1_zscore, colour = mutation_type)) +
geom_histogram(size=3,  position = pos) +
labs(title= "PC9 - prime editing screens", x = "osimertinib vs t7 exp.1 (z-score)", y = "count", colour = "consequence") +
#geom_text_repel(aes(label=ifelse(L2FC_PC9_osi_t7_1_zscore >1 & L2FC_PC9_osi_t7_2_zscore >1, paste(mutation), "")), size = 3, position = pos, max.overlaps = Inf) +
theme_bw(base_size = 15)
plot1

plot2 <- ggplot(pegRNA_counts, aes(x= L2FC_PC9_osi_t7_2_zscore, colour = mutation_type)) +
geom_histogram(size=3,  position = pos) +
labs(title= "PC9 - prime editing screens", x = "osimertinib vs t7 exp.2 (z-score)", y = "count", colour = "consequence") +
#geom_text_repel(aes(label=ifelse(L2FC_PC9_osi_t7_1_zscore >1 & L2FC_PC9_osi_t7_2_zscore >1, paste(mutation), "")), size = 3, position = pos, max.overlaps = Inf) +
theme_bw(base_size = 15)
plot2

plot3 <- ggplot(pegRNA_counts, aes(x= L2FC_PC9_gefit_t7_1_zscore, colour = mutation_type)) +
geom_histogram(size=3,  position = pos) +
labs(title= "PC9 - prime editing screens", x = "gefit vs t7 exp.1 (z-score)", y = "count", colour = "consequence") +
#geom_text_repel(aes(label=ifelse(L2FC_PC9_osi_t7_1_zscore >1 & L2FC_PC9_osi_t7_2_zscore >1, paste(mutation), "")), size = 3, position = pos, max.overlaps = Inf) +
theme_bw(base_size = 15)
plot3

plot4 <- ggplot(pegRNA_counts, aes(x= L2FC_PC9_gefit_t7_2_zscore, colour = mutation_type)) +
geom_histogram(size=3,  position = pos) +
labs(title= "PC9 - prime editing screens", x = "gefit vs t7 exp.2 (z-score)", y = "count", colour = "consequence") +
#geom_text_repel(aes(label=ifelse(L2FC_PC9_osi_t7_1_zscore >1 & L2FC_PC9_osi_t7_2_zscore >1, paste(mutation), "")), size = 3, position = pos, max.overlaps = Inf) +
theme_bw(base_size = 15)
plot4

plot5 <- ggplot(pegRNA_counts, aes(x= L2FC_PC9_control_plasmid_1_zscore, colour = mutation_type)) +
geom_histogram(size=3,  position = pos) +
labs(title= "PC9 - prime editing screens", x = "control vs plasmid exp.1 (z-score)", y = "count", colour = "consequence") +
#geom_text_repel(aes(label=ifelse(L2FC_PC9_osi_t7_1_zscore >1 & L2FC_PC9_osi_t7_2_zscore >1, paste(mutation), "")), size = 3, position = pos, max.overlaps = Inf) +
theme_bw(base_size = 15)
plot5

plot6 <- ggplot(pegRNA_counts, aes(x= L2FC_PC9_control_plasmid_2_zscore, colour = mutation_type)) +
geom_histogram(size=3,  position = pos) +
labs(title= "PC9 - prime editing screens", x = "control vs plasmid exp.2 (z-score)", y = "count", colour = "consequence") +
#geom_text_repel(aes(label=ifelse(L2FC_PC9_osi_t7_1_zscore >1 & L2FC_PC9_osi_t7_2_zscore >1, paste(mutation), "")), size = 3, position = pos, max.overlaps = Inf) +
theme_bw(base_size = 15)
plot6
```